// txl/dropdowns/_DropdownArrow.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ARROW_STYLE = exports.HOLDER_STYLE = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _helpers = require('../styles/helpers');

var _layers = require('../styles/mixins/layers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ARROW_SIZE = (0, _helpers.gridUnits)(4);

var HOLDER_STYLE = exports.HOLDER_STYLE = {
  alignItems: 'center',
  display: 'flex',
  height: 0,
  justifyContent: 'center',
  position: 'relative',
  width: 'auto',
  zIndex: 10
};

var ARROW_STYLE = exports.ARROW_STYLE = (0, _extends3.default)({}, _layers.LAYER_2, (0, _helpers.square)(ARROW_SIZE), {
  backgroundColor: 'white',
  borderRadius: 2,
  flex: '0 0 auto',
  transform: 'rotate(-45deg)'
});
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ARROW_SIZE, 'ARROW_SIZE', 'src/dropdowns/_DropdownArrow.style.js');

  __REACT_HOT_LOADER__.register(HOLDER_STYLE, 'HOLDER_STYLE', 'src/dropdowns/_DropdownArrow.style.js');

  __REACT_HOT_LOADER__.register(ARROW_STYLE, 'ARROW_STYLE', 'src/dropdowns/_DropdownArrow.style.js');
}();

;
