// txl/dropdowns/_Dropdown.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CONTENT_STYLE = exports.CONTAINER_STYLES = exports.WRAPPER_STYLE = exports.DROPDOWN_BASE = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _helpers = require('../styles/helpers');

var _layers = require('../styles/mixins/layers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DROPDOWN_BASE = {
  left: 0,
  top: (0, _helpers.gridUnits)(5)
};

var WRAPPER_STYLE = {
  position: 'relative'
};

var CONTAINER_STYLES = (0, _extends3.default)({}, _layers.LAYER_2, {
  marginTop: (0, _helpers.gridUnits)(1),
  position: 'absolute',
  right: 0,
  top: 0
});

var CONTENT_STYLE = {
  backgroundColor: 'white'
};

exports.DROPDOWN_BASE = DROPDOWN_BASE;
exports.WRAPPER_STYLE = WRAPPER_STYLE;
exports.CONTAINER_STYLES = CONTAINER_STYLES;
exports.CONTENT_STYLE = CONTENT_STYLE;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(DROPDOWN_BASE, 'DROPDOWN_BASE', 'src/dropdowns/_Dropdown.style.js');

  __REACT_HOT_LOADER__.register(WRAPPER_STYLE, 'WRAPPER_STYLE', 'src/dropdowns/_Dropdown.style.js');

  __REACT_HOT_LOADER__.register(CONTAINER_STYLES, 'CONTAINER_STYLES', 'src/dropdowns/_Dropdown.style.js');

  __REACT_HOT_LOADER__.register(CONTENT_STYLE, 'CONTENT_STYLE', 'src/dropdowns/_Dropdown.style.js');
}();

;
