// txl/dropdowns/_DropdownContent.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _helpers = require('../styles/helpers');

var _layers = require('../styles/mixins/layers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CONTAINER_STYLES = (0, _extends3.default)({}, _layers.LAYER_2, {
  backgroundColor: 'white'
});

var TxlDropdownContent = (0, _radium2.default)(_class = function (_React$Component) {
  (0, _inherits3.default)(TxlDropdownContent, _React$Component);

  function TxlDropdownContent() {
    (0, _classCallCheck3.default)(this, TxlDropdownContent);
    return (0, _possibleConstructorReturn3.default)(this, (TxlDropdownContent.__proto__ || (0, _getPrototypeOf2.default)(TxlDropdownContent)).apply(this, arguments));
  }

  (0, _createClass3.default)(TxlDropdownContent, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          style = _props.style,
          children = _props.children,
          otherProps = (0, _objectWithoutProperties3.default)(_props, ['style', 'children']);


      var containerStyles = (0, _helpers.combineStyles)(CONTAINER_STYLES, style);

      var component = _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, otherProps, {
          'data-component': 'TxlDropdownContent',
          style: containerStyles
        }),
        children
      );

      return this.props.open ? component : null;
    }
  }]);
  return TxlDropdownContent;
}(_react2.default.Component)) || _class;

exports.default = TxlDropdownContent;


TxlDropdownContent.propTypes = {
  children: _react.PropTypes.node,
  open: _react.PropTypes.bool,
  style: _react.PropTypes.object
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(CONTAINER_STYLES, 'CONTAINER_STYLES', 'src/dropdowns/_DropdownContent.jsx');

  __REACT_HOT_LOADER__.register(TxlDropdownContent, 'TxlDropdownContent', 'src/dropdowns/_DropdownContent.jsx');
}();

;
module.exports = exports['default'];
