// txl/dropdowns/_DropdownListItem.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LIST_ITEM_SELECTED_STYLE = exports.LIST_ITEM_HOVER_STYLE = exports.LIST_ITEM_STYLE = exports.ITEM_HEIGHT = undefined;

var _theme = require('../styles/theme');

var _helpers = require('../styles/helpers');

var ITEM_HEIGHT = exports.ITEM_HEIGHT = (0, _helpers.gridUnitsRaw)(7); // in grid units

var LIST_ITEM_STYLE = exports.LIST_ITEM_STYLE = {
  backgroundColor: 'white',
  color: _theme.COLOR_NEUTRAL['50c'],
  cursor: 'pointer',
  height: ITEM_HEIGHT,
  lineHeight: ITEM_HEIGHT + 'px', // px required for line-height since unitless valid by default
  paddingLeft: (0, _helpers.gridUnits)(3),
  paddingRight: (0, _helpers.gridUnits)(3),
  whiteSpace: 'nowrap'
};

var LIST_ITEM_HOVER_STYLE = exports.LIST_ITEM_HOVER_STYLE = {
  backgroundColor: _theme.COLOR_NEUTRAL['50'],
  color: _theme.COLOR_NEUTRAL['50c']
};

var LIST_ITEM_SELECTED_STYLE = exports.LIST_ITEM_SELECTED_STYLE = {
  backgroundColor: _theme.COLOR_ACCENT['500'],
  color: _theme.COLOR_ACCENT['500c']
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ITEM_HEIGHT, 'ITEM_HEIGHT', 'src/dropdowns/_DropdownListItem.style.js');

  __REACT_HOT_LOADER__.register(LIST_ITEM_STYLE, 'LIST_ITEM_STYLE', 'src/dropdowns/_DropdownListItem.style.js');

  __REACT_HOT_LOADER__.register(LIST_ITEM_HOVER_STYLE, 'LIST_ITEM_HOVER_STYLE', 'src/dropdowns/_DropdownListItem.style.js');

  __REACT_HOT_LOADER__.register(LIST_ITEM_SELECTED_STYLE, 'LIST_ITEM_SELECTED_STYLE', 'src/dropdowns/_DropdownListItem.style.js');
}();

;
