// txl/dropdowns/_DropdownListItem.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _clone = require('lodash/clone');

var _clone2 = _interopRequireDefault(_clone);

var _TruncatedText = require('../text/TruncatedText');

var _TruncatedText2 = _interopRequireDefault(_TruncatedText);

var _helpers = require('../styles/helpers');

var _DropdownListItem = require('./_DropdownListItem.style');

var _Base = require('../base/Base');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlDropdownListItem = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlDropdownListItem, _React$Component);

  function TxlDropdownListItem() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlDropdownListItem);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlDropdownListItem.__proto__ || (0, _getPrototypeOf2.default)(TxlDropdownListItem)).call.apply(_ref, [this].concat(args)));

    _this.state = { hovered: false };
    return _this;
  }

  (0, _createClass3.default)(TxlDropdownListItem, [{
    key: '_handleMouseEnter',
    value: function _handleMouseEnter() {
      this.setState({ hovered: true });
    }
  }, {
    key: '_handleMouseLeave',
    value: function _handleMouseLeave() {
      this.setState({ hovered: false });
    }
  }, {
    key: '_handleClick',
    value: function _handleClick() {
      this.props.onClick((0, _clone2.default)(this.props.item, true));
    }
  }, {
    key: 'render',
    value: function render() {
      var styles = (0, _helpers.combineStyles)(_DropdownListItem.LIST_ITEM_STYLE, this.state.hovered && _DropdownListItem.LIST_ITEM_HOVER_STYLE, this.props.selected && _DropdownListItem.LIST_ITEM_SELECTED_STYLE);

      return _react2.default.createElement(
        'li',
        {
          id: this.props.id,
          onMouseEnter: this._handleMouseEnter,
          onMouseLeave: this._handleMouseLeave,
          onClick: this._handleClick,
          style: styles
        },
        _react2.default.createElement(_TruncatedText2.default, {
          text: this.props.item.label
        })
      );
    }
  }]);
  return TxlDropdownListItem;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleMouseEnter', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseEnter'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleMouseLeave', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseLeave'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleClick'), _class2.prototype)), _class2)) || _class;

exports.default = TxlDropdownListItem;


TxlDropdownListItem.propTypes = {
  id: _react.PropTypes.string,
  item: _react.PropTypes.shape({
    label: _react.PropTypes.node
  }).isRequired,
  onClick: _react.PropTypes.func,
  selected: _react.PropTypes.bool
};

TxlDropdownListItem.defaultProps = {
  onClick: _Base.NOOP,
  selected: false
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlDropdownListItem, 'TxlDropdownListItem', 'src/dropdowns/_DropdownListItem.jsx');
}();

;
module.exports = exports['default'];
