// txl/dropdowns/_DropdownList.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _list = require('../styles/mixins/list');

var _DropdownListItem = require('./_DropdownListItem');

var _DropdownListItem2 = _interopRequireDefault(_DropdownListItem);

var _DropdownListItem3 = require('./_DropdownListItem.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var NO_ITEMS_FOUND = 'No Items Found';
var ITEMS_TO_SHOW = 10;

var LIST_STYLE = (0, _extends3.default)({}, _list.UNSTYLED_LIST, {
  maxHeight: _DropdownListItem3.ITEM_HEIGHT * ITEMS_TO_SHOW,
  overflowX: 'hidden',
  overflowY: 'auto'
});

var TxlDropdownList = function TxlDropdownList(props) {
  var items = void 0;
  if (props.items.length) {
    items = props.items.map(function (item, index) {
      return _react2.default.createElement(_DropdownListItem2.default, {
        id: props.id && props.id + '-' + item.value,
        item: item,
        key: index,
        selected: props.selectedValues.indexOf(item.value) > -1,
        onClick: function onClick(i) {
          return props.onClick(i);
        }
      });
    });
  } else {
    items = _react2.default.createElement(
      'div',
      { style: _DropdownListItem3.LIST_ITEM_STYLE },
      NO_ITEMS_FOUND
    );
  }

  return _react2.default.createElement(
    'ul',
    {
      'data-component': 'TxlDropdownList',
      style: LIST_STYLE
    },
    items
  );
};

TxlDropdownList.propTypes = {
  id: _react.PropTypes.string,
  items: _react.PropTypes.arrayOf(_react.PropTypes.shape({
    label: _react.PropTypes.node,
    value: _react.PropTypes.string
  })),
  onClick: _react.PropTypes.func.isRequired,
  selectedValues: _react.PropTypes.arrayOf(_react.PropTypes.string)
};

TxlDropdownList.defaultProps = {
  onClick: _Base.NOOP,
  selectedValues: []
};

var _default = (0, _radium2.default)(TxlDropdownList);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(NO_ITEMS_FOUND, 'NO_ITEMS_FOUND', 'src/dropdowns/_DropdownList.jsx');

  __REACT_HOT_LOADER__.register(ITEMS_TO_SHOW, 'ITEMS_TO_SHOW', 'src/dropdowns/_DropdownList.jsx');

  __REACT_HOT_LOADER__.register(LIST_STYLE, 'LIST_STYLE', 'src/dropdowns/_DropdownList.jsx');

  __REACT_HOT_LOADER__.register(TxlDropdownList, 'TxlDropdownList', 'src/dropdowns/_DropdownList.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/dropdowns/_DropdownList.jsx');
}();

;
module.exports = exports['default'];
