// txl/dropdowns/_DropdownContentWithArrow.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Dropdown = require('./_Dropdown.style');

var _DropdownArrow = require('./_DropdownArrow');

var _DropdownArrow2 = _interopRequireDefault(_DropdownArrow);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TxlDropdownContentWithArrow = (0, _radium2.default)(_class = function (_React$Component) {
  (0, _inherits3.default)(TxlDropdownContentWithArrow, _React$Component);

  function TxlDropdownContentWithArrow() {
    (0, _classCallCheck3.default)(this, TxlDropdownContentWithArrow);
    return (0, _possibleConstructorReturn3.default)(this, (TxlDropdownContentWithArrow.__proto__ || (0, _getPrototypeOf2.default)(TxlDropdownContentWithArrow)).apply(this, arguments));
  }

  (0, _createClass3.default)(TxlDropdownContentWithArrow, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          arrowProps = _props.arrowProps,
          children = _props.children,
          contentStyle = _props.contentStyle,
          style = _props.style,
          otherProps = (0, _objectWithoutProperties3.default)(_props, ['arrowProps', 'children', 'contentStyle', 'style']);

      var component = _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlDropdownContentWithArrow',
          style: [_Dropdown.CONTAINER_STYLES, style]
        },
        _react2.default.createElement(_DropdownArrow2.default, arrowProps),
        _react2.default.createElement(
          'div',
          (0, _extends3.default)({}, otherProps, {
            'data-component': 'TxlDropdownContent',
            style: [_Dropdown.CONTENT_STYLE, contentStyle]
          }),
          children
        )
      );

      return this.props.open ? component : null;
    }
  }]);
  return TxlDropdownContentWithArrow;
}(_react2.default.Component)) || _class;

exports.default = TxlDropdownContentWithArrow;


TxlDropdownContentWithArrow.propTypes = {
  arrowProps: _react.PropTypes.object,
  children: _react.PropTypes.node,
  contentStyle: _react.PropTypes.object,
  open: _react.PropTypes.bool,
  style: _react.PropTypes.object
};

TxlDropdownContentWithArrow.defaultProps = {
  arrowProps: {}
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlDropdownContentWithArrow, 'TxlDropdownContentWithArrow', 'src/dropdowns/_DropdownContentWithArrow.jsx');
}();

;
module.exports = exports['default'];
