// txl/dropdowns/_FlyoutTargetBase.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _TetheredPopoverTarget = require('../_popover/_TetheredPopoverTarget');

var _TetheredPopoverTarget2 = _interopRequireDefault(_TetheredPopoverTarget);

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TETHER_CONFIGURATION = {
  attachment: 'top center',
  constraints: [{
    attachment: 'none',
    pin: true,
    to: 'window' }],
  offset: '-' + (0, _helpers.gridUnits)(1) + ' 0',
  optimizations: {
    moveElement: false
  },
  targetAttachment: 'bottom center',
  targetOffset: '0 0'
};

var TxlFlyoutTargetBase = function TxlFlyoutTargetBase(props) {
  var children = props.children,
      content = props.content,
      style = props.style,
      visible = props.visible,
      otherProps = (0, _objectWithoutProperties3.default)(props, ['children', 'content', 'style', 'visible']);


  return _react2.default.createElement(
    _TetheredPopoverTarget2.default,
    (0, _extends3.default)({}, otherProps, {
      content: content,
      style: style,
      visible: visible,
      popoverZIndex: props.popoverZIndex,
      configuration: TETHER_CONFIGURATION
    }),
    children
  );
};

TxlFlyoutTargetBase.propTypes = {
  children: _TetheredPopoverTarget2.default.propTypes.children,
  componentName: _TetheredPopoverTarget2.default.propTypes.componentName,
  content: _TetheredPopoverTarget2.default.propTypes.content,
  popoverZIndex: _TetheredPopoverTarget2.default.propTypes.popoverZIndex,
  style: _TetheredPopoverTarget2.default.propTypes.style,
  visible: _TetheredPopoverTarget2.default.propTypes.visible
};

var _default = TxlFlyoutTargetBase;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TETHER_CONFIGURATION, 'TETHER_CONFIGURATION', 'src/dropdowns/_FlyoutTargetBase.jsx');

  __REACT_HOT_LOADER__.register(TxlFlyoutTargetBase, 'TxlFlyoutTargetBase', 'src/dropdowns/_FlyoutTargetBase.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/dropdowns/_FlyoutTargetBase.jsx');
}();

;
module.exports = exports['default'];
