// txl/dropdowns/_TourPointTargetBase.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _desc, _value, _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _omit = require('lodash/omit');

var _omit2 = _interopRequireDefault(_omit);

var _Base = require('../base/Base');

var _TetheredPopoverTarget = require('../_popover/_TetheredPopoverTarget');

var _TetheredPopoverTarget2 = _interopRequireDefault(_TetheredPopoverTarget);

var _helpers = require('../styles/helpers');

var _HintDot = require('../notifications/_HintDot');

var _HintDot2 = _interopRequireDefault(_HintDot);

var _eventTargetAncestryComparator = require('../utils/eventTargetAncestryComparator');

var _eventTargetAncestryComparator2 = _interopRequireDefault(_eventTargetAncestryComparator);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TETHER_CONFIGURATION = {
  attachment: 'center left',
  constraints: [{
    attachment: 'both',
    pin: true,
    to: 'window'
  }],
  optimizations: {
    moveElement: true
  },
  targetAttachment: 'center middle',
  targetOffset: '0 ' + (0, _helpers.gridUnits)(3)
};

var TxlTourPointTarget = (_class = function (_React$Component) {
  (0, _inherits3.default)(TxlTourPointTarget, _React$Component);

  function TxlTourPointTarget() {
    (0, _classCallCheck3.default)(this, TxlTourPointTarget);
    return (0, _possibleConstructorReturn3.default)(this, (TxlTourPointTarget.__proto__ || (0, _getPrototypeOf2.default)(TxlTourPointTarget)).apply(this, arguments));
  }

  (0, _createClass3.default)(TxlTourPointTarget, [{
    key: '_handleAwayClick',
    value: function _handleAwayClick(e) {
      if (!(0, _eventTargetAncestryComparator2.default)(e, this._popup)) {
        this.props.onAwayClick(e);
      }
    }
  }, {
    key: '_handleRef',
    value: function _handleRef(ref) {
      this._popup = ref;
      if (ref) {
        window.addEventListener('click', this._handleAwayClick, true);
      } else {
        window.removeEventListener('click', this._handleAwayClick, true);
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          content = _props.content,
          onHintClick = _props.onHintClick,
          pulsing = _props.pulsing,
          style = _props.style,
          visible = _props.visible,
          otherProps = (0, _objectWithoutProperties3.default)(_props, ['content', 'onHintClick', 'pulsing', 'style', 'visible']);


      var safeProps = (0, _omit2.default)(otherProps, 'onAwayClick');

      var wrappedContent = _react2.default.createElement(
        'div',
        { ref: this._handleRef },
        content
      );
      return _react2.default.createElement(
        _TetheredPopoverTarget2.default,
        (0, _extends3.default)({}, safeProps, {
          componentName: 'TxlTourPointTargetBase',
          configuration: TETHER_CONFIGURATION,
          content: wrappedContent,
          onClick: onHintClick,
          style: style,
          visible: visible
        }),
        _react2.default.createElement(_HintDot2.default, { pulsing: pulsing })
      );
    }
  }]);
  return TxlTourPointTarget;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class.prototype, '_handleAwayClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleAwayClick'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_handleRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleRef'), _class.prototype)), _class);
exports.default = TxlTourPointTarget;


TxlTourPointTarget.defaultProps = {
  onAwayClick: Function.prototype,
  onHintClick: Function.prototype,
  visible: false
};

TxlTourPointTarget.propTypes = {
  content: _TetheredPopoverTarget2.default.propTypes.content,
  onAwayClick: _react.PropTypes.func,
  onHintClick: _react.PropTypes.func,
  pulsing: _react.PropTypes.bool,
  style: _TetheredPopoverTarget2.default.propTypes.style,
  visible: _TetheredPopoverTarget2.default.propTypes.visible
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TETHER_CONFIGURATION, 'TETHER_CONFIGURATION', 'src/dropdowns/_TourPointTargetBase.jsx');

  __REACT_HOT_LOADER__.register(TxlTourPointTarget, 'TxlTourPointTarget', 'src/dropdowns/_TourPointTargetBase.jsx');
}();

;
module.exports = exports['default'];
