// txl/dropdowns/_UncontrolledSingleSelect.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _desc, _value, _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _SingleSelect = require('./_SingleSelect');

var _SingleSelect2 = _interopRequireDefault(_SingleSelect);

var _Base = require('../base/Base');

var _listHelpers = require('../lists/listHelpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlUncontrolledSingleSelect = (_class = function (_React$Component) {
  (0, _inherits3.default)(TxlUncontrolledSingleSelect, _React$Component);

  function TxlUncontrolledSingleSelect(args) {
    (0, _classCallCheck3.default)(this, TxlUncontrolledSingleSelect);

    var _this = (0, _possibleConstructorReturn3.default)(this, (TxlUncontrolledSingleSelect.__proto__ || (0, _getPrototypeOf2.default)(TxlUncontrolledSingleSelect)).call(this, args));

    var _this$props = _this.props,
        items = _this$props.items,
        defaultValue = _this$props.defaultValue;

    var matchedItem = (0, _listHelpers.getItemByValue)(items, defaultValue ? defaultValue.value : undefined);

    _this.state = {
      open: false,
      value: matchedItem
    };
    return _this;
  }

  (0, _createClass3.default)(TxlUncontrolledSingleSelect, [{
    key: '_handleChange',
    value: function _handleChange(_ref) {
      var _this2 = this;

      var value = _ref.value;

      this.setState({ value: value }, function () {
        _this2.props.onChange(value);
      });
    }
  }, {
    key: '_openDropdown',
    value: function _openDropdown() {
      this.setState({ open: true });
    }
  }, {
    key: '_closeDropdown',
    value: function _closeDropdown() {
      this.setState({ open: false });
    }
  }, {
    key: 'setValue',
    value: function setValue() {
      var value = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : '';

      this.setState({
        value: value
      });
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          allowNone = _props.allowNone,
          dropdownHeight = _props.dropdownHeight,
          dropdownWidth = _props.dropdownWidth,
          grouper = _props.grouper,
          items = _props.items,
          label = _props.label,
          loading = _props.loading,
          noContentView = _props.noContentView,
          onItemCreate = _props.onItemCreate,
          onSearchChange = _props.onSearchChange,
          placeholder = _props.placeholder,
          secondaryText = _props.secondaryText,
          showCreateNew = _props.showCreateNew,
          showCreateNewLabel = _props.showCreateNewLabel,
          showSearchToViewMoreMessage = _props.showSearchToViewMoreMessage,
          validationState = _props.validationState,
          width = _props.width;


      return _react2.default.createElement(_SingleSelect2.default, {
        allowNone: allowNone,
        closeDropdown: this._closeDropdown,
        dropdownHeight: dropdownHeight,
        dropdownWidth: dropdownWidth,
        grouper: grouper,
        items: items,
        label: label,
        loading: loading,
        noContentView: noContentView,
        onChange: this._handleChange,
        onItemCreate: onItemCreate,
        onSearchChange: onSearchChange,
        open: this.state.open,
        openDropdown: this._openDropdown,
        placeholder: placeholder,
        secondaryText: secondaryText,
        showCreateNew: showCreateNew,
        showCreateNewLabel: showCreateNewLabel,
        showSearchToViewMoreMessage: showSearchToViewMoreMessage,
        validationState: validationState,
        value: this.state.value,
        width: width
      });
    }
  }]);
  return TxlUncontrolledSingleSelect;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class.prototype, '_handleChange', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleChange'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_openDropdown', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_openDropdown'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_closeDropdown', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_closeDropdown'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, 'setValue', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, 'setValue'), _class.prototype)), _class);
exports.default = TxlUncontrolledSingleSelect;


TxlUncontrolledSingleSelect.displayName = 'TxlUncontrolledSingleSelect';

TxlUncontrolledSingleSelect.propTypes = {
  allowNone: _SingleSelect2.default.propTypes.allowNone,
  defaultValue: _SingleSelect2.default.propTypes.value,
  dropdownHeight: _SingleSelect2.default.propTypes.dropdownHeight,
  dropdownWidth: _SingleSelect2.default.propTypes.dropdownWidth,
  grouper: _SingleSelect2.default.propTypes.grouper,
  items: _SingleSelect2.default.propTypes.items,
  label: _SingleSelect2.default.propTypes.label,
  loading: _SingleSelect2.default.propTypes.loading,
  noContentView: _SingleSelect2.default.propTypes.noContentView,
  onChange: _SingleSelect2.default.propTypes.onChange,
  onItemCreate: _SingleSelect2.default.propTypes.onItemCreate,
  onSearchChange: _SingleSelect2.default.propTypes.onSearchChange,
  placeholder: _SingleSelect2.default.propTypes.placeholder,
  secondaryText: _SingleSelect2.default.propTypes.secondaryText,
  showCreateNew: _SingleSelect2.default.propTypes.showCreateNew,
  showCreateNewLabel: _SingleSelect2.default.propTypes.showCreateNewLabel,
  showSearchToViewMoreMessage: _SingleSelect2.default.propTypes.showSearchToViewMoreMessage,
  validationState: _SingleSelect2.default.propTypes.validationState,
  width: _SingleSelect2.default.propTypes.width
};

TxlUncontrolledSingleSelect.defaultProps = {
  onChange: _Base.NOOP
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlUncontrolledSingleSelect, 'TxlUncontrolledSingleSelect', 'src/dropdowns/_UncontrolledSingleSelect.jsx');
}();

;
module.exports = exports['default'];
