// txl/dropdowns/_UncontrolledMultiSelect.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _desc, _value, _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _MultiSelect = require('./_MultiSelect');

var _MultiSelect2 = _interopRequireDefault(_MultiSelect);

var _Base = require('../base/Base');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlUncontrolledMultiSelect = (_class = function (_React$Component) {
  (0, _inherits3.default)(TxlUncontrolledMultiSelect, _React$Component);

  function TxlUncontrolledMultiSelect(args) {
    (0, _classCallCheck3.default)(this, TxlUncontrolledMultiSelect);

    var _this = (0, _possibleConstructorReturn3.default)(this, (TxlUncontrolledMultiSelect.__proto__ || (0, _getPrototypeOf2.default)(TxlUncontrolledMultiSelect)).call(this, args));

    _this.state = {
      open: false,
      value: []
    };
    return _this;
  }

  (0, _createClass3.default)(TxlUncontrolledMultiSelect, [{
    key: '_handleChange',
    value: function _handleChange(_ref) {
      var _this2 = this;

      var value = _ref.value;

      this.setState({ value: value }, function () {
        _this2.props.onChange(value);
      });
    }
  }, {
    key: '_openDropdown',
    value: function _openDropdown() {
      this.setState({ open: true });
    }
  }, {
    key: '_closeDropdown',
    value: function _closeDropdown() {
      this.setState({ open: false });
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          dropdownHeight = _props.dropdownHeight,
          dropdownWidth = _props.dropdownWidth,
          enableLocalFiltering = _props.enableLocalFiltering,
          items = _props.items,
          label = _props.label,
          loading = _props.loading,
          noContentView = _props.noContentView,
          onItemCreate = _props.onItemCreate,
          onSearchChange = _props.onSearchChange,
          placeholder = _props.placeholder,
          secondaryText = _props.secondaryText,
          showCreateNew = _props.showCreateNew,
          showSearchToViewMoreMessage = _props.showSearchToViewMoreMessage,
          validationState = _props.validationState,
          width = _props.width;


      return _react2.default.createElement(_MultiSelect2.default, {
        closeDropdown: this._closeDropdown,
        dropdownHeight: dropdownHeight,
        dropdownWidth: dropdownWidth,
        enableLocalFiltering: enableLocalFiltering,
        items: items,
        label: label,
        loading: loading,
        noContentView: noContentView,
        onChange: this._handleChange,
        onItemCreate: onItemCreate,
        onSearchChange: onSearchChange,
        open: this.state.open,
        openDropdown: this._openDropdown,
        placeholder: placeholder,
        secondaryText: secondaryText,
        showCreateNew: showCreateNew,
        showSearchToViewMoreMessage: showSearchToViewMoreMessage,
        validationState: validationState,
        value: this.state.value,
        width: width
      });
    }
  }]);
  return TxlUncontrolledMultiSelect;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class.prototype, '_handleChange', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleChange'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_openDropdown', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_openDropdown'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_closeDropdown', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_closeDropdown'), _class.prototype)), _class);
exports.default = TxlUncontrolledMultiSelect;


TxlUncontrolledMultiSelect.displayName = 'TxlUncontrolledMultiSelect';

TxlUncontrolledMultiSelect.propTypes = {
  dropdownHeight: _MultiSelect2.default.propTypes.dropdownHeight,
  dropdownWidth: _MultiSelect2.default.propTypes.dropdownWidth,
  enableLocalFiltering: _MultiSelect2.default.propTypes.enableLocalFiltering,
  items: _MultiSelect2.default.propTypes.items,
  label: _MultiSelect2.default.propTypes.label,
  loading: _MultiSelect2.default.propTypes.loading,
  noContentView: _MultiSelect2.default.propTypes.noContentView,
  onChange: _MultiSelect2.default.propTypes.onChange,
  onItemCreate: _MultiSelect2.default.propTypes.onItemCreate,
  onSearchChange: _MultiSelect2.default.propTypes.onSearchChange,
  placeholder: _MultiSelect2.default.propTypes.placeholder,
  secondaryText: _MultiSelect2.default.propTypes.secondaryText,
  showCreateNew: _MultiSelect2.default.propTypes.showCreateNew,
  showSearchToViewMoreMessage: _MultiSelect2.default.propTypes.showSearchToViewMoreMessage,
  validationState: _MultiSelect2.default.propTypes.validationState,
  width: _MultiSelect2.default.propTypes.width
};

TxlUncontrolledMultiSelect.defaultProps = {
  enableLocalFiltering: true,
  onChange: _Base.NOOP
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlUncontrolledMultiSelect, 'TxlUncontrolledMultiSelect', 'src/dropdowns/_UncontrolledMultiSelect.jsx');
}();

;
module.exports = exports['default'];
