// txl/dropdowns/ButtonSelect.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _helpers = require('../styles/helpers');

var _ArrowDown = require('../icons/ArrowDown');

var _ArrowDown2 = _interopRequireDefault(_ArrowDown);

var _Button = require('../buttons/Button');

var _Button2 = _interopRequireDefault(_Button);

var _Container = require('../containers/Container');

var _Container2 = _interopRequireDefault(_Container);

var _DropdownTargetBase = require('./_DropdownTargetBase');

var _DropdownTargetBase2 = _interopRequireDefault(_DropdownTargetBase);

var _SelectableItemList = require('../lists/_SelectableItemList');

var _SelectableItemList2 = _interopRequireDefault(_SelectableItemList);

var _SearchBox = require('../input-fields/SearchBox');

var _SearchBox2 = _interopRequireDefault(_SearchBox);

var _TruncatedText = require('../text/TruncatedText');

var _TruncatedText2 = _interopRequireDefault(_TruncatedText);

var _lists = require('../lists/lists.style');

var _ButtonSelect = require('./ButtonSelect.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

// @TODO-TRANSLATE
var babelPluginFlowReactPropTypes_proptype_ButtonSize = require('../buttons/Button').babelPluginFlowReactPropTypes_proptype_ButtonSize || require('react').PropTypes.any;

var NO_MATCHING_ITEMS_TEXT = 'No matches found';

Object.defineProperty(module.exports, 'babelPluginFlowReactPropTypes_proptype_Props', require('react').PropTypes.shape({
  anchorComponent: require('react').PropTypes.any,
  buttonText: require('react').PropTypes.string.isRequired,
  dropdownWidth: require('react').PropTypes.any,
  dropdownAttachmentLocation: require('react').PropTypes.any,
  id: require('react').PropTypes.string,
  itemHeight: require('react').PropTypes.string,
  items: require('react').PropTypes.arrayOf(require('react').PropTypes.object).isRequired,
  onButtonClick: require('react').PropTypes.func.isRequired,
  onItemSelect: require('react').PropTypes.func.isRequired,
  onSearchChange: require('react').PropTypes.func.isRequired,
  open: require('react').PropTypes.bool.isRequired,
  popoverZIndex: require('react').PropTypes.string,
  searchPlaceholder: require('react').PropTypes.string,
  searchValue: require('react').PropTypes.string,
  showSearch: require('react').PropTypes.bool.isRequired,
  size: babelPluginFlowReactPropTypes_proptype_ButtonSize,
  value: require('react').PropTypes.string,
  visibleItemCount: require('react').PropTypes.number
}));

var TxlButtonSelect = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlButtonSelect, _React$Component);

  function TxlButtonSelect() {
    (0, _classCallCheck3.default)(this, TxlButtonSelect);
    return (0, _possibleConstructorReturn3.default)(this, (TxlButtonSelect.__proto__ || (0, _getPrototypeOf2.default)(TxlButtonSelect)).apply(this, arguments));
  }

  (0, _createClass3.default)(TxlButtonSelect, [{
    key: 'componentDidUpdate',
    value: function componentDidUpdate(prevProps) {
      var searchBox = this.refs.searchBox;
      // only focus text field if we're transitioning to open state otherwise each render, e.g.,
      // user typing in search box, will "re-focus" text field

      if (!prevProps.open && this.props.open && searchBox) {
        searchBox.focus();
      }
    }
  }, {
    key: '_handleSearchClear',
    value: function _handleSearchClear() {
      this.props.onSearchChange({ value: '' });
      // put cursor back in text field after user clears search term
      if (this.refs.searchBox) {
        this.refs.searchBox.focus();
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          anchorComponent = _props.anchorComponent,
          buttonText = _props.buttonText,
          dropdownAttachmentLocation = _props.dropdownAttachmentLocation,
          dropdownWidth = _props.dropdownWidth,
          id = _props.id,
          itemHeight = _props.itemHeight,
          items = _props.items,
          onButtonClick = _props.onButtonClick,
          onItemSelect = _props.onItemSelect,
          onSearchChange = _props.onSearchChange,
          open = _props.open,
          popoverZIndex = _props.popoverZIndex,
          searchPlaceholder = _props.searchPlaceholder,
          searchValue = _props.searchValue,
          showSearch = _props.showSearch,
          size = _props.size,
          value = _props.value,
          visibleItemCount = _props.visibleItemCount;


      var buttonArrowStyle = open ? _ButtonSelect.BUTTON_ARROW_OPEN : _ButtonSelect.BUTTON_ARROW;
      var hasItems = !!(items && items.length);
      var noItemsMatching = !hasItems && searchValue;

      var dropdownContent = _react2.default.createElement(
        _Container2.default,
        null,
        showSearch && _react2.default.createElement(_SearchBox2.default, {
          onClear: this._handleSearchClear,
          onChange: onSearchChange,
          placeholder: searchPlaceholder,
          ref: 'searchBox',
          value: searchValue
        }),
        hasItems && _react2.default.createElement(_SelectableItemList2.default, {
          itemHeight: itemHeight,
          items: items,
          onClick: onItemSelect,
          ref: 'itemList',
          selectedValues: [value],
          visibleItemCount: visibleItemCount
        }),
        noItemsMatching && _react2.default.createElement(
          'div',
          {
            ref: 'noMatchingItemsContent',
            style: _ButtonSelect.NO_MATCHING_ITEMS_STYLE
          },
          NO_MATCHING_ITEMS_TEXT
        ),
        !!anchorComponent && _react2.default.createElement(
          'div',
          { style: _lists.ANCHOR_COMPONENT_STYLE },
          anchorComponent
        )
      );

      return _react2.default.createElement(
        _DropdownTargetBase2.default,
        {
          attachmentLocation: dropdownAttachmentLocation,
          componentName: 'TxlButtonSelect',
          content: dropdownContent,
          id: id,
          popoverZIndex: popoverZIndex,
          ref: 'dropdownTargetBase',
          visible: open,
          width: dropdownWidth,
          yOffset: (0, _helpers.gridUnitsRaw)(-1)
        },
        _react2.default.createElement(
          _Button2.default,
          {
            _style: _ButtonSelect.BUTTON_EXTRA,
            onClick: onButtonClick,
            ref: 'button',
            size: size,
            variant: 'muted'
          },
          _react2.default.createElement(
            'span',
            {
              'data-test-button-content': true,
              style: _ButtonSelect.BUTTON_CONTENT_WRAPPER
            },
            _react2.default.createElement(_TruncatedText2.default, { text: buttonText }),
            _react2.default.createElement(
              'span',
              { style: buttonArrowStyle },
              _react2.default.createElement(_ArrowDown2.default, null)
            )
          )
        )
      );
    }
  }]);
  return TxlButtonSelect;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleSearchClear', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleSearchClear'), _class2.prototype)), _class2)) || _class;

TxlButtonSelect.propTypes = {
  anchorComponent: require('react').PropTypes.any,
  buttonText: require('react').PropTypes.string.isRequired,
  dropdownWidth: require('react').PropTypes.any,
  dropdownAttachmentLocation: require('react').PropTypes.any,
  id: require('react').PropTypes.string,
  itemHeight: require('react').PropTypes.string,
  items: require('react').PropTypes.arrayOf(require('react').PropTypes.object).isRequired,
  onButtonClick: require('react').PropTypes.func.isRequired,
  onItemSelect: require('react').PropTypes.func.isRequired,
  onSearchChange: require('react').PropTypes.func.isRequired,
  open: require('react').PropTypes.bool.isRequired,
  popoverZIndex: require('react').PropTypes.string,
  searchPlaceholder: require('react').PropTypes.string,
  searchValue: require('react').PropTypes.string,
  showSearch: require('react').PropTypes.bool.isRequired,
  size: babelPluginFlowReactPropTypes_proptype_ButtonSize,
  value: require('react').PropTypes.string,
  visibleItemCount: require('react').PropTypes.number
};


TxlButtonSelect.defaultProps = {
  buttonText: '',
  onButtonClick: _Base.NOOP,
  onItemSelect: _Base.NOOP,
  onSearchChange: _Base.NOOP,
  open: false,
  searchPlaceholder: _SearchBox2.default.defaultProps.placeholder,
  showSearch: false,
  size: _Button2.default.defaultProps.size
};

var _default = TxlButtonSelect;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(NO_MATCHING_ITEMS_TEXT, 'NO_MATCHING_ITEMS_TEXT', 'src/dropdowns/ButtonSelect.jsx');

  __REACT_HOT_LOADER__.register(TxlButtonSelect, 'TxlButtonSelect', 'src/dropdowns/ButtonSelect.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/dropdowns/ButtonSelect.jsx');
}();

;
module.exports = exports['default'];
