// txl/dropdowns/ButtonSelect.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NO_MATCHING_ITEMS_STYLE = exports.BUTTON_EXTRA = exports.BUTTON_CONTENT_WRAPPER = exports.BUTTON_ARROW_OPEN = exports.BUTTON_ARROW = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _theme = require('../styles/theme');

var _lists = require('../lists/lists.style');

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BUTTON_ARROW = (0, _extends3.default)({}, (0, _helpers.square)((0, _helpers.gridUnits)(5)), {
  alignItems: 'center',
  display: 'flex',
  flex: '0 0 ' + (0, _helpers.gridUnits)(5),
  justifyContent: 'center',
  marginRight: (0, _helpers.gridUnits)(-2)
});

var BUTTON_ARROW_OPEN = (0, _extends3.default)({}, BUTTON_ARROW, {
  transform: 'rotate(180deg)'
});

var BUTTON_CONTENT_WRAPPER = {
  alignItems: 'center',
  display: 'flex',
  height: '100%',
  justifyContent: 'space-between',
  textAlign: 'left',
  width: '100%'
};

var BUTTON_EXTRA = {
  paddingRight: 0,
  width: '100%'
};

var NO_MATCHING_ITEMS_STYLE = (0, _extends3.default)({}, _lists.LIST_ITEM_STYLE, {
  color: _theme.COLOR_NEUTRAL['200']
});

exports.BUTTON_ARROW = BUTTON_ARROW;
exports.BUTTON_ARROW_OPEN = BUTTON_ARROW_OPEN;
exports.BUTTON_CONTENT_WRAPPER = BUTTON_CONTENT_WRAPPER;
exports.BUTTON_EXTRA = BUTTON_EXTRA;
exports.NO_MATCHING_ITEMS_STYLE = NO_MATCHING_ITEMS_STYLE;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(BUTTON_ARROW, 'BUTTON_ARROW', 'src/dropdowns/ButtonSelect.style.js');

  __REACT_HOT_LOADER__.register(BUTTON_ARROW_OPEN, 'BUTTON_ARROW_OPEN', 'src/dropdowns/ButtonSelect.style.js');

  __REACT_HOT_LOADER__.register(BUTTON_CONTENT_WRAPPER, 'BUTTON_CONTENT_WRAPPER', 'src/dropdowns/ButtonSelect.style.js');

  __REACT_HOT_LOADER__.register(BUTTON_EXTRA, 'BUTTON_EXTRA', 'src/dropdowns/ButtonSelect.style.js');

  __REACT_HOT_LOADER__.register(NO_MATCHING_ITEMS_STYLE, 'NO_MATCHING_ITEMS_STYLE', 'src/dropdowns/ButtonSelect.style.js');
}();

;
