// txl/dropdowns/MultiSelectDropdownHeader.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _helpers = require('../styles/helpers');

var _type = require('../styles/mixins/type');

var _theme = require('../styles/theme');

var _Underline = require('../layout/_Underline');

var _Underline2 = _interopRequireDefault(_Underline);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SELECT_BUTTONS_CONTAINER_STYLE = {
  height: (0, _helpers.gridUnits)(7),
  position: 'relative'
};

var SELECT_BUTTON_CONTAINER_STYLE = {
  alignItems: 'center',
  display: 'flex',
  height: '100%',
  paddingLeft: (0, _helpers.gridUnits)(2)
};

var SELECT_BUTTON_STYLE = (0, _extends3.default)({}, _type.DEFAULT_FONT_FAMILY, _type.TYPE_SMALL, {
  color: _theme.COLOR_ACCENT['600'],
  cursor: 'pointer'
});

var SELECT_BUTTON_DIVIDER_STYLE = {
  backgroundColor: _theme.COLOR_NEUTRAL['200'],
  height: (0, _helpers.gridUnits)(2),
  marginLeft: (0, _helpers.gridUnits)(1),
  marginRight: (0, _helpers.gridUnits)(1),
  width: '1px'
};

var UNDERLINE_STYLE = {
  bottom: 0,
  position: 'absolute'
};

var MultiSelectDropdownHeader = function MultiSelectDropdownHeader(props) {
  return _react2.default.createElement(
    'div',
    { style: SELECT_BUTTONS_CONTAINER_STYLE },
    _react2.default.createElement(
      'div',
      { style: SELECT_BUTTON_CONTAINER_STYLE },
      _react2.default.createElement(
        'div',
        {
          onClick: props.onSelectAll,
          style: SELECT_BUTTON_STYLE
        },
        'All'
      ),
      _react2.default.createElement('div', { style: SELECT_BUTTON_DIVIDER_STYLE }),
      _react2.default.createElement(
        'div',
        {
          onClick: props.onSelectNone,
          style: SELECT_BUTTON_STYLE
        },
        'None'
      )
    ),
    _react2.default.createElement(_Underline2.default, {
      color: _theme.COLOR_NEUTRAL['100'],
      style: UNDERLINE_STYLE
    })
  );
};

MultiSelectDropdownHeader.propTypes = {
  onSelectAll: _react.PropTypes.func.isRequired,
  onSelectNone: _react.PropTypes.func.isRequired
};

MultiSelectDropdownHeader.defaultProps = {
  onSelectAll: Function.prototype,
  onSelectNone: Function.prototype
};

var _default = MultiSelectDropdownHeader;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(SELECT_BUTTONS_CONTAINER_STYLE, 'SELECT_BUTTONS_CONTAINER_STYLE', 'src/dropdowns/MultiSelectDropdownHeader.jsx');

  __REACT_HOT_LOADER__.register(SELECT_BUTTON_CONTAINER_STYLE, 'SELECT_BUTTON_CONTAINER_STYLE', 'src/dropdowns/MultiSelectDropdownHeader.jsx');

  __REACT_HOT_LOADER__.register(SELECT_BUTTON_STYLE, 'SELECT_BUTTON_STYLE', 'src/dropdowns/MultiSelectDropdownHeader.jsx');

  __REACT_HOT_LOADER__.register(SELECT_BUTTON_DIVIDER_STYLE, 'SELECT_BUTTON_DIVIDER_STYLE', 'src/dropdowns/MultiSelectDropdownHeader.jsx');

  __REACT_HOT_LOADER__.register(UNDERLINE_STYLE, 'UNDERLINE_STYLE', 'src/dropdowns/MultiSelectDropdownHeader.jsx');

  __REACT_HOT_LOADER__.register(MultiSelectDropdownHeader, 'MultiSelectDropdownHeader', 'src/dropdowns/MultiSelectDropdownHeader.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/dropdowns/MultiSelectDropdownHeader.jsx');
}();

;
module.exports = exports['default'];
