// txl/dropdowns/SingleSelectDropdownHeader.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _helpers = require('../styles/helpers');

var _type = require('../styles/mixins/type');

var _DropdownListItem = require('./_DropdownListItem.style');

var _theme = require('../styles/theme');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var SELECT_BUTTONS_CONTAINER_STYLE = {
  height: (0, _helpers.gridUnits)(7),
  position: 'relative'
};

var SELECT_BUTTON_CONTAINER_STYLE = {
  alignItems: 'center',
  display: 'flex',
  height: '100%',
  paddingLeft: (0, _helpers.gridUnits)(2)
};

var SELECT_NONE_BUTTON_STYLE = (0, _extends3.default)({}, _type.DEFAULT_FONT_FAMILY, _type.TYPE_BASE, {
  color: _theme.COLOR_NEUTRAL['200'],
  cursor: 'pointer'
});

var TxlSingleSelectDropdownHeader = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlSingleSelectDropdownHeader, _React$Component);

  function TxlSingleSelectDropdownHeader(args) {
    (0, _classCallCheck3.default)(this, TxlSingleSelectDropdownHeader);

    var _this = (0, _possibleConstructorReturn3.default)(this, (TxlSingleSelectDropdownHeader.__proto__ || (0, _getPrototypeOf2.default)(TxlSingleSelectDropdownHeader)).call(this, args));

    _this.state = { hovered: false };
    return _this;
  }

  (0, _createClass3.default)(TxlSingleSelectDropdownHeader, [{
    key: '_handleMouseEnter',
    value: function _handleMouseEnter() {
      this.setState({ hovered: true });
    }
  }, {
    key: '_handleMouseLeave',
    value: function _handleMouseLeave() {
      this.setState({ hovered: false });
    }
  }, {
    key: 'getStyles',
    value: function getStyles() {
      return (0, _helpers.combineStyles)(SELECT_BUTTON_CONTAINER_STYLE, this.state.hovered && _DropdownListItem.LIST_ITEM_HOVER_STYLE, this.props.keySelectedItemIndex === -1 && _DropdownListItem.LIST_ITEM_HOVER_STYLE);
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        {
          onClick: this.props.onSelectNone,
          onMouseEnter: this._handleMouseEnter,
          onMouseLeave: this._handleMouseLeave,
          style: SELECT_BUTTONS_CONTAINER_STYLE
        },
        _react2.default.createElement(
          'div',
          { style: this.getStyles() },
          _react2.default.createElement(
            'div',
            { style: SELECT_NONE_BUTTON_STYLE },
            'None'
          )
        )
      );
    }
  }]);
  return TxlSingleSelectDropdownHeader;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleMouseEnter', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseEnter'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleMouseLeave', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseLeave'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, 'getStyles', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, 'getStyles'), _class2.prototype)), _class2)) || _class;

TxlSingleSelectDropdownHeader.propTypes = {
  keySelectedItemIndex: _react.PropTypes.number,
  onSelectNone: _react.PropTypes.func.isRequired
};

TxlSingleSelectDropdownHeader.defaultProps = {
  onSelectNone: Function.prototype
};

var _default = TxlSingleSelectDropdownHeader;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(SELECT_BUTTONS_CONTAINER_STYLE, 'SELECT_BUTTONS_CONTAINER_STYLE', 'src/dropdowns/SingleSelectDropdownHeader.jsx');

  __REACT_HOT_LOADER__.register(SELECT_BUTTON_CONTAINER_STYLE, 'SELECT_BUTTON_CONTAINER_STYLE', 'src/dropdowns/SingleSelectDropdownHeader.jsx');

  __REACT_HOT_LOADER__.register(SELECT_NONE_BUTTON_STYLE, 'SELECT_NONE_BUTTON_STYLE', 'src/dropdowns/SingleSelectDropdownHeader.jsx');

  __REACT_HOT_LOADER__.register(TxlSingleSelectDropdownHeader, 'TxlSingleSelectDropdownHeader', 'src/dropdowns/SingleSelectDropdownHeader.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/dropdowns/SingleSelectDropdownHeader.jsx');
}();

;
module.exports = exports['default'];
