// txl/dropdowns/TourPointControlled.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _TourPointTargetBase = require('./_TourPointTargetBase');

var _TourPointTargetBase2 = _interopRequireDefault(_TourPointTargetBase);

var _TourCardContainer = require('../containers/_TourCardContainer');

var _TourCardContainer2 = _interopRequireDefault(_TourCardContainer);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* ***************************

+-TourPointControlled--------+
| +-TourPointTargetBase----+ |
| |                        | |
| | +TourCard-+ +HintDot-+ | |
| | |         | |        | | |
| | |         | |        | | |
| | |         | |        | | |
| | |         | |        | | |
| | |         | |        | | |
| | |         | |        | | |
| | |         | |        | | |
| | |         | |        | | |
| | +---------+ +--------+ | |
| |                        | |
| +------------------------+ |
|                            |
+----------------------------+

****************************/

var TourPointControlled = function TourPointControlled(props) {
  var content = props.content,
      footer = props.footer,
      header = props.header,
      icon = props.icon,
      onAwayClick = props.onAwayClick,
      onDismiss = props.onDismiss,
      onHintClick = props.onHintClick,
      pulsing = props.pulsing,
      style = props.style,
      visible = props.visible,
      width = props.width;

  return _react2.default.createElement(_TourPointTargetBase2.default, {
    content: _react2.default.createElement(_TourCardContainer2.default, {
      icon: icon,
      header: header,
      content: content,
      width: width,
      onDismiss: onDismiss,
      footer: footer
    }),
    style: style,
    pulsing: pulsing,
    onAwayClick: onAwayClick,
    onHintClick: onHintClick,
    visible: visible
  });
};

TourPointControlled.defaultProps = {
  icon: _TourCardContainer2.default.defaultProps.icon,
  onAwayClick: _TourPointTargetBase2.default.defaultProps.onAwayClick,
  onDismiss: _TourCardContainer2.default.defaultProps.onDismiss,
  onHintClick: _TourPointTargetBase2.default.defaultProps.onHintClick,
  visible: _TourPointTargetBase2.default.defaultProps.visible,
  width: _TourCardContainer2.default.defaultProps.width
};

TourPointControlled.propTypes = {
  content: _TourPointTargetBase2.default.propTypes.content,
  footer: _TourCardContainer2.default.propTypes.footer,
  header: _TourCardContainer2.default.propTypes.header,
  icon: _TourCardContainer2.default.propTypes.icon,
  onAwayClick: _TourPointTargetBase2.default.propTypes.onAwayClick,
  onDismiss: _TourCardContainer2.default.propTypes.onDismiss,
  onHintClick: _TourPointTargetBase2.default.propTypes.onHintClick,
  pulsing: _TourPointTargetBase2.default.propTypes.pulsing,
  style: _TourPointTargetBase2.default.propTypes.style,
  visible: _TourPointTargetBase2.default.propTypes.visible,
  width: _TourCardContainer2.default.propTypes.width
};

var _default = TourPointControlled;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TourPointControlled, 'TourPointControlled', 'src/dropdowns/TourPointControlled.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/dropdowns/TourPointControlled.jsx');
}();

;
module.exports = exports['default'];
