// txl/dropdowns/TourPoint.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _desc, _value, _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Base = require('../base/Base');

var _TourPointControlled = require('./TourPointControlled');

var _TourPointControlled2 = _interopRequireDefault(_TourPointControlled);

var _TourPointTargetBase = require('./_TourPointTargetBase');

var _TourPointTargetBase2 = _interopRequireDefault(_TourPointTargetBase);

var _TourCardContainer = require('../containers/_TourCardContainer');

var _TourCardContainer2 = _interopRequireDefault(_TourCardContainer);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

/* *****************************

+-TourPoint------------------------+
|                                  |
| +-TourPointControlled----------+ |
| |                              | |
| | +--TourPointTargetBase-----+ | |
| | |                          | | |
| | | +-TetheredPopoverTarget+ | | |     +-new Tether()------+
| | | |                      | | | |     |                   |
| | | |  +-HintDot--------+  | | | |     |                   |
| | | |  |                |  | | | |     |  +-TourCard-----+ |
| | | |  |                |  | | | |     |  |              | |
| | | |  |                +--------------+  |              | |
| | | |  |                |  | | | |     |  |              | |
| | | |  +----------------+  | | | |     |  |              | |
| | | |                      | | | |     |  |              | |
| | | |                      | | | |     |  |              | |
| | | |                      | | | |     |  |              | |
| | | |                      | | | |     |  |              | |
| | | |                      | | | |     |  +--------------+ |
| | | |                      | | | |     |                   |
| | | |                      | | | |     +-------------------+
| | | |                      | | | |
| | | +----------------------+ | | |
| | |                          | | |
| | +--------------------------+ | |
| +------------------------------+ |
|                                  |
+----------------------------------+


*******************************/

var TxlTourPoint = (_class = function (_React$Component) {
  (0, _inherits3.default)(TxlTourPoint, _React$Component);

  function TxlTourPoint() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlTourPoint);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlTourPoint.__proto__ || (0, _getPrototypeOf2.default)(TxlTourPoint)).call.apply(_ref, [this].concat(args)));

    _this.state = {
      pulsingState: true,
      visibleState: false
    };
    return _this;
  }

  (0, _createClass3.default)(TxlTourPoint, [{
    key: '_handleClick',
    value: function _handleClick(e) {
      // This is a strategy to ensure that when the HintDot is clicked, its treated
      // as an "awayClick" when the TourCardContainer is showing. Otherwise the content will go
      // from off to on in an instant.
      if (this._lastAwayClickTimeStamp !== e.timeStamp) {
        this.setState({
          pulsingState: this.state.visibleState,
          visibleState: !this.state.visibleState
        });
      }
    }
  }, {
    key: '_hideContent',
    value: function _hideContent() {
      this.setState({ visibleState: false });
    }
  }, {
    key: '_handleDismiss',
    value: function _handleDismiss() {
      this._hideContent();
      this.setState({
        pulsingState: false
      });
      this.props.onDismiss.apply(this, [{}]);
    }
  }, {
    key: '_handleAwayClick',
    value: function _handleAwayClick(e) {
      this._lastAwayClickTimeStamp = e.timeStamp;
      this._hideContent();
      this.setState({
        pulsingState: true
      });
    }
  }, {
    key: 'render',
    value: function render() {
      var _handleAwayClick = this._handleAwayClick,
          _handleClick = this._handleClick,
          _handleDismiss = this._handleDismiss,
          _props = this.props,
          content = _props.content,
          footer = _props.footer,
          header = _props.header,
          icon = _props.icon,
          style = _props.style,
          width = _props.width,
          _state = this.state,
          visibleState = _state.visibleState,
          pulsingState = _state.pulsingState;

      return _react2.default.createElement(_TourPointControlled2.default, {
        content: content,
        footer: footer,
        header: header,
        icon: icon,
        onAwayClick: _handleAwayClick,
        onDismiss: _handleDismiss,
        onHintClick: _handleClick,
        pulsing: pulsingState,
        style: style,
        visible: visibleState,
        width: width
      });
    }
  }]);
  return TxlTourPoint;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class.prototype, '_handleClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleClick'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_hideContent', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_hideContent'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_handleDismiss', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleDismiss'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_handleAwayClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleAwayClick'), _class.prototype)), _class);


TxlTourPoint.defaultProps = {
  icon: _TourCardContainer2.default.defaultProps.icon,
  onAwayClick: _TourPointTargetBase2.default.defaultProps.onAwayClick,
  onDismiss: _TourCardContainer2.default.defaultProps.onDismiss,
  onHintClick: _TourPointTargetBase2.default.defaultProps.onHintClick,
  visible: _TourPointTargetBase2.default.defaultProps.visible,
  width: _TourCardContainer2.default.defaultProps.width
};

TxlTourPoint.propTypes = {
  content: _TourPointTargetBase2.default.propTypes.content,
  footer: _TourCardContainer2.default.propTypes.footer,
  header: _TourCardContainer2.default.propTypes.header,
  icon: _TourCardContainer2.default.propTypes.icon,
  onDismiss: _TourCardContainer2.default.propTypes.onDismiss,
  pulsing: _TourPointTargetBase2.default.propTypes.pulsing,
  style: _TourPointTargetBase2.default.propTypes.style,
  visible: _TourPointTargetBase2.default.propTypes.visible,
  width: _TourCardContainer2.default.propTypes.width
};

var _default = TxlTourPoint;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlTourPoint, 'TxlTourPoint', 'src/dropdowns/TourPoint.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/dropdowns/TourPoint.jsx');
}();

;
module.exports = exports['default'];
