// txl/dropdowns/UncontrolledButtonSelect.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _createDropdown = require('./_createDropdown');

var _createDropdown2 = _interopRequireDefault(_createDropdown);

var _listHelpers = require('../lists/listHelpers');

var _ButtonSelect = require('./ButtonSelect');

var _ButtonSelect2 = _interopRequireDefault(_ButtonSelect);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var babelPluginFlowReactPropTypes_proptype_AttachmentLocation = require('./ButtonSelect').babelPluginFlowReactPropTypes_proptype_AttachmentLocation || require('react').PropTypes.any;

// @TODO Maybe this should be stored as some sort of global
// value so all lists have the same threshold?
var babelPluginFlowReactPropTypes_proptype_ButtonSize = require('../buttons/Button').babelPluginFlowReactPropTypes_proptype_ButtonSize || require('react').PropTypes.any;

var MAX_VISIBLE_ITEMS = 10;
var ITEM_PLACEHOLDER = 'Select...';

var TxlUncontrolledButtonSelect = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlUncontrolledButtonSelect, _React$Component);

  function TxlUncontrolledButtonSelect(props, context) {
    (0, _classCallCheck3.default)(this, TxlUncontrolledButtonSelect);

    var _this = (0, _possibleConstructorReturn3.default)(this, (TxlUncontrolledButtonSelect.__proto__ || (0, _getPrototypeOf2.default)(TxlUncontrolledButtonSelect)).call(this, props, context));

    var _this$props = _this.props,
        defaultValue = _this$props.defaultValue,
        items = _this$props.items;

    // will return undefined if defaultValue is undefined

    var matchedItem = (0, _listHelpers.getItemByValue)(items, defaultValue);
    _this.state = {
      searchValue: '',
      value: matchedItem ? matchedItem.value : ''
    };
    return _this;
  }

  (0, _createClass3.default)(TxlUncontrolledButtonSelect, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(_ref) {
      var open = _ref.open;

      // Clear search text when dropdown closes. Setting here because we don't have a way of
      // subscribing to 'click outside of dropdown' from within this component.
      if (!open) {
        this.setState({ searchValue: '' });
      }
    }
  }, {
    key: '_handleItemSelect',
    value: function _handleItemSelect(obj) {
      var _this2 = this;

      this.setState({ value: obj.value }, function () {
        return _this2.props.onItemSelect(obj);
      });
      this._toggleDropdown();
    }
  }, {
    key: '_handleSearchChange',
    value: function _handleSearchChange(_ref2) {
      var value = _ref2.value;

      this.setState({ searchValue: value });
    }
  }, {
    key: '_toggleDropdown',
    value: function _toggleDropdown() {
      this.props.updateDropdown && this.props.updateDropdown({ open: !this.props.open });
    }
  }, {
    key: 'setValue',
    value: function setValue() {
      var value = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : '';

      this.setState({ value: value });
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          dropdownAttachmentLocation = _props.dropdownAttachmentLocation,
          dropdownWidth = _props.dropdownWidth,
          items = _props.items,
          open = _props.open,
          placeholder = _props.placeholder,
          size = _props.size;
      var _state = this.state,
          searchValue = _state.searchValue,
          value = _state.value;


      var buttonText = (0, _listHelpers.getItemLabelByValue)(items, value) || placeholder;
      var filteredItems = (0, _listHelpers.getFilteredFlatItemList)(items, searchValue);
      // per UX spec
      var shouldShowSearch = items && items.length > MAX_VISIBLE_ITEMS;

      return _react2.default.createElement(_ButtonSelect2.default, {
        buttonText: buttonText,
        dropdownAttachmentLocation: dropdownAttachmentLocation,
        dropdownWidth: dropdownWidth,
        items: filteredItems,
        onButtonClick: this._toggleDropdown,
        onItemSelect: this._handleItemSelect,
        onSearchChange: this._handleSearchChange,
        open: open,
        ref: 'buttonSelect',
        searchValue: searchValue,
        showSearch: shouldShowSearch,
        size: size,
        value: value,
        visibleItemCount: MAX_VISIBLE_ITEMS
      });
    }
  }]);
  return TxlUncontrolledButtonSelect;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleItemSelect', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleItemSelect'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleSearchChange', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleSearchChange'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_toggleDropdown', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_toggleDropdown'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, 'setValue', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, 'setValue'), _class2.prototype)), _class2)) || _class;

TxlUncontrolledButtonSelect.propTypes = {
  defaultValue: require('react').PropTypes.string,
  dropdownAttachmentLocation: babelPluginFlowReactPropTypes_proptype_AttachmentLocation,
  dropdownWidth: require('react').PropTypes.any,
  items: require('react').PropTypes.arrayOf(require('react').PropTypes.object).isRequired,
  onItemSelect: require('react').PropTypes.func.isRequired,
  open: require('react').PropTypes.bool,
  placeholder: require('react').PropTypes.any,
  size: babelPluginFlowReactPropTypes_proptype_ButtonSize,
  updateDropdown: require('react').PropTypes.func
};


TxlUncontrolledButtonSelect.defaultProps = {
  placeholder: ITEM_PLACEHOLDER
};

var _default = (0, _createDropdown2.default)(TxlUncontrolledButtonSelect);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(MAX_VISIBLE_ITEMS, 'MAX_VISIBLE_ITEMS', 'src/dropdowns/UncontrolledButtonSelect.jsx');

  __REACT_HOT_LOADER__.register(ITEM_PLACEHOLDER, 'ITEM_PLACEHOLDER', 'src/dropdowns/UncontrolledButtonSelect.jsx');

  __REACT_HOT_LOADER__.register(TxlUncontrolledButtonSelect, 'TxlUncontrolledButtonSelect', 'src/dropdowns/UncontrolledButtonSelect.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/dropdowns/UncontrolledButtonSelect.jsx');
}();

;
module.exports = exports['default'];
