// txl/badge/Badge.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BADGE_VARIANT_PROPS = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Tag = require('../tag/Tag');

var _Tag2 = _interopRequireDefault(_Tag);

var _theme = require('../styles/theme');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function truncate(number, digits) {
  var numberToRender = '';
  if (('' + number).length > digits) {
    for (var i = 0; i < digits; i++) {
      numberToRender += '9';
    }
    numberToRender += '+';
  } else {
    numberToRender = number;
  }
  return numberToRender;
}

var BADGE_VARIANT_PROPS = exports.BADGE_VARIANT_PROPS = {
  accent: {
    backgroundColor: _theme.COLOR_ACCENT['500'],
    textColor: _theme.COLOR_ACCENT['500c']
  },
  neutral: {
    backgroundColor: _theme.COLOR_NEUTRAL['100'],
    textColor: _theme.COLOR_NEUTRAL['100c']
  },
  plain: {
    backgroundColor: _theme.COLOR_ACCENT['500c'],
    textColor: _theme.COLOR_ACCENT['500']
  }
};

var TxlBadge = function TxlBadge(props) {
  var number = props.number,
      variant = props.variant,
      id = props.id,
      digits = props.digits;


  return _react2.default.createElement(
    _Tag2.default,
    (0, _extends3.default)({}, BADGE_VARIANT_PROPS[variant], {
      id: id
    }),
    truncate(number, digits)
  );
};

TxlBadge.propTypes = {
  digits: _react.PropTypes.number,
  id: _react.PropTypes.string,
  number: _react.PropTypes.number.isRequired,
  variant: _react.PropTypes.oneOf(['accent', 'neutral', 'plain'])
};

TxlBadge.defaultProps = {
  digits: 2,
  variant: 'neutral'
};

var _default = (0, _radium2.default)(TxlBadge);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(truncate, 'truncate', 'src/badge/Badge.jsx');

  __REACT_HOT_LOADER__.register(BADGE_VARIANT_PROPS, 'BADGE_VARIANT_PROPS', 'src/badge/Badge.jsx');

  __REACT_HOT_LOADER__.register(TxlBadge, 'TxlBadge', 'src/badge/Badge.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/badge/Badge.jsx');
}();

;
