// txl/header/_HeaderFlyout.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _FlyoutTargetBase = require('../dropdowns/_FlyoutTargetBase');

var _FlyoutTargetBase2 = _interopRequireDefault(_FlyoutTargetBase);

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

var _layers = require('../styles/mixins/layers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ICON_BASE_STYLES = (0, _extends3.default)({}, (0, _helpers.square)((0, _helpers.gridUnits)(5)), {
  alignItems: 'center',
  backgroundColor: 'transparent',
  border: 'none',
  cursor: 'pointer',
  display: 'flex',
  justifyContent: 'center',
  outline: 'none',
  padding: 0
});

var WRAPPER_STYLES = {
  marginLeft: (0, _helpers.gridUnits)(2),
  marginRight: (0, _helpers.gridUnits)(2)
};

var ICON_ACTIVE_STYLES = {
  backgroundColor: _theme.COLOR_NEUTRAL['800']
};

var FLYOUT_CONTENT_CONTAINER_STYLES = (0, _extends3.default)({}, _layers.LAYER_2, {
  backgroundColor: 'white',
  margin: (0, _helpers.gridUnits)(1) });

var TxlHeaderFlyout = (0, _radium2.default)(_class = function (_React$Component) {
  (0, _inherits3.default)(TxlHeaderFlyout, _React$Component);

  function TxlHeaderFlyout() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlHeaderFlyout);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlHeaderFlyout.__proto__ || (0, _getPrototypeOf2.default)(TxlHeaderFlyout)).call.apply(_ref, [this].concat(args)));

    _this._handleFlyoutTargetClick = _this._handleFlyoutTargetClick.bind(_this);
    return _this;
  }

  (0, _createClass3.default)(TxlHeaderFlyout, [{
    key: '_handleFlyoutTargetClick',
    value: function _handleFlyoutTargetClick(e) {
      e.preventDefault();
      this.props.onTargetClick(e);
    }
  }, {
    key: 'render',
    value: function render() {
      var contentComponent = _react2.default.createElement(
        'div',
        {
          'data-test-header-flyout-content': true,
          style: FLYOUT_CONTENT_CONTAINER_STYLES,
          onClick: this.props.onContentClick
        },
        this.props.content
      );

      var iconStyles = (0, _helpers.combineStyles)(ICON_BASE_STYLES, this.props.visible && ICON_ACTIVE_STYLES);

      var iconComponent = _react2.default.createElement(
        'button',
        {
          'data-test-header-flyout-target': true,
          id: this.props.id,
          style: iconStyles
        },
        this.props.icon
      );

      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlHeaderFlyout',
          style: WRAPPER_STYLES
        },
        _react2.default.createElement(
          _FlyoutTargetBase2.default,
          {
            componentName: 'TxlHeaderFlyoutTarget',
            content: contentComponent,
            visible: this.props.visible,
            onClick: this._handleFlyoutTargetClick
          },
          iconComponent
        )
      );
    }
  }]);
  return TxlHeaderFlyout;
}(_react2.default.Component)) || _class;

exports.default = TxlHeaderFlyout;


TxlHeaderFlyout.propTypes = {
  content: _react.PropTypes.node.isRequired,
  icon: _react.PropTypes.node.isRequired,
  id: _react.PropTypes.string,
  onContentClick: _react.PropTypes.func,
  onTargetClick: _react.PropTypes.func.isRequired,
  visible: _react.PropTypes.bool
};

TxlHeaderFlyout.defaultProps = {
  onContentClick: _Base.NOOP
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ICON_BASE_STYLES, 'ICON_BASE_STYLES', 'src/header/_HeaderFlyout.jsx');

  __REACT_HOT_LOADER__.register(WRAPPER_STYLES, 'WRAPPER_STYLES', 'src/header/_HeaderFlyout.jsx');

  __REACT_HOT_LOADER__.register(ICON_ACTIVE_STYLES, 'ICON_ACTIVE_STYLES', 'src/header/_HeaderFlyout.jsx');

  __REACT_HOT_LOADER__.register(FLYOUT_CONTENT_CONTAINER_STYLES, 'FLYOUT_CONTENT_CONTAINER_STYLES', 'src/header/_HeaderFlyout.jsx');

  __REACT_HOT_LOADER__.register(TxlHeaderFlyout, 'TxlHeaderFlyout', 'src/header/_HeaderFlyout.jsx');
}();

;
module.exports = exports['default'];
