// txl/header/HeaderFlyoutsControlled.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _HeaderFlyout = require('./_HeaderFlyout');

var _HeaderFlyout2 = _interopRequireDefault(_HeaderFlyout);

var _helpers = require('../styles/helpers');

var _Base = require('../base/Base');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CONTAINER_STYLES = {
  display: 'flex',
  marginRight: (0, _helpers.gridUnits)(4)
};

var TxlHeaderFlyoutsControlled = function TxlHeaderFlyoutsControlled(props) {
  var items = props.items,
      selectedFlyoutIndex = props.selectedFlyoutIndex,
      onFlyoutTargetClick = props.onFlyoutTargetClick,
      onFlyoutContentClick = props.onFlyoutContentClick;


  return _react2.default.createElement(
    'div',
    {
      'data-component': 'TxlHeaderFlyoutsControlled',
      style: CONTAINER_STYLES
    },
    items.map(function (item, index) {
      var isSelected = index === selectedFlyoutIndex;

      return _react2.default.createElement(_HeaderFlyout2.default, {
        key: index,
        id: item.id,
        content: item.content,
        visible: isSelected,
        icon: item.icon,
        onContentClick: function onContentClick(event) {
          onFlyoutContentClick({ event: event });
        },
        onTargetClick: function onTargetClick(event) {
          onFlyoutTargetClick({ event: event, index: index });
        }
      });
    })
  );
};

TxlHeaderFlyoutsControlled.propTypes = {
  items: _react.PropTypes.arrayOf(_react.PropTypes.shape({
    content: _HeaderFlyout2.default.propTypes.content,
    icon: _HeaderFlyout2.default.propTypes.icon,
    id: _HeaderFlyout2.default.propTypes.id
  })).isRequired,
  onFlyoutContentClick: _react.PropTypes.func,
  onFlyoutTargetClick: _react.PropTypes.func.isRequired,
  selectedFlyoutIndex: _react.PropTypes.number
};

TxlHeaderFlyoutsControlled.defaultProps = {
  onFlyoutContentClick: _Base.NOOP,
  selectedFlyoutIndex: null
};

var _default = (0, _radium2.default)(TxlHeaderFlyoutsControlled);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(CONTAINER_STYLES, 'CONTAINER_STYLES', 'src/header/HeaderFlyoutsControlled.jsx');

  __REACT_HOT_LOADER__.register(TxlHeaderFlyoutsControlled, 'TxlHeaderFlyoutsControlled', 'src/header/HeaderFlyoutsControlled.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/header/HeaderFlyoutsControlled.jsx');
}();

;
module.exports = exports['default'];
