// txl/header/HeaderFlyouts.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _desc, _value, _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _HeaderFlyoutsControlled = require('./HeaderFlyoutsControlled');

var _HeaderFlyoutsControlled2 = _interopRequireDefault(_HeaderFlyoutsControlled);

var _Base = require('../base/Base');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlHeaderFlyouts = (_class = function (_React$Component) {
  (0, _inherits3.default)(TxlHeaderFlyouts, _React$Component);

  function TxlHeaderFlyouts() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlHeaderFlyouts);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlHeaderFlyouts.__proto__ || (0, _getPrototypeOf2.default)(TxlHeaderFlyouts)).call.apply(_ref, [this].concat(args)));

    _this.state = {
      selectedFlyoutIndex: null
    };
    return _this;
  }

  (0, _createClass3.default)(TxlHeaderFlyouts, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      window.addEventListener('click', this._handleWindowClick);
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      window.removeEventListener('click', this._handleWindowClick);
    }
  }, {
    key: '_handleWindowClick',
    value: function _handleWindowClick() {
      this._setSelectedFlyoutIndex(null);
    }
  }, {
    key: '_handleFlyoutContentClick',
    value: function _handleFlyoutContentClick(obj) {
      if (obj.event.target.getAttribute('data-close-header-flyout') !== null) {
        this.setState({ selectedFlyoutIndex: null });
      } else {
        obj.event.stopPropagation();
      }
    }
  }, {
    key: '_handleFlyoutTargetClick',
    value: function _handleFlyoutTargetClick(obj) {
      var index = obj.index;
      var shouldClose = this.state.selectedFlyoutIndex === index;

      obj.event.stopPropagation();

      this._setSelectedFlyoutIndex(shouldClose ? null : index);
    }
  }, {
    key: '_setSelectedFlyoutIndex',
    value: function _setSelectedFlyoutIndex(selectedFlyoutIndex) {
      this.setState({ selectedFlyoutIndex: selectedFlyoutIndex });
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(_HeaderFlyoutsControlled2.default, {
        items: this.props.items,
        selectedFlyoutIndex: this.state.selectedFlyoutIndex,
        onFlyoutContentClick: this._handleFlyoutContentClick,
        onFlyoutTargetClick: this._handleFlyoutTargetClick
      });
    }
  }]);
  return TxlHeaderFlyouts;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class.prototype, '_handleWindowClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleWindowClick'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_handleFlyoutContentClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleFlyoutContentClick'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_handleFlyoutTargetClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleFlyoutTargetClick'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_setSelectedFlyoutIndex', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_setSelectedFlyoutIndex'), _class.prototype)), _class);
exports.default = TxlHeaderFlyouts;


TxlHeaderFlyouts.propTypes = {
  items: _HeaderFlyoutsControlled2.default.propTypes.items
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlHeaderFlyouts, 'TxlHeaderFlyouts', 'src/header/HeaderFlyouts.jsx');
}();

;
module.exports = exports['default'];
