// txl/header/Header.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _theme = require('../styles/theme');

var _helpers = require('../styles/helpers');

var _layers = require('../styles/mixins/layers');

var _type = require('../styles/mixins/type');

var _Base = require('../base/Base');

var _TuneLogo = require('../logos/TuneLogo');

var _TuneLogo2 = _interopRequireDefault(_TuneLogo);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CONTAINER_STYLES = (0, _extends3.default)({}, _layers.LAYER_2, _type.DEFAULT_FONT_FAMILY, {
  alignItems: 'center',
  backgroundColor: _theme.COLOR_NEUTRAL['700'],
  borderBottomStyle: 'solid',
  borderBottomWidth: 3,
  boxSizing: 'border-box',
  color: _theme.COLOR_NEUTRAL['700c'],
  display: 'flex',
  height: (0, _helpers.gridUnits)(13),
  justifyContent: 'space-between',
  position: 'relative'
});

var TITLE_STYLES = (0, _extends3.default)({}, _type.HEADING_MEDIUM);

var SUBTITLE_STYLES = (0, _extends3.default)({}, _type.TYPE_BASE, {
  marginRight: (0, _helpers.gridUnits)(4)
});

var LOGO_STYLES = {
  margin: '0 ' + (0, _helpers.gridUnits)(4)
};

var CLICKABLE_LOGO_STYLES = {
  cursor: 'pointer'
};

var FLEX_ITEM_STYLES = {
  alignItems: 'center',
  display: 'flex',
  flex: '0 0 auto'
};

var TxlHeader = function TxlHeader(props) {
  var containerStyles = (0, _helpers.combineStyles)(CONTAINER_STYLES, { borderColor: props.accentColor });

  var logoStyles = (0, _helpers.combineStyles)(LOGO_STYLES, props.onLogoClick !== _Base.NOOP && CLICKABLE_LOGO_STYLES);

  var logoColor = _theme.COLOR_NEUTRAL['50'];
  var logo = props.logo || _react2.default.createElement(_TuneLogo2.default, { color: logoColor });

  return _react2.default.createElement(
    'div',
    {
      'data-component': 'TxlHeader',
      style: containerStyles
    },
    _react2.default.createElement(
      'div',
      { style: FLEX_ITEM_STYLES },
      _react2.default.createElement(
        'h1',
        {
          style: logoStyles,
          onClick: props.onLogoClick
        },
        logo
      ),
      _react2.default.createElement(
        'h2',
        {
          style: TITLE_STYLES,
          'data-test-header-title': true
        },
        props.title
      )
    ),
    _react2.default.createElement(
      'div',
      { style: FLEX_ITEM_STYLES },
      !!props.subtitle && _react2.default.createElement(
        'div',
        {
          style: SUBTITLE_STYLES,
          'data-test-header-subtitle': true
        },
        props.subtitle
      ),
      _react2.default.createElement(
        'div',
        { 'data-test-header-right-content': true },
        props.rightContent
      )
    )
  );
};

TxlHeader.propTypes = {
  accentColor: _react.PropTypes.string,
  logo: _react.PropTypes.node,
  onLogoClick: _react.PropTypes.func,
  rightContent: _react.PropTypes.node,
  subtitle: _react.PropTypes.node,
  title: _react.PropTypes.node.isRequired
};

TxlHeader.defaultProps = {
  accentColor: 'transparent',
  onLogoClick: _Base.NOOP,
  subtitle: ''
};

var _default = (0, _radium2.default)(TxlHeader);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(CONTAINER_STYLES, 'CONTAINER_STYLES', 'src/header/Header.jsx');

  __REACT_HOT_LOADER__.register(TITLE_STYLES, 'TITLE_STYLES', 'src/header/Header.jsx');

  __REACT_HOT_LOADER__.register(SUBTITLE_STYLES, 'SUBTITLE_STYLES', 'src/header/Header.jsx');

  __REACT_HOT_LOADER__.register(LOGO_STYLES, 'LOGO_STYLES', 'src/header/Header.jsx');

  __REACT_HOT_LOADER__.register(CLICKABLE_LOGO_STYLES, 'CLICKABLE_LOGO_STYLES', 'src/header/Header.jsx');

  __REACT_HOT_LOADER__.register(FLEX_ITEM_STYLES, 'FLEX_ITEM_STYLES', 'src/header/Header.jsx');

  __REACT_HOT_LOADER__.register(TxlHeader, 'TxlHeader', 'src/header/Header.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/header/Header.jsx');
}();

;
module.exports = exports['default'];
