// txl/buttons/_ButtonStyleHelpers.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BUTTON_BASE_STYLE = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

exports.getButtonStylesFromProps = getButtonStylesFromProps;
exports.getIconButtonStylesFromProps = getIconButtonStylesFromProps;

var _helpers = require('../styles/helpers');

var _resets = require('../styles/mixins/resets');

var _transitions = require('../styles/mixins/transitions');

var _type = require('../styles/mixins/type');

var _theme = require('../styles/theme');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BUTTON_BASE_STYLE = exports.BUTTON_BASE_STYLE = (0, _extends3.default)({}, _type.TYPE_BASE, _resets.RESET_PADDING, _resets.RESET_MARGIN, (0, _helpers.allBorderRadius)(2), {
  borderColor: 'transparent',
  borderStyle: 'none',
  borderWidth: 0,
  cursor: 'pointer',
  display: 'inline-block',
  lineHeight: 1,
  textAlign: 'center',
  textTransform: 'none',
  transition: (0, _transitions.cubicTransition)(['background-color', 'color']),
  verticalAlign: 'bottom',
  whiteSpace: 'nowrap'
});

var BUTTON_VARIANT_STYLE = {
  muted: {
    backgroundColor: _theme.COLOR_NEUTRAL['100'],
    color: _theme.COLOR_NEUTRAL['100c']
  },
  'muted-hover': {
    backgroundColor: _theme.COLOR_NEUTRAL['200'],
    color: _theme.COLOR_NEUTRAL['200c']
  },
  neutral: {
    backgroundColor: _theme.COLOR_NEUTRAL['600'],
    color: _theme.COLOR_NEUTRAL['600c']
  },
  'neutral-hover': {
    backgroundColor: _theme.COLOR_NEUTRAL['700'],
    color: _theme.COLOR_NEUTRAL['700c']
  },
  plain: {
    backgroundColor: 'transparent',
    color: _theme.COLOR_NEUTRAL['300']
  },
  'plain-hover': {
    backgroundColor: 'transparent',
    color: _theme.COLOR_NEUTRAL['900']
  }
};

var BUTTON_DISABLED_STYLE = {
  backgroundColor: _theme.COLOR_NEUTRAL['100'],
  color: _theme.COLOR_NEUTRAL['200'],
  cursor: 'not-allowed'
};

var ICON_BUTTONS_SIZE_STANDARD = (0, _extends3.default)({}, (0, _helpers.square)((0, _helpers.gridUnits)(5)));

var ICON_BUTTONS_SIZE_LARGE = (0, _extends3.default)({}, (0, _helpers.square)((0, _helpers.gridUnits)(7)));

function getDisabledStyles(variant) {
  return (0, _helpers.combineStyles)(BUTTON_DISABLED_STYLE,
  // plain buttons shouldn't have a background, even when disabled
  variant === 'plain' && { backgroundColor: BUTTON_VARIANT_STYLE[variant].backgroundColor });
}

function getButtonStylesFromProps(props, options, palette) {
  var variant = props.variant;

  if (options.hovered) {
    variant = variant + '-hover';
  }

  var COLOR_APP_THEME = (0, _theme.determineColorAppTheme)(palette);
  BUTTON_VARIANT_STYLE.accent = {
    backgroundColor: COLOR_APP_THEME['500'],
    color: COLOR_APP_THEME['500c']
  };
  BUTTON_VARIANT_STYLE['accent-hover'] = {
    backgroundColor: COLOR_APP_THEME['600'],
    color: COLOR_APP_THEME['600c']
  };

  return (0, _helpers.combineStyles)(BUTTON_BASE_STYLE, props.disabled && getDisabledStyles(props.variant), !props.disabled && BUTTON_VARIANT_STYLE[variant],

  // not documented but super useful for internal button consumers <Button _style={myExtra}... />
  props._style);
}

function getIconButtonStylesFromProps(props, options, palette) {
  return (0, _helpers.combineStyles)(getButtonStylesFromProps(props, { hovered: options.hovered }, palette), props.size === 'large' ? ICON_BUTTONS_SIZE_LARGE : ICON_BUTTONS_SIZE_STANDARD);
}
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(BUTTON_BASE_STYLE, 'BUTTON_BASE_STYLE', 'src/buttons/_ButtonStyleHelpers.js');

  __REACT_HOT_LOADER__.register(BUTTON_VARIANT_STYLE, 'BUTTON_VARIANT_STYLE', 'src/buttons/_ButtonStyleHelpers.js');

  __REACT_HOT_LOADER__.register(BUTTON_DISABLED_STYLE, 'BUTTON_DISABLED_STYLE', 'src/buttons/_ButtonStyleHelpers.js');

  __REACT_HOT_LOADER__.register(ICON_BUTTONS_SIZE_STANDARD, 'ICON_BUTTONS_SIZE_STANDARD', 'src/buttons/_ButtonStyleHelpers.js');

  __REACT_HOT_LOADER__.register(ICON_BUTTONS_SIZE_LARGE, 'ICON_BUTTONS_SIZE_LARGE', 'src/buttons/_ButtonStyleHelpers.js');

  __REACT_HOT_LOADER__.register(getDisabledStyles, 'getDisabledStyles', 'src/buttons/_ButtonStyleHelpers.js');

  __REACT_HOT_LOADER__.register(getButtonStylesFromProps, 'getButtonStylesFromProps', 'src/buttons/_ButtonStyleHelpers.js');

  __REACT_HOT_LOADER__.register(getIconButtonStylesFromProps, 'getIconButtonStylesFromProps', 'src/buttons/_ButtonStyleHelpers.js');
}();

;
