// txl/buttons/Button.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _helpers = require('../styles/helpers');

var _ButtonStyleHelpers = require('./_ButtonStyleHelpers');

var _theme = require('../styles/theme');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var BUTTON_SIZES = {
  large: {
    height: (0, _helpers.gridUnits)(7),
    lineHeight: (0, _helpers.gridUnits)(7 * 0.9), // height tweak, looks slightly off vertically otherwise
    paddingLeft: (0, _helpers.gridUnits)(3),
    paddingRight: (0, _helpers.gridUnits)(3)
  },
  standard: {
    height: (0, _helpers.gridUnits)(5),
    lineHeight: (0, _helpers.gridUnits)(5 * 0.9), // height tweak, looks slightly off vertically otherwise
    paddingLeft: (0, _helpers.gridUnits)(3),
    paddingRight: (0, _helpers.gridUnits)(3)
  }
};

var TxlButton = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlButton, _React$Component);

  function TxlButton() {
    var _ref;

    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, TxlButton);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlButton.__proto__ || (0, _getPrototypeOf2.default)(TxlButton)).call.apply(_ref, [this].concat(args))), _this), _this.state = { hovered: false }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  (0, _createClass3.default)(TxlButton, [{
    key: '_handleClick',
    value: function _handleClick() {
      if (!this.props.disabled) {
        this.props.onClick && this.props.onClick({});
      }
    }
  }, {
    key: '_handleMouseEnter',
    value: function _handleMouseEnter() {
      this.setState({ hovered: true });
    }
  }, {
    key: '_handleMouseLeave',
    value: function _handleMouseLeave() {
      this.setState({ hovered: false });
    }
  }, {
    key: 'render',
    value: function render() {
      var hovered = this.state.hovered && !this.props.disabled;

      var buttonStyles = (0, _helpers.combineStyles)((0, _ButtonStyleHelpers.getButtonStylesFromProps)(this.props, { hovered: hovered }, this.context.txlAppTheme), // contains resets, needs to come first
      BUTTON_SIZES[this.props.size]);

      return _react2.default.createElement(
        'button',
        {
          'data-component': 'TxlButton',
          onMouseEnter: this._handleMouseEnter,
          onMouseLeave: this._handleMouseLeave,
          onClick: this._handleClick,
          disabled: this.props.disabled,
          style: buttonStyles,
          id: this.props.id,
          tabIndex: this.props.tabIndex,
          type: this.props.type
        },
        this.props.children
      );
    }
  }]);
  return TxlButton;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleClick'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleMouseEnter', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseEnter'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleMouseLeave', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseLeave'), _class2.prototype)), _class2)) || _class;

TxlButton.propTypes = {
  _style: require('react').PropTypes.object,
  children: require('react').PropTypes.any,
  disabled: require('react').PropTypes.bool,
  id: require('react').PropTypes.string,
  onClick: require('react').PropTypes.func,
  size: require('react').PropTypes.any,
  tabIndex: require('react').PropTypes.number,
  type: require('react').PropTypes.any,
  variant: require('react').PropTypes.any
};
exports.default = TxlButton;


TxlButton.defaultProps = {
  disabled: false,
  onClick: _Base.NOOP,
  size: 'standard',
  tabIndex: 0,
  type: 'button',
  variant: 'neutral'
};

TxlButton.contextTypes = {
  txlAppTheme: _theme.contextThemePropType
};
;

var _temp2 = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(BUTTON_SIZES, 'BUTTON_SIZES', 'src/buttons/Button.jsx');

  __REACT_HOT_LOADER__.register(TxlButton, 'TxlButton', 'src/buttons/Button.jsx');
}();

;
module.exports = exports['default'];
