// txl/text/MiddleTruncatedText.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _TruncatedText = require('./TruncatedText.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var measurementTextStyles = (0, _extends3.default)({}, _TruncatedText.ABSOLUTE_TEXT_STYLES, _TruncatedText.INVISIBLE_TEXT_STYLES);

var CHARS_IN_ELLIPSES = 3;

var TxlMiddleTruncatedText = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlMiddleTruncatedText, _React$Component);

  function TxlMiddleTruncatedText() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlMiddleTruncatedText);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlMiddleTruncatedText.__proto__ || (0, _getPrototypeOf2.default)(TxlMiddleTruncatedText)).call.apply(_ref, [this].concat(args)));

    _this.state = {
      truncatedText: undefined
    };
    return _this;
  }

  (0, _createClass3.default)(TxlMiddleTruncatedText, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps() {
      this.setState({ truncatedText: undefined });
    }
  }, {
    key: 'componentDidMount',
    value: function componentDidMount() {
      if (this._isTruncationRequired()) {
        this._truncateText();
      }
    }
  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate() {
      if (this._isTruncationRequired()) {
        this._truncateText();
      }
    }
  }, {
    key: '_getExcessLengthInPixels',
    value: function _getExcessLengthInPixels() {
      var containerWidth = this._container.offsetWidth;
      var textWidth = this._measurementText.offsetWidth;
      var maxLength = containerWidth * this.props.numberOfLines - 1; // Give 1px of wiggle room

      return textWidth - maxLength;
    }
  }, {
    key: '_isTruncationRequired',
    value: function _isTruncationRequired() {
      return this._getExcessLengthInPixels() > 0;
    }
  }, {
    key: '_getInitialTruncation',
    value: function _getInitialTruncation() {
      var toTruncate = this.props.text;

      var pixelsTooLong = this._getExcessLengthInPixels();
      var textWidth = this._measurementText.offsetWidth;
      var percentTooLong = pixelsTooLong / textWidth;

      var numberOfCharsToTruncate = Math.ceil(percentTooLong * toTruncate.length);
      var middleCharacterIndex = toTruncate.length / 2;
      var firstHalfTruncated = toTruncate.substring(0, middleCharacterIndex - numberOfCharsToTruncate / 2);
      var secondHalfTruncated = toTruncate.substring(middleCharacterIndex + numberOfCharsToTruncate / 2);

      return firstHalfTruncated + '...' + secondHalfTruncated;
    }
  }, {
    key: '_getSubsequentTruncation',
    value: function _getSubsequentTruncation() {
      var toTruncate = this.state.truncatedText;
      var firstPeriodInEllipses = toTruncate.indexOf('.');
      var firstHalfTruncated = toTruncate.substring(0, firstPeriodInEllipses + CHARS_IN_ELLIPSES);
      var secondHalfTruncated = toTruncate.substring(firstPeriodInEllipses + CHARS_IN_ELLIPSES + 1);

      return firstHalfTruncated.concat(secondHalfTruncated);
    }
  }, {
    key: '_truncateText',
    value: function _truncateText() {
      var truncatedText = this.state.truncatedText ? this._getSubsequentTruncation() : this._getInitialTruncation();

      this.setState({ truncatedText: truncatedText });
    }
  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      var text = this.state.truncatedText || this.props.text;

      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlMiddleTruncatedText',
          style: _TruncatedText.CONTAINER_STYLES,
          ref: function ref(_ref3) {
            _this2._container = _ref3;
          }
        },
        _react2.default.createElement(
          'span',
          {
            ref: function ref(_ref2) {
              _this2._measurementText = _ref2;
            },
            style: measurementTextStyles
          },
          text
        ),
        _react2.default.createElement(
          'span',
          null,
          text
        )
      );
    }
  }]);
  return TxlMiddleTruncatedText;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_getExcessLengthInPixels', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_getExcessLengthInPixels'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_isTruncationRequired', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_isTruncationRequired'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_getInitialTruncation', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_getInitialTruncation'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_getSubsequentTruncation', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_getSubsequentTruncation'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_truncateText', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_truncateText'), _class2.prototype)), _class2)) || _class;

exports.default = TxlMiddleTruncatedText;


TxlMiddleTruncatedText.propTypes = {
  numberOfLines: _react.PropTypes.number,
  text: _react.PropTypes.string.isRequired
};

TxlMiddleTruncatedText.defaultProps = {
  numberOfLines: 1
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(measurementTextStyles, 'measurementTextStyles', 'src/text/MiddleTruncatedText.jsx');

  __REACT_HOT_LOADER__.register(CHARS_IN_ELLIPSES, 'CHARS_IN_ELLIPSES', 'src/text/MiddleTruncatedText.jsx');

  __REACT_HOT_LOADER__.register(TxlMiddleTruncatedText, 'TxlMiddleTruncatedText', 'src/text/MiddleTruncatedText.jsx');
}();

;
module.exports = exports['default'];
