// txl/text/TruncatedText.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _helpers = require('../styles/helpers');

var _TruncatedText = require('./TruncatedText.style');

var _FixedTooltipTarget = require('../_tooltips/FixedTooltipTarget');

var _FixedTooltipTarget2 = _interopRequireDefault(_FixedTooltipTarget);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TRUNCATION_STYLES = {
  overflow: 'hidden',
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap',
  width: '100%'
};

var TxlTruncatedText = (0, _radium2.default)(_class = function (_React$Component) {
  (0, _inherits3.default)(TxlTruncatedText, _React$Component);

  function TxlTruncatedText() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlTruncatedText);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlTruncatedText.__proto__ || (0, _getPrototypeOf2.default)(TxlTruncatedText)).call.apply(_ref, [this].concat(args)));

    _this._containerWidth = 0;
    _this._textWidth = 0;

    _this._handleMouseEnter = _this._handleMouseEnter.bind(_this);

    _this.state = {
      showTooltip: false
    };
    return _this;
  }

  (0, _createClass3.default)(TxlTruncatedText, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      this._setWidths();
    }
  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate() {
      this._setWidths();
    }
  }, {
    key: '_setWidths',
    value: function _setWidths() {
      this._containerWidth = this._container.offsetWidth;
      this._textWidth = this._measurementText.offsetWidth;
    }
  }, {
    key: '_handleMouseEnter',
    value: function _handleMouseEnter() {
      if (this._containerWidth >= this._textWidth) {
        this.setState({ showTooltip: false });
      } else {
        this.setState({ showTooltip: true });
      }
    }
  }, {
    key: '_renderMeasurementText',
    value: function _renderMeasurementText() {
      var _this2 = this;

      var measurementTextStyles = (0, _helpers.combineStyles)(_TruncatedText.ABSOLUTE_TEXT_STYLES, _TruncatedText.INVISIBLE_TEXT_STYLES);

      return _react2.default.createElement(
        'div',
        {
          ref: function ref(n) {
            _this2._measurementText = n;
          },
          style: measurementTextStyles
        },
        this.props.text
      );
    }
  }, {
    key: '_renderText',
    value: function _renderText() {
      var text = this.props.text;
      var showTooltip = this.state.showTooltip;


      if (showTooltip) {
        return _react2.default.createElement(
          _FixedTooltipTarget2.default,
          {
            style: TRUNCATION_STYLES,
            message: text
          },
          text
        );
      }

      return _react2.default.createElement(
        'div',
        { style: TRUNCATION_STYLES },
        text
      );
    }
  }, {
    key: 'render',
    value: function render() {
      var _this3 = this;

      return _react2.default.createElement(
        'div',
        {
          style: _TruncatedText.CONTAINER_STYLES,
          ref: function ref(n) {
            _this3._container = n;
          },
          onMouseEnter: this._handleMouseEnter
        },
        this._renderMeasurementText(),
        this._renderText()
      );
    }
  }]);
  return TxlTruncatedText;
}(_react2.default.Component)) || _class;

exports.default = TxlTruncatedText;


TxlTruncatedText.propTypes = {
  text: _react.PropTypes.string.isRequired
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TRUNCATION_STYLES, 'TRUNCATION_STYLES', 'src/text/TruncatedText.jsx');

  __REACT_HOT_LOADER__.register(TxlTruncatedText, 'TxlTruncatedText', 'src/text/TruncatedText.jsx');
}();

;
module.exports = exports['default'];
