// txl/lists/_BufferedListItems.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _ItemBuffer = require('../utils/ItemBuffer');

var _ItemBuffer2 = _interopRequireDefault(_ItemBuffer);

var _BufferedListItem = require('./_BufferedListItem');

var _BufferedListItem2 = _interopRequireDefault(_BufferedListItem);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ROW_STYLE = {
  position: 'relative'
};

function bufferPropsMapper(props) {
  return {
    defaultItemSize: props.itemHeight,
    itemsCount: props.itemsCount,
    viewportSize: props.viewportHeight
  };
}

/**
 * ** DO NOT MAKE this pure-rendered component.
 */

var TxlBufferedListItems = function (_Component) {
  (0, _inherits3.default)(TxlBufferedListItems, _Component);

  function TxlBufferedListItems() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlBufferedListItems);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlBufferedListItems.__proto__ || (0, _getPrototypeOf2.default)(TxlBufferedListItems)).call.apply(_ref, [this].concat(args)));

    var _this$props = _this.props,
        firstItemIndex = _this$props.firstItemIndex,
        firstItemOffset = _this$props.firstItemOffset;

    _this.buffer = new _ItemBuffer2.default(bufferPropsMapper(_this.props));
    _this.staticRowsArray = [];
    _this.state = {
      rowsToRender: _this.buffer.getItems(firstItemIndex, firstItemOffset)
    };
    return _this;
  }

  (0, _createClass3.default)(TxlBufferedListItems, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(props) {
      if (this.props.itemsCount !== props.itemsCount || this.props.itemHeight !== props.itemHeight) {
        this.buffer = new _ItemBuffer2.default(bufferPropsMapper(props));
      }

      this.setState({
        rowsToRender: this.buffer.getItems(props.firstItemIndex, props.firstItemOffset)
      });
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      this.staticRowsArray = [];
    }
  }, {
    key: 'render',
    value: function render() {
      var rowsToRender = this.state.rowsToRender;
      var _props = this.props,
          itemRenderer = _props.itemRenderer,
          itemHeight = _props.itemHeight,
          offsetTop = _props.offsetTop;


      this.staticRowsArray = rowsToRender.map(function (itemIndex, rowsToRenderIndex) {
        var itemOffsetTop = offsetTop + itemIndex * itemHeight;
        return _react2.default.createElement(_BufferedListItem2.default, {
          key: rowsToRenderIndex,
          itemRenderer: itemRenderer,
          index: itemIndex,
          height: itemHeight,
          offsetTop: Math.round(itemOffsetTop)
        });
      });

      var style = (0, _extends3.default)({}, ROW_STYLE, {
        height: this.props.scrollHeight
      });

      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlBufferedListItems',
          style: style
        },
        this.staticRowsArray
      );
    }
  }]);
  return TxlBufferedListItems;
}(_react.Component);

TxlBufferedListItems.displayName = 'TxlBufferedListItems';
TxlBufferedListItems.propTypes = {
  firstItemIndex: _react.PropTypes.number.isRequired,
  firstItemOffset: _react.PropTypes.number.isRequired,
  itemHeight: _react.PropTypes.number.isRequired,
  itemRenderer: _react.PropTypes.func.isRequired,
  itemsCount: _react.PropTypes.number.isRequired,
  offsetTop: _react.PropTypes.number.isRequired,
  scrollHeight: _react.PropTypes.number.isRequired,
  viewportHeight: _react.PropTypes.number.isRequired
};

var _default = TxlBufferedListItems;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ROW_STYLE, 'ROW_STYLE', 'src/lists/_BufferedListItems.jsx');

  __REACT_HOT_LOADER__.register(bufferPropsMapper, 'bufferPropsMapper', 'src/lists/_BufferedListItems.jsx');

  __REACT_HOT_LOADER__.register(TxlBufferedListItems, 'TxlBufferedListItems', 'src/lists/_BufferedListItems.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/lists/_BufferedListItems.jsx');
}();

;
module.exports = exports['default'];
