// txl/lists/_BufferedListItem.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function getStyle(props) {
  return {
    backfaceVisibility: 'hidden',
    height: props.height,
    position: 'absolute',
    transform: 'translate3d(' + 0 + 'px,' + props.offsetTop + 'px,0)',
    width: '100%'
  };
}

// DO NOT USE pure renderer
var TxlBufferedListItem = function TxlBufferedListItem(props) {
  var height = props.height,
      index = props.index,
      itemRenderer = props.itemRenderer;

  var renderedItem = itemRenderer({
    height: height,
    index: index
  });

  return _react2.default.createElement(
    'div',
    {
      'data-component': 'TxlBufferedListItem',
      style: getStyle(props)
    },
    renderedItem
  );
};

TxlBufferedListItem.displayName = 'TxlBufferedListItem';
TxlBufferedListItem.propTypes = {
  height: _react.PropTypes.oneOfType([_react.PropTypes.number, _react.PropTypes.string]),
  index: _react.PropTypes.number.isRequired,
  itemRenderer: _react.PropTypes.func.isRequired,
  offsetTop: _react.PropTypes.number.isRequired
};

var _default = (0, _radium2.default)(TxlBufferedListItem);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(getStyle, 'getStyle', 'src/lists/_BufferedListItem.jsx');

  __REACT_HOT_LOADER__.register(TxlBufferedListItem, 'TxlBufferedListItem', 'src/lists/_BufferedListItem.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/lists/_BufferedListItem.jsx');
}();

;
module.exports = exports['default'];
