// txl/lists/_FlatSelectableItemList.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _list = require('../styles/mixins/list');

var _Base = require('../base/Base');

var _SelectItemBase = require('../_selects/SelectItemBase');

var _SelectItemBase2 = _interopRequireDefault(_SelectItemBase);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlFlatSelectableItemList = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlFlatSelectableItemList, _React$Component);

  function TxlFlatSelectableItemList(props) {
    (0, _classCallCheck3.default)(this, TxlFlatSelectableItemList);

    var _this = (0, _possibleConstructorReturn3.default)(this, (TxlFlatSelectableItemList.__proto__ || (0, _getPrototypeOf2.default)(TxlFlatSelectableItemList)).call(this, props));

    _this.state = {
      activeIndex: -1
    };
    return _this;
  }

  (0, _createClass3.default)(TxlFlatSelectableItemList, [{
    key: '_handleClick',
    value: function _handleClick(_ref) {
      var item = _ref.item;

      this.props.onClick(item);
    }
  }, {
    key: '_handleItemActivate',
    value: function _handleItemActivate(_ref2) {
      var index = _ref2.index;

      this.setState({ activeIndex: index });
    }
  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      var _props = this.props,
          itemHeight = _props.itemHeight,
          id = _props.id,
          items = _props.items,
          selectedValues = _props.selectedValues;


      return _react2.default.createElement(
        'ul',
        {
          'data-component': 'TxlFlatSelectableItemList',
          id: id,
          style: _list.UNSTYLED_LIST
        },
        items.map(function (item, index) {
          return _react2.default.createElement(
            _SelectItemBase2.default,
            {
              active: index === _this2.state.activeIndex,
              height: itemHeight,
              id: id && id + '-' + item.value,
              index: index,
              item: item,
              key: item.value,
              selected: selectedValues.indexOf(item.value) > -1,
              onClick: _this2._handleClick,
              onItemActivate: _this2._handleItemActivate
            },
            item.label
          );
        })
      );
    }
  }]);
  return TxlFlatSelectableItemList;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleClick'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleItemActivate', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleItemActivate'), _class2.prototype)), _class2)) || _class;

exports.default = TxlFlatSelectableItemList;


TxlFlatSelectableItemList.displayName = 'TxlFlatSelectableItemList';
TxlFlatSelectableItemList.propTypes = {
  id: _react.PropTypes.string,
  itemHeight: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.number]),
  items: _react.PropTypes.arrayOf(_react.PropTypes.shape({
    label: _react.PropTypes.node,
    value: _react.PropTypes.string
  })),
  onClick: _react.PropTypes.func.isRequired,
  selectedValues: _react.PropTypes.arrayOf(_react.PropTypes.string)
};

TxlFlatSelectableItemList.defaultProps = {
  selectedValues: []
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlFlatSelectableItemList, 'TxlFlatSelectableItemList', 'src/lists/_FlatSelectableItemList.jsx');
}();

;
module.exports = exports['default'];
