// txl/lists/_GroupedBufferedListView.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _desc, _value, _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Base = require('../base/Base');

var _BufferedListView = require('../lists/_BufferedListView');

var _BufferedListView2 = _interopRequireDefault(_BufferedListView);

var _GroupedItemsWrapper = require('../lists/GroupedItemsWrapper');

var _GroupedItemsWrapper2 = _interopRequireDefault(_GroupedItemsWrapper);

var _GroupHeader = require('../lists/_GroupHeader');

var _GroupHeader2 = _interopRequireDefault(_GroupHeader);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var defaultGroupRenderer = _react2.default.createFactory(_GroupHeader2.default);

function isNotEqual(props, nextProps) {
  return props.items !== nextProps.items || props.grouper !== nextProps.grouper || props.itemRenderer !== nextProps.itemRenderer || props.groupRenderer !== nextProps.groupRenderer;
}

/**
 * *Should not be pure rendered.
 */
var GroupedBufferedListView = (_class = function (_Component) {
  (0, _inherits3.default)(GroupedBufferedListView, _Component);

  function GroupedBufferedListView() {
    var _ref;

    (0, _classCallCheck3.default)(this, GroupedBufferedListView);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = GroupedBufferedListView.__proto__ || (0, _getPrototypeOf2.default)(GroupedBufferedListView)).call.apply(_ref, [this].concat(args)));

    _this.state = {
      groupedItems: new _GroupedItemsWrapper2.default(_this.props)
    };
    return _this;
  }

  (0, _createClass3.default)(GroupedBufferedListView, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      if (isNotEqual(this.props, nextProps)) {
        this.setState({
          groupedItems: new _GroupedItemsWrapper2.default(nextProps)
        });
      }
    }
  }, {
    key: '_splitProps',
    value: function _splitProps() {
      var _props = this.props,
          items = _props.items,
          grouper = _props.grouper,
          others = (0, _objectWithoutProperties3.default)(_props, ['items', 'grouper']);


      return {
        grouper: grouper,
        items: items,
        others: others
      };
    }
  }, {
    key: 'render',
    value: function render() {
      var _splitProps2 = this._splitProps(),
          others = _splitProps2.others;

      var groupedItems = this.state.groupedItems;

      var itemsCount = groupedItems.getSize();
      var itemRenderer = groupedItems.render;
      var bufferedListViewProps = (0, _extends3.default)({}, others, {
        itemRenderer: itemRenderer,
        itemsCount: itemsCount,
        noContent: others.noContentView
      });

      return _react2.default.createElement(_BufferedListView2.default, bufferedListViewProps);
    }
  }]);
  return GroupedBufferedListView;
}(_react.Component), (_applyDecoratedDescriptor(_class.prototype, '_splitProps', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_splitProps'), _class.prototype)), _class);


GroupedBufferedListView.propTypes = {
  activateItem: _react.PropTypes.func,
  footer: _BufferedListView2.default.propTypes.footer,
  groupRenderer: _react.PropTypes.func,
  grouper: _react.PropTypes.func,
  header: _BufferedListView2.default.propTypes.header,
  height: _BufferedListView2.default.propTypes.height,
  itemRenderer: _react.PropTypes.func,
  items: _react.PropTypes.array,
  keySelectedItemClear: _react.PropTypes.func,
  keySelectedItemIndex: _react.PropTypes.number,
  maxHeight: _BufferedListView2.default.propTypes.maxHeight,
  noContentView: _BufferedListView2.default.propTypes.noContent,
  showSearchToViewMoreMessage: _BufferedListView2.default.propTypes.showSearchToViewMoreMessage
};

GroupedBufferedListView.defaultProps = {
  groupRenderer: defaultGroupRenderer
};

var _default = GroupedBufferedListView;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(defaultGroupRenderer, 'defaultGroupRenderer', 'src/lists/_GroupedBufferedListView.jsx');

  __REACT_HOT_LOADER__.register(isNotEqual, 'isNotEqual', 'src/lists/_GroupedBufferedListView.jsx');

  __REACT_HOT_LOADER__.register(GroupedBufferedListView, 'GroupedBufferedListView', 'src/lists/_GroupedBufferedListView.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/lists/_GroupedBufferedListView.jsx');
}();

;
module.exports = exports['default'];
