// txl/lists/_GroupHeader.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var HEADER_STYLE = {
  color: 'grey',
  fontSize: 12,
  padding: 5
};

var GroupItemType = _react.PropTypes.shape({
  groupKey: _react.PropTypes.string
});

function stopPropagation(e) {
  e.stopPropagation();
}

var TxlGroupHeader = function TxlGroupHeader(props) {
  return _react2.default.createElement(
    'h3',
    {
      style: HEADER_STYLE,
      onClick: stopPropagation
    },
    props.item.groupKey
  );
};

TxlGroupHeader.propTypes = {
  height: _react.PropTypes.number,
  index: _react.PropTypes.number,
  item: GroupItemType.isRequired
};

var _default = TxlGroupHeader;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(HEADER_STYLE, 'HEADER_STYLE', 'src/lists/_GroupHeader.jsx');

  __REACT_HOT_LOADER__.register(GroupItemType, 'GroupItemType', 'src/lists/_GroupHeader.jsx');

  __REACT_HOT_LOADER__.register(stopPropagation, 'stopPropagation', 'src/lists/_GroupHeader.jsx');

  __REACT_HOT_LOADER__.register(TxlGroupHeader, 'TxlGroupHeader', 'src/lists/_GroupHeader.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/lists/_GroupHeader.jsx');
}();

;
module.exports = exports['default'];
