// txl/lists/_GroupedSelectableItemList.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _FlatSelectableItemList = require('./_FlatSelectableItemList');

var _FlatSelectableItemList2 = _interopRequireDefault(_FlatSelectableItemList);

var _lists = require('./lists.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlGroupedSelectableItemList = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlGroupedSelectableItemList, _React$Component);

  function TxlGroupedSelectableItemList() {
    (0, _classCallCheck3.default)(this, TxlGroupedSelectableItemList);
    return (0, _possibleConstructorReturn3.default)(this, (TxlGroupedSelectableItemList.__proto__ || (0, _getPrototypeOf2.default)(TxlGroupedSelectableItemList)).apply(this, arguments));
  }

  (0, _createClass3.default)(TxlGroupedSelectableItemList, [{
    key: '_handleItemClick',
    value: function _handleItemClick(item) {
      this.props.onClick(item);
    }
  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      var _props = this.props,
          id = _props.id,
          itemHeight = _props.itemHeight,
          items = _props.items,
          selectedValues = _props.selectedValues;


      return _react2.default.createElement(
        'div',
        { 'data-component': 'TxlGroupedSelectableItemList' },
        items.map(function (group) {
          return _react2.default.createElement(
            'section',
            { key: group.label },
            _react2.default.createElement(
              'header',
              {
                'data-test-selectable-item-list-header': true,
                style: _lists.LIST_HEADER_STYLE
              },
              group.label
            ),
            _react2.default.createElement(_FlatSelectableItemList2.default, {
              id: id,
              itemHeight: itemHeight,
              selectedValues: selectedValues,
              items: group.items,
              onClick: _this2._handleItemClick
            })
          );
        })
      );
    }
  }]);
  return TxlGroupedSelectableItemList;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleItemClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleItemClick'), _class2.prototype)), _class2)) || _class;

exports.default = TxlGroupedSelectableItemList;


TxlGroupedSelectableItemList.displayName = 'TxlGroupedSelectableItemList';
TxlGroupedSelectableItemList.propTypes = {
  id: _react.PropTypes.string,
  itemHeight: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.number]),
  items: _react.PropTypes.arrayOf(_react.PropTypes.shape({
    items: _FlatSelectableItemList2.default.propTypes.items,
    label: _react.PropTypes.node
  })),
  onClick: _react.PropTypes.func.isRequired,
  selectedValues: _FlatSelectableItemList2.default.propTypes.selectedValues
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlGroupedSelectableItemList, 'TxlGroupedSelectableItemList', 'src/lists/_GroupedSelectableItemList.jsx');
}();

;
module.exports = exports['default'];
