// txl/lists/_SelectableItemList.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _FlatSelectableItemList = require('./_FlatSelectableItemList');

var _FlatSelectableItemList2 = _interopRequireDefault(_FlatSelectableItemList);

var _GroupedSelectableItemList = require('./_GroupedSelectableItemList');

var _GroupedSelectableItemList2 = _interopRequireDefault(_GroupedSelectableItemList);

var _listHelpers = require('../lists/listHelpers');

var _lists = require('./lists.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// @TODO-TRANSLATE
var NO_ITEMS_FOUND = 'No Items Found';

var CONTAINER_STYLES = {
  overflowX: 'hidden',
  overflowY: 'auto'
};

var TxlSelectableItemList = (0, _radium2.default)(_class = function (_React$Component) {
  (0, _inherits3.default)(TxlSelectableItemList, _React$Component);

  function TxlSelectableItemList() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlSelectableItemList);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlSelectableItemList.__proto__ || (0, _getPrototypeOf2.default)(TxlSelectableItemList)).call.apply(_ref, [this].concat(args)));

    _this.calculateVisibleStyles = function () {
      return _this.__calculateVisibleStyles__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._groupedItems = (0, _listHelpers.hasGroupedItems)(_this.props.items);
    _this.calculateVisibleStyles(_this.props.visibleItemCount);
    return _this;
  }

  (0, _createClass3.default)(TxlSelectableItemList, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      if (nextProps.visibleItemCount !== this.props.visibleItemCount) {
        this.calculateVisibleStyles(nextProps.visibleItemCount);
      }
    }
  }, {
    key: '__calculateVisibleStyles__REACT_HOT_LOADER__',
    value: function __calculateVisibleStyles__REACT_HOT_LOADER__() {
      return this.__calculateVisibleStyles__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__calculateVisibleStyles__REACT_HOT_LOADER__',
    value: function __calculateVisibleStyles__REACT_HOT_LOADER__() {
      return this.__calculateVisibleStyles__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__calculateVisibleStyles__REACT_HOT_LOADER__',
    value: function __calculateVisibleStyles__REACT_HOT_LOADER__(visibleItemCount) {
      this._maxHeight = _lists.ITEM_HEIGHT * visibleItemCount;
      this._containerStyles = (0, _extends3.default)({}, CONTAINER_STYLES, {
        maxHeight: this._maxHeight
      });
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          id = _props.id,
          itemHeight = _props.itemHeight,
          items = _props.items,
          onClick = _props.onClick,
          selectedValues = _props.selectedValues;


      var ItemListComponent = this._groupedItems ? _GroupedSelectableItemList2.default : _FlatSelectableItemList2.default;

      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlSelectableItemList',
          id: id,
          style: this._containerStyles
        },
        items.length ? _react2.default.createElement(ItemListComponent, {
          id: id,
          itemHeight: itemHeight,
          items: items,
          onClick: onClick,
          selectedValues: selectedValues
        }) : _react2.default.createElement(
          'div',
          { style: _lists.LIST_ITEM_STYLE },
          NO_ITEMS_FOUND
        )
      );
    }
  }]);
  return TxlSelectableItemList;
}(_react2.default.Component)) || _class;

exports.default = TxlSelectableItemList;


TxlSelectableItemList.displayName = 'TxlSelectableItemList';
TxlSelectableItemList.propTypes = {
  id: _react.PropTypes.string,
  itemHeight: _react.PropTypes.oneOfType([_FlatSelectableItemList2.default.propTypes.itemHeight, _GroupedSelectableItemList2.default.propTypes.itemHeight]),
  items: _react.PropTypes.oneOfType([_FlatSelectableItemList2.default.propTypes.items, _GroupedSelectableItemList2.default.propTypes.items]),
  onClick: _react.PropTypes.func.isRequired,
  selectedValues: _react.PropTypes.arrayOf(_react.PropTypes.string),
  visibleItemCount: _react.PropTypes.number
};

TxlSelectableItemList.defaultProps = {
  itemHeight: 'auto',
  selectedValues: [],
  visibleItemCount: 12
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(NO_ITEMS_FOUND, 'NO_ITEMS_FOUND', 'src/lists/_SelectableItemList.jsx');

  __REACT_HOT_LOADER__.register(CONTAINER_STYLES, 'CONTAINER_STYLES', 'src/lists/_SelectableItemList.jsx');

  __REACT_HOT_LOADER__.register(TxlSelectableItemList, 'TxlSelectableItemList', 'src/lists/_SelectableItemList.jsx');
}();

;
module.exports = exports['default'];
