// txl/lists/listHelpers.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.hasGroupedItems = exports.getItemLabelByValue = exports.getItemByValue = exports.getFilteredGroupItemList = exports.getFilteredFlatItemList = undefined;

var _toConsumableArray2 = require('babel-runtime/helpers/toConsumableArray');

var _toConsumableArray3 = _interopRequireDefault(_toConsumableArray2);

var _find = require('lodash/find');

var _find2 = _interopRequireDefault(_find);

var _get2 = require('lodash/get');

var _get3 = _interopRequireDefault(_get2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * A grouping of option items
 * @typedef {Object} OptionGroup
 * @property {string} label - Label of group in list
 * @property {OptionItem[]} items - Collection of options
 */

/**
 * An option item to be used in a list
 * @typedef {Object} OptionItem
 * @property {string} label - Label of item in list
 * @property {string} value - Value used if option selected
 */

/**
 * Is array a flat list of items or a group of lists of items?
 * @param  {Object[]} list Collection of options or option groups
 * @return {Boolean}
 */
function hasGroupedItems(list) {
  return !!(0, _get3.default)(list, '[0].items');
}

/**
 * Find item where item.value === value arg
 * @param  {(OptionItem[]|OptionGroup[])} [list=[]] - Collection of options or option groups
 * @param  {string} value - Value to match upon
 * @return {(Object|undefined)} - Item with matching value or `undefined`
 */
function getItemByValue() {
  var list = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : [];
  var value = arguments[1];

  if (value === undefined) return undefined;
  var items = list;
  if (hasGroupedItems(list)) {
    items = list.reduce(function (allItems, group) {
      return allItems.concat(group.items);
    }, []);
  }
  // find option with matching value prop
  return (0, _find2.default)(items, { value: value });
}

/**
 * Find label of item where item.value === value arg
 * @param  {(OptionItem[]|OptionGroup[])} [list=[]] - Collection of options or option groups
 * @param  {string} value - Value to match upon
 * @return {(string|undefined)} - Label of item where value matches or `undefined`
 */
function getItemLabelByValue() {
  var list = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : [];
  var value = arguments[1];

  return (0, _get3.default)(getItemByValue(list, value), 'label');
}

/**
 * Filter a list of options by searching their labels for a value (case insensitive)
 * @param  {OptionItem[]} list - List of options
 * @param  {string} value - The pattern to search for
 * @return {OptionItem[]} - Array of matching options
 */
function getFilteredFlatItemList(list, value) {
  var matchRegExp = new RegExp(value, 'i');
  return list.filter(function (item) {
    return matchRegExp.test(item.label);
  });
}

/**
 * Find option groups with items that contain `value`
 * @param  {Object[]} list - Option groups to search through
 * @param  {string} list[].label - Label visible to user
 * @param  {string} list[].value - Value of option used if option selected
 * @param  {string} value - The pattern to search for
 * @return {Object[]} - Collection of option groups where item.label contains value
 */
function getFilteredGroupItemList(list, value) {
  return list.reduce(function (accum, curr) {
    var group = {
      items: getFilteredFlatItemList(curr.items, value),
      label: curr.label
    };
    return group.items.length ? [].concat((0, _toConsumableArray3.default)(accum), [group]) : accum;
  }, []);
}

exports.getFilteredFlatItemList = getFilteredFlatItemList;
exports.getFilteredGroupItemList = getFilteredGroupItemList;
exports.getItemByValue = getItemByValue;
exports.getItemLabelByValue = getItemLabelByValue;
exports.hasGroupedItems = hasGroupedItems;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(hasGroupedItems, 'hasGroupedItems', 'src/lists/listHelpers.js');

  __REACT_HOT_LOADER__.register(getItemByValue, 'getItemByValue', 'src/lists/listHelpers.js');

  __REACT_HOT_LOADER__.register(getItemLabelByValue, 'getItemLabelByValue', 'src/lists/listHelpers.js');

  __REACT_HOT_LOADER__.register(getFilteredFlatItemList, 'getFilteredFlatItemList', 'src/lists/listHelpers.js');

  __REACT_HOT_LOADER__.register(getFilteredGroupItemList, 'getFilteredGroupItemList', 'src/lists/listHelpers.js');
}();

;
