// txl/lists/lists.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LIST_ITEM_STYLE = exports.LIST_ITEM_SELECTED_SUBTEXT_STYLE = exports.LIST_ITEM_SELECTED_STYLE = exports.LIST_ITEM_SELECT_NONE_STYLE = exports.LIST_ITEM_HOVER_STYLE = exports.LIST_ITEM_DISABLED_STYLE = exports.LIST_HEADER_STYLE = exports.ITEM_HEIGHT = exports.ANCHOR_COMPONENT_STYLE = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _theme = require('../styles/theme');

var _type = require('../styles/mixins/type');

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// in grid units
var ITEM_HEIGHT = (0, _helpers.gridUnitsRaw)(7);
var ITEM_INDENT = (0, _helpers.gridUnitsRaw)(3);

var ANCHOR_COMPONENT_STYLE = {
  borderTop: '1px solid ' + _theme.COLOR_NEUTRAL['100']
};

var LIST_ITEM_BASE_STYLE = (0, _extends3.default)({}, _type.DEFAULT_FONT_FAMILY, _type.TYPE_BASE, {
  backgroundColor: 'white',
  // px required for line-height since unitless valid by default
  lineHeight: ITEM_HEIGHT + 'px',
  whiteSpace: 'nowrap'
});

var LIST_HEADER_STYLE = (0, _extends3.default)({}, LIST_ITEM_BASE_STYLE, {
  color: _theme.COLOR_NEUTRAL['300'],
  marginLeft: ITEM_INDENT / 2 + 'px'
});

var LIST_ITEM_HOVER_STYLE = {
  backgroundColor: _theme.COLOR_NEUTRAL['50'],
  color: _theme.COLOR_NEUTRAL['50c']
};

var LIST_ITEM_SELECTED_STYLE = {
  backgroundColor: _theme.COLOR_ACCENT['500'],
  color: _theme.COLOR_ACCENT['500c']
};

var LIST_ITEM_SELECTED_SUBTEXT_STYLE = {
  color: _theme.COLOR_NEUTRAL['200']
};

var LIST_ITEM_STYLE = (0, _extends3.default)({}, LIST_ITEM_BASE_STYLE, {
  color: _theme.COLOR_NEUTRAL['50c'],
  cursor: 'pointer',
  paddingLeft: ITEM_INDENT + 'px',
  paddingRight: ITEM_INDENT + 'px'
});

var LIST_ITEM_DISABLED_STYLE = {
  color: _theme.COLOR_NEUTRAL['200'],
  cursor: 'not-allowed'
};

var LIST_ITEM_SELECT_NONE_STYLE = {
  color: _theme.COLOR_NEUTRAL['200']
};

exports.ANCHOR_COMPONENT_STYLE = ANCHOR_COMPONENT_STYLE;
exports.ITEM_HEIGHT = ITEM_HEIGHT;
exports.LIST_HEADER_STYLE = LIST_HEADER_STYLE;
exports.LIST_ITEM_DISABLED_STYLE = LIST_ITEM_DISABLED_STYLE;
exports.LIST_ITEM_HOVER_STYLE = LIST_ITEM_HOVER_STYLE;
exports.LIST_ITEM_SELECT_NONE_STYLE = LIST_ITEM_SELECT_NONE_STYLE;
exports.LIST_ITEM_SELECTED_STYLE = LIST_ITEM_SELECTED_STYLE;
exports.LIST_ITEM_SELECTED_SUBTEXT_STYLE = LIST_ITEM_SELECTED_SUBTEXT_STYLE;
exports.LIST_ITEM_STYLE = LIST_ITEM_STYLE;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ITEM_HEIGHT, 'ITEM_HEIGHT', 'src/lists/lists.style.js');

  __REACT_HOT_LOADER__.register(ITEM_INDENT, 'ITEM_INDENT', 'src/lists/lists.style.js');

  __REACT_HOT_LOADER__.register(ANCHOR_COMPONENT_STYLE, 'ANCHOR_COMPONENT_STYLE', 'src/lists/lists.style.js');

  __REACT_HOT_LOADER__.register(LIST_ITEM_BASE_STYLE, 'LIST_ITEM_BASE_STYLE', 'src/lists/lists.style.js');

  __REACT_HOT_LOADER__.register(LIST_HEADER_STYLE, 'LIST_HEADER_STYLE', 'src/lists/lists.style.js');

  __REACT_HOT_LOADER__.register(LIST_ITEM_HOVER_STYLE, 'LIST_ITEM_HOVER_STYLE', 'src/lists/lists.style.js');

  __REACT_HOT_LOADER__.register(LIST_ITEM_SELECTED_STYLE, 'LIST_ITEM_SELECTED_STYLE', 'src/lists/lists.style.js');

  __REACT_HOT_LOADER__.register(LIST_ITEM_SELECTED_SUBTEXT_STYLE, 'LIST_ITEM_SELECTED_SUBTEXT_STYLE', 'src/lists/lists.style.js');

  __REACT_HOT_LOADER__.register(LIST_ITEM_STYLE, 'LIST_ITEM_STYLE', 'src/lists/lists.style.js');

  __REACT_HOT_LOADER__.register(LIST_ITEM_DISABLED_STYLE, 'LIST_ITEM_DISABLED_STYLE', 'src/lists/lists.style.js');

  __REACT_HOT_LOADER__.register(LIST_ITEM_SELECT_NONE_STYLE, 'LIST_ITEM_SELECT_NONE_STYLE', 'src/lists/lists.style.js');
}();

;
