// txl/_popover/_TetheredPopoverTarget.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _desc, _value, _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _tether = require('tether');

var _tether2 = _interopRequireDefault(_tether);

var _pickBy = require('lodash/pickBy');

var _pickBy2 = _interopRequireDefault(_pickBy);

var _omit = require('lodash/omit');

var _omit2 = _interopRequireDefault(_omit);

var _ReactUtils = require('../ReactUtils');

var _Base = require('../base/Base');

var _StyleRoot = require('../styles/StyleRoot');

var _StyleRoot2 = _interopRequireDefault(_StyleRoot);

var _jsStyleHelpers = require('../styles/_helpers/jsStyleHelpers');

var _constants = require('../styles/constants');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlTetheredPopoverTarget = (_class = function (_React$Component) {
  (0, _inherits3.default)(TxlTetheredPopoverTarget, _React$Component);

  function TxlTetheredPopoverTarget() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlTetheredPopoverTarget);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlTetheredPopoverTarget.__proto__ || (0, _getPrototypeOf2.default)(TxlTetheredPopoverTarget)).call.apply(_ref, [this].concat(args)));

    _this._popover = null;
    _this._DOMNode = null;
    _this._bootstrapped = false;
    return _this;
  }

  (0, _createClass3.default)(TxlTetheredPopoverTarget, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      if (!this.props.visible) {
        return;
      }

      this._bootstrapPopover();
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      if (!this._bootstrapped) {
        return;
      }

      this._teardownPopover();
    }

    /**
     * @TODO: May need a special shouldComponentUpdate method so that we don't
     * rerender on width change.
     */

  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate() {
      if (this.props.visible) {
        if (this._bootstrapped) {
          this._renderPopover();
        } else {
          this._bootstrapPopover();
        }
      } else {
        if (!this._bootstrapped) {
          return;
        }

        this._teardownPopover();
      }
    }
  }, {
    key: '_bootstrapPopover',
    value: function _bootstrapPopover() {
      this._bootstrapped = true;
      this._createPopover();
      this._renderPopover();
    }
  }, {
    key: '_teardownPopover',
    value: function _teardownPopover() {
      this._tether.destroy();

      if (this._bootstrapped) {
        (0, _ReactUtils.unmountComponentAtNode)(this._popover);
        document.querySelector('body').removeChild(this._popover);
      }

      this._bootstrapped = false;
    }
  }, {
    key: '_getTetherOptions',
    value: function _getTetherOptions() {
      return (0, _extends3.default)({}, this.props.configuration, {
        element: this._popover,
        target: this._DOMNode
      });
    }
  }, {
    key: '_createPopover',
    value: function _createPopover() {
      var popover = document.createElement('div');
      // NOTE: we're relying on this 'data-test-tethered-popover' attribute for detecting clicks
      // within dropdown containers. Deleting this will break dropdowns.
      popover.setAttribute('data-test-tethered-popover', true);
      popover.setAttribute('id', this.props.id + '-popover');

      (0, _jsStyleHelpers.setStylesForNode)(popover, {
        opacity: 0,
        zIndex: this.props.popoverZIndex
      });

      this._popover = popover;
      document.body.appendChild(this._popover);
    }
  }, {
    key: '_setTetherOptions',
    value: function _setTetherOptions() {
      if (this._tether) {
        this._tether.setOptions(this._getTetherOptions());
      } else {
        this._tether = new _tether2.default(this._getTetherOptions());
      }
    }
  }, {
    key: '_renderPopover',
    value: function _renderPopover() {
      var _this2 = this;

      if (this.props.popoverWidth) {
        (0, _jsStyleHelpers.setStylesForNode)(this._popover, {
          width: this.props.popoverWidth
        });
      }

      (0, _ReactUtils.render)(_react2.default.createElement(
        _StyleRoot2.default,
        null,
        _react2.default.createElement(
          'div',
          { onClick: this.props.onContentClick },
          this.props.content
        )
      ), this._popover);
      this._setTetherOptions();

      this.props.tetherAdjustFunction(this._tether);
      window.requestAnimationFrame(function () {
        _this2._tether.position();

        (0, _jsStyleHelpers.setStylesForNode)(_this2._popover, { opacity: 1 });
      });
    }
  }, {
    key: 'render',
    value: function render() {
      var _this3 = this;

      var _props = this.props,
          children = _props.children,
          componentName = _props.componentName,
          id = _props.id,
          style = _props.style,
          others = (0, _objectWithoutProperties3.default)(_props, ['children', 'componentName', 'id', 'style']);

      // only pass along on* event handlers to native <div />
      // but do not pass onContentClick, which is special handler.

      var rest = (0, _omit2.default)(others, 'onContentClick');
      var eventProps = (0, _pickBy2.default)(rest, function (value, key) {
        return key.indexOf('on') === 0;
      });

      return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, eventProps, {
          ref: function ref(node) {
            _this3._DOMNode = node;
          },
          'data-component': componentName,
          id: id && id + '-target',
          style: style
        }),
        children
      );
    }
  }]);
  return TxlTetheredPopoverTarget;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class.prototype, '_bootstrapPopover', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_bootstrapPopover'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_teardownPopover', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_teardownPopover'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_getTetherOptions', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_getTetherOptions'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_createPopover', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_createPopover'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_setTetherOptions', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_setTetherOptions'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_renderPopover', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_renderPopover'), _class.prototype)), _class);
exports.default = TxlTetheredPopoverTarget;


TxlTetheredPopoverTarget.defaultProps = {
  popoverZIndex: _constants.Z_INDEXES.popover,
  tetherAdjustFunction: _Base.NOOP
};

TxlTetheredPopoverTarget.propTypes = {
  children: _react.PropTypes.node.isRequired,
  componentName: _react.PropTypes.string.isRequired,
  configuration: _react.PropTypes.object.isRequired,
  content: _react.PropTypes.node.isRequired,
  id: _react.PropTypes.string,
  onContentClick: _react.PropTypes.func,
  popoverWidth: _react.PropTypes.oneOfType([_react.PropTypes.number, _react.PropTypes.string]),
  popoverZIndex: _react.PropTypes.string,
  style: _react.PropTypes.object,
  tetherAdjustFunction: _react.PropTypes.func,
  visible: _react.PropTypes.bool
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlTetheredPopoverTarget, 'TxlTetheredPopoverTarget', 'src/_popover/_TetheredPopoverTarget.jsx');
}();

;
module.exports = exports['default'];
