// txl/input-fields/_ControlledDateRangeSelector.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _slicedToArray2 = require('babel-runtime/helpers/slicedToArray');

var _slicedToArray3 = _interopRequireDefault(_slicedToArray2);

var _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _isEqual = require('lodash/isEqual');

var _isEqual2 = _interopRequireDefault(_isEqual);

var _Base = require('../base/Base');

var _CalendarUtils = require('../calendar-picker/_CalendarUtils');

var _propTypes = require('../utils/propTypes');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Button = require('../buttons/Button');

var _Button2 = _interopRequireDefault(_Button);

var _CalendarOutline = require('../icons/CalendarOutline');

var _CalendarOutline2 = _interopRequireDefault(_CalendarOutline);

var _CalendarPickerWithPresets = require('../calendar-picker/CalendarPickerWithPresets');

var _CalendarPickerWithPresets2 = _interopRequireDefault(_CalendarPickerWithPresets);

var _DropdownContent = require('../dropdowns/_DropdownContent');

var _DropdownContent2 = _interopRequireDefault(_DropdownContent);

var _IconButton = require('../buttons/IconButton');

var _IconButton2 = _interopRequireDefault(_IconButton);

var _InputUnderline = require('../layout/_InputUnderline');

var _InputUnderline2 = _interopRequireDefault(_InputUnderline);

var _TextField = require('../input-fields/TextField');

var _TextField2 = _interopRequireDefault(_TextField);

var _TimeZonePicker = require('../input-fields/_TimeZonePicker');

var _TimeZonePicker2 = _interopRequireDefault(_TimeZonePicker);

var _TimePickerControlled = require('../input-fields/TimePickerControlled');

var _TimePickerControlled2 = _interopRequireDefault(_TimePickerControlled);

var _DateRangeSelector = require('./DateRangeSelector.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _DATE_RANGE_PRESETS$f = _CalendarUtils.DATE_RANGE_PRESETS.filter(function (preset) {
  return preset.key === 'today';
}),
    _DATE_RANGE_PRESETS$f2 = (0, _slicedToArray3.default)(_DATE_RANGE_PRESETS$f, 1),
    todayPreset = _DATE_RANGE_PRESETS$f2[0];

var DEFAULT_DATE_INPUT_PATTERN = 'YYYY-MM-DD';
var CONDENSED_DATE_FORMAT = 'MMM D, YYYY';
var TIME_PATTERN = 'hh:mm:ss A';

// @TODO translate
var APPLY_BUTTON_TEXT = 'Apply';
var CANCEL_BUTTON_TEXT = 'Cancel';
var TIME_RANGE_PICKER_LABEL = 'Time range';
var TIME_ZONE_PICKER_LABEL = 'Time zone';
var TIME_DELIMITER_TEXT = 'to';

// Convert time string that matches pattern to { hour, minute, second } obj
var _mapStringToTime = function _mapStringToTime(timeStr) {
  var timeMoment = (0, _moment2.default)(timeStr, TIME_PATTERN, true);
  return timeMoment.isValid() ? (0, _CalendarUtils.timeObjFromMoment)(timeMoment) : null;
};

// value prop comes through from "Create New", m/h/s obj comes from item select handler
var _getTimeFromItemSelect = function _getTimeFromItemSelect(_ref) {
  var hour = _ref.hour,
      minute = _ref.minute,
      second = _ref.second,
      value = _ref.value;
  return value ? _mapStringToTime(value) : { hour: hour, minute: minute, second: second };
};

var _formatTime = function _formatTime(timeObj) {
  return timeObj !== null ? (0, _moment2.default)(timeObj).format(TIME_PATTERN) : '';
};

function hasDateRangeChanged(existingRange, activeRange) {
  var _existingRange = (0, _slicedToArray3.default)(existingRange, 2),
      existingStart = _existingRange[0],
      existingEnd = _existingRange[1];

  var _activeRange = (0, _slicedToArray3.default)(activeRange, 2),
      activeStart = _activeRange[0],
      activeEnd = _activeRange[1];

  return existingStart && existingEnd && (!existingStart.isSame(activeStart, 'day') || !existingEnd.isSame(activeEnd, 'day'));
}

function hasTimeRangeChanged(_ref2, _ref3) {
  var startTime = _ref2.startTime,
      endTime = _ref2.endTime;
  var activeStartTime = _ref3.activeStartTime,
      activeEndTime = _ref3.activeEndTime;

  return !(0, _isEqual2.default)(startTime, activeStartTime) || !(0, _isEqual2.default)(endTime, activeEndTime);
}

function hasNewSelectedValue(props, state) {
  var timeZone = props.timeZone,
      value = props.value;
  var activeRange = state.activeRange,
      activeTimeZone = state.activeTimeZone;


  return hasDateRangeChanged(value.map(_CalendarUtils.dateStringToMoment), activeRange) || hasTimeRangeChanged(props, state) || timeZone !== activeTimeZone;
}

function buildDisplayValueFromDates(startDate, endDate, pattern) {
  var isValidRange = startDate && endDate && (0, _CalendarUtils.isValidDateRange)(startDate, endDate);
  return isValidRange ? [startDate, endDate].map(function (date) {
    return (0, _moment2.default)(date).format(pattern);
  }).join(' - ') : '';
}

function buildTimeRangeSecondaryText(dateString) {
  return dateString ? (0, _moment2.default)(dateString).format(CONDENSED_DATE_FORMAT) : '';
}

function areDatesInRange(_ref4) {
  var start = _ref4.start,
      end = _ref4.end,
      rangeEndBoundary = _ref4.rangeEndBoundary,
      rangeStartBoundary = _ref4.rangeStartBoundary;

  return (0, _CalendarUtils.isDateWithinRange)(start, rangeStartBoundary, rangeEndBoundary, true) && (0, _CalendarUtils.isDateWithinRange)(end, rangeStartBoundary, rangeEndBoundary, true);
}

function isApplyButtonDisabled(props, state) {
  var rangeEndBoundary = props.rangeEndBoundary,
      rangeStartBoundary = props.rangeStartBoundary;

  var activeEndTime = state.activeEndTime,
      activeStartTime = state.activeStartTime,
      _state$activeRange = (0, _slicedToArray3.default)(state.activeRange, 2),
      activeStartDate = _state$activeRange[0],
      activeEndDate = _state$activeRange[1];

  // no new value, button still disabled


  if (!hasNewSelectedValue(props, state)) {
    return true;
  }

  // new value with no boundaries, button enabled
  if (!rangeEndBoundary || !rangeStartBoundary) {
    return false;
  }

  // new value with boundaries provided, check that date falls in range
  return !areDatesInRange({
    end: activeEndDate.clone().set(activeEndTime),
    rangeEndBoundary: rangeEndBoundary,
    rangeStartBoundary: rangeStartBoundary,
    start: activeStartDate.clone().set(activeStartTime)
  });
}

// As date range selection changes, we need to update the boundaries that we send to each
// TimePicker for validating dates *and* times are in range
function buildTimeBoundaries(_ref5) {
  var startDate = _ref5.startDate,
      endDate = _ref5.endDate,
      rangeStartBoundary = _ref5.rangeStartBoundary,
      rangeEndBoundary = _ref5.rangeEndBoundary;

  if (!rangeStartBoundary || !rangeEndBoundary) {
    return {};
  }

  // init with defaults: all hours of day
  var boundaries = {
    endTimeBoundaryEnd: (0, _CalendarUtils.timeObjFromMoment)((0, _moment2.default)().endOf('day')),
    endTimeBoundaryStart: (0, _CalendarUtils.timeObjFromMoment)((0, _moment2.default)().startOf('day')),
    startTimeBoundaryEnd: (0, _CalendarUtils.timeObjFromMoment)((0, _moment2.default)().endOf('day')),
    startTimeBoundaryStart: (0, _CalendarUtils.timeObjFromMoment)((0, _moment2.default)().startOf('day'))
  };

  if (startDate.isSame(rangeStartBoundary, 'day')) {
    boundaries.startTimeBoundaryStart = (0, _CalendarUtils.timeObjFromMoment)(rangeStartBoundary);
  }
  if (startDate.isSame(rangeEndBoundary, 'day')) {
    boundaries.startTimeBoundaryEnd = (0, _CalendarUtils.timeObjFromMoment)(rangeEndBoundary);
  }
  if (endDate.isSame(rangeStartBoundary, 'day')) {
    boundaries.endTimeBoundaryStart = (0, _CalendarUtils.timeObjFromMoment)(rangeStartBoundary);
  }
  if (endDate.isSame(rangeEndBoundary, 'day')) {
    boundaries.endTimeBoundaryEnd = (0, _CalendarUtils.timeObjFromMoment)(rangeEndBoundary);
  }
  return boundaries;
}

var TxlDateRangeSelector = (0, _radium2.default)(_class = function (_React$Component) {
  (0, _inherits3.default)(TxlDateRangeSelector, _React$Component);

  function TxlDateRangeSelector() {
    var _ref6;

    (0, _classCallCheck3.default)(this, TxlDateRangeSelector);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref6 = TxlDateRangeSelector.__proto__ || (0, _getPrototypeOf2.default)(TxlDateRangeSelector)).call.apply(_ref6, [this].concat(args)));

    _this._handleDateSelect = function () {
      return _this.___handleDateSelect__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleTimeZoneSelect = function () {
      return _this.___handleTimeZoneSelect__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleApplyClick = function () {
      return _this.___handleApplyClick__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleDateChange = function () {
      return _this.___handleDateChange__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._toggleDropdown = function () {
      return _this.___toggleDropdown__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._closeDropdown = function () {
      return _this.___closeDropdown__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._openDropdown = function () {
      return _this.___openDropdown__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleEndTimeClose = function () {
      return _this.___handleEndTimeClose__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleEndTimeOpen = function () {
      return _this.___handleEndTimeOpen__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleEndTimeRef = function () {
      return _this.___handleEndTimeRef__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleEndTimeSelect = function () {
      return _this.___handleEndTimeSelect__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleStartTimeClose = function () {
      return _this.___handleStartTimeClose__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleStartTimeOpen = function () {
      return _this.___handleStartTimeOpen__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleStartTimeRef = function () {
      return _this.___handleStartTimeRef__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleStartTimeSelect = function () {
      return _this.___handleStartTimeSelect__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    var _this$props = _this.props,
        endTime = _this$props.endTime,
        presetKey = _this$props.presetKey,
        rangeEndBoundary = _this$props.rangeEndBoundary,
        rangeStartBoundary = _this$props.rangeStartBoundary,
        startTime = _this$props.startTime,
        timeZone = _this$props.timeZone,
        value = _this$props.value,
        _this$props$value = (0, _slicedToArray3.default)(_this$props.value, 2),
        startDate = _this$props$value[0],
        endDate = _this$props$value[1];

    // maintain 'active' date within component for updating Calendars when not auto-updating.


    _this.state = {
      activeEndTime: endTime,
      activePreset: presetKey,
      activeRange: value.map(_CalendarUtils.dateStringToMoment),
      activeStartTime: startTime,
      activeTimeZone: timeZone,
      endTimeOpen: false,
      startTimeOpen: false,
      timeBoundaries: buildTimeBoundaries({
        endDate: endDate,
        rangeEndBoundary: rangeEndBoundary,
        rangeStartBoundary: rangeStartBoundary,
        startDate: startDate
      })
    };
    return _this;
  }

  (0, _createClass3.default)(TxlDateRangeSelector, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      this.setState({
        activeEndTime: nextProps.endTime,
        activePreset: nextProps.presetKey,
        activeRange: nextProps.value.map(_CalendarUtils.dateStringToMoment),
        activeStartTime: nextProps.startTime,
        activeTimeZone: nextProps.timeZone
      });
    }
  }, {
    key: '___handleDateSelect__REACT_HOT_LOADER__',
    value: function ___handleDateSelect__REACT_HOT_LOADER__() {
      return this.___handleDateSelect__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleDateSelect__REACT_HOT_LOADER__',
    value: function ___handleDateSelect__REACT_HOT_LOADER__() {
      return this.___handleDateSelect__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleDateSelect__REACT_HOT_LOADER__',
    value: function ___handleDateSelect__REACT_HOT_LOADER__(_ref7) {
      var _this2 = this;

      var presetKey = _ref7.presetKey,
          startDate = _ref7.startDate,
          endDate = _ref7.endDate;

      var newDateState = {
        activePreset: presetKey,
        activeRange: [startDate, endDate]
      };
      if (presetKey !== 'custom') {
        newDateState = (0, _extends3.default)({}, newDateState, {
          activeEndTime: (0, _CalendarUtils.timeObjFromMoment)(endDate),
          activeStartTime: (0, _CalendarUtils.timeObjFromMoment)(startDate)
        });
      }
      this.setState(newDateState, function () {
        if (_this2.props.mode === 'simple') {
          // apply date immediately
          _this2._handleDateChange({ endDate: endDate, presetKey: presetKey, startDate: startDate });
        }
      });
    }
  }, {
    key: '___handleTimeZoneSelect__REACT_HOT_LOADER__',
    value: function ___handleTimeZoneSelect__REACT_HOT_LOADER__() {
      return this.___handleTimeZoneSelect__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleTimeZoneSelect__REACT_HOT_LOADER__',
    value: function ___handleTimeZoneSelect__REACT_HOT_LOADER__() {
      return this.___handleTimeZoneSelect__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleTimeZoneSelect__REACT_HOT_LOADER__',
    value: function ___handleTimeZoneSelect__REACT_HOT_LOADER__(_ref8) {
      var value = _ref8.value;
      return this.setState({ activeTimeZone: value });
    }
  }, {
    key: '___handleApplyClick__REACT_HOT_LOADER__',
    value: function ___handleApplyClick__REACT_HOT_LOADER__() {
      return this.___handleApplyClick__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleApplyClick__REACT_HOT_LOADER__',
    value: function ___handleApplyClick__REACT_HOT_LOADER__() {
      return this.___handleApplyClick__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleApplyClick__REACT_HOT_LOADER__',
    value: function ___handleApplyClick__REACT_HOT_LOADER__() {
      var _state = this.state,
          endTime = _state.activeEndTime,
          presetKey = _state.activePreset,
          _state$activeRange2 = (0, _slicedToArray3.default)(_state.activeRange, 2),
          startDate = _state$activeRange2[0],
          endDate = _state$activeRange2[1],
          startTime = _state.activeStartTime,
          timeZone = _state.activeTimeZone;

      this._handleDateChange({
        endDate: endDate,
        endTime: endTime,
        presetKey: presetKey,
        startDate: startDate,
        startTime: startTime,
        timeZone: timeZone
      });
    }
  }, {
    key: '___handleDateChange__REACT_HOT_LOADER__',
    value: function ___handleDateChange__REACT_HOT_LOADER__() {
      return this.___handleDateChange__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleDateChange__REACT_HOT_LOADER__',
    value: function ___handleDateChange__REACT_HOT_LOADER__() {
      return this.___handleDateChange__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleDateChange__REACT_HOT_LOADER__',
    value: function ___handleDateChange__REACT_HOT_LOADER__(_ref9) {
      var presetKey = _ref9.presetKey,
          startDate = _ref9.startDate,
          endDate = _ref9.endDate,
          startTime = _ref9.startTime,
          endTime = _ref9.endTime,
          timeZone = _ref9.timeZone;
      var _props = this.props,
          onChange = _props.onChange,
          pattern = _props.pattern;


      onChange({
        endTime: endTime,
        presetKey: presetKey,
        startTime: startTime,
        timeZone: timeZone,
        value: [startDate.format(pattern), endDate.format(pattern)]
      });
      this._closeDropdown();
    }
  }, {
    key: '___toggleDropdown__REACT_HOT_LOADER__',
    value: function ___toggleDropdown__REACT_HOT_LOADER__() {
      return this.___toggleDropdown__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___toggleDropdown__REACT_HOT_LOADER__',
    value: function ___toggleDropdown__REACT_HOT_LOADER__() {
      return this.___toggleDropdown__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___toggleDropdown__REACT_HOT_LOADER__',
    value: function ___toggleDropdown__REACT_HOT_LOADER__() {
      return this.props.updateDropdown({ open: !this.props.open });
    }
  }, {
    key: '___closeDropdown__REACT_HOT_LOADER__',
    value: function ___closeDropdown__REACT_HOT_LOADER__() {
      return this.___closeDropdown__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___closeDropdown__REACT_HOT_LOADER__',
    value: function ___closeDropdown__REACT_HOT_LOADER__() {
      return this.___closeDropdown__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___closeDropdown__REACT_HOT_LOADER__',
    value: function ___closeDropdown__REACT_HOT_LOADER__() {
      return this.props.updateDropdown({ open: false });
    }
  }, {
    key: '___openDropdown__REACT_HOT_LOADER__',
    value: function ___openDropdown__REACT_HOT_LOADER__() {
      return this.___openDropdown__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___openDropdown__REACT_HOT_LOADER__',
    value: function ___openDropdown__REACT_HOT_LOADER__() {
      return this.___openDropdown__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___openDropdown__REACT_HOT_LOADER__',
    value: function ___openDropdown__REACT_HOT_LOADER__() {
      return this.props.updateDropdown({ open: true });
    }
  }, {
    key: '___handleEndTimeClose__REACT_HOT_LOADER__',
    value: function ___handleEndTimeClose__REACT_HOT_LOADER__() {
      return this.___handleEndTimeClose__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleEndTimeClose__REACT_HOT_LOADER__',
    value: function ___handleEndTimeClose__REACT_HOT_LOADER__() {
      return this.___handleEndTimeClose__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleEndTimeClose__REACT_HOT_LOADER__',
    value: function ___handleEndTimeClose__REACT_HOT_LOADER__() {
      return this.setState({ endTimeOpen: false });
    }
  }, {
    key: '___handleEndTimeOpen__REACT_HOT_LOADER__',
    value: function ___handleEndTimeOpen__REACT_HOT_LOADER__() {
      return this.___handleEndTimeOpen__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleEndTimeOpen__REACT_HOT_LOADER__',
    value: function ___handleEndTimeOpen__REACT_HOT_LOADER__() {
      return this.___handleEndTimeOpen__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleEndTimeOpen__REACT_HOT_LOADER__',
    value: function ___handleEndTimeOpen__REACT_HOT_LOADER__() {
      return this.setState({
        endTimeOpen: true,
        startTimeOpen: false
      });
    }
  }, {
    key: '___handleEndTimeRef__REACT_HOT_LOADER__',
    value: function ___handleEndTimeRef__REACT_HOT_LOADER__() {
      return this.___handleEndTimeRef__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleEndTimeRef__REACT_HOT_LOADER__',
    value: function ___handleEndTimeRef__REACT_HOT_LOADER__() {
      return this.___handleEndTimeRef__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleEndTimeRef__REACT_HOT_LOADER__',
    value: function ___handleEndTimeRef__REACT_HOT_LOADER__(ref) {
      this._endTimePicker = ref;
    }
  }, {
    key: '___handleEndTimeSelect__REACT_HOT_LOADER__',
    value: function ___handleEndTimeSelect__REACT_HOT_LOADER__() {
      return this.___handleEndTimeSelect__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleEndTimeSelect__REACT_HOT_LOADER__',
    value: function ___handleEndTimeSelect__REACT_HOT_LOADER__() {
      return this.___handleEndTimeSelect__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleEndTimeSelect__REACT_HOT_LOADER__',
    value: function ___handleEndTimeSelect__REACT_HOT_LOADER__(item) {
      var time = _getTimeFromItemSelect(item);
      time && this.setState({ activeEndTime: time });
    }
  }, {
    key: '___handleStartTimeClose__REACT_HOT_LOADER__',
    value: function ___handleStartTimeClose__REACT_HOT_LOADER__() {
      return this.___handleStartTimeClose__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleStartTimeClose__REACT_HOT_LOADER__',
    value: function ___handleStartTimeClose__REACT_HOT_LOADER__() {
      return this.___handleStartTimeClose__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleStartTimeClose__REACT_HOT_LOADER__',
    value: function ___handleStartTimeClose__REACT_HOT_LOADER__() {
      return this.setState({ startTimeOpen: false });
    }
  }, {
    key: '___handleStartTimeOpen__REACT_HOT_LOADER__',
    value: function ___handleStartTimeOpen__REACT_HOT_LOADER__() {
      return this.___handleStartTimeOpen__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleStartTimeOpen__REACT_HOT_LOADER__',
    value: function ___handleStartTimeOpen__REACT_HOT_LOADER__() {
      return this.___handleStartTimeOpen__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleStartTimeOpen__REACT_HOT_LOADER__',
    value: function ___handleStartTimeOpen__REACT_HOT_LOADER__() {
      return this.setState({
        endTimeOpen: false,
        startTimeOpen: true
      });
    }
  }, {
    key: '___handleStartTimeRef__REACT_HOT_LOADER__',
    value: function ___handleStartTimeRef__REACT_HOT_LOADER__() {
      return this.___handleStartTimeRef__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleStartTimeRef__REACT_HOT_LOADER__',
    value: function ___handleStartTimeRef__REACT_HOT_LOADER__() {
      return this.___handleStartTimeRef__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleStartTimeRef__REACT_HOT_LOADER__',
    value: function ___handleStartTimeRef__REACT_HOT_LOADER__(ref) {
      this._startTimePicker = ref;
    }
  }, {
    key: '___handleStartTimeSelect__REACT_HOT_LOADER__',
    value: function ___handleStartTimeSelect__REACT_HOT_LOADER__() {
      return this.___handleStartTimeSelect__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleStartTimeSelect__REACT_HOT_LOADER__',
    value: function ___handleStartTimeSelect__REACT_HOT_LOADER__() {
      return this.___handleStartTimeSelect__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleStartTimeSelect__REACT_HOT_LOADER__',
    value: function ___handleStartTimeSelect__REACT_HOT_LOADER__(item) {
      var time = _getTimeFromItemSelect(item);
      time && this.setState({ activeStartTime: time });
    }
  }, {
    key: 'render',
    value: function render() {
      var props = this.props,
          state = this.state;
      var displayPattern = props.displayPattern,
          footerContent = props.footerContent,
          mode = props.mode,
          open = props.open,
          outOfRangeTooltip = props.outOfRangeTooltip,
          rangeEndBoundary = props.rangeEndBoundary,
          rangeStartBoundary = props.rangeStartBoundary,
          secondaryText = props.secondaryText,
          value = props.value,
          onChange = props.onChange;

      var activeEndTime = state.activeEndTime,
          activePreset = state.activePreset,
          activeRange = state.activeRange,
          _state$activeRange3 = (0, _slicedToArray3.default)(state.activeRange, 2),
          activeStartDate = _state$activeRange3[0],
          activeEndDate = _state$activeRange3[1],
          activeStartTime = state.activeStartTime,
          activeTimeZone = state.activeTimeZone,
          endTimeOpen = state.endTimeOpen,
          startTimeOpen = state.startTimeOpen;

      // TimePicker boundaries have to update with range selection


      var _buildTimeBoundaries = buildTimeBoundaries({
        endDate: activeEndDate,
        rangeEndBoundary: rangeEndBoundary,
        rangeStartBoundary: rangeStartBoundary,
        startDate: activeStartDate
      }),
          endTimeBoundaryEnd = _buildTimeBoundaries.endTimeBoundaryEnd,
          endTimeBoundaryStart = _buildTimeBoundaries.endTimeBoundaryStart,
          startTimeBoundaryEnd = _buildTimeBoundaries.startTimeBoundaryEnd,
          startTimeBoundaryStart = _buildTimeBoundaries.startTimeBoundaryStart;

      var displayValue = buildDisplayValueFromDates(value[0], value[1], displayPattern);
      var showTimeAndFooter = mode !== 'simple';
      var startTimeTextValue = _formatTime(activeStartTime);
      var endTimeTextValue = _formatTime(activeEndTime);

      // only enable button when new date has been selected and values are in range
      var applyButtonIsDisabled = isApplyButtonDisabled(props, state);

      var dropdownIcon = _react2.default.createElement(_IconButton2.default, {
        icon: _CalendarOutline2.default,
        onClick: this._toggleDropdown,
        variant: 'plain'
      });

      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlDateRangeSelector',
          style: _DateRangeSelector.WRAPPER_STYLE
        },
        _react2.default.createElement(_TextField2.default, {
          _afterContent: dropdownIcon
          // onClick/onChange aren't explicit TextField props;
          // they're passed to input elem via 'otherProps'
          , onChange: onChange,
          onClick: this._openDropdown,
          readOnly: true,
          ref: 'textField',
          secondaryText: secondaryText,
          type: 'text',
          value: displayValue
        }),
        _react2.default.createElement(_InputUnderline2.default, { validationState: 'base' }),
        _react2.default.createElement(
          _DropdownContent2.default,
          {
            style: _DateRangeSelector.DROPDOWN_CONTAINER_STYLE,
            onMouseDown: this._disableBlur,
            onClick: this._enableBlur,
            open: open
          },
          _react2.default.createElement(_CalendarPickerWithPresets2.default, {
            onDateSelect: this._handleDateSelect,
            outOfRangeTooltip: outOfRangeTooltip,
            presetKey: activePreset,
            range: activeRange,
            rangeEndBoundary: rangeEndBoundary,
            rangeStartBoundary: rangeStartBoundary,
            ref: 'calPicker'
          }),
          showTimeAndFooter && _react2.default.createElement(
            'section',
            null,
            _react2.default.createElement(
              'div',
              { style: _DateRangeSelector.TIME_WRAPPER_STYLE },
              _react2.default.createElement(
                'div',
                { style: _DateRangeSelector.TIME_ZONE_WRAPPER_STYLE },
                _react2.default.createElement(_TimeZonePicker2.default, {
                  label: TIME_ZONE_PICKER_LABEL,
                  onItemSelect: this._handleTimeZoneSelect,
                  ref: 'timeZonePicker',
                  value: activeTimeZone
                })
              ),
              _react2.default.createElement(
                'div',
                null,
                _react2.default.createElement(
                  'div',
                  { style: _DateRangeSelector.LABEL_STYLE },
                  TIME_RANGE_PICKER_LABEL
                ),
                _react2.default.createElement(
                  'div',
                  { style: _DateRangeSelector.TIME_RANGE_WRAPPER_STYLE },
                  _react2.default.createElement(
                    'div',
                    { style: _DateRangeSelector.TIME_PICKER_WRAPPER_STYLE },
                    _react2.default.createElement(_TimePickerControlled2.default, {
                      boundaryEnd: startTimeBoundaryEnd,
                      boundaryStart: startTimeBoundaryStart,
                      closeDropdown: this._handleStartTimeClose,
                      onItemSelect: this._handleStartTimeSelect,
                      open: startTimeOpen,
                      openDropdown: this._handleStartTimeOpen,
                      pattern: TIME_PATTERN,
                      ref: this._handleStartTimeRef,
                      secondaryText: buildTimeRangeSecondaryText(activeStartDate),
                      textValue: startTimeTextValue,
                      time: activeStartTime
                    })
                  ),
                  _react2.default.createElement(
                    'div',
                    { style: _DateRangeSelector.INPUT_SEPARATOR_STYLE },
                    TIME_DELIMITER_TEXT
                  ),
                  _react2.default.createElement(
                    'div',
                    { style: _DateRangeSelector.TIME_PICKER_WRAPPER_STYLE },
                    _react2.default.createElement(_TimePickerControlled2.default, {
                      boundaryEnd: endTimeBoundaryEnd,
                      boundaryStart: endTimeBoundaryStart,
                      closeDropdown: this._handleEndTimeClose,
                      onItemSelect: this._handleEndTimeSelect,
                      open: endTimeOpen,
                      openDropdown: this._handleEndTimeOpen,
                      pattern: TIME_PATTERN,
                      ref: this._handleEndTimeRef,
                      secondaryText: buildTimeRangeSecondaryText(activeEndDate),
                      textValue: endTimeTextValue,
                      time: activeEndTime
                    })
                  )
                )
              )
            ),
            _react2.default.createElement(
              'footer',
              { style: _DateRangeSelector.FOOTER_STYLE },
              _react2.default.createElement(
                'div',
                { style: _DateRangeSelector.BUTTON_GROUP_STYLE },
                _react2.default.createElement(
                  _Button2.default,
                  {
                    _style: _DateRangeSelector.BUTTON_STYLE,
                    disabled: applyButtonIsDisabled,
                    onClick: this._handleApplyClick,
                    ref: 'applyButton',
                    size: 'large',
                    variant: 'accent'
                  },
                  APPLY_BUTTON_TEXT
                ),
                _react2.default.createElement(
                  _Button2.default,
                  {
                    _style: _DateRangeSelector.BUTTON_STYLE,
                    onClick: this._closeDropdown,
                    ref: 'cancelButton',
                    size: 'large',
                    variant: 'muted'
                  },
                  CANCEL_BUTTON_TEXT
                )
              ),
              footerContent && _react2.default.createElement(
                'div',
                { style: _DateRangeSelector.FOOTER_CONTENT_STYLE },
                footerContent
              )
            )
          )
        )
      );
    }
  }]);
  return TxlDateRangeSelector;
}(_react2.default.Component)) || _class;

exports.default = TxlDateRangeSelector;


TxlDateRangeSelector.propTypes = {
  displayPattern: _react.PropTypes.string,
  endTime: _TimePickerControlled2.default.propTypes.time,
  footerContent: _react.PropTypes.node,
  mode: _react.PropTypes.oneOf(['simple', 'full']),
  onChange: _react.PropTypes.func.isRequired,
  open: _react.PropTypes.bool,
  outOfRangeTooltip: _CalendarPickerWithPresets2.default.propTypes.outOfRangeTooltip,
  pattern: _react.PropTypes.string,
  presetKey: _CalendarPickerWithPresets2.default.propTypes.presetKey,
  rangeEndBoundary: _CalendarPickerWithPresets2.default.propTypes.rangeEndBoundary,
  rangeStartBoundary: _CalendarPickerWithPresets2.default.propTypes.rangeStartBoundary,
  secondaryText: _react.PropTypes.node,
  startTime: _TimePickerControlled2.default.propTypes.time,
  timeZone: _TimeZonePicker2.default.propTypes.value,
  updateDropdown: _react.PropTypes.func,
  value: _propTypes2.default.dateRange
};

TxlDateRangeSelector.defaultProps = {
  displayPattern: CONDENSED_DATE_FORMAT,
  endTime: (0, _CalendarUtils.timeObjFromMoment)(todayPreset.endDate),
  mode: 'full',
  open: false,
  pattern: DEFAULT_DATE_INPUT_PATTERN,
  presetKey: 'today',
  startTime: (0, _CalendarUtils.timeObjFromMoment)(todayPreset.startDate),
  timeZone: _moment2.default.tz.guess(),
  updateDropdown: _Base.NOOP,
  value: [todayPreset.startDate, todayPreset.endDate]
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(todayPreset, 'todayPreset', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(DEFAULT_DATE_INPUT_PATTERN, 'DEFAULT_DATE_INPUT_PATTERN', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(CONDENSED_DATE_FORMAT, 'CONDENSED_DATE_FORMAT', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(TIME_PATTERN, 'TIME_PATTERN', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(APPLY_BUTTON_TEXT, 'APPLY_BUTTON_TEXT', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(CANCEL_BUTTON_TEXT, 'CANCEL_BUTTON_TEXT', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(TIME_RANGE_PICKER_LABEL, 'TIME_RANGE_PICKER_LABEL', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(TIME_ZONE_PICKER_LABEL, 'TIME_ZONE_PICKER_LABEL', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(TIME_DELIMITER_TEXT, 'TIME_DELIMITER_TEXT', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(_mapStringToTime, '_mapStringToTime', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(_getTimeFromItemSelect, '_getTimeFromItemSelect', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(_formatTime, '_formatTime', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(hasDateRangeChanged, 'hasDateRangeChanged', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(hasTimeRangeChanged, 'hasTimeRangeChanged', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(hasNewSelectedValue, 'hasNewSelectedValue', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(buildDisplayValueFromDates, 'buildDisplayValueFromDates', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(buildTimeRangeSecondaryText, 'buildTimeRangeSecondaryText', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(areDatesInRange, 'areDatesInRange', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(isApplyButtonDisabled, 'isApplyButtonDisabled', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(buildTimeBoundaries, 'buildTimeBoundaries', 'src/input-fields/_ControlledDateRangeSelector.jsx');

  __REACT_HOT_LOADER__.register(TxlDateRangeSelector, 'TxlDateRangeSelector', 'src/input-fields/_ControlledDateRangeSelector.jsx');
}();

;
module.exports = exports['default'];
