// txl/input-fields/_ControlledDatePicker.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _Dropdown = require('../dropdowns/_Dropdown.style');

var _DatePicker = require('./DatePicker.style');

var _Base = require('../base/Base');

var _SingleCalendarPicker = require('../calendar-picker/_SingleCalendarPicker');

var _SingleCalendarPicker2 = _interopRequireDefault(_SingleCalendarPicker);

var _DropdownContentWithArrow = require('../dropdowns/_DropdownContentWithArrow');

var _DropdownContentWithArrow2 = _interopRequireDefault(_DropdownContentWithArrow);

var _IconButton = require('../buttons/IconButton');

var _IconButton2 = _interopRequireDefault(_IconButton);

var _CalendarOutline = require('../icons/CalendarOutline');

var _CalendarOutline2 = _interopRequireDefault(_CalendarOutline);

var _TextField = require('./TextField');

var _TextField2 = _interopRequireDefault(_TextField);

var _CalendarUtils = require('../calendar-picker/_CalendarUtils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var DEFAULT_DATE_PATTERN = 'MM/DD/YYYY';
var DEFAULT_ERROR_TEXT_PREFIX = 'Invalid date:';
var NOT_WITHIN_RANGE_ERROR_TEXT = 'Date is not within valid range';

var TxlDatePicker = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlDatePicker, _React$Component);

  function TxlDatePicker() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlDatePicker);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlDatePicker.__proto__ || (0, _getPrototypeOf2.default)(TxlDatePicker)).call.apply(_ref, [this].concat(args)));

    _this._isBlurDisabled = false;
    return _this;
  }

  (0, _createClass3.default)(TxlDatePicker, [{
    key: '_handleBlur',
    value: function _handleBlur() {
      if (!this._isBlurDisabled) {
        this.props.updateDropdown({ open: false });
      }
      this.props.onBlur({});
      this._isBlurDisabled = false;
    }
  }, {
    key: '_handleFocus',
    value: function _handleFocus() {
      this.props.updateDropdown({ open: true });
      this.props.onFocus({});
    }
  }, {
    key: '_handleCalendarClick',
    value: function _handleCalendarClick() {
      this.props.updateDropdown({ open: !this.props.open });
      this.props.onFocus({});
    }
  }, {
    key: '_handleDateClick',
    value: function _handleDateClick(_ref2) {
      var date = _ref2.date;
      var pattern = this.props.pattern;

      this.props.updateDropdown({ open: false });
      this.props.onChange({ value: date.format(pattern) });
      this._isBlurDisabled = false;
    }
  }, {
    key: '_disableBlur',
    value: function _disableBlur() {
      this._isBlurDisabled = true;
    }
  }, {
    key: '_getValidationText',
    value: function _getValidationText(isWithinValidRange) {
      var _props = this.props,
          pattern = _props.pattern,
          validationText = _props.validationText;


      if (validationText) {
        return validationText;
      }
      // TODO Make this component no longer control its own validation text
      if (!isWithinValidRange) {
        return NOT_WITHIN_RANGE_ERROR_TEXT;
      }

      return DEFAULT_ERROR_TEXT_PREFIX + ' ' + pattern;
    }
  }, {
    key: 'render',
    value: function render() {
      var _props2 = this.props,
          isValid = _props2.isValid,
          label = _props2.label,
          onChange = _props2.onChange,
          open = _props2.open,
          pattern = _props2.pattern,
          rangeEndBoundary = _props2.rangeEndBoundary,
          rangeStartBoundary = _props2.rangeStartBoundary,
          secondaryText = _props2.secondaryText,
          value = _props2.value;

      var date = (0, _moment2.default)(value, pattern, true);
      var isValidDate = isValid !== undefined ? isValid : date.isValid();

      var rangeIsDefined = rangeEndBoundary || rangeStartBoundary;
      var isWithinValidRange = rangeIsDefined ? (0, _CalendarUtils.isDateWithinRange)(date, rangeStartBoundary, rangeEndBoundary) : true;

      var validationState = isValidDate && isWithinValidRange ? 'base' : 'error';
      // TODO Make this component not control its own validation text
      var secondaryTextVal = isValidDate && isWithinValidRange ? secondaryText : this._getValidationText(isWithinValidRange);

      var dropdownIcon = _react2.default.createElement(_IconButton2.default, {
        icon: _CalendarOutline2.default,
        onClick: this._handleCalendarClick,
        size: 'large',
        variant: 'plain'
      });

      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlDatePicker',
          style: _Dropdown.WRAPPER_STYLE
        },
        _react2.default.createElement(_TextField2.default, {
          _afterContent: dropdownIcon,
          label: label,
          onBlur: this._handleBlur,
          onChange: onChange,
          onFocus: this._handleFocus,
          ref: 'dateTextField',
          secondaryText: secondaryTextVal,
          type: 'text',
          validationState: validationState,
          value: value
        }),
        _react2.default.createElement(
          _DropdownContentWithArrow2.default,
          {
            style: _DatePicker.DROPDOWN_CONTAINER_STYLE,
            contentStyle: _DatePicker.DROPDOWN_CONTENT_STYLE,
            onMouseEnter: this._disableBlur,
            open: open
          },
          _react2.default.createElement(_SingleCalendarPicker2.default, {
            date: isValidDate ? date : (0, _moment2.default)(),
            rangeEndBoundary: rangeEndBoundary,
            rangeStartBoundary: rangeStartBoundary,
            onDateClick: this._handleDateClick
          })
        )
      );
    }
  }]);
  return TxlDatePicker;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleBlur', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleBlur'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleFocus', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleFocus'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleCalendarClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleCalendarClick'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDateClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDateClick'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_disableBlur', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_disableBlur'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_getValidationText', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_getValidationText'), _class2.prototype)), _class2)) || _class;

exports.default = TxlDatePicker;


TxlDatePicker.propTypes = {
  isValid: _react.PropTypes.bool,
  label: _react.PropTypes.string,
  onBlur: _react.PropTypes.func,
  onChange: _react.PropTypes.func,
  onFocus: _react.PropTypes.func,
  open: _react.PropTypes.bool,
  pattern: _react.PropTypes.string,
  rangeEndBoundary: _SingleCalendarPicker2.default.propTypes.rangeEndBoundary,
  rangeStartBoundary: _SingleCalendarPicker2.default.propTypes.rangeStartBoundary,
  secondaryText: _react.PropTypes.node,
  updateDropdown: _react.PropTypes.func,
  validationText: _react.PropTypes.string,
  value: _react.PropTypes.string
};

TxlDatePicker.defaultProps = {
  onBlur: _Base.NOOP,
  onChange: _Base.NOOP,
  onFocus: _Base.NOOP,
  open: false,
  pattern: DEFAULT_DATE_PATTERN,
  updateDropdown: _Base.NOOP,
  value: (0, _moment2.default)().format(DEFAULT_DATE_PATTERN)
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(DEFAULT_DATE_PATTERN, 'DEFAULT_DATE_PATTERN', 'src/input-fields/_ControlledDatePicker.jsx');

  __REACT_HOT_LOADER__.register(DEFAULT_ERROR_TEXT_PREFIX, 'DEFAULT_ERROR_TEXT_PREFIX', 'src/input-fields/_ControlledDatePicker.jsx');

  __REACT_HOT_LOADER__.register(NOT_WITHIN_RANGE_ERROR_TEXT, 'NOT_WITHIN_RANGE_ERROR_TEXT', 'src/input-fields/_ControlledDatePicker.jsx');

  __REACT_HOT_LOADER__.register(TxlDatePicker, 'TxlDatePicker', 'src/input-fields/_ControlledDatePicker.jsx');
}();

;
module.exports = exports['default'];
