// txl/input-fields/_InputFieldBase.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BEFORE_CONTENT_LEFT_MARGIN = undefined;

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _InputUnderline = require('../layout/_InputUnderline');

var _InputUnderline2 = _interopRequireDefault(_InputUnderline);

var _FlexContainer = require('../layout/FlexContainer');

var _FlexContainer2 = _interopRequireDefault(_FlexContainer);

var _SecondaryText = require('./_SecondaryText');

var _SecondaryText2 = _interopRequireDefault(_SecondaryText);

var _InputFieldsPropTypes = require('./_InputFieldsPropTypes');

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

var _type = require('../styles/mixins/type');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BEFORE_CONTENT_LEFT_MARGIN = exports.BEFORE_CONTENT_LEFT_MARGIN = (0, _helpers.gridUnits)(2);

var LABEL_STYLE = (0, _extends3.default)({}, _type.LABEL, _helpers.TRUNCATE, _type.TYPE_SMALL, {
  color: _theme.COLOR_NEUTRAL['400'],
  minHeight: (0, _helpers.gridUnits)(4)
});

var LABEL_ASTERISK_STYLE = {
  color: _theme.COLOR_ERROR['400']
};

var FIELD_CONTAINER_STYLE = {
  lineHeight: 0,
  minHeight: (0, _helpers.gridUnits)(7),
  position: 'relative'
};

var BEFORE_CONTAINER_STYLE = {
  marginLeft: BEFORE_CONTENT_LEFT_MARGIN,
  marginRight: (0, _helpers.gridUnits)(2)
};

var AFTER_CONTAINER_STYLE = {
  marginLeft: (0, _helpers.gridUnits)(2)
};

function renderExtraContent(content, style) {
  if (!content) {
    return null;
  }

  if (_react2.default.isValidElement(content)) {
    return _react2.default.createElement(
      'div',
      { style: style },
      content
    );
  }

  return content({ style: style });
}
/**
 * Purpose of the TxlInputFieldBase is to provide Sensible Base for most of
 * Txl input fields which often involves with label, underline, and secondaryText
 * +---------------------+
 * | InputFieldBase      |
 * | +-----------------+ |
 * | | label           | |
 * | +-----------------+ |
 * | +-----------------+ |
 * | | children        | |
 * | +-----------------+ |
 * | +-----------------+ |
 * | | InputUnderline  | |
 * | +-----------------+ |
 * | +-----------------+ |
 * | | SecondaryText   | |
 * | +-----------------+ |
 * +---------------------+
 *
 * **This component is controlled component.
 */
var TxlInputFieldBase = function TxlInputFieldBase(props) {
  var _afterContent = props._afterContent,
      _beforeContent = props._beforeContent,
      _containerProps = props._containerProps,
      children = props.children,
      disabled = props.disabled,
      focused = props.focused,
      label = props.label,
      readOnly = props.readOnly,
      required = props.required,
      secondaryText = props.secondaryText,
      validationState = props.validationState,
      others = (0, _objectWithoutProperties3.default)(props, ['_afterContent', '_beforeContent', '_containerProps', 'children', 'disabled', 'focused', 'label', 'readOnly', 'required', 'secondaryText', 'validationState']);


  return _react2.default.createElement(
    'div',
    (0, _extends3.default)({}, others, {
      'data-txl-input-field': true
    }),
    !!label && _react2.default.createElement(
      'div',
      { style: LABEL_STYLE },
      label,
      required && _react2.default.createElement(
        'span',
        { style: LABEL_ASTERISK_STYLE },
        ' *'
      )
    ),
    _react2.default.createElement(
      _FlexContainer2.default,
      (0, _extends3.default)({
        alignItems: 'center',
        flexWrap: 'nowrap',
        style: FIELD_CONTAINER_STYLE
      }, _containerProps),
      renderExtraContent(_beforeContent, BEFORE_CONTAINER_STYLE),
      children,
      renderExtraContent(_afterContent, AFTER_CONTAINER_STYLE)
    ),
    _react2.default.createElement(_InputUnderline2.default, {
      disabled: disabled,
      focused: focused,
      readOnly: readOnly,
      validationState: validationState
    }),
    secondaryText && _react2.default.createElement(
      _SecondaryText2.default,
      { validationState: validationState },
      secondaryText
    )
  );
};

TxlInputFieldBase.propTypes = {
  _afterContent: _react.PropTypes.oneOfType([_react.PropTypes.node, _react.PropTypes.func]),
  _beforeContent: _react.PropTypes.oneOfType([_react.PropTypes.node, _react.PropTypes.func]),
  _containerProps: _react.PropTypes.object,
  children: _react.PropTypes.node.isRequired,
  disabled: _react.PropTypes.bool,
  focused: _react.PropTypes.bool,
  label: _react.PropTypes.node,
  readOnly: _react.PropTypes.bool,
  required: _react.PropTypes.bool,
  secondaryText: _react.PropTypes.node,
  validationState: _InputFieldsPropTypes.validationStateType
};

TxlInputFieldBase.defaultProps = {
  _containerProps: {}
};

var _default = (0, _radium2.default)(TxlInputFieldBase);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(BEFORE_CONTENT_LEFT_MARGIN, 'BEFORE_CONTENT_LEFT_MARGIN', 'src/input-fields/_InputFieldBase.jsx');

  __REACT_HOT_LOADER__.register(LABEL_STYLE, 'LABEL_STYLE', 'src/input-fields/_InputFieldBase.jsx');

  __REACT_HOT_LOADER__.register(LABEL_ASTERISK_STYLE, 'LABEL_ASTERISK_STYLE', 'src/input-fields/_InputFieldBase.jsx');

  __REACT_HOT_LOADER__.register(FIELD_CONTAINER_STYLE, 'FIELD_CONTAINER_STYLE', 'src/input-fields/_InputFieldBase.jsx');

  __REACT_HOT_LOADER__.register(BEFORE_CONTAINER_STYLE, 'BEFORE_CONTAINER_STYLE', 'src/input-fields/_InputFieldBase.jsx');

  __REACT_HOT_LOADER__.register(AFTER_CONTAINER_STYLE, 'AFTER_CONTAINER_STYLE', 'src/input-fields/_InputFieldBase.jsx');

  __REACT_HOT_LOADER__.register(renderExtraContent, 'renderExtraContent', 'src/input-fields/_InputFieldBase.jsx');

  __REACT_HOT_LOADER__.register(TxlInputFieldBase, 'TxlInputFieldBase', 'src/input-fields/_InputFieldBase.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/input-fields/_InputFieldBase.jsx');
}();

;
