// txl/input-fields/_InputStyleHelpers.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DISABLED_INPUT_ITEM_LABEL_STYLES = exports.INPUT_ITEM_LABEL_STYLES = exports.INPUT_GROUP_LABEL_STYLES = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

exports.getSecondaryTextStyle = getSecondaryTextStyle;

var _helpers = require('../styles/helpers');

var _type = require('../styles/mixins/type');

var _theme = require('../styles/theme');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SECONDARY_TEXT_STYLES = {
  base: (0, _extends3.default)({}, _type.TYPE_SMALL, {
    color: _theme.COLOR_NEUTRAL['300'],
    minHeight: (0, _helpers.gridUnits)(5),
    padding: '2.5px 0' }),
  error: {
    color: _theme.COLOR_ERROR['500']
  },
  success: {
    color: _theme.COLOR_SUCCESS['500']
  },
  warning: {
    color: _theme.COLOR_WARNING['500']
  }
};

var INPUT_GROUP_LABEL_STYLES = exports.INPUT_GROUP_LABEL_STYLES = (0, _extends3.default)({}, _type.LABEL, {
  color: _theme.COLOR_NEUTRAL['400'],
  lineHeight: (0, _helpers.gridUnits)(5)
});

var INPUT_ITEM_LABEL_STYLES = exports.INPUT_ITEM_LABEL_STYLES = (0, _extends3.default)({}, _type.LABEL, _type.TYPE_BASE, {
  color: _theme.COLOR_NEUTRAL['800'],
  lineHeight: (0, _helpers.gridUnits)(5),
  wordWrap: 'break-word'
});

var DISABLED_INPUT_ITEM_LABEL_STYLES = exports.DISABLED_INPUT_ITEM_LABEL_STYLES = (0, _extends3.default)({}, INPUT_ITEM_LABEL_STYLES, {
  color: _theme.COLOR_NEUTRAL['200']
});

function getSecondaryTextStyle(validationState) {
  return (0, _helpers.combineStyles)(SECONDARY_TEXT_STYLES.base, SECONDARY_TEXT_STYLES[validationState]);
}
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(SECONDARY_TEXT_STYLES, 'SECONDARY_TEXT_STYLES', 'src/input-fields/_InputStyleHelpers.js');

  __REACT_HOT_LOADER__.register(INPUT_GROUP_LABEL_STYLES, 'INPUT_GROUP_LABEL_STYLES', 'src/input-fields/_InputStyleHelpers.js');

  __REACT_HOT_LOADER__.register(INPUT_ITEM_LABEL_STYLES, 'INPUT_ITEM_LABEL_STYLES', 'src/input-fields/_InputStyleHelpers.js');

  __REACT_HOT_LOADER__.register(DISABLED_INPUT_ITEM_LABEL_STYLES, 'DISABLED_INPUT_ITEM_LABEL_STYLES', 'src/input-fields/_InputStyleHelpers.js');

  __REACT_HOT_LOADER__.register(getSecondaryTextStyle, 'getSecondaryTextStyle', 'src/input-fields/_InputStyleHelpers.js');
}();

;
