// txl/input-fields/_PillButton.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _theme = require('../styles/theme');

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var PILL_BUTTON_STYLES = {
  base: {
    ':hover': { // @TODO: Replace with manual state management
      background: _theme.COLOR_NEUTRAL['200'],
      color: _theme.COLOR_NEUTRAL['200c']
    },
    background: _theme.COLOR_NEUTRAL['800c'],
    borderBottom: 'none',
    borderLeft: 'none',
    borderRadius: '0',
    borderRight: 'none',
    borderTop: 'none',
    color: _theme.COLOR_NEUTRAL['800'],
    cursor: 'pointer',
    height: (0, _helpers.gridUnits)(5),
    margin: '0',
    maxWidth: (0, _helpers.gridUnits)(60),
    outline: _theme.COLOR_NEUTRAL['100'],
    overflow: 'hidden',
    padding: '0 ' + (0, _helpers.gridUnits)(3),
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap'
  },
  border: {
    borderLeft: '1px solid ' + _theme.COLOR_NEUTRAL['100']
  },
  selected: {
    background: _theme.COLOR_NEUTRAL['100'],
    color: _theme.COLOR_NEUTRAL['100c']
  }
};

/**
 * @TODO: Since we are using overflow: hidden and ellipsis we should
 * use the TruncatedText component with this.props.children
 */
var TxlPillButton = function TxlPillButton(props) {
  var pillButtonStyles = (0, _helpers.combineStyles)(PILL_BUTTON_STYLES.base, props.selected && PILL_BUTTON_STYLES.selected, props.index && PILL_BUTTON_STYLES.border);

  return _react2.default.createElement(
    'button',
    {
      'data-component': 'TxlPillButton',
      onClick: function onClick() {
        return props.onChange(props);
      },
      style: pillButtonStyles
    },
    props.children
  );
};

TxlPillButton.propTypes = {
  children: _react.PropTypes.node.isRequired,
  id: _react.PropTypes.string,
  index: _react.PropTypes.number.isRequired,
  onChange: _react.PropTypes.func.isRequired,
  selected: _react.PropTypes.bool
};

TxlPillButton.defaultProps = {
  selected: false
};

var _default = (0, _radium2.default)(TxlPillButton);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(PILL_BUTTON_STYLES, 'PILL_BUTTON_STYLES', 'src/input-fields/_PillButton.jsx');

  __REACT_HOT_LOADER__.register(TxlPillButton, 'TxlPillButton', 'src/input-fields/_PillButton.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/input-fields/_PillButton.jsx');
}();

;
module.exports = exports['default'];
