// txl/input-fields/_TimeZonePicker.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _find = require('lodash/find');

var _find2 = _interopRequireDefault(_find);

var _TimeZones = require('../constants/_TimeZones');

var _Dropdown = require('../dropdowns/_Dropdown.style');

var _UncontrolledSingleSelect = require('../dropdowns/_UncontrolledSingleSelect');

var _UncontrolledSingleSelect2 = _interopRequireDefault(_UncontrolledSingleSelect);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DEFAULT_PLACEHOLDER_TEXT = 'Select a Time Zone';

var TxlTimeZonePicker = (0, _radium2.default)(_class = function (_React$Component) {
  (0, _inherits3.default)(TxlTimeZonePicker, _React$Component);

  function TxlTimeZonePicker() {
    var _ref;

    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, TxlTimeZonePicker);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlTimeZonePicker.__proto__ || (0, _getPrototypeOf2.default)(TxlTimeZonePicker)).call.apply(_ref, [this].concat(args))), _this), _this._handleItemSelect = function () {
      var _this2;

      return (_this2 = _this).___handleItemSelect__REACT_HOT_LOADER__.apply(_this2, arguments);
    }, _this._handleRef = function () {
      var _this3;

      return (_this3 = _this).___handleRef__REACT_HOT_LOADER__.apply(_this3, arguments);
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  (0, _createClass3.default)(TxlTimeZonePicker, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      // map timeZone val to label to match items in dropdown
      var value = this.props.value;

      var matchingTimeZoneItem = (0, _find2.default)(_TimeZones.TIME_ZONE_LIST_ITEMS, function (_ref2) {
        var timeZone = _ref2.timeZone;
        return timeZone === value;
      });
      if (matchingTimeZoneItem) {
        this._singleSelect.setValue({ value: matchingTimeZoneItem.label });
      }
    }
  }, {
    key: '___handleItemSelect__REACT_HOT_LOADER__',
    value: function ___handleItemSelect__REACT_HOT_LOADER__() {
      return this.___handleItemSelect__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleItemSelect__REACT_HOT_LOADER__',
    value: function ___handleItemSelect__REACT_HOT_LOADER__() {
      return this.___handleItemSelect__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleItemSelect__REACT_HOT_LOADER__',
    value: function ___handleItemSelect__REACT_HOT_LOADER__(selectedItem) {
      if (selectedItem) {
        var label = selectedItem.label,
            timeZone = selectedItem.timeZone;

        this.props.onItemSelect({ label: label, value: timeZone });
      }
    }
  }, {
    key: '___handleRef__REACT_HOT_LOADER__',
    value: function ___handleRef__REACT_HOT_LOADER__() {
      return this.___handleRef__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleRef__REACT_HOT_LOADER__',
    value: function ___handleRef__REACT_HOT_LOADER__() {
      return this.___handleRef__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleRef__REACT_HOT_LOADER__',
    value: function ___handleRef__REACT_HOT_LOADER__(ref) {
      this._singleSelect = ref;
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          label = _props.label,
          placeholder = _props.placeholder,
          secondaryText = _props.secondaryText,
          validationState = _props.validationState;


      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlTimeZonePicker',
          style: _Dropdown.WRAPPER_STYLE
        },
        _react2.default.createElement(_UncontrolledSingleSelect2.default, {
          items: _TimeZones.TIME_ZONE_LIST_ITEMS,
          label: label,
          onChange: this._handleItemSelect,
          placeholder: placeholder,
          ref: this._handleRef,
          secondaryText: secondaryText,
          validationState: validationState
        })
      );
    }
  }]);
  return TxlTimeZonePicker;
}(_react2.default.Component)) || _class;

exports.default = TxlTimeZonePicker;


TxlTimeZonePicker.displayName = 'TxlTimeZonePicker';

TxlTimeZonePicker.propTypes = {
  label: _UncontrolledSingleSelect2.default.propTypes.label,
  onItemSelect: _react.PropTypes.func.isRequired,
  placeholder: _UncontrolledSingleSelect2.default.propTypes.placeholder,
  secondaryText: _UncontrolledSingleSelect2.default.propTypes.secondaryText,
  validationState: _UncontrolledSingleSelect2.default.propTypes.validationState,
  value: _react.PropTypes.oneOf(_TimeZones.TIME_ZONES)
};

TxlTimeZonePicker.defaultProps = {
  placeholder: DEFAULT_PLACEHOLDER_TEXT
};
;

var _temp2 = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(DEFAULT_PLACEHOLDER_TEXT, 'DEFAULT_PLACEHOLDER_TEXT', 'src/input-fields/_TimeZonePicker.jsx');

  __REACT_HOT_LOADER__.register(TxlTimeZonePicker, 'TxlTimeZonePicker', 'src/input-fields/_TimeZonePicker.jsx');
}();

;
module.exports = exports['default'];
