// txl/input-fields/_TimeRangePickerControlled.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TIME_WRAPPER_STYLE = exports.PICKER_WRAPPER_STYLE = exports.LABEL_STYLE = exports.INPUT_SEPARATOR_STYLE = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _helpers = require('../styles/helpers');

var _type = require('../styles/mixins/type');

var _theme = require('../styles/theme');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var INPUT_SEPARATOR_STYLE = (0, _extends3.default)({}, _type.LABEL, {
  alignSelf: 'center',
  color: _theme.COLOR_NEUTRAL['400'],
  paddingLeft: (0, _helpers.gridUnits)(2),
  paddingRight: (0, _helpers.gridUnits)(2)
});

var LABEL_STYLE = (0, _extends3.default)({}, _type.LABEL, {
  color: _theme.COLOR_NEUTRAL['400']
});

var PICKER_WRAPPER_STYLE = {
  maxWidth: (0, _helpers.gridUnits)(24),
  minWidth: (0, _helpers.gridUnits)(21)
};

var TIME_WRAPPER_STYLE = {
  alignItems: 'stretch',
  display: 'flex',
  flexWrap: 'nowrap'
};

exports.INPUT_SEPARATOR_STYLE = INPUT_SEPARATOR_STYLE;
exports.LABEL_STYLE = LABEL_STYLE;
exports.PICKER_WRAPPER_STYLE = PICKER_WRAPPER_STYLE;
exports.TIME_WRAPPER_STYLE = TIME_WRAPPER_STYLE;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(INPUT_SEPARATOR_STYLE, 'INPUT_SEPARATOR_STYLE', 'src/input-fields/_TimeRangePickerControlled.style.js');

  __REACT_HOT_LOADER__.register(LABEL_STYLE, 'LABEL_STYLE', 'src/input-fields/_TimeRangePickerControlled.style.js');

  __REACT_HOT_LOADER__.register(PICKER_WRAPPER_STYLE, 'PICKER_WRAPPER_STYLE', 'src/input-fields/_TimeRangePickerControlled.style.js');

  __REACT_HOT_LOADER__.register(TIME_WRAPPER_STYLE, 'TIME_WRAPPER_STYLE', 'src/input-fields/_TimeRangePickerControlled.style.js');
}();

;
