// txl/input-fields/_TimeRangePickerControlled.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _propTypes = require('../utils/propTypes');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _TimeRangePickerControlled = require('./_TimeRangePickerControlled.style');

var _TimePicker = require('./TimePicker');

var _TimePicker2 = _interopRequireDefault(_TimePicker);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DEFAULT_DELIMITER_TEXT = 'to'; // @TODO translate

var TxlTimeRangePickerControlled = function TxlTimeRangePickerControlled(props) {
  var endTime = props.endTime,
      endTimeSecondaryText = props.endTimeSecondaryText,
      inputSeparator = props.inputSeparator,
      label = props.label,
      onEndTimeChange = props.onEndTimeChange,
      onStartTimeChange = props.onStartTimeChange,
      pattern = props.pattern,
      startTime = props.startTime,
      startTimeSecondaryText = props.startTimeSecondaryText;


  return _react2.default.createElement(
    'div',
    null,
    !!label && _react2.default.createElement(
      'div',
      { style: _TimeRangePickerControlled.LABEL_STYLE },
      label
    ),
    _react2.default.createElement(
      'div',
      { style: _TimeRangePickerControlled.TIME_WRAPPER_STYLE },
      _react2.default.createElement(
        'div',
        { style: _TimeRangePickerControlled.PICKER_WRAPPER_STYLE },
        _react2.default.createElement(_TimePicker2.default, {
          onItemSelect: onStartTimeChange,
          pattern: pattern,
          secondaryText: startTimeSecondaryText,
          time: startTime
        })
      ),
      _react2.default.createElement(
        'div',
        {
          'data-test-input-separator': true,
          style: _TimeRangePickerControlled.INPUT_SEPARATOR_STYLE
        },
        inputSeparator
      ),
      _react2.default.createElement(
        'div',
        { style: _TimeRangePickerControlled.PICKER_WRAPPER_STYLE },
        _react2.default.createElement(_TimePicker2.default, {
          onItemSelect: onEndTimeChange,
          pattern: pattern,
          secondaryText: endTimeSecondaryText,
          time: endTime
        })
      )
    )
  );
};

TxlTimeRangePickerControlled.propTypes = {
  endTime: _propTypes2.default.timeObj,
  endTimeSecondaryText: _react.PropTypes.string,
  inputSeparator: _react.PropTypes.string,
  label: _react.PropTypes.string,
  onEndTimeChange: _react.PropTypes.func,
  onStartTimeChange: _react.PropTypes.func,
  pattern: _react.PropTypes.string,
  startTime: _propTypes2.default.timeObj,
  startTimeSecondaryText: _react.PropTypes.string
};

TxlTimeRangePickerControlled.defaultProps = {
  inputSeparator: DEFAULT_DELIMITER_TEXT,
  onEndTimeChange: _Base.NOOP,
  onStartTimeChange: _Base.NOOP
};

var _default = (0, _radium2.default)(TxlTimeRangePickerControlled);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(DEFAULT_DELIMITER_TEXT, 'DEFAULT_DELIMITER_TEXT', 'src/input-fields/_TimeRangePickerControlled.jsx');

  __REACT_HOT_LOADER__.register(TxlTimeRangePickerControlled, 'TxlTimeRangePickerControlled', 'src/input-fields/_TimeRangePickerControlled.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/input-fields/_TimeRangePickerControlled.jsx');
}();

;
module.exports = exports['default'];
