// txl/input-fields/_TimeRangePicker.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _TimeZones = require('../constants/_TimeZones');

var _TimeRangePickerControlled = require('./_TimeRangePickerControlled');

var _TimeRangePickerControlled2 = _interopRequireDefault(_TimeRangePickerControlled);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlTimeRangePicker = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlTimeRangePicker, _React$Component);

  function TxlTimeRangePicker() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlTimeRangePicker);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlTimeRangePicker.__proto__ || (0, _getPrototypeOf2.default)(TxlTimeRangePicker)).call.apply(_ref, [this].concat(args)));

    var _this$props = _this.props,
        endTime = _this$props.endTime,
        startTime = _this$props.startTime;

    _this.state = { endTime: endTime, startTime: startTime };
    return _this;
  }

  (0, _createClass3.default)(TxlTimeRangePicker, [{
    key: '_handleChange',
    value: function _handleChange(startTime, endTime) {
      var _this2 = this;

      this.setState({ endTime: endTime, startTime: startTime }, function () {
        return _this2.props.onChange({ endTime: endTime, startTime: startTime });
      });
    }
  }, {
    key: '_handleEndTimeChange',
    value: function _handleEndTimeChange(endTime) {
      this._handleChange(this.state.startTime, endTime);
    }
  }, {
    key: '_handleStartTimeChange',
    value: function _handleStartTimeChange(startTime) {
      this._handleChange(startTime, this.state.endTime);
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          endTimeSecondaryText = _props.endTimeSecondaryText,
          inputSeparator = _props.inputSeparator,
          label = _props.label,
          pattern = _props.pattern,
          startTimeSecondaryText = _props.startTimeSecondaryText;
      var _state = this.state,
          endTime = _state.endTime,
          startTime = _state.startTime;


      return _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(_TimeRangePickerControlled2.default, {
          endTime: endTime,
          endTimeSecondaryText: endTimeSecondaryText,
          inputSeparator: inputSeparator,
          label: label,
          onEndTimeChange: this._handleEndTimeChange,
          onStartTimeChange: this._handleStartTimeChange,
          pattern: pattern,
          startTime: startTime,
          startTimeSecondaryText: startTimeSecondaryText
        })
      );
    }
  }]);
  return TxlTimeRangePicker;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleChange', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleChange'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleEndTimeChange', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleEndTimeChange'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleStartTimeChange', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleStartTimeChange'), _class2.prototype)), _class2)) || _class;

// destructuring shared prop types


var _TimeRangePickerContr = _TimeRangePickerControlled2.default.propTypes,
    endTime = _TimeRangePickerContr.endTime,
    endTimeSecondaryText = _TimeRangePickerContr.endTimeSecondaryText,
    inputSeparator = _TimeRangePickerContr.inputSeparator,
    label = _TimeRangePickerContr.label,
    pattern = _TimeRangePickerContr.pattern,
    startTime = _TimeRangePickerContr.startTime,
    startTimeSecondaryText = _TimeRangePickerContr.startTimeSecondaryText;


TxlTimeRangePicker.propTypes = {
  endTime: endTime,
  endTimeSecondaryText: endTimeSecondaryText,
  inputSeparator: inputSeparator,
  label: label,
  onChange: _react.PropTypes.func.isRequired,
  pattern: pattern,
  startTime: startTime,
  startTimeSecondaryText: startTimeSecondaryText
};

TxlTimeRangePicker.defaultProps = {
  endTime: { hour: 23, minute: 59, second: 59 },
  endTimeSecondaryText: '',
  pattern: _TimeZones.DEFAULT_TIME_PATTERN,
  startTime: { hour: 0, minute: 0, second: 0 },
  startTimeSecondaryText: ''
};

var _default = TxlTimeRangePicker;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlTimeRangePicker, 'TxlTimeRangePicker', 'src/input-fields/_TimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(endTime, 'endTime', 'src/input-fields/_TimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(endTimeSecondaryText, 'endTimeSecondaryText', 'src/input-fields/_TimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(inputSeparator, 'inputSeparator', 'src/input-fields/_TimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(label, 'label', 'src/input-fields/_TimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(pattern, 'pattern', 'src/input-fields/_TimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(startTime, 'startTime', 'src/input-fields/_TimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(startTimeSecondaryText, 'startTimeSecondaryText', 'src/input-fields/_TimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/input-fields/_TimeRangePicker.jsx');
}();

;
module.exports = exports['default'];
