// txl/input-fields/ContentEditable.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _isEmpty = require('lodash/isEmpty');

var _isEmpty2 = _interopRequireDefault(_isEmpty);

var _Base = require('../base/Base');

var _theme = require('../styles/theme');

var _type = require('../styles/mixins/type');

var _helpers = require('../styles/helpers');

var _KeyCodes = require('../constants/KeyCodes');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var INPUT_CONTAINER_STYLE = {
  alignItems: 'flex-start',
  display: 'flex'
};

var INPUT_STYLE = {
  backgroundColor: 'transparent',
  border: 'none',
  color: _theme.COLOR_NEUTRAL['800'],
  flexGrow: 0,
  height: (0, _helpers.gridUnits)(7),
  lineHeight: (0, _helpers.gridUnits)(7),
  minWidth: (0, _helpers.gridUnits)(0.2),
  outline: 'none',
  overflow: 'hidden',
  paddingBottom: 0,
  paddingRight: 0,
  paddingTop: 0,
  textIndent: 0
};

var PLACE_HOLDER_STYLE = (0, _extends3.default)({}, _type.TYPE_BASE, _type.DEFAULT_FONT_FAMILY, INPUT_STYLE, {
  color: _theme.COLOR_NEUTRAL['200']
});

var ENTITY_MAP = {
  '"': '&quot;',
  '&': '&amp;',
  '\'': '&#39;',
  '/': '&#x2F;',
  '<': '&lt;',
  '>': '&gt;'
};

function getInitialState(props) {
  return {
    focus: false,
    value: props.defaultValue || props.value || ''
  };
}

function canShowPlaceholder(props) {
  return !(0, _isEmpty2.default)(props.placeholder) && (0, _isEmpty2.default)(props.value);
}

// TODO:: make this generic util function or object
function escapeHtml(string) {
  return String(string).replace(/[&<>"'/]/g, function (s) {
    return ENTITY_MAP[s];
  });
}

// TODO:: make this a generic util function or object
function hasStyledElement() {
  var children = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : [];

  var childrenArray = Array.prototype.slice.call(children, 0);
  // FireFox inserts <br> for "space" Input
  // Following will filter through such children
  return childrenArray.filter(function (child) {
    return child && child.tagName !== 'BR';
  }).length !== 0;
}

// DO NOT USE purerendering with this component

var TxlContentEditable = (0, _radium2.default)(_class = function (_Component) {
  (0, _inherits3.default)(TxlContentEditable, _Component);

  function TxlContentEditable() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlContentEditable);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlContentEditable.__proto__ || (0, _getPrototypeOf2.default)(TxlContentEditable)).call.apply(_ref, [this].concat(args)));

    _this.moveCursorToEnd = function () {
      return _this.__moveCursorToEnd__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this.didChange = function () {
      return _this.__didChange__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this.focus = function () {
      return _this.__focus__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this.blur = function () {
      return _this.__blur__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this.handlePaste = function () {
      return _this.__handlePaste__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this.handleClick = function () {
      return _this.__handleClick__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this.handleOnInput = function () {
      return _this.__handleOnInput__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this.handleKeyDown = function () {
      return _this.__handleKeyDown__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this.handleFocus = function () {
      return _this.__handleFocus__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this.handleBlur = function () {
      return _this.__handleBlur__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleInputRef = function () {
      return _this.___handleInputRef__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this.state = getInitialState.apply(undefined, args);
    return _this;
  }

  (0, _createClass3.default)(TxlContentEditable, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      if (this.props.value !== nextProps.value) {
        var value = this.state.value;

        if (value !== nextProps.value) {
          this.setState({
            value: nextProps.value
          });
        }
      }
    }
  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate() {
      var node = this._inputRef;
      if (node) {
        var escapedValue = escapeHtml(this.state.value);
        if (escapedValue !== node.textContent || hasStyledElement(node.children)) {
          node.innerHTML = escapedValue;
        }
      }
    }
  }, {
    key: '__moveCursorToEnd__REACT_HOT_LOADER__',
    value: function __moveCursorToEnd__REACT_HOT_LOADER__() {
      return this.__moveCursorToEnd__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__moveCursorToEnd__REACT_HOT_LOADER__',
    value: function __moveCursorToEnd__REACT_HOT_LOADER__() {
      return this.__moveCursorToEnd__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__moveCursorToEnd__REACT_HOT_LOADER__',
    value: function __moveCursorToEnd__REACT_HOT_LOADER__() {
      var range = document.createRange();
      range.selectNodeContents(this._inputRef);
      range.collapse(false);
      var selection = window.getSelection();
      selection.removeAllRanges();
      selection.addRange(range);
    }
  }, {
    key: '__didChange__REACT_HOT_LOADER__',
    value: function __didChange__REACT_HOT_LOADER__() {
      return this.__didChange__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__didChange__REACT_HOT_LOADER__',
    value: function __didChange__REACT_HOT_LOADER__() {
      return this.__didChange__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__didChange__REACT_HOT_LOADER__',
    value: function __didChange__REACT_HOT_LOADER__() {
      var value = this.state.value;

      this.props.onChange({ value: value });
    }
  }, {
    key: '__focus__REACT_HOT_LOADER__',
    value: function __focus__REACT_HOT_LOADER__() {
      return this.__focus__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__focus__REACT_HOT_LOADER__',
    value: function __focus__REACT_HOT_LOADER__() {
      return this.__focus__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__focus__REACT_HOT_LOADER__',
    value: function __focus__REACT_HOT_LOADER__() {
      this._inputRef && this._inputRef.focus();
    }
  }, {
    key: '__blur__REACT_HOT_LOADER__',
    value: function __blur__REACT_HOT_LOADER__() {
      return this.__blur__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__blur__REACT_HOT_LOADER__',
    value: function __blur__REACT_HOT_LOADER__() {
      return this.__blur__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__blur__REACT_HOT_LOADER__',
    value: function __blur__REACT_HOT_LOADER__() {
      this._inputRef && this._inputRef.blur();
    }
  }, {
    key: '__handlePaste__REACT_HOT_LOADER__',
    value: function __handlePaste__REACT_HOT_LOADER__() {
      return this.__handlePaste__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__handlePaste__REACT_HOT_LOADER__',
    value: function __handlePaste__REACT_HOT_LOADER__() {
      return this.__handlePaste__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__handlePaste__REACT_HOT_LOADER__',
    value: function __handlePaste__REACT_HOT_LOADER__() {
      var node = this._inputRef;
      if (node) {
        var value = node.textContent;
        this.setState({ value: value }, this.didChange);
      }
    }
  }, {
    key: '__handleClick__REACT_HOT_LOADER__',
    value: function __handleClick__REACT_HOT_LOADER__() {
      return this.__handleClick__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__handleClick__REACT_HOT_LOADER__',
    value: function __handleClick__REACT_HOT_LOADER__() {
      return this.__handleClick__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__handleClick__REACT_HOT_LOADER__',
    value: function __handleClick__REACT_HOT_LOADER__(e) {
      this.focus();
      this.props.onClick(e);
    }
  }, {
    key: '__handleOnInput__REACT_HOT_LOADER__',
    value: function __handleOnInput__REACT_HOT_LOADER__() {
      return this.__handleOnInput__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__handleOnInput__REACT_HOT_LOADER__',
    value: function __handleOnInput__REACT_HOT_LOADER__() {
      return this.__handleOnInput__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__handleOnInput__REACT_HOT_LOADER__',
    value: function __handleOnInput__REACT_HOT_LOADER__(e) {
      var value = e.target.textContent;
      this.setState({ value: value }, this.didChange);
    }
  }, {
    key: '__handleKeyDown__REACT_HOT_LOADER__',
    value: function __handleKeyDown__REACT_HOT_LOADER__() {
      return this.__handleKeyDown__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__handleKeyDown__REACT_HOT_LOADER__',
    value: function __handleKeyDown__REACT_HOT_LOADER__() {
      return this.__handleKeyDown__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__handleKeyDown__REACT_HOT_LOADER__',
    value: function __handleKeyDown__REACT_HOT_LOADER__(e) {
      var keyCode = e.keyCode;

      switch (keyCode) {
        case _KeyCodes.ENTER:
        case _KeyCodes.ESCAPE:
        case _KeyCodes.TAB:
          e.preventDefault();
          this.blur();
          break;
        default:
      }
      this.props.onKeyDown && this.props.onKeyDown(e);
    }
  }, {
    key: '__handleFocus__REACT_HOT_LOADER__',
    value: function __handleFocus__REACT_HOT_LOADER__() {
      return this.__handleFocus__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__handleFocus__REACT_HOT_LOADER__',
    value: function __handleFocus__REACT_HOT_LOADER__() {
      return this.__handleFocus__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__handleFocus__REACT_HOT_LOADER__',
    value: function __handleFocus__REACT_HOT_LOADER__(e) {
      var _this2 = this;

      this.setState({ focus: true }, function () {
        return _this2.props.onFocus(e);
      });
    }
  }, {
    key: '__handleBlur__REACT_HOT_LOADER__',
    value: function __handleBlur__REACT_HOT_LOADER__() {
      return this.__handleBlur__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__handleBlur__REACT_HOT_LOADER__',
    value: function __handleBlur__REACT_HOT_LOADER__() {
      return this.__handleBlur__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__handleBlur__REACT_HOT_LOADER__',
    value: function __handleBlur__REACT_HOT_LOADER__(e) {
      var _this3 = this;

      this.setState({ focus: false }, function () {
        return _this3.props.onBlur(e);
      });
    }
  }, {
    key: '___handleInputRef__REACT_HOT_LOADER__',
    value: function ___handleInputRef__REACT_HOT_LOADER__() {
      return this.___handleInputRef__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleInputRef__REACT_HOT_LOADER__',
    value: function ___handleInputRef__REACT_HOT_LOADER__() {
      return this.___handleInputRef__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleInputRef__REACT_HOT_LOADER__',
    value: function ___handleInputRef__REACT_HOT_LOADER__(ref) {
      this._inputRef = ref;
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          disabled = _props.disabled,
          id = _props.id,
          spellCheck = _props.spellCheck,
          style = _props.style,
          tagName = _props.tagName,
          placeholder = _props.placeholder,
          tabIndex = _props.tabIndex;
      var value = this.state.value;


      var handlers = {
        onBlur: !disabled ? this.handleBlur : _Base.NOOP,
        onFocus: !disabled ? this.handleFocus : _Base.NOOP
      };

      var showPlaceholder = canShowPlaceholder({ placeholder: placeholder, value: value });

      var elemStyle = (0, _extends3.default)({}, INPUT_STYLE, {
        paddingLeft: showPlaceholder ? 0 : (0, _helpers.gridUnits)(0.2)
      });

      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlContentEditable',
          onClick: this.handleClick,
          style: style
        },
        _react2.default.createElement(
          'div',
          { style: INPUT_CONTAINER_STYLE },
          _react2.default.DOM[tagName]((0, _extends3.default)({}, handlers, {
            contentEditable: !disabled,
            id: id,
            onInput: this.handleOnInput,
            onKeyDown: this.handleKeyDown,
            onPaste: this.handlePaste,
            ref: this._handleInputRef,
            spellCheck: spellCheck,
            style: elemStyle,
            tabIndex: tabIndex
          })),
          _react2.default.DOM[tagName]({
            style: PLACE_HOLDER_STYLE
          }, showPlaceholder ? placeholder : undefined)
        )
      );
    }
  }]);
  return TxlContentEditable;
}(_react.Component)) || _class;

TxlContentEditable.propTypes = {
  defaultValue: _react.PropTypes.string,
  disabled: _react.PropTypes.bool,
  id: _react.PropTypes.string,
  onBlur: _react.PropTypes.func,
  onChange: _react.PropTypes.func,
  onClick: _react.PropTypes.func,
  onFocus: _react.PropTypes.func,
  onKeyDown: _react.PropTypes.func,
  placeholder: _react.PropTypes.string,
  spellCheck: _react.PropTypes.bool,
  style: _react.PropTypes.object,
  tabIndex: _react.PropTypes.number,
  tagName: _react.PropTypes.string,
  value: _react.PropTypes.string
};

TxlContentEditable.defaultProps = {
  onBlur: _Base.NOOP,
  onChange: _Base.NOOP,
  onClick: _Base.NOOP,
  onFocus: _Base.NOOP,
  spellCheck: true,
  tabIndex: 0,
  tagName: 'div',
  value: ''
};

var _default = TxlContentEditable;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(INPUT_CONTAINER_STYLE, 'INPUT_CONTAINER_STYLE', 'src/input-fields/ContentEditable.jsx');

  __REACT_HOT_LOADER__.register(INPUT_STYLE, 'INPUT_STYLE', 'src/input-fields/ContentEditable.jsx');

  __REACT_HOT_LOADER__.register(PLACE_HOLDER_STYLE, 'PLACE_HOLDER_STYLE', 'src/input-fields/ContentEditable.jsx');

  __REACT_HOT_LOADER__.register(ENTITY_MAP, 'ENTITY_MAP', 'src/input-fields/ContentEditable.jsx');

  __REACT_HOT_LOADER__.register(getInitialState, 'getInitialState', 'src/input-fields/ContentEditable.jsx');

  __REACT_HOT_LOADER__.register(canShowPlaceholder, 'canShowPlaceholder', 'src/input-fields/ContentEditable.jsx');

  __REACT_HOT_LOADER__.register(escapeHtml, 'escapeHtml', 'src/input-fields/ContentEditable.jsx');

  __REACT_HOT_LOADER__.register(hasStyledElement, 'hasStyledElement', 'src/input-fields/ContentEditable.jsx');

  __REACT_HOT_LOADER__.register(TxlContentEditable, 'TxlContentEditable', 'src/input-fields/ContentEditable.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/input-fields/ContentEditable.jsx');
}();

;
module.exports = exports['default'];
