// txl/input-fields/ContentEditableField.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _InputFieldBase = require('./_InputFieldBase');

var _InputFieldBase2 = _interopRequireDefault(_InputFieldBase);

var _ContentEditable = require('./ContentEditable');

var _ContentEditable2 = _interopRequireDefault(_ContentEditable);

var _InputFieldsPropTypes = require('./_InputFieldsPropTypes');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

function getInitialState(props) {
  var focused = typeof props.focused !== 'undefined' ? props.focused : false;
  return { focused: focused };
}

var TxlContentEditableField = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlContentEditableField, _React$Component);

  function TxlContentEditableField() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlContentEditableField);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlContentEditableField.__proto__ || (0, _getPrototypeOf2.default)(TxlContentEditableField)).call.apply(_ref, [this].concat(args)));

    _this.state = getInitialState(_this.props);
    return _this;
  }

  (0, _createClass3.default)(TxlContentEditableField, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      if (typeof nextProps.focused !== 'undefined' && this.props.focused !== nextProps.focused && nextProps.focused !== this.state.focused) {
        // Implements controlled behaviour
        this.setState({ focused: nextProps.focused });
      }
    }
  }, {
    key: 'focus',
    value: function focus() {
      this.refs.input && this.refs.input.focus();
    }
  }, {
    key: 'blur',
    value: function blur() {
      this.refs.input && this.refs.input.blur();
    }
  }, {
    key: '_handleFocus',
    value: function _handleFocus(e) {
      var _this2 = this;

      this.setState({
        focused: true
      }, function () {
        return _this2.props.onFocus(e);
      });
    }
  }, {
    key: '_handleBlur',
    value: function _handleBlur(e) {
      var _this3 = this;

      this.setState({
        focused: false
      }, function () {
        return _this3.props.onBlur(e);
      });
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          _afterContent = _props._afterContent,
          _beforeContent = _props._beforeContent,
          _containerProps = _props._containerProps,
          children = _props.children,
          disabled = _props.disabled,
          id = _props.id,
          label = _props.label,
          onChange = _props.onChange,
          onClick = _props.onClick,
          placeholder = _props.placeholder,
          secondaryText = _props.secondaryText,
          tabIndex = _props.tabIndex,
          validationState = _props.validationState,
          value = _props.value;
      var focused = this.state.focused;


      return _react2.default.createElement(
        _InputFieldBase2.default,
        {
          _afterContent: _afterContent,
          _beforeContent: _beforeContent,
          _containerProps: _containerProps,
          disabled: disabled,
          focused: focused,
          label: label,
          onClick: onClick,
          secondaryText: secondaryText,
          validationState: validationState
        },
        children,
        _react2.default.createElement(_ContentEditable2.default, {
          disabled: disabled,
          id: id,
          onBlur: this._handleBlur,
          onChange: onChange,
          onFocus: this._handleFocus,
          placeholder: placeholder,
          ref: 'input',
          tabIndex: tabIndex,
          value: value
        })
      );
    }
  }]);
  return TxlContentEditableField;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, 'focus', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, 'focus'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, 'blur', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, 'blur'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleFocus', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleFocus'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleBlur', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleBlur'), _class2.prototype)), _class2)) || _class;

TxlContentEditableField.propTypes = {
  _afterContent: _react.PropTypes.node,
  _beforeContent: _react.PropTypes.node,
  _containerProps: _InputFieldBase2.default.propTypes._containerProps,
  children: _react.PropTypes.node,
  disabled: _react.PropTypes.bool,
  focused: _react.PropTypes.bool,
  id: _react.PropTypes.string,
  label: _react.PropTypes.node,
  onBlur: _react.PropTypes.func,
  onChange: _react.PropTypes.func,
  onClick: _react.PropTypes.func,
  onFocus: _react.PropTypes.func,
  placeholder: _react.PropTypes.string,
  secondaryText: _react.PropTypes.node,
  tabIndex: _react.PropTypes.number,
  validationState: _InputFieldsPropTypes.validationStateType,
  value: _react.PropTypes.node
};

TxlContentEditableField.defaultProps = {
  onBlur: _Base.NOOP,
  onClick: _Base.NOOP,
  onFocus: _Base.NOOP
};

var _default = TxlContentEditableField;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(getInitialState, 'getInitialState', 'src/input-fields/ContentEditableField.jsx');

  __REACT_HOT_LOADER__.register(TxlContentEditableField, 'TxlContentEditableField', 'src/input-fields/ContentEditableField.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/input-fields/ContentEditableField.jsx');
}();

;
module.exports = exports['default'];
