// txl/input-fields/CurrencyField.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _CurrencyFieldControlled = require('./CurrencyFieldControlled');

var _CurrencyFieldControlled2 = _interopRequireDefault(_CurrencyFieldControlled);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlCurrencyField = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlCurrencyField, _React$Component);

  function TxlCurrencyField(props) {
    (0, _classCallCheck3.default)(this, TxlCurrencyField);

    var _this = (0, _possibleConstructorReturn3.default)(this, (TxlCurrencyField.__proto__ || (0, _getPrototypeOf2.default)(TxlCurrencyField)).call(this, props));

    _this.state = {
      value: props.defaultValue
    };
    return _this;
  }

  (0, _createClass3.default)(TxlCurrencyField, [{
    key: '_handleBlur',
    value: function _handleBlur(_ref) {
      var value = _ref.value;
      var _props = this.props,
          onBlur = _props.onBlur,
          onChange = _props.onChange;


      this.setState({ value: value }, function () {
        onBlur && onBlur({ value: value });
        onChange({ value: value });
      });
    }
  }, {
    key: '_handleChange',
    value: function _handleChange(_ref2) {
      var _this2 = this;

      var value = _ref2.value;

      this.setState({ value: value }, function () {
        _this2.props.onChange({ value: value });
      });
    }
  }, {
    key: 'focus',
    value: function focus() {
      this._currencyFieldRef.focus();
    }
  }, {
    key: '_handleCurrencyFieldRef',
    value: function _handleCurrencyFieldRef(ref) {
      this._currencyFieldRef = ref;
    }
  }, {
    key: 'render',
    value: function render() {
      var _props2 = this.props,
          currencyCode = _props2.currencyCode,
          label = _props2.label,
          maximumFractionDigits = _props2.maximumFractionDigits,
          minimumFractionDigits = _props2.minimumFractionDigits,
          onFocus = _props2.onFocus,
          secondaryText = _props2.secondaryText,
          validationState = _props2.validationState;


      return _react2.default.createElement(_CurrencyFieldControlled2.default, {
        currencyCode: currencyCode,
        label: label,
        maximumFractionDigits: maximumFractionDigits,
        minimumFractionDigits: minimumFractionDigits,
        onBlur: this._handleBlur,
        onChange: this._handleChange,
        onFocus: onFocus,
        ref: this._handleCurrencyFieldRef,
        secondaryText: secondaryText,
        validationState: validationState,
        value: this.state.value
      });
    }
  }]);
  return TxlCurrencyField;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleBlur', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleBlur'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleChange', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleChange'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, 'focus', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, 'focus'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleCurrencyFieldRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleCurrencyFieldRef'), _class2.prototype)), _class2)) || _class;

exports.default = TxlCurrencyField;


TxlCurrencyField.displayName = 'TxlCurrencyField';

TxlCurrencyField.propTypes = {
  currencyCode: _CurrencyFieldControlled2.default.propTypes.currencyCode,
  defaultValue: _CurrencyFieldControlled2.default.propTypes.value,
  label: _CurrencyFieldControlled2.default.propTypes.label,
  maximumFractionDigits: _react.PropTypes.oneOf([0, 1, 2, 3, 4, 5]),
  minimumFractionDigits: _react.PropTypes.oneOf([0, 1, 2, 3, 4, 5]),
  onBlur: _CurrencyFieldControlled2.default.propTypes.onBlur,
  onChange: _CurrencyFieldControlled2.default.propTypes.onChange,
  onFocus: _CurrencyFieldControlled2.default.propTypes.onFocus,
  secondaryText: _CurrencyFieldControlled2.default.propTypes.secondaryText,
  validationState: _CurrencyFieldControlled2.default.propTypes.validationState
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlCurrencyField, 'TxlCurrencyField', 'src/input-fields/CurrencyField.jsx');
}();

;
module.exports = exports['default'];
