// txl/input-fields/CurrencyFieldControlled.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _isNan = require('babel-runtime/core-js/number/is-nan');

var _isNan2 = _interopRequireDefault(_isNan);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _pick = require('lodash/pick');

var _pick2 = _interopRequireDefault(_pick);

var _keys = require('lodash/keys');

var _keys2 = _interopRequireDefault(_keys);

var _Base = require('../base/Base');

var _TextField = require('./TextField');

var _TextField2 = _interopRequireDefault(_TextField);

var _currencyCodeToSymbolMap = require('../utils/currencyCodeToSymbolMap');

var _currencyCodeToSymbolMap2 = _interopRequireDefault(_currencyCodeToSymbolMap);

var _helpers = require('../styles/helpers');

var _InputStyleHelpers = require('./_InputStyleHelpers');

var _InputFieldBase = require('./_InputFieldBase');

var _formatter = require('../utils/formatter');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var CURRENCY_LABEL_STYLE = (0, _extends3.default)({}, _InputStyleHelpers.INPUT_GROUP_LABEL_STYLES, {
  alignItems: 'center',
  display: 'flex',
  marginLeft: '-' + _InputFieldBase.BEFORE_CONTENT_LEFT_MARGIN,
  marginRight: (0, _helpers.gridUnits)(2),
  whiteSpace: 'nowrap'
});

var TxlCurrencyFieldControlled = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlCurrencyFieldControlled, _React$Component);

  function TxlCurrencyFieldControlled(props, context) {
    (0, _classCallCheck3.default)(this, TxlCurrencyFieldControlled);

    var _this = (0, _possibleConstructorReturn3.default)(this, (TxlCurrencyFieldControlled.__proto__ || (0, _getPrototypeOf2.default)(TxlCurrencyFieldControlled)).call(this, props, context));

    _this._placeholder = (0, _formatter.formatCurrencyStringForLocale)('123456789.12345', {
      maximumFractionDigits: props.maximumFractionDigits,
      minimumFractionDigits: props.minimumFractionDigits
    });
    return _this;
  }

  (0, _createClass3.default)(TxlCurrencyFieldControlled, [{
    key: 'focus',
    value: function focus() {
      this._textFieldRef.focus();
    }
  }, {
    key: '_handleTextFieldRef',
    value: function _handleTextFieldRef(ref) {
      this._textFieldRef = ref;
    }
  }, {
    key: '_handleChange',
    value: function _handleChange(_ref) {
      var value = _ref.value;
      var onChange = this.props.onChange;


      var decimalValue = (0, _formatter.convertCurrencyToDecimalValue)(value);
      if (!(0, _isNan2.default)(decimalValue)) {
        onChange({ value: value });
      }
    }
  }, {
    key: '_handleBlur',
    value: function _handleBlur(_ref2) {
      var value = _ref2.value;
      var _props = this.props,
          maximumFractionDigits = _props.maximumFractionDigits,
          minimumFractionDigits = _props.minimumFractionDigits,
          onChange = _props.onChange,
          onBlur = _props.onBlur;


      var formattedValue = value ? (0, _formatter.formatCurrencyStringForLocale)(value, {
        maximumFractionDigits: maximumFractionDigits,
        minimumFractionDigits: minimumFractionDigits
      }) : '';
      onBlur && onBlur({ value: formattedValue });

      onChange({ value: formattedValue });
    }
  }, {
    key: 'render',
    value: function render() {
      var _props2 = this.props,
          currencyCode = _props2.currencyCode,
          currencySymbol = _props2.currencySymbol,
          others = (0, _objectWithoutProperties3.default)(_props2, ['currencyCode', 'currencySymbol']);


      var currencyLabel = _react2.default.createElement(
        'div',
        { style: CURRENCY_LABEL_STYLE },
        currencyCode + ' ' + (currencySymbol || _currencyCodeToSymbolMap2.default[currencyCode])
      );

      return _react2.default.createElement(_TextField2.default, (0, _extends3.default)({}, (0, _pick2.default)(others, (0, _keys2.default)(_TextField2.default.propTypes)), {
        _beforeContent: currencyLabel,
        onBlur: this._handleBlur,
        onChange: this._handleChange,
        placeholder: this.props.placeholder || this._placeholder,
        ref: this._handleTextFieldRef,
        textAlign: 'right'
      }));
    }
  }]);
  return TxlCurrencyFieldControlled;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, 'focus', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, 'focus'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleTextFieldRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleTextFieldRef'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleChange', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleChange'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleBlur', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleBlur'), _class2.prototype)), _class2)) || _class;

exports.default = TxlCurrencyFieldControlled;


TxlCurrencyFieldControlled.displayName = 'TxlCurrencyFieldControlled';

TxlCurrencyFieldControlled.propTypes = (0, _extends3.default)({}, _TextField2.default.propTypes, {
  currencyCode: _react.PropTypes.string.isRequired,
  currencySymbol: _react.PropTypes.string,
  label: _TextField2.default.propTypes.label,
  maximumFractionDigits: _react.PropTypes.oneOf([0, 1, 2, 3, 4, 5]),
  minimumFractionDigits: _react.PropTypes.oneOf([0, 1, 2, 3, 4, 5]),
  onBlur: _TextField2.default.propTypes.onBlur,
  onChange: _TextField2.default.propTypes.onChange,
  onFocus: _TextField2.default.propTypes.onFocus,
  secondaryText: _TextField2.default.propTypes.secondaryText,
  validationState: _TextField2.default.propTypes.validationState,
  value: _react.PropTypes.string
});

TxlCurrencyFieldControlled.defaultProps = {
  maximumFractionDigits: 2,
  minimumFractionDigits: 2
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(CURRENCY_LABEL_STYLE, 'CURRENCY_LABEL_STYLE', 'src/input-fields/CurrencyFieldControlled.jsx');

  __REACT_HOT_LOADER__.register(TxlCurrencyFieldControlled, 'TxlCurrencyFieldControlled', 'src/input-fields/CurrencyFieldControlled.jsx');
}();

;
module.exports = exports['default'];
