// txl/input-fields/DateRangeSelector.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WRAPPER_STYLE = exports.TIME_ZONE_WRAPPER_STYLE = exports.TIME_WRAPPER_STYLE = exports.TIME_RANGE_WRAPPER_STYLE = exports.TIME_PICKER_WRAPPER_STYLE = exports.LABEL_STYLE = exports.INPUT_SEPARATOR_STYLE = exports.FOOTER_STYLE = exports.FOOTER_CONTENT_STYLE = exports.DROPDOWN_CONTAINER_STYLE = exports.BUTTON_STYLE = exports.BUTTON_GROUP_STYLE = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

var _constants = require('../styles/constants');

var _type = require('../styles/mixins/type');

var _Dropdown = require('../dropdowns/_Dropdown.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DROPDOWN_CONTAINER_STYLE = (0, _extends3.default)({}, _Dropdown.DROPDOWN_BASE, {
  left: 0,
  position: 'absolute',
  top: (0, _helpers.gridUnits)(8),
  zIndex: _constants.Z_INDEXES.popover
});

var TIME_WRAPPER_STYLE = {
  borderTop: '1px solid ' + _theme.COLOR_NEUTRAL['100'],
  display: 'flex',
  justifyContent: 'space-between',
  padding: (0, _helpers.gridUnits)(4)
};

var TIME_RANGE_WRAPPER_STYLE = {
  alignItems: 'stretch',
  display: 'flex',
  flexWrap: 'nowrap',
  paddingRight: (0, _helpers.gridUnits)(10)
};

var INPUT_SEPARATOR_STYLE = (0, _extends3.default)({}, _type.LABEL, {
  alignSelf: 'center',
  color: _theme.COLOR_NEUTRAL['400'],
  paddingLeft: (0, _helpers.gridUnits)(2),
  paddingRight: (0, _helpers.gridUnits)(2)
});

var LABEL_STYLE = (0, _extends3.default)({}, _type.LABEL, {
  color: _theme.COLOR_NEUTRAL['400']
});

var TIME_PICKER_WRAPPER_STYLE = {
  maxWidth: (0, _helpers.gridUnits)(24),
  minWidth: (0, _helpers.gridUnits)(21)
};

var TIME_ZONE_WRAPPER_STYLE = {
  flex: '0 1 ' + (0, _helpers.gridUnits)(55)
};

var FOOTER_CONTENT_STYLE = {
  display: 'inline-flex',
  flex: '0 1 auto',
  flexDirection: 'column',
  justifyContent: 'center',
  padding: '0 ' + (0, _helpers.gridUnits)(4)
};

var FOOTER_STYLE = (0, _extends3.default)({}, _type.TYPE_SMALL, {
  borderTop: '1px solid ' + _theme.COLOR_NEUTRAL['100'],
  color: _theme.COLOR_NEUTRAL['400'],
  display: 'flex',
  justifyContent: 'space-between'
});

var BUTTON_STYLE = {
  marginRight: (0, _helpers.gridUnits)(2)
};

var BUTTON_GROUP_STYLE = {
  boxSizing: 'border-box',
  padding: (0, _helpers.gridUnits)(4)
};

exports.BUTTON_GROUP_STYLE = BUTTON_GROUP_STYLE;
exports.BUTTON_STYLE = BUTTON_STYLE;
exports.DROPDOWN_CONTAINER_STYLE = DROPDOWN_CONTAINER_STYLE;
exports.FOOTER_CONTENT_STYLE = FOOTER_CONTENT_STYLE;
exports.FOOTER_STYLE = FOOTER_STYLE;
exports.INPUT_SEPARATOR_STYLE = INPUT_SEPARATOR_STYLE;
exports.LABEL_STYLE = LABEL_STYLE;
exports.TIME_PICKER_WRAPPER_STYLE = TIME_PICKER_WRAPPER_STYLE;
exports.TIME_RANGE_WRAPPER_STYLE = TIME_RANGE_WRAPPER_STYLE;
exports.TIME_WRAPPER_STYLE = TIME_WRAPPER_STYLE;
exports.TIME_ZONE_WRAPPER_STYLE = TIME_ZONE_WRAPPER_STYLE;
exports.WRAPPER_STYLE = _Dropdown.WRAPPER_STYLE;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(DROPDOWN_CONTAINER_STYLE, 'DROPDOWN_CONTAINER_STYLE', 'src/input-fields/DateRangeSelector.style.js');

  __REACT_HOT_LOADER__.register(TIME_WRAPPER_STYLE, 'TIME_WRAPPER_STYLE', 'src/input-fields/DateRangeSelector.style.js');

  __REACT_HOT_LOADER__.register(TIME_RANGE_WRAPPER_STYLE, 'TIME_RANGE_WRAPPER_STYLE', 'src/input-fields/DateRangeSelector.style.js');

  __REACT_HOT_LOADER__.register(INPUT_SEPARATOR_STYLE, 'INPUT_SEPARATOR_STYLE', 'src/input-fields/DateRangeSelector.style.js');

  __REACT_HOT_LOADER__.register(LABEL_STYLE, 'LABEL_STYLE', 'src/input-fields/DateRangeSelector.style.js');

  __REACT_HOT_LOADER__.register(TIME_PICKER_WRAPPER_STYLE, 'TIME_PICKER_WRAPPER_STYLE', 'src/input-fields/DateRangeSelector.style.js');

  __REACT_HOT_LOADER__.register(TIME_ZONE_WRAPPER_STYLE, 'TIME_ZONE_WRAPPER_STYLE', 'src/input-fields/DateRangeSelector.style.js');

  __REACT_HOT_LOADER__.register(FOOTER_CONTENT_STYLE, 'FOOTER_CONTENT_STYLE', 'src/input-fields/DateRangeSelector.style.js');

  __REACT_HOT_LOADER__.register(FOOTER_STYLE, 'FOOTER_STYLE', 'src/input-fields/DateRangeSelector.style.js');

  __REACT_HOT_LOADER__.register(BUTTON_STYLE, 'BUTTON_STYLE', 'src/input-fields/DateRangeSelector.style.js');

  __REACT_HOT_LOADER__.register(BUTTON_GROUP_STYLE, 'BUTTON_GROUP_STYLE', 'src/input-fields/DateRangeSelector.style.js');
}();

;
