// txl/input-fields/DatePicker.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DROPDOWN_CONTENT_STYLE = exports.DROPDOWN_CONTAINER_STYLE = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _helpers = require('../styles/helpers');

var _Dropdown = require('../dropdowns/_Dropdown.style');

var DropdownStyles = _interopRequireWildcard(_Dropdown);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DROPDOWN_CONTENT_WIDTH = (0, _helpers.gridUnitsRaw)(7.8); // 39

var ICON_BUTTON_LARGE_WIDTH = (0, _helpers.gridUnitsRaw)(1.4); // 7

var DROPDOWN_CONTAINER_STYLE = exports.DROPDOWN_CONTAINER_STYLE = (0, _extends3.default)({}, DropdownStyles.DROPDOWN_BASE, {
  left: 'auto',
  // Moving dropdown content half size of its content size and icon button
  // so that DropdownArrow can be positioned at the center of IconButton
  right: (0, _helpers.gridUnits)((-1 * DROPDOWN_CONTENT_WIDTH + ICON_BUTTON_LARGE_WIDTH) / 2),
  top: (0, _helpers.gridUnits)(10.5),
  width: (0, _helpers.gridUnits)(DROPDOWN_CONTENT_WIDTH)
});

var DROPDOWN_CONTENT_STYLE = exports.DROPDOWN_CONTENT_STYLE = {
  padding: (0, _helpers.gridUnits)(2),
  position: 'relative',
  zIndex: 20
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(DROPDOWN_CONTENT_WIDTH, 'DROPDOWN_CONTENT_WIDTH', 'src/input-fields/DatePicker.style.js');

  __REACT_HOT_LOADER__.register(ICON_BUTTON_LARGE_WIDTH, 'ICON_BUTTON_LARGE_WIDTH', 'src/input-fields/DatePicker.style.js');

  __REACT_HOT_LOADER__.register(DROPDOWN_CONTAINER_STYLE, 'DROPDOWN_CONTAINER_STYLE', 'src/input-fields/DatePicker.style.js');

  __REACT_HOT_LOADER__.register(DROPDOWN_CONTENT_STYLE, 'DROPDOWN_CONTENT_STYLE', 'src/input-fields/DatePicker.style.js');
}();

;
