// txl/input-fields/DayPickerControlled.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _helpers = require('../styles/helpers');

var _DayPickerItemControlled = require('./DayPickerItemControlled');

var _DayPickerItemControlled2 = _interopRequireDefault(_DayPickerItemControlled);

var _DayPicker = require('./DayPicker.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

// we'll always show seven days, but allow for custom labels for L10n
var DEFAULT_DAY_LABELS = ['S', 'M', 'T', 'W', 'Th', 'F', 'Sa'];

var TxlDayPickerControlled = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlDayPickerControlled, _React$Component);

  function TxlDayPickerControlled() {
    (0, _classCallCheck3.default)(this, TxlDayPickerControlled);
    return (0, _possibleConstructorReturn3.default)(this, (TxlDayPickerControlled.__proto__ || (0, _getPrototypeOf2.default)(TxlDayPickerControlled)).apply(this, arguments));
  }

  (0, _createClass3.default)(TxlDayPickerControlled, [{
    key: '_handleClick',
    value: function _handleClick(_ref) {
      var day = _ref.day,
          index = _ref.index;

      this.props.onClick({ day: day, index: index });
    }
  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      var _props = this.props,
          arrangement = _props.arrangement,
          dayLabels = _props.dayLabels,
          disabledDayIndices = _props.disabledDayIndices,
          selectedDayIndices = _props.selectedDayIndices;


      var listStyle = (0, _helpers.combineStyles)(_DayPicker.LIST_STYLES.base, _DayPicker.LIST_STYLES[arrangement]);

      var listItemStyle = (0, _helpers.combineStyles)(_DayPicker.ITEM_STYLES.base, arrangement === 'horizontal' ? _DayPicker.ITEM_STYLES[arrangement] : {});

      return _react2.default.createElement(
        'section',
        { 'data-component': 'TxlDayPickerControlled' },
        _react2.default.createElement(
          'ol',
          { style: listStyle },
          dayLabels.map(function (label, idx) {
            return _react2.default.createElement(
              'li',
              {
                key: idx,
                style: listItemStyle
              },
              _react2.default.createElement(_DayPickerItemControlled2.default, {
                day: label,
                disabled: disabledDayIndices.indexOf(idx) >= 0,
                index: idx,
                onClick: _this2._handleClick,
                selected: selectedDayIndices.indexOf(idx) >= 0
              })
            );
          })
        )
      );
    }
  }]);
  return TxlDayPickerControlled;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleClick'), _class2.prototype)), _class2)) || _class;

exports.default = TxlDayPickerControlled;


TxlDayPickerControlled.propTypes = {
  arrangement: _react.PropTypes.oneOf(['horizontal', 'vertical']),
  dayLabels: _react.PropTypes.arrayOf(_react.PropTypes.string),
  disabledDayIndices: _react.PropTypes.arrayOf(_react.PropTypes.number),
  onClick: _react.PropTypes.func,
  selectedDayIndices: _react.PropTypes.arrayOf(_react.PropTypes.number)
};

TxlDayPickerControlled.defaultProps = {
  arrangement: 'horizontal',
  dayLabels: DEFAULT_DAY_LABELS,
  disabledDayIndices: [],
  onClick: _Base.NOOP,
  selectedDayIndices: []
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(DEFAULT_DAY_LABELS, 'DEFAULT_DAY_LABELS', 'src/input-fields/DayPickerControlled.jsx');

  __REACT_HOT_LOADER__.register(TxlDayPickerControlled, 'TxlDayPickerControlled', 'src/input-fields/DayPickerControlled.jsx');
}();

;
module.exports = exports['default'];
