// txl/input-fields/DayPickerItemControlled.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _helpers = require('../styles/helpers');

var _DayPickerItemControlled = require('./DayPickerItemControlled.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlDayPickerItemControlled = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlDayPickerItemControlled, _React$Component);

  function TxlDayPickerItemControlled() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlDayPickerItemControlled);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlDayPickerItemControlled.__proto__ || (0, _getPrototypeOf2.default)(TxlDayPickerItemControlled)).call.apply(_ref, [this].concat(args)));

    _this.state = {
      focused: false,
      hovered: false
    };
    return _this;
  }

  (0, _createClass3.default)(TxlDayPickerItemControlled, [{
    key: '_handleChange',
    value: function _handleChange() {
      var _props = this.props,
          day = _props.day,
          disabled = _props.disabled,
          index = _props.index;

      if (!disabled) {
        this.props.onClick({ day: day, index: index });
      }
    }
  }, {
    key: '_handleBlur',
    value: function _handleBlur() {
      this.setState({ focused: false });
    }
  }, {
    key: '_handleFocus',
    value: function _handleFocus() {
      this.setState({ focused: true });
    }
  }, {
    key: '_handleMouseEnter',
    value: function _handleMouseEnter() {
      this.setState({ hovered: true });
    }
  }, {
    key: '_handleMouseLeave',
    value: function _handleMouseLeave() {
      this.setState({
        // Need to make sure to not cause the visual hover/focus
        // state to stick after interacting with the input.
        focused: false,
        hovered: false
      });
    }
  }, {
    key: 'render',
    value: function render() {
      var _state = this.state,
          focused = _state.focused,
          hovered = _state.hovered;
      var _props2 = this.props,
          disabled = _props2.disabled,
          day = _props2.day,
          selected = _props2.selected;


      var showHoverStyles = hovered || focused;
      var enabledState = showHoverStyles ? 'focused' : 'resting';
      var disabledState = selected ? 'selected' : 'unselected';

      var dayPickerItemStyle = _DayPickerItemControlled.SELECTED_DAY_PICKER_ITEM_STYLES[enabledState];

      var selectedStateStyles = !disabled && !selected ? _DayPickerItemControlled.UNSELECTED_DAY_PICKER_ITEM_STYLES[enabledState] : dayPickerItemStyle;

      var dayPickerItemStyles = (0, _helpers.combineStyles)(_DayPickerItemControlled.BASE_DAY_PICKER_ITEM_STYLES, selectedStateStyles, disabled && _DayPickerItemControlled.DISABLED_DAY_PICKER_ITEM_STYLES[disabledState]);

      var inputStyles = (0, _helpers.combineStyles)(_DayPickerItemControlled.INPUT_STYLES, disabled && _DayPickerItemControlled.DISABLED_INPUT_STYLES);

      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlDayPickerItemControlled',
          style: _DayPickerItemControlled.DAY_PICKER_ITEM_CONTAINER_STYLES
        },
        _react2.default.createElement(
          'div',
          {
            ref: 'checkboxWrapper',
            style: dayPickerItemStyles
          },
          day,
          _react2.default.createElement('input', {
            checked: selected,
            disabled: disabled,
            onBlur: this._handleBlur,
            onChange: this._handleChange,
            onFocus: this._handleFocus,
            onMouseEnter: this._handleMouseEnter,
            onMouseLeave: this._handleMouseLeave,
            ref: 'checkbox',
            style: inputStyles,
            type: 'checkbox'
          })
        )
      );
    }
  }]);
  return TxlDayPickerItemControlled;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleChange', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleChange'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleBlur', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleBlur'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleFocus', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleFocus'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleMouseEnter', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseEnter'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleMouseLeave', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseLeave'), _class2.prototype)), _class2)) || _class;

exports.default = TxlDayPickerItemControlled;


TxlDayPickerItemControlled.propTypes = {
  day: _react.PropTypes.string.isRequired,
  disabled: _react.PropTypes.bool,
  index: _react.PropTypes.number.isRequired,
  onClick: _react.PropTypes.func.isRequired,
  selected: _react.PropTypes.bool
};

TxlDayPickerItemControlled.defaultProps = {
  disabled: false,
  selected: false
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlDayPickerItemControlled, 'TxlDayPickerItemControlled', 'src/input-fields/DayPickerItemControlled.jsx');
}();

;
module.exports = exports['default'];
