// txl/input-fields/DayPicker.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _toConsumableArray2 = require('babel-runtime/helpers/toConsumableArray');

var _toConsumableArray3 = _interopRequireDefault(_toConsumableArray2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _DayPickerControlled = require('./DayPickerControlled');

var _DayPickerControlled2 = _interopRequireDefault(_DayPickerControlled);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var addOrRemove = function addOrRemove(arr, val) {
  return arr.indexOf(val) >= 0
  // remove
  ? arr.filter(function (v) {
    return v !== val;
  })
  // add
  : [].concat((0, _toConsumableArray3.default)(arr), [val]);
};

var buildDayItems = function buildDayItems(labels, indices) {
  return indices.map(function (index) {
    return {
      day: labels[index],
      index: index
    };
  });
};

var TxlDayPicker = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlDayPicker, _React$Component);

  function TxlDayPicker() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlDayPicker);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlDayPicker.__proto__ || (0, _getPrototypeOf2.default)(TxlDayPicker)).call.apply(_ref, [this].concat(args)));

    _this.state = {
      selectedDayIndices: _this.props.defaultSelectedDayIndices.sort()
    };
    return _this;
  }

  (0, _createClass3.default)(TxlDayPicker, [{
    key: '_handleClick',
    value: function _handleClick(_ref2) {
      var index = _ref2.index;
      var selectedDayIndices = this.state.selectedDayIndices;
      var _props = this.props,
          dayLabels = _props.dayLabels,
          onChange = _props.onChange,
          singleDayLimit = _props.singleDayLimit;

      var newIndices = singleDayLimit ? [index] : addOrRemove(selectedDayIndices, index).sort();
      this.setState({
        selectedDayIndices: newIndices
      }, function () {
        return onChange({
          days: buildDayItems(dayLabels, newIndices)
        });
      });
    }
  }, {
    key: 'render',
    value: function render() {
      var _props2 = this.props,
          arrangement = _props2.arrangement,
          dayLabels = _props2.dayLabels,
          disabledDayIndices = _props2.disabledDayIndices;
      var selectedDayIndices = this.state.selectedDayIndices;


      return _react2.default.createElement(
        'section',
        { 'data-component': 'TxlDayPicker' },
        _react2.default.createElement(_DayPickerControlled2.default, {
          arrangement: arrangement,
          dayLabels: dayLabels,
          disabledDayIndices: disabledDayIndices,
          onClick: this._handleClick,
          ref: 'dayPickerControlled',
          selectedDayIndices: selectedDayIndices
        })
      );
    }
  }]);
  return TxlDayPicker;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleClick'), _class2.prototype)), _class2)) || _class;

exports.default = TxlDayPicker;


TxlDayPicker.propTypes = {
  arrangement: _DayPickerControlled2.default.propTypes.arrangement,
  dayLabels: _DayPickerControlled2.default.propTypes.dayLabels,
  defaultSelectedDayIndices: _DayPickerControlled2.default.propTypes.selectedDayIndices,
  disabledDayIndices: _DayPickerControlled2.default.propTypes.disabledDayIndices,
  onChange: _react.PropTypes.func,
  singleDayLimit: _react.PropTypes.bool
};

TxlDayPicker.defaultProps = {
  dayLabels: _DayPickerControlled2.default.defaultProps.dayLabels,
  defaultSelectedDayIndices: [],
  disabledDayIndices: [],
  onChange: _Base.NOOP,
  singleDayLimit: false
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(addOrRemove, 'addOrRemove', 'src/input-fields/DayPicker.jsx');

  __REACT_HOT_LOADER__.register(buildDayItems, 'buildDayItems', 'src/input-fields/DayPicker.jsx');

  __REACT_HOT_LOADER__.register(TxlDayPicker, 'TxlDayPicker', 'src/input-fields/DayPicker.jsx');
}();

;
module.exports = exports['default'];
