// txl/input-fields/DayPicker.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ITEM_STYLES = exports.LIST_STYLES = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _helpers = require('../styles/helpers');

var _list = require('../styles/mixins/list');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var HORIZONTAL_SPACING = (0, _helpers.gridUnits)(2);

var LIST_STYLES = {
  base: (0, _extends3.default)({}, _list.UNSTYLED_LIST, {
    display: 'flex',
    flexWrap: 'wrap'
  }),
  horizontal: {
    flexDirection: 'row',
    marginLeft: '-' + HORIZONTAL_SPACING
  },
  vertical: {
    flexDirection: 'column'
  }
};

var ITEM_STYLES = {
  base: {
    alignItems: 'center',
    display: 'flex',
    marginBottom: (0, _helpers.gridUnits)(1),
    marginTop: (0, _helpers.gridUnits)(1)
  },
  horizontal: {
    marginLeft: HORIZONTAL_SPACING
  }
};

exports.LIST_STYLES = LIST_STYLES;
exports.ITEM_STYLES = ITEM_STYLES;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(HORIZONTAL_SPACING, 'HORIZONTAL_SPACING', 'src/input-fields/DayPicker.style.js');

  __REACT_HOT_LOADER__.register(LIST_STYLES, 'LIST_STYLES', 'src/input-fields/DayPicker.style.js');

  __REACT_HOT_LOADER__.register(ITEM_STYLES, 'ITEM_STYLES', 'src/input-fields/DayPicker.style.js');
}();

;
