// txl/input-fields/DayPickerItemControlled.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UNSELECTED_DAY_PICKER_ITEM_STYLES = exports.SELECTED_DAY_PICKER_ITEM_STYLES = exports.INPUT_STYLES = exports.DISABLED_INPUT_STYLES = exports.DISABLED_DAY_PICKER_ITEM_STYLES = exports.DAY_PICKER_ITEM_CONTAINER_STYLES = exports.BASE_DAY_PICKER_ITEM_STYLES = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _transitions = require('../styles/mixins/transitions');

var _type = require('../styles/mixins/type');

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BASE_DAY_PICKER_ITEM_STYLES = (0, _extends3.default)({}, (0, _helpers.square)((0, _helpers.gridUnits)(5)), _type.TYPE_SMALL, {
  alignItems: 'center',
  backgroundColor: _theme.COLOR_NEUTRAL['800c'],
  borderColor: 'transparent',
  borderRadius: 2,
  borderStyle: 'solid',
  borderWidth: '1px',
  display: 'flex',
  justifyContent: 'center',
  overflow: 'hidden',
  position: 'relative',
  transition: (0, _transitions.cubicTransition)('background')
});

var DAY_PICKER_ITEM_CONTAINER_STYLES = (0, _extends3.default)({}, (0, _helpers.square)((0, _helpers.gridUnits)(5)), {
  alignItems: 'center',
  display: 'flex',
  justifyContent: 'center'
});

var DISABLED_DAY_PICKER_ITEM_STYLES = {
  selected: {
    backgroundColor: _theme.COLOR_NEUTRAL['200'],
    borderColor: _theme.COLOR_NEUTRAL['200'],
    color: 'white',
    cursor: 'not-allowed'
  },
  unselected: {
    backgroundColor: _theme.COLOR_NEUTRAL['100'],
    borderColor: _theme.COLOR_NEUTRAL['200'],
    color: _theme.COLOR_NEUTRAL['800'],
    cursor: 'not-allowed'
  }
};

var DISABLED_INPUT_STYLES = {
  display: 'none'
};

var INPUT_STYLES = {
  border: 'none',
  cursor: 'pointer',
  height: '100%',
  left: 0,
  margin: 0,
  opacity: 0,
  position: 'absolute',
  top: 0,
  width: '100%'
};

var SELECTED_DAY_PICKER_ITEM_STYLES = {
  focused: {
    backgroundColor: _theme.COLOR_ACCENT['600'],
    borderColor: _theme.COLOR_ACCENT['600'],
    color: _theme.COLOR_ACCENT['600c']
  },
  resting: {
    backgroundColor: _theme.COLOR_ACCENT['500'],
    borderColor: _theme.COLOR_ACCENT['500'],
    color: _theme.COLOR_ACCENT['500c']
  }
};

var UNSELECTED_DAY_PICKER_ITEM_STYLES = {
  focused: {
    backgroundColor: _theme.COLOR_NEUTRAL['800c'],
    borderColor: _theme.COLOR_NEUTRAL['400'],
    color: _theme.COLOR_NEUTRAL['800']
  },
  resting: {
    backgroundColor: _theme.COLOR_NEUTRAL['800c'],
    borderColor: _theme.COLOR_NEUTRAL['300'],
    color: _theme.COLOR_NEUTRAL['800']
  }
};

exports.BASE_DAY_PICKER_ITEM_STYLES = BASE_DAY_PICKER_ITEM_STYLES;
exports.DAY_PICKER_ITEM_CONTAINER_STYLES = DAY_PICKER_ITEM_CONTAINER_STYLES;
exports.DISABLED_DAY_PICKER_ITEM_STYLES = DISABLED_DAY_PICKER_ITEM_STYLES;
exports.DISABLED_INPUT_STYLES = DISABLED_INPUT_STYLES;
exports.INPUT_STYLES = INPUT_STYLES;
exports.SELECTED_DAY_PICKER_ITEM_STYLES = SELECTED_DAY_PICKER_ITEM_STYLES;
exports.UNSELECTED_DAY_PICKER_ITEM_STYLES = UNSELECTED_DAY_PICKER_ITEM_STYLES;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(BASE_DAY_PICKER_ITEM_STYLES, 'BASE_DAY_PICKER_ITEM_STYLES', 'src/input-fields/DayPickerItemControlled.style.js');

  __REACT_HOT_LOADER__.register(DAY_PICKER_ITEM_CONTAINER_STYLES, 'DAY_PICKER_ITEM_CONTAINER_STYLES', 'src/input-fields/DayPickerItemControlled.style.js');

  __REACT_HOT_LOADER__.register(DISABLED_DAY_PICKER_ITEM_STYLES, 'DISABLED_DAY_PICKER_ITEM_STYLES', 'src/input-fields/DayPickerItemControlled.style.js');

  __REACT_HOT_LOADER__.register(DISABLED_INPUT_STYLES, 'DISABLED_INPUT_STYLES', 'src/input-fields/DayPickerItemControlled.style.js');

  __REACT_HOT_LOADER__.register(INPUT_STYLES, 'INPUT_STYLES', 'src/input-fields/DayPickerItemControlled.style.js');

  __REACT_HOT_LOADER__.register(SELECTED_DAY_PICKER_ITEM_STYLES, 'SELECTED_DAY_PICKER_ITEM_STYLES', 'src/input-fields/DayPickerItemControlled.style.js');

  __REACT_HOT_LOADER__.register(UNSELECTED_DAY_PICKER_ITEM_STYLES, 'UNSELECTED_DAY_PICKER_ITEM_STYLES', 'src/input-fields/DayPickerItemControlled.style.js');
}();

;
