// txl/input-fields/FileDropZone.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PLAIN_DROP_AREA_STYLE = exports.getUploadIconProps = exports.getBrowseLabelStyle = exports.FILE_UPLOAD_STYLE = exports.COMPACT_DROP_AREA_STYLE = exports.BROWSE_LINK_STYLE = exports.BROWSE_LINK_DISABLED_STYLE = exports.BORDERLESS_DROP_AREA_STYLE = exports.BORDERED_DROP_AREA_STYLE = exports.BASE_DROP_AREA_STYLE = exports.ACCENTED_DROP_AREA_STYLE = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _theme = require('../styles/theme');

var _InputStyleHelpers = require('./_InputStyleHelpers');

var _link = require('../styles/mixins/link');

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FILE_UPLOAD_STYLE = {
  display: 'none'
};

var BROWSE_LINK_STYLE = _link.LINK_STYLE;
var BROWSE_LINK_DISABLED_STYLE = { color: _theme.COLOR_NEUTRAL['100'] };

function getBrowseLabelStyle(_ref) {
  var disabled = _ref.disabled,
      showsBorder = _ref.showsBorder,
      compact = _ref.compact;

  var BASE_BROWSE_LABEL_STYLE = (0, _extends3.default)({}, _InputStyleHelpers.INPUT_GROUP_LABEL_STYLES, {
    lineHeight: (0, _helpers.gridUnits)(4),
    paddingBottom: compact ? (0, _helpers.gridUnits)(1) : 0,
    paddingTop: showsBorder && !compact ? (0, _helpers.gridUnits)(4) : (0, _helpers.gridUnits)(1)
  });

  var BROWSE_LABEL_DISABLED_STYLE = (0, _extends3.default)({}, BASE_BROWSE_LABEL_STYLE, {
    color: _theme.COLOR_NEUTRAL['100']
  });

  return disabled ? BROWSE_LABEL_DISABLED_STYLE : BASE_BROWSE_LABEL_STYLE;
}

function getUploadIconProps(_ref2) {
  var disabled = _ref2.disabled,
      dragging = _ref2.dragging,
      showsBorder = _ref2.showsBorder,
      compact = _ref2.compact;

  var iconColor = dragging && !disabled ? _theme.COLOR_ACCENT['200'] : _theme.COLOR_NEUTRAL['100'];
  var iconScale = showsBorder && !compact ? 2 : 1;

  return {
    color: iconColor,
    scale: iconScale
  };
}

var BASE_DROP_AREA_STYLE = {
  boxSizing: 'border-box',
  height: '100%',
  textAlign: 'center',
  width: '100%'
};

var ACCENTED_DROP_AREA_STYLE = {
  backgroundColor: _theme.COLOR_ACCENT['50'],
  outlineColor: _theme.COLOR_ACCENT['400']
};

var PLAIN_DROP_AREA_STYLE = {
  backgroundColor: 'transparent',
  outlineColor: _theme.COLOR_NEUTRAL['200']
};

var BORDERED_DROP_AREA_STYLE = {
  alignItems: 'center',
  display: 'flex',
  flexFlow: 'column wrap',
  justifyContent: 'center',
  outlineStyle: 'dashed',
  outlineWidth: '1px',
  paddingBottom: (0, _helpers.gridUnits)(4),
  paddingLeft: (0, _helpers.gridUnits)(4),
  paddingRight: (0, _helpers.gridUnits)(4),
  paddingTop: (0, _helpers.gridUnits)(4)
};

var COMPACT_DROP_AREA_STYLE = {
  paddingBottom: (0, _helpers.gridUnits)(2),
  paddingLeft: (0, _helpers.gridUnits)(2),
  paddingRight: (0, _helpers.gridUnits)(2),
  paddingTop: (0, _helpers.gridUnits)(2)
};

var BORDERLESS_DROP_AREA_STYLE = {
  borderWidth: '0',
  outlineStyle: 'none',
  paddingBottom: (0, _helpers.gridUnits)(4),
  paddingLeft: (0, _helpers.gridUnits)(4),
  paddingRight: (0, _helpers.gridUnits)(4),
  paddingTop: (0, _helpers.gridUnits)(12)
};

exports.ACCENTED_DROP_AREA_STYLE = ACCENTED_DROP_AREA_STYLE;
exports.BASE_DROP_AREA_STYLE = BASE_DROP_AREA_STYLE;
exports.BORDERED_DROP_AREA_STYLE = BORDERED_DROP_AREA_STYLE;
exports.BORDERLESS_DROP_AREA_STYLE = BORDERLESS_DROP_AREA_STYLE;
exports.BROWSE_LINK_DISABLED_STYLE = BROWSE_LINK_DISABLED_STYLE;
exports.BROWSE_LINK_STYLE = BROWSE_LINK_STYLE;
exports.COMPACT_DROP_AREA_STYLE = COMPACT_DROP_AREA_STYLE;
exports.FILE_UPLOAD_STYLE = FILE_UPLOAD_STYLE;
exports.getBrowseLabelStyle = getBrowseLabelStyle;
exports.getUploadIconProps = getUploadIconProps;
exports.PLAIN_DROP_AREA_STYLE = PLAIN_DROP_AREA_STYLE;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(FILE_UPLOAD_STYLE, 'FILE_UPLOAD_STYLE', 'src/input-fields/FileDropZone.style.js');

  __REACT_HOT_LOADER__.register(BROWSE_LINK_STYLE, 'BROWSE_LINK_STYLE', 'src/input-fields/FileDropZone.style.js');

  __REACT_HOT_LOADER__.register(BROWSE_LINK_DISABLED_STYLE, 'BROWSE_LINK_DISABLED_STYLE', 'src/input-fields/FileDropZone.style.js');

  __REACT_HOT_LOADER__.register(getBrowseLabelStyle, 'getBrowseLabelStyle', 'src/input-fields/FileDropZone.style.js');

  __REACT_HOT_LOADER__.register(getUploadIconProps, 'getUploadIconProps', 'src/input-fields/FileDropZone.style.js');

  __REACT_HOT_LOADER__.register(BASE_DROP_AREA_STYLE, 'BASE_DROP_AREA_STYLE', 'src/input-fields/FileDropZone.style.js');

  __REACT_HOT_LOADER__.register(ACCENTED_DROP_AREA_STYLE, 'ACCENTED_DROP_AREA_STYLE', 'src/input-fields/FileDropZone.style.js');

  __REACT_HOT_LOADER__.register(PLAIN_DROP_AREA_STYLE, 'PLAIN_DROP_AREA_STYLE', 'src/input-fields/FileDropZone.style.js');

  __REACT_HOT_LOADER__.register(BORDERED_DROP_AREA_STYLE, 'BORDERED_DROP_AREA_STYLE', 'src/input-fields/FileDropZone.style.js');

  __REACT_HOT_LOADER__.register(COMPACT_DROP_AREA_STYLE, 'COMPACT_DROP_AREA_STYLE', 'src/input-fields/FileDropZone.style.js');

  __REACT_HOT_LOADER__.register(BORDERLESS_DROP_AREA_STYLE, 'BORDERLESS_DROP_AREA_STYLE', 'src/input-fields/FileDropZone.style.js');
}();

;
