// txl/input-fields/FileDropZone.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _FileDropZoneControlled = require('./FileDropZoneControlled');

var _FileDropZoneControlled2 = _interopRequireDefault(_FileDropZoneControlled);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlFileDropZone = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlFileDropZone, _React$Component);

  function TxlFileDropZone() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlFileDropZone);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlFileDropZone.__proto__ || (0, _getPrototypeOf2.default)(TxlFileDropZone)).call.apply(_ref, [this].concat(args)));

    _this.state = {
      dragging: false
    };
    return _this;
  }

  (0, _createClass3.default)(TxlFileDropZone, [{
    key: '_handleDragStart',
    value: function _handleDragStart(_ref2) {
      var event = _ref2.event;

      event.preventDefault();
    }
  }, {
    key: '_handleDragEnter',
    value: function _handleDragEnter(_ref3) {
      var event = _ref3.event;

      event.preventDefault();
      this.lastEntered = event.target;
      this.setState({ dragging: true });
    }
  }, {
    key: '_handleDragLeave',
    value: function _handleDragLeave(_ref4) {
      var event = _ref4.event;

      event.preventDefault();
      if (this.lastEntered === event.target) {
        this.setState({ dragging: false });
      }
    }
  }, {
    key: '_handleDragOver',
    value: function _handleDragOver(_ref5) {
      var event = _ref5.event;

      event.preventDefault();
    }
  }, {
    key: '_handleDrop',
    value: function _handleDrop(_ref6) {
      var event = _ref6.event;

      event.preventDefault();
      this.setState({ dragging: false });

      this._handleFilesSelected({ event: event });
    }
  }, {
    key: '_handleFilesSelected',
    value: function _handleFilesSelected(_ref7) {
      var event = _ref7.event;
      var dataTransfer = event.dataTransfer,
          target = event.target;

      var selectedFiles = dataTransfer ? dataTransfer.files : target.files;
      var selectedFilesList = [];
      var numberOfFilesToSelect = this.props.multiFile ? selectedFiles.length : 1;
      for (var i = 0; i < numberOfFilesToSelect; i++) {
        selectedFilesList.push(selectedFiles.item(i));
      }

      this.props.onFilesSelected({ files: selectedFilesList });
    }
  }, {
    key: 'render',
    value: function render() {
      var dragging = this.state.dragging;
      var _props = this.props,
          compact = _props.compact,
          disabled = _props.disabled,
          multiFile = _props.multiFile,
          showsBorder = _props.showsBorder;


      return _react2.default.createElement(_FileDropZoneControlled2.default, {
        'data-component': 'TxlFileDropZoneControlled',
        compact: compact,
        disabled: disabled,
        dragging: dragging,
        multiFile: multiFile,
        onDragEnter: this._handleDragEnter,
        onDragLeave: this._handleDragLeave,
        onDragOver: this._handleDragOver,
        onDragStart: this._handleDragStart,
        onDrop: this._handleDrop,
        onFilesSelected: this._handleFilesSelected,
        showsBorder: showsBorder
      });
    }
  }]);
  return TxlFileDropZone;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleDragStart', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDragStart'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDragEnter', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDragEnter'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDragLeave', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDragLeave'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDragOver', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDragOver'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDrop', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDrop'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleFilesSelected', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleFilesSelected'), _class2.prototype)), _class2)) || _class;

exports.default = TxlFileDropZone;


TxlFileDropZone.displayName = 'TxlFileDropZone';

TxlFileDropZone.propTypes = {
  compact: _react.PropTypes.bool,
  disabled: _react.PropTypes.bool,
  multiFile: _react.PropTypes.bool,
  onFilesSelected: _react.PropTypes.func.isRequired,
  showsBorder: _react.PropTypes.bool
};

TxlFileDropZone.defaultProps = {
  compact: false,
  disabled: false,
  multiFile: false,
  showsBorder: true
};

TxlFileDropZone.displayName = 'TxlFileDropZone';
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlFileDropZone, 'TxlFileDropZone', 'src/input-fields/FileDropZone.jsx');
}();

;
module.exports = exports['default'];
