// txl/input-fields/FileUploader.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MULTIFILE_DROP_ZONE_HEIGHT = exports.SINGLE_FILE_DROP_ZONE_WIDTH = exports.SINGLE_FILE_DROP_ZONE_HEIGHT = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _helpers = require('../styles/helpers');

var _WithLabel = require('./_WithLabel');

var _WithLabel2 = _interopRequireDefault(_WithLabel);

var _FileDropZone = require('./FileDropZone');

var _FileDropZone2 = _interopRequireDefault(_FileDropZone);

var _FilePreviewWithDetailsModal = require('../file-preview/FilePreviewWithDetailsModal');

var _FilePreviewWithDetailsModal2 = _interopRequireDefault(_FilePreviewWithDetailsModal);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SINGLE_FILE_DROP_ZONE_HEIGHT = exports.SINGLE_FILE_DROP_ZONE_HEIGHT = (0, _helpers.gridUnitsRaw)(26);
var SINGLE_FILE_DROP_ZONE_WIDTH = exports.SINGLE_FILE_DROP_ZONE_WIDTH = (0, _helpers.gridUnitsRaw)(40);

var MULTIFILE_DROP_ZONE_HEIGHT = exports.MULTIFILE_DROP_ZONE_HEIGHT = (0, _helpers.gridUnitsRaw)(57);
var FILE_PREVIEW_VERTICAL_PADDING = (0, _helpers.gridUnits)(4);
var FILE_PREVIEW_HORIZONTAL_PADDING = (0, _helpers.gridUnits)(2);

var PREVIEW_STYLE = {
  paddingLeft: FILE_PREVIEW_HORIZONTAL_PADDING,
  paddingRight: FILE_PREVIEW_HORIZONTAL_PADDING,
  paddingTop: FILE_PREVIEW_VERTICAL_PADDING
};

var PREVIEWS_CONTAINER_STYLE = {
  display: 'flex',
  flexFlow: 'row wrap',
  marginLeft: '-' + FILE_PREVIEW_HORIZONTAL_PADDING,
  marginTop: '-' + FILE_PREVIEW_VERTICAL_PADDING
};

var TxlFileUploader = function TxlFileUploader(props) {
  var requestedDropZoneHeight = props.dropZoneHeight,
      fileList = props.fileList,
      multiFile = props.multiFile,
      onFileCancelled = props.onFileCancelled,
      onFileDeleted = props.onFileDeleted,
      onFileReplaced = props.onFileReplaced,
      onFilesSelected = props.onFilesSelected,
      width = props.width;


  var hasSelectedFiles = fileList.length > 0;

  var dropZoneHeight = void 0;
  if (requestedDropZoneHeight) {
    dropZoneHeight = requestedDropZoneHeight + 'px';
  } else {
    dropZoneHeight = multiFile ? MULTIFILE_DROP_ZONE_HEIGHT + 'px' : SINGLE_FILE_DROP_ZONE_HEIGHT + 'px';
  }

  var uploaderWidth = void 0;
  if (width) {
    uploaderWidth = width + 'px';
  } else {
    uploaderWidth = multiFile ? '100%' : SINGLE_FILE_DROP_ZONE_WIDTH + 'px';
  }

  var isCompact = requestedDropZoneHeight < SINGLE_FILE_DROP_ZONE_HEIGHT || width < SINGLE_FILE_DROP_ZONE_WIDTH;

  var shouldDisableDropZone = false;
  var filePreviews = fileList.map(function (fileInfo, index) {
    if (fileInfo.uploading) {
      shouldDisableDropZone = true;
    }

    return _react2.default.createElement(
      'div',
      {
        style: PREVIEW_STYLE,
        key: index
      },
      _react2.default.createElement(_FilePreviewWithDetailsModal2.default, {
        fileInfo: fileInfo,
        onDelete: onFileDeleted,
        onEdit: onFileReplaced,
        onCancel: onFileCancelled
      })
    );
  });

  var fileDropZone = _react2.default.createElement(
    'div',
    {
      id: 'DropZoneContainer',
      style: { height: dropZoneHeight }
    },
    _react2.default.createElement(_FileDropZone2.default, {
      compact: isCompact,
      disabled: shouldDisableDropZone,
      multiFile: multiFile,
      onFilesSelected: onFilesSelected,
      showsBorder: !hasSelectedFiles
    })
  );

  var isSingleFileWithoutSelectedFiles = !multiFile && !hasSelectedFiles;
  var shouldShowDropZone = multiFile || isSingleFileWithoutSelectedFiles;
  return _react2.default.createElement(
    'div',
    {
      'data-component': 'TxlFileUploader',
      style: { width: uploaderWidth }
    },
    filePreviews.length > 0 && _react2.default.createElement(
      'div',
      {
        id: 'FilePreviewsContainer',
        style: PREVIEWS_CONTAINER_STYLE
      },
      filePreviews
    ),
    shouldShowDropZone && _react2.default.createElement(
      'div',
      null,
      fileDropZone
    )
  );
};

TxlFileUploader.displayName = 'TxlFileUploader';

TxlFileUploader.propTypes = {
  dropZoneHeight: _react.PropTypes.number,
  fileList: _react.PropTypes.array,
  labelText: _react.PropTypes.string,
  multiFile: _react.PropTypes.bool,
  onFileCancelled: _react.PropTypes.func,
  onFileDeleted: _react.PropTypes.func,
  onFileReplaced: _react.PropTypes.func,
  onFilesSelected: _react.PropTypes.func,
  secondaryText: _react.PropTypes.string,
  validationState: _react.PropTypes.oneOf(['base', 'error']),
  width: _react.PropTypes.number
};

TxlFileUploader.defaultProps = {
  fileList: [],
  multiFile: false,
  onFileCancelled: _Base.NOOP,
  onFileDeleted: _Base.NOOP,
  onFileReplaced: _Base.NOOP,
  onFilesSelected: _Base.NOOP
};

var _default = (0, _WithLabel2.default)({
  defaultPosition: 'top',
  padding: (0, _helpers.gridUnits)(3)
}, (0, _radium2.default)(TxlFileUploader));

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(SINGLE_FILE_DROP_ZONE_HEIGHT, 'SINGLE_FILE_DROP_ZONE_HEIGHT', 'src/input-fields/FileUploader.jsx');

  __REACT_HOT_LOADER__.register(SINGLE_FILE_DROP_ZONE_WIDTH, 'SINGLE_FILE_DROP_ZONE_WIDTH', 'src/input-fields/FileUploader.jsx');

  __REACT_HOT_LOADER__.register(MULTIFILE_DROP_ZONE_HEIGHT, 'MULTIFILE_DROP_ZONE_HEIGHT', 'src/input-fields/FileUploader.jsx');

  __REACT_HOT_LOADER__.register(FILE_PREVIEW_VERTICAL_PADDING, 'FILE_PREVIEW_VERTICAL_PADDING', 'src/input-fields/FileUploader.jsx');

  __REACT_HOT_LOADER__.register(FILE_PREVIEW_HORIZONTAL_PADDING, 'FILE_PREVIEW_HORIZONTAL_PADDING', 'src/input-fields/FileUploader.jsx');

  __REACT_HOT_LOADER__.register(PREVIEW_STYLE, 'PREVIEW_STYLE', 'src/input-fields/FileUploader.jsx');

  __REACT_HOT_LOADER__.register(PREVIEWS_CONTAINER_STYLE, 'PREVIEWS_CONTAINER_STYLE', 'src/input-fields/FileUploader.jsx');

  __REACT_HOT_LOADER__.register(TxlFileUploader, 'TxlFileUploader', 'src/input-fields/FileUploader.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/input-fields/FileUploader.jsx');
}();

;
