// txl/input-fields/FileDropZoneControlled.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _Upload = require('../icons/Upload');

var _Upload2 = _interopRequireDefault(_Upload);

var _FileDropZone = require('./FileDropZone.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

function preventDefault(event) {
  event && event.preventDefault();
}

var TxlFileDropZoneControlled = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlFileDropZoneControlled, _React$Component);

  function TxlFileDropZoneControlled() {
    (0, _classCallCheck3.default)(this, TxlFileDropZoneControlled);
    return (0, _possibleConstructorReturn3.default)(this, (TxlFileDropZoneControlled.__proto__ || (0, _getPrototypeOf2.default)(TxlFileDropZoneControlled)).apply(this, arguments));
  }

  (0, _createClass3.default)(TxlFileDropZoneControlled, [{
    key: '_handleDragEnter',
    value: function _handleDragEnter(event) {
      this.props.onDragEnter({ event: event });
    }
  }, {
    key: '_handleDragLeave',
    value: function _handleDragLeave(event) {
      this.props.onDragLeave({ event: event });
    }
  }, {
    key: '_handleDragOver',
    value: function _handleDragOver(event) {
      this.props.onDragOver({ event: event });
    }
  }, {
    key: '_handleDragStart',
    value: function _handleDragStart(event) {
      this.props.onDragStart({ event: event });
    }
  }, {
    key: '_handleDrop',
    value: function _handleDrop(event) {
      this.props.onDrop({ event: event });
    }
  }, {
    key: '_handleFilesSelected',
    value: function _handleFilesSelected(event) {
      this.props.onFilesSelected({ event: event });
    }
  }, {
    key: '_handleDropZoneRef',
    value: function _handleDropZoneRef(ref) {
      this._dropZone = ref;
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          compact = _props.compact,
          disabled = _props.disabled,
          dragging = _props.dragging,
          multiFile = _props.multiFile,
          showsBorder = _props.showsBorder;


      var browseLinkStyle = disabled ? _FileDropZone.BROWSE_LINK_DISABLED_STYLE : _FileDropZone.BROWSE_LINK_STYLE;
      var browseLabelStyle = (0, _FileDropZone.getBrowseLabelStyle)(this.props);
      var uploadIconProps = (0, _FileDropZone.getUploadIconProps)(this.props);

      var accentStyle = dragging && showsBorder ? _FileDropZone.ACCENTED_DROP_AREA_STYLE : _FileDropZone.PLAIN_DROP_AREA_STYLE;
      var borderStyle = showsBorder ? _FileDropZone.BORDERED_DROP_AREA_STYLE : _FileDropZone.BORDERLESS_DROP_AREA_STYLE;
      var compactStyle = compact && _FileDropZone.COMPACT_DROP_AREA_STYLE;
      var dropAreaStyle = (0, _extends3.default)({}, _FileDropZone.BASE_DROP_AREA_STYLE, accentStyle, borderStyle, compactStyle);

      var BrowseLinkEl = disabled ? 'label' : 'a';
      var fileText = multiFile ? 'files' : 'file';

      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlFileDropZoneControlled',
          onDragEnter: disabled ? preventDefault : this._handleDragEnter,
          onDragLeave: disabled ? preventDefault : this._handleDragLeave,
          onDragOver: disabled ? preventDefault : this._handleDragOver,
          onDragStart: disabled ? preventDefault : this._handleDragStart,
          onDrop: disabled ? preventDefault : this._handleDrop,
          ref: this._handleDropZoneRef,
          style: dropAreaStyle
        },
        !compact && _react2.default.createElement(_Upload2.default, uploadIconProps),
        _react2.default.createElement(
          'label',
          { style: browseLabelStyle },
          'Drop ',
          fileText,
          ' here or',
          _react2.default.createElement(
            BrowseLinkEl,
            { style: browseLinkStyle },
            ' browse '
          ),
          _react2.default.createElement('input', {
            multiple: multiFile,
            onChange: this._handleFilesSelected,
            style: _FileDropZone.FILE_UPLOAD_STYLE,
            type: 'file'
          }),
          'to upload'
        ),
        compact && _react2.default.createElement(_Upload2.default, uploadIconProps)
      );
    }
  }]);
  return TxlFileDropZoneControlled;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleDragEnter', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDragEnter'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDragLeave', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDragLeave'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDragOver', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDragOver'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDragStart', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDragStart'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDrop', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDrop'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleFilesSelected', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleFilesSelected'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDropZoneRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDropZoneRef'), _class2.prototype)), _class2)) || _class;

exports.default = TxlFileDropZoneControlled;


TxlFileDropZoneControlled.displayName = 'TxlFileDropZoneControlled';

TxlFileDropZoneControlled.propTypes = {
  compact: _react.PropTypes.bool,
  disabled: _react.PropTypes.bool,
  dragging: _react.PropTypes.bool,
  multiFile: _react.PropTypes.bool,
  onDragEnter: _react.PropTypes.func,
  onDragLeave: _react.PropTypes.func,
  onDragOver: _react.PropTypes.func,
  onDragStart: _react.PropTypes.func,
  onDrop: _react.PropTypes.func,
  onFilesSelected: _react.PropTypes.func.isRequired,
  showsBorder: _react.PropTypes.bool
};

TxlFileDropZoneControlled.defaultProps = {
  compact: false,
  disabled: false,
  dragging: false,
  multiFile: false,
  onDragEnter: _Base.NOOP,
  onDragLeave: _Base.NOOP,
  onDragOver: _Base.NOOP,
  onDragStart: _Base.NOOP,
  onDrop: _Base.NOOP,
  showsBorder: true
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(preventDefault, 'preventDefault', 'src/input-fields/FileDropZoneControlled.jsx');

  __REACT_HOT_LOADER__.register(TxlFileDropZoneControlled, 'TxlFileDropZoneControlled', 'src/input-fields/FileDropZoneControlled.jsx');
}();

;
module.exports = exports['default'];
