// txl/input-fields/PillBox.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _PillButton = require('./_PillButton');

var _PillButton2 = _interopRequireDefault(_PillButton);

var _theme = require('../styles/theme');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var PILLBOX_STYLES = {
  border: '1px solid ' + _theme.COLOR_NEUTRAL['100'],
  borderRadius: '2px', // @TODO: Share from button styles
  display: 'inline-flex',
  overflow: 'hidden'
};

var TxlPillBox = function TxlPillBox(props) {
  return _react2.default.createElement(
    'div',
    {
      style: PILLBOX_STYLES,
      'data-component': 'TxlPillBox'
    },
    props.buttons.map(function (pill, idx) {
      var selected = idx === props.selectedIndex;

      return _react2.default.createElement(
        _PillButton2.default,
        (0, _extends3.default)({
          key: idx,
          index: idx,
          selected: selected,
          onChange: props.onChange
        }, pill),
        pill.labelText
      );
    })
  );
};

TxlPillBox.propTypes = {
  buttons: _react.PropTypes.arrayOf(_react.PropTypes.shape({
    id: _PillButton2.default.propTypes.id,
    labelText: _react.PropTypes.node
  })).isRequired,
  onChange: _react.PropTypes.func.isRequired,
  selectedIndex: _react.PropTypes.number
};

var _default = (0, _radium2.default)(TxlPillBox);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(PILLBOX_STYLES, 'PILLBOX_STYLES', 'src/input-fields/PillBox.jsx');

  __REACT_HOT_LOADER__.register(TxlPillBox, 'TxlPillBox', 'src/input-fields/PillBox.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/input-fields/PillBox.jsx');
}();

;
module.exports = exports['default'];
