// txl/input-fields/RadioGroup.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Radio = require('./Radio');

var _Radio2 = _interopRequireDefault(_Radio);

var _WithLabel = require('./_WithLabel');

var _WithLabel2 = _interopRequireDefault(_WithLabel);

var _helpers = require('../styles/helpers');

var _list = require('../styles/mixins/list');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var HORIZONTAL_SPACING = (0, _helpers.gridUnits)(6);

var LIST_STYLES = {
  base: (0, _extends3.default)({}, _list.UNSTYLED_LIST, {
    display: 'flex',
    flexWrap: 'wrap'
  }),
  horizontal: {
    flexDirection: 'row',
    marginLeft: '-' + HORIZONTAL_SPACING
  },
  vertical: {
    flexDirection: 'column'
  }
};

var ITEM_STYLES = {
  base: {
    alignItems: 'center',
    display: 'flex',
    marginBottom: (0, _helpers.gridUnits)(1),
    marginTop: (0, _helpers.gridUnits)(1)
  },
  horizontal: {
    marginLeft: HORIZONTAL_SPACING
  }
};

var TxlRadioGroup = function TxlRadioGroup(props) {
  var listStyle = (0, _helpers.combineStyles)(LIST_STYLES.base, LIST_STYLES[props.arrangement]);

  var listItemStyle = (0, _helpers.combineStyles)(ITEM_STYLES.base, props.arrangement === 'horizontal' ? ITEM_STYLES[props.arrangement] : {});

  return _react2.default.createElement(
    'section',
    { 'data-component': 'TxlRadioGroup' },
    _react2.default.createElement(
      'ul',
      {
        style: listStyle,
        'data-test-radio-group-list': true
      },
      props.items.map(function (item) {
        return _react2.default.createElement(
          'li',
          {
            style: listItemStyle,
            key: item.value
          },
          _react2.default.createElement(_Radio2.default, {
            disabled: item.disabled,
            id: item.id,
            labelText: item.labelText,
            name: props.name,
            checked: item.checked,
            onChange: props.onChange,
            tabIndex: props.tabIndex,
            value: item.value
          })
        );
      })
    )
  );
};

TxlRadioGroup.propTypes = {
  arrangement: _react2.default.PropTypes.oneOf(['horizontal', 'vertical']),
  items: _react2.default.PropTypes.arrayOf(_react2.default.PropTypes.shape({
    // We will be passing these props to Radio so use the props
    // that we would like to expose from RadioGroup
    checked: _Radio2.default.propTypes.checked,
    disabled: _Radio2.default.propTypes.disabled,
    id: _Radio2.default.propTypes.id,
    labelText: _Radio2.default.propTypes.labelText,
    value: _Radio2.default.propTypes.value
  })).isRequired,
  labelText: _react2.default.PropTypes.string,
  name: _react2.default.PropTypes.string.isRequired,
  onChange: _react2.default.PropTypes.func.isRequired,
  secondaryText: _react2.default.PropTypes.string,
  tabIndex: _react2.default.PropTypes.number,
  validationState: _react2.default.PropTypes.oneOf(['base', 'error', 'warning', 'success'])
};

TxlRadioGroup.defaultProps = {
  arrangement: 'vertical',
  tabIndex: 0,
  validationState: 'base'
};

var _default = (0, _WithLabel2.default)({
  defaultPosition: 'top',
  padding: (0, _helpers.gridUnits)(1)
}, (0, _radium2.default)(TxlRadioGroup));

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(HORIZONTAL_SPACING, 'HORIZONTAL_SPACING', 'src/input-fields/RadioGroup.jsx');

  __REACT_HOT_LOADER__.register(LIST_STYLES, 'LIST_STYLES', 'src/input-fields/RadioGroup.jsx');

  __REACT_HOT_LOADER__.register(ITEM_STYLES, 'ITEM_STYLES', 'src/input-fields/RadioGroup.jsx');

  __REACT_HOT_LOADER__.register(TxlRadioGroup, 'TxlRadioGroup', 'src/input-fields/RadioGroup.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/input-fields/RadioGroup.jsx');
}();

;
module.exports = exports['default'];
