// txl/input-fields/SearchBox.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

var _Close = require('../icons/Close');

var _Close2 = _interopRequireDefault(_Close);

var _IconButton = require('../buttons/IconButton');

var _IconButton2 = _interopRequireDefault(_IconButton);

var _Search = require('../icons/Search');

var _Search2 = _interopRequireDefault(_Search);

var _TextField = require('./TextField');

var _TextField2 = _interopRequireDefault(_TextField);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var SEARCH_ICON_COLOR = _theme.COLOR_NEUTRAL['300'];

var FORM_STYLES = {
  position: 'relative'
};

var CLEAR_BUTTON_STYLES = (0, _extends3.default)({}, (0, _helpers.square)((0, _helpers.gridUnits)(5)));

var TxlSearchBox = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlSearchBox, _React$Component);

  function TxlSearchBox() {
    (0, _classCallCheck3.default)(this, TxlSearchBox);
    return (0, _possibleConstructorReturn3.default)(this, (TxlSearchBox.__proto__ || (0, _getPrototypeOf2.default)(TxlSearchBox)).apply(this, arguments));
  }

  (0, _createClass3.default)(TxlSearchBox, [{
    key: 'focus',
    value: function focus() {
      this.refs.textField.focus();
    }
  }, {
    key: 'select',
    value: function select() {
      this.refs.textField.select();
    }
  }, {
    key: '_handleClear',
    value: function _handleClear() {
      this.props.onClear({
        value: this.props.value
      });
    }
  }, {
    key: '_handleSubmit',
    value: function _handleSubmit(e) {
      this.props.onSubmit({
        value: this.props.value
      });
      e.preventDefault();
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          id = _props.id,
          onChange = _props.onChange,
          onClear = _props.onClear,
          placeholder = _props.placeholder,
          value = _props.value;

      var clearButton = void 0;

      if (value && onClear) {
        clearButton = _react2.default.createElement(
          'div',
          {
            'data-test': 'clear-button-wrap',
            style: CLEAR_BUTTON_STYLES
          },
          _react2.default.createElement(_IconButton2.default, {
            onClick: this._handleClear,
            icon: _Close2.default,
            variant: 'plain'
          })
        );
      }

      var BEFORE_CONTENT = _react2.default.createElement(_Search2.default, { color: SEARCH_ICON_COLOR });

      return _react2.default.createElement(
        'form',
        {
          'data-component': 'TxlSearchBox',
          onSubmit: this._handleSubmit,
          ref: 'searchForm',
          style: FORM_STYLES
        },
        _react2.default.createElement(_TextField2.default, {
          _afterContent: clearButton,
          _beforeContent: BEFORE_CONTENT,
          id: id,
          onChange: onChange,
          value: value,
          ref: 'textField',
          placeholder: placeholder
        })
      );
    }
  }]);
  return TxlSearchBox;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, 'focus', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, 'focus'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, 'select', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, 'select'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleClear', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleClear'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleSubmit', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleSubmit'), _class2.prototype)), _class2)) || _class;

exports.default = TxlSearchBox;


TxlSearchBox.propTypes = {
  id: _TextField2.default.propTypes.id,
  onChange: _TextField2.default.propTypes.onChange,
  onClear: _react.PropTypes.func,
  onSubmit: _react.PropTypes.func,
  placeholder: _TextField2.default.propTypes.placeholder,
  value: _TextField2.default.propTypes.value
};

TxlSearchBox.defaultProps = {
  onChange: _Base.NOOP,
  onSubmit: _Base.NOOP,
  placeholder: 'Search'
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(SEARCH_ICON_COLOR, 'SEARCH_ICON_COLOR', 'src/input-fields/SearchBox.jsx');

  __REACT_HOT_LOADER__.register(FORM_STYLES, 'FORM_STYLES', 'src/input-fields/SearchBox.jsx');

  __REACT_HOT_LOADER__.register(CLEAR_BUTTON_STYLES, 'CLEAR_BUTTON_STYLES', 'src/input-fields/SearchBox.jsx');

  __REACT_HOT_LOADER__.register(TxlSearchBox, 'TxlSearchBox', 'src/input-fields/SearchBox.jsx');
}();

;
module.exports = exports['default'];
