// txl/input-fields/TextFieldWithSwitch.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _TextField = require('./TextField');

var _TextField2 = _interopRequireDefault(_TextField);

var _ToggleSwitch = require('./ToggleSwitch');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlTextFieldWithSwitch = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlTextFieldWithSwitch, _React$Component);

  function TxlTextFieldWithSwitch() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlTextFieldWithSwitch);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlTextFieldWithSwitch.__proto__ || (0, _getPrototypeOf2.default)(TxlTextFieldWithSwitch)).call.apply(_ref, [this].concat(args)));

    _this.state = {};
    return _this;
  }

  (0, _createClass3.default)(TxlTextFieldWithSwitch, [{
    key: '_handleToggleChange',
    value: function _handleToggleChange(obj) {
      this.props.onChange({
        name: this.props.name,
        on: obj.on,
        value: this.props.value
      });
    }
  }, {
    key: '_handleTextChange',
    value: function _handleTextChange(obj) {
      this.props.onChange({
        name: this.props.name,
        on: this.props.on,
        value: obj.value
      });
    }
  }, {
    key: 'render',
    value: function render() {
      var toggleSwitch = _react2.default.createElement(_ToggleSwitch.TxlToggleSwitch, {
        disabled: this.props.disabled || this.props.readOnly,
        name: this.props.name,
        on: this.props.on,
        onChange: this._handleToggleChange,
        tabIndex: this.props.tabIndex
      });
      // @TODO: Maybe use TextField.propTypes to get the props to pass through?
      return _react2.default.createElement(
        'div',
        { 'data-component': 'TxlTextFieldWithSwitch' },
        _react2.default.createElement(_TextField2.default, {
          disabled: this.props.disabled || !this.props.on,
          readOnly: this.props.readOnly,
          _afterContent: toggleSwitch,
          label: this.props.label,
          multiLine: this.props.multiLine,
          onChange: this._handleTextChange,
          placeholder: this.props.placeholder,
          secondaryText: this.props.secondaryText,
          tabIndex: this.props.tabIndex,
          validationState: this.props.validationState,
          value: this.props.value
        })
      );
    }
  }]);
  return TxlTextFieldWithSwitch;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleToggleChange', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleToggleChange'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleTextChange', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleTextChange'), _class2.prototype)), _class2)) || _class;

exports.default = TxlTextFieldWithSwitch;


TxlTextFieldWithSwitch.defaultProps = (0, _extends3.default)({}, _TextField2.default.defaultProps, _ToggleSwitch.TxlToggleSwitch.defaultProps, {
  disabled: false,
  onChange: _Base.NOOP,
  readOnly: false,
  tabIndex: 0
});

TxlTextFieldWithSwitch.propTypes = (0, _extends3.default)({}, _TextField2.default.propTypes, _ToggleSwitch.TxlToggleSwitch.propTypes, {
  disabled: _react2.default.PropTypes.bool,
  onChange: _react2.default.PropTypes.func.isRequired,
  readOnly: _react2.default.PropTypes.bool,
  tabIndex: _react2.default.PropTypes.number
});
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlTextFieldWithSwitch, 'TxlTextFieldWithSwitch', 'src/input-fields/TextFieldWithSwitch.jsx');
}();

;
module.exports = exports['default'];
