// txl/input-fields/TimePicker.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _Base = require('../base/Base');

var _CalendarUtils = require('../calendar-picker/_CalendarUtils');

var _TimePickerControlled = require('./TimePickerControlled');

var _TimePickerControlled2 = _interopRequireDefault(_TimePickerControlled);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var OUT_OF_RANGE_INVALID_TEXT = 'Time is out of accepted range';

function getValidationState(_ref) {
  var time = _ref.time,
      boundaryEnd = _ref.boundaryEnd,
      boundaryStart = _ref.boundaryStart;

  var state = {
    validationState: 'base',
    validationText: null
  };

  // Don't validate empty field
  // note: this would be an issue if this field is required
  if (time === undefined) return state;

  // time falls outside of boundaries
  if (!(0, _CalendarUtils.isTimeWithinRange)(time, boundaryStart, boundaryEnd)) {
    state.validationState = 'error';
    state.validationText = OUT_OF_RANGE_INVALID_TEXT;
  }
  return state;
}

function buildState(_ref2) {
  var boundaryStart = _ref2.boundaryStart,
      boundaryEnd = _ref2.boundaryEnd,
      time = _ref2.time;

  return (0, _extends3.default)({}, getValidationState({ boundaryEnd: boundaryEnd, boundaryStart: boundaryStart, time: time }), {
    time: time
  });
}

var TxlTimePicker = (0, _radium2.default)(_class = function (_React$Component) {
  (0, _inherits3.default)(TxlTimePicker, _React$Component);

  function TxlTimePicker() {
    var _ref3;

    (0, _classCallCheck3.default)(this, TxlTimePicker);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref3 = TxlTimePicker.__proto__ || (0, _getPrototypeOf2.default)(TxlTimePicker)).call.apply(_ref3, [this].concat(args)));

    _this.setTime = function () {
      return _this.__setTime__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._update = function () {
      return _this.___update__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleChildTimePickerRef = function () {
      return _this.___handleChildTimePickerRef__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleCloseDropdown = function () {
      return _this.___handleCloseDropdown__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleItemSelect = function () {
      return _this.___handleItemSelect__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleOpenDropdown = function () {
      return _this.___handleOpenDropdown__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    var defaultTime = _this.props.defaultTime;

    _this.state = (0, _extends3.default)({}, buildState((0, _extends3.default)({ time: defaultTime }, _this.props)), {
      open: false
    });
    return _this;
  }

  (0, _createClass3.default)(TxlTimePicker, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      // if boundaries or time change, we need to revalidate
      var boundaryEnd = nextProps.boundaryEnd,
          boundaryStart = nextProps.boundaryStart;
      var time = this.state.time;

      this.setState(buildState((0, _extends3.default)({}, this.props, {
        boundaryEnd: boundaryEnd,
        boundaryStart: boundaryStart,
        time: time
      })));
    }
  }, {
    key: '__setTime__REACT_HOT_LOADER__',
    value: function __setTime__REACT_HOT_LOADER__() {
      return this.__setTime__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__setTime__REACT_HOT_LOADER__',
    value: function __setTime__REACT_HOT_LOADER__() {
      return this.__setTime__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__setTime__REACT_HOT_LOADER__',
    value: function __setTime__REACT_HOT_LOADER__(_ref4) {
      var hour = _ref4.hour,
          minute = _ref4.minute,
          second = _ref4.second;
      return this.setState({ time: { hour: hour, minute: minute, second: second } });
    }
  }, {
    key: '___update__REACT_HOT_LOADER__',
    value: function ___update__REACT_HOT_LOADER__() {
      return this.___update__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___update__REACT_HOT_LOADER__',
    value: function ___update__REACT_HOT_LOADER__() {
      return this.___update__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___update__REACT_HOT_LOADER__',
    value: function ___update__REACT_HOT_LOADER__(time) {
      var _this2 = this;

      this.setState((0, _extends3.default)({}, buildState((0, _extends3.default)({}, this.props, { time: time })), {
        open: false
      }), function () {
        if (_this2.state.validationState !== 'error') {
          _this2.props.onItemSelect(time);
        }
      });
    }
  }, {
    key: '___handleChildTimePickerRef__REACT_HOT_LOADER__',
    value: function ___handleChildTimePickerRef__REACT_HOT_LOADER__() {
      return this.___handleChildTimePickerRef__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleChildTimePickerRef__REACT_HOT_LOADER__',
    value: function ___handleChildTimePickerRef__REACT_HOT_LOADER__() {
      return this.___handleChildTimePickerRef__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleChildTimePickerRef__REACT_HOT_LOADER__',
    value: function ___handleChildTimePickerRef__REACT_HOT_LOADER__(ref) {
      this._childTimePicker = ref;
    }
  }, {
    key: '___handleCloseDropdown__REACT_HOT_LOADER__',
    value: function ___handleCloseDropdown__REACT_HOT_LOADER__() {
      return this.___handleCloseDropdown__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleCloseDropdown__REACT_HOT_LOADER__',
    value: function ___handleCloseDropdown__REACT_HOT_LOADER__() {
      return this.___handleCloseDropdown__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleCloseDropdown__REACT_HOT_LOADER__',
    value: function ___handleCloseDropdown__REACT_HOT_LOADER__() {
      return this.setState({ open: false });
    }
  }, {
    key: '___handleItemSelect__REACT_HOT_LOADER__',
    value: function ___handleItemSelect__REACT_HOT_LOADER__() {
      return this.___handleItemSelect__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleItemSelect__REACT_HOT_LOADER__',
    value: function ___handleItemSelect__REACT_HOT_LOADER__() {
      return this.___handleItemSelect__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleItemSelect__REACT_HOT_LOADER__',
    value: function ___handleItemSelect__REACT_HOT_LOADER__(_ref5) {
      var value = _ref5.value;

      // validate string from user or selected item
      var time = (0, _moment2.default)(value, this.props.pattern, true);
      if (time.isValid()) {
        this._update({
          hour: time.hours(),
          minute: time.minutes(),
          second: time.seconds()
        });
      }
    }
  }, {
    key: '___handleOpenDropdown__REACT_HOT_LOADER__',
    value: function ___handleOpenDropdown__REACT_HOT_LOADER__() {
      return this.___handleOpenDropdown__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleOpenDropdown__REACT_HOT_LOADER__',
    value: function ___handleOpenDropdown__REACT_HOT_LOADER__() {
      return this.___handleOpenDropdown__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleOpenDropdown__REACT_HOT_LOADER__',
    value: function ___handleOpenDropdown__REACT_HOT_LOADER__() {
      return this.setState({ open: true });
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          boundaryEnd = _props.boundaryEnd,
          boundaryStart = _props.boundaryStart,
          label = _props.label,
          pattern = _props.pattern,
          placeholder = _props.placeholder,
          secondaryText = _props.secondaryText;
      var _state = this.state,
          open = _state.open,
          time = _state.time,
          validationState = _state.validationState,
          validationText = _state.validationText;


      var textValue = time ? (0, _moment2.default)(time).format(pattern) : '';

      return _react2.default.createElement(
        'div',
        { 'data-component': 'TxlTimePicker' },
        _react2.default.createElement(_TimePickerControlled2.default, {
          boundaryEnd: boundaryEnd,
          boundaryStart: boundaryStart,
          closeDropdown: this._handleCloseDropdown,
          label: label,
          onItemSelect: this._handleItemSelect,
          open: open,
          openDropdown: this._handleOpenDropdown,
          pattern: pattern,
          placeholder: placeholder,
          secondaryText: validationText || secondaryText,
          validationState: validationState,
          textValue: textValue,
          time: time
        })
      );
    }
  }]);
  return TxlTimePicker;
}(_react2.default.Component)) || _class;

exports.default = TxlTimePicker;


TxlTimePicker.propTypes = {
  boundaryEnd: _TimePickerControlled2.default.propTypes.boundaryEnd,
  boundaryStart: _TimePickerControlled2.default.propTypes.boundaryStart,
  defaultTime: _TimePickerControlled2.default.propTypes.time,
  label: _TimePickerControlled2.default.propTypes.label,
  onItemSelect: _react.PropTypes.func,
  pattern: _TimePickerControlled2.default.propTypes.pattern,
  placeholder: _TimePickerControlled2.default.propTypes.placeholder,
  secondaryText: _TimePickerControlled2.default.propTypes.secondaryText
};

TxlTimePicker.defaultProps = {
  boundaryEnd: _TimePickerControlled2.default.defaultProps.boundaryEnd,
  boundaryStart: _TimePickerControlled2.default.defaultProps.boundaryStart,
  onItemSelect: _Base.NOOP,
  pattern: _TimePickerControlled2.default.defaultProps.pattern,
  placeholder: 'Select a time'
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(OUT_OF_RANGE_INVALID_TEXT, 'OUT_OF_RANGE_INVALID_TEXT', 'src/input-fields/TimePicker.jsx');

  __REACT_HOT_LOADER__.register(getValidationState, 'getValidationState', 'src/input-fields/TimePicker.jsx');

  __REACT_HOT_LOADER__.register(buildState, 'buildState', 'src/input-fields/TimePicker.jsx');

  __REACT_HOT_LOADER__.register(TxlTimePicker, 'TxlTimePicker', 'src/input-fields/TimePicker.jsx');
}();

;
module.exports = exports['default'];
