// txl/input-fields/TitleEditableControlled.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _class, _class2, _temp2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _FlexContainer = require('../layout/FlexContainer');

var _FlexContainer2 = _interopRequireDefault(_FlexContainer);

var _InputFieldBase = require('./_InputFieldBase');

var _InputFieldBase2 = _interopRequireDefault(_InputFieldBase);

var _LoadingIndicator = require('../progress-indicators/LoadingIndicator');

var _LoadingIndicator2 = _interopRequireDefault(_LoadingIndicator);

var _IconButton = require('../buttons/IconButton');

var _IconButton2 = _interopRequireDefault(_IconButton);

var _EditOutline = require('../icons/EditOutline');

var _EditOutline2 = _interopRequireDefault(_EditOutline);

var _Check = require('../icons/Check');

var _Check2 = _interopRequireDefault(_Check);

var _Close = require('../icons/Close');

var _Close2 = _interopRequireDefault(_Close);

var _type = require('../styles/mixins/type');

var _theme = require('../styles/theme');

var _helpers = require('../styles/helpers');

var _KeyCodes = require('../constants/KeyCodes');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BUTTON_SPACING = {
  marginLeft: (0, _helpers.gridUnits)(2)
};
var BOTTOM_BORDER_WIDTH = '1px';
var INPUT_STYLES = (0, _extends3.default)({}, _type.DEFAULT_FONT_FAMILY, {
  backgroundColor: 'transparent',
  border: 'none',
  boxSizing: 'border-box',
  flexBasis: '100%',
  flexGrow: 1,
  marginBottom: '-' + BOTTOM_BORDER_WIDTH,
  minWidth: 0, // for MS Edge
  outline: 'none',
  overflow: 'hidden',
  textIndent: 0
});

var getHeadingStyles = function getHeadingStyles(type, placeholder) {
  return (0, _extends3.default)({}, _type.DEFAULT_FONT_FAMILY, type === 'h1' ? _type.HEADING_LARGE : _type.HEADING_MEDIUM, placeholder ? { color: _theme.COLOR_NEUTRAL['200'] } : {});
};

// TODO: use SyntheticInputEvent instead of this, when eslint no longer thinks it's undefined. TXLENG-370
Object.defineProperty(module.exports, 'babelPluginFlowReactPropTypes_proptype_CallbackArgument', require('react').PropTypes.shape({
  value: require('react').PropTypes.string.isRequired
}));

var TxlTitleEditableControlled = (0, _radium2.default)(_class = (_temp2 = _class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlTitleEditableControlled, _React$Component);

  function TxlTitleEditableControlled() {
    var _ref;

    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, TxlTitleEditableControlled);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlTitleEditableControlled.__proto__ || (0, _getPrototypeOf2.default)(TxlTitleEditableControlled)).call.apply(_ref, [this].concat(args))), _this), _this._assignInputRef = function () {
      var _this2;

      return (_this2 = _this).___assignInputRef__REACT_HOT_LOADER__.apply(_this2, arguments);
    }, _this._handleBlur = function () {
      var _this3;

      return (_this3 = _this).___handleBlur__REACT_HOT_LOADER__.apply(_this3, arguments);
    }, _this._handleCancel = function () {
      var _this4;

      return (_this4 = _this).___handleCancel__REACT_HOT_LOADER__.apply(_this4, arguments);
    }, _this._handleChange = function () {
      var _this5;

      return (_this5 = _this).___handleChange__REACT_HOT_LOADER__.apply(_this5, arguments);
    }, _this._handleEdit = function () {
      var _this6;

      return (_this6 = _this).___handleEdit__REACT_HOT_LOADER__.apply(_this6, arguments);
    }, _this._handleFocus = function () {
      var _this7;

      return (_this7 = _this).___handleFocus__REACT_HOT_LOADER__.apply(_this7, arguments);
    }, _this._handleKeyDown = function () {
      var _this8;

      return (_this8 = _this).___handleKeyDown__REACT_HOT_LOADER__.apply(_this8, arguments);
    }, _this._handleSave = function () {
      var _this9;

      return (_this9 = _this).___handleSave__REACT_HOT_LOADER__.apply(_this9, arguments);
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  (0, _createClass3.default)(TxlTitleEditableControlled, [{
    key: 'componentDidUpdate',
    value: function componentDidUpdate(prevProps) {
      if (this.props.focused && !prevProps.focused) {
        this._input && this._input.focus();
      }
    }
  }, {
    key: '___assignInputRef__REACT_HOT_LOADER__',


    // users responsibility to handle 'focused' prop
    value: function ___assignInputRef__REACT_HOT_LOADER__() {
      return this.___assignInputRef__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___assignInputRef__REACT_HOT_LOADER__',
    value: function ___assignInputRef__REACT_HOT_LOADER__() {
      return this.___assignInputRef__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___assignInputRef__REACT_HOT_LOADER__',
    value: function ___assignInputRef__REACT_HOT_LOADER__(node) {
      this._input = node;
    }
  }, {
    key: '___handleBlur__REACT_HOT_LOADER__',


    // Button to cancel active changes. Users responsibility to restore previous state.
    value: function ___handleBlur__REACT_HOT_LOADER__() {
      return this.___handleBlur__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleBlur__REACT_HOT_LOADER__',
    value: function ___handleBlur__REACT_HOT_LOADER__() {
      return this.___handleBlur__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleBlur__REACT_HOT_LOADER__',
    value: function ___handleBlur__REACT_HOT_LOADER__(_ref2) {
      var target = _ref2.target;
      var value = target.value;

      this.props.onBlur({ value: value });
    }
  }, {
    key: '___handleCancel__REACT_HOT_LOADER__',
    value: function ___handleCancel__REACT_HOT_LOADER__() {
      return this.___handleCancel__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleCancel__REACT_HOT_LOADER__',
    value: function ___handleCancel__REACT_HOT_LOADER__() {
      return this.___handleCancel__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleCancel__REACT_HOT_LOADER__',
    value: function ___handleCancel__REACT_HOT_LOADER__() {
      this.props.onCancel();
    }
  }, {
    key: '___handleChange__REACT_HOT_LOADER__',


    // Button to trigger edit mode. Users responsibility to handle editing/focused props.
    value: function ___handleChange__REACT_HOT_LOADER__() {
      return this.___handleChange__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleChange__REACT_HOT_LOADER__',
    value: function ___handleChange__REACT_HOT_LOADER__() {
      return this.___handleChange__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleChange__REACT_HOT_LOADER__',
    value: function ___handleChange__REACT_HOT_LOADER__(_ref3) {
      var target = _ref3.target;
      var value = target.value;

      this.props.onChange({ value: value });
    }
  }, {
    key: '___handleEdit__REACT_HOT_LOADER__',


    // users responsibility to handle 'focused' prop
    value: function ___handleEdit__REACT_HOT_LOADER__() {
      return this.___handleEdit__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleEdit__REACT_HOT_LOADER__',
    value: function ___handleEdit__REACT_HOT_LOADER__() {
      return this.___handleEdit__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleEdit__REACT_HOT_LOADER__',
    value: function ___handleEdit__REACT_HOT_LOADER__() {
      this.props.onEdit();
    }
  }, {
    key: '___handleFocus__REACT_HOT_LOADER__',
    value: function ___handleFocus__REACT_HOT_LOADER__() {
      return this.___handleFocus__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleFocus__REACT_HOT_LOADER__',
    value: function ___handleFocus__REACT_HOT_LOADER__() {
      return this.___handleFocus__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleFocus__REACT_HOT_LOADER__',
    value: function ___handleFocus__REACT_HOT_LOADER__(_ref4) {
      var target = _ref4.target;
      var value = target.value;

      this.props.onFocus({ value: value });
    }
  }, {
    key: '___handleKeyDown__REACT_HOT_LOADER__',


    // Button to save current value. Passes back value prop.
    value: function ___handleKeyDown__REACT_HOT_LOADER__() {
      return this.___handleKeyDown__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleKeyDown__REACT_HOT_LOADER__',
    value: function ___handleKeyDown__REACT_HOT_LOADER__() {
      return this.___handleKeyDown__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleKeyDown__REACT_HOT_LOADER__',
    value: function ___handleKeyDown__REACT_HOT_LOADER__(_ref5) {
      var keyCode = _ref5.keyCode;

      switch (keyCode) {
        case _KeyCodes.ENTER:
          return this._handleSave();
        case _KeyCodes.ESCAPE:
          return this._handleCancel();
        default:
          return undefined;
      }
    }
  }, {
    key: '___handleSave__REACT_HOT_LOADER__',
    value: function ___handleSave__REACT_HOT_LOADER__() {
      return this.___handleSave__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleSave__REACT_HOT_LOADER__',
    value: function ___handleSave__REACT_HOT_LOADER__() {
      return this.___handleSave__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleSave__REACT_HOT_LOADER__',
    value: function ___handleSave__REACT_HOT_LOADER__() {
      this.props.onSave({ value: this.props.value });
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          editing = _props.editing,
          focused = _props.focused,
          loading = _props.loading,
          placeholder = _props.placeholder,
          secondaryText = _props.secondaryText,
          tabIndex = _props.tabIndex,
          type = _props.type,
          validationState = _props.validationState,
          value = _props.value;


      return _react2.default.createElement(
        'div',
        { 'data-component': 'TxlTitleEditable' },
        editing && _react2.default.createElement(
          _FlexContainer2.default,
          { alignItems: 'baseline' },
          _react2.default.createElement(
            _InputFieldBase2.default,
            {
              focused: focused,
              secondaryText: secondaryText,
              validationState: validationState
            },
            _react2.default.createElement('input', {
              style: (0, _extends3.default)({}, INPUT_STYLES, getHeadingStyles(type, value.length === 0)),
              type: 'text',
              placeholder: this.props.placeholder,
              onBlur: this._handleBlur,
              onChange: this._handleChange,
              onFocus: this._handleFocus,
              onKeyDown: this._handleKeyDown,
              ref: this._assignInputRef,
              tabIndex: tabIndex,
              value: value
            })
          ),
          loading && _react2.default.createElement(
            'div',
            { style: BUTTON_SPACING },
            _react2.default.createElement(_LoadingIndicator2.default, { size: 'small' })
          ),
          !loading && _react2.default.createElement(
            _FlexContainer2.default,
            null,
            _react2.default.createElement(_IconButton2.default, {
              variant: 'plain',
              altText: 'Save',
              icon: _Check2.default,
              onClick: this._handleSave,
              _style: BUTTON_SPACING
            }),
            _react2.default.createElement(_IconButton2.default, {
              variant: 'plain',
              altText: 'Cancel editing',
              icon: _Close2.default,
              onClick: this._handleCancel,
              _style: BUTTON_SPACING
            })
          )
        ),
        !editing && _react2.default.createElement(
          _FlexContainer2.default,
          { alignItems: 'baseline' },
          _react2.default.createElement(type, {
            onClick: this._handleEdit,
            style: getHeadingStyles(type, value.length === 0)
          }, value || placeholder),
          _react2.default.createElement(_IconButton2.default, {
            variant: 'plain',
            altText: 'Edit',
            icon: _EditOutline2.default,
            onClick: this._handleEdit,
            _style: BUTTON_SPACING
          })
        )
      );
    }
  }]);
  return TxlTitleEditableControlled;
}(_react2.default.Component), _class2.displayName = 'TitleEditableControlled', _class2.defaultProps = {
  defaultValue: '',
  editing: false,
  focused: false,
  loading: false,
  onBlur: _Base.NOOP,
  onChange: _Base.NOOP,
  onFocus: _Base.NOOP,
  tabIndex: 0,
  type: 'h1',
  value: ''
}, _temp2)) || _class;

TxlTitleEditableControlled.propTypes = {
  defaultValue: require('react').PropTypes.string.isRequired,
  editing: require('react').PropTypes.bool.isRequired,
  focused: require('react').PropTypes.bool.isRequired,
  loading: require('react').PropTypes.bool.isRequired,
  onBlur: require('react').PropTypes.func.isRequired,
  onCancel: require('react').PropTypes.func.isRequired,
  onChange: require('react').PropTypes.func.isRequired,
  onFocus: require('react').PropTypes.func.isRequired,
  onEdit: require('react').PropTypes.func.isRequired,
  onSave: require('react').PropTypes.func.isRequired,
  placeholder: require('react').PropTypes.string,
  secondaryText: require('react').PropTypes.oneOfType([require('react').PropTypes.string, require('react').PropTypes.any]),
  tabIndex: require('react').PropTypes.number.isRequired,
  type: require('react').PropTypes.oneOf(['h1', 'h2']).isRequired,
  validationState: require('react').PropTypes.oneOf(['base', 'error', 'warning', 'success']),
  value: require('react').PropTypes.string.isRequired
};
var _default = TxlTitleEditableControlled;
exports.default = _default;
;

var _temp3 = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(BUTTON_SPACING, 'BUTTON_SPACING', 'src/input-fields/TitleEditableControlled.jsx');

  __REACT_HOT_LOADER__.register(BOTTOM_BORDER_WIDTH, 'BOTTOM_BORDER_WIDTH', 'src/input-fields/TitleEditableControlled.jsx');

  __REACT_HOT_LOADER__.register(INPUT_STYLES, 'INPUT_STYLES', 'src/input-fields/TitleEditableControlled.jsx');

  __REACT_HOT_LOADER__.register(getHeadingStyles, 'getHeadingStyles', 'src/input-fields/TitleEditableControlled.jsx');

  __REACT_HOT_LOADER__.register(TxlTitleEditableControlled, 'TxlTitleEditableControlled', 'src/input-fields/TitleEditableControlled.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/input-fields/TitleEditableControlled.jsx');
}();

;
module.exports = exports['default'];
