// txl/input-fields/TitleEditable.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Base = require('../base/Base');

var _TitleEditableControlled = require('./TitleEditableControlled');

var _TitleEditableControlled2 = _interopRequireDefault(_TitleEditableControlled);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var babelPluginFlowReactPropTypes_proptype_CallbackArgument = require('./TitleEditableControlled').babelPluginFlowReactPropTypes_proptype_CallbackArgument || require('react').PropTypes.any;

var TxlTitleEditable = (_temp = _class = function (_React$Component) {
  (0, _inherits3.default)(TxlTitleEditable, _React$Component);

  function TxlTitleEditable(props) {
    (0, _classCallCheck3.default)(this, TxlTitleEditable);

    var _this = (0, _possibleConstructorReturn3.default)(this, (TxlTitleEditable.__proto__ || (0, _getPrototypeOf2.default)(TxlTitleEditable)).call(this, props));

    _this._handleCancel = function () {
      return _this.___handleCancel__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleChange = function () {
      return _this.___handleChange__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleEdit = function () {
      return _this.___handleEdit__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleSave = function () {
      return _this.___handleSave__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleBlur = function () {
      return _this.___handleBlur__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleFocus = function () {
      return _this.___handleFocus__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this.state = {
      cachedValue: props.defaultValue || '',
      editing: false,
      focused: false,
      value: props.defaultValue || ''
    };
    return _this;
  }

  // Exit edit mode, and restore value to previously cached value.
  // Caching value happens on entering edit mode & saving.


  (0, _createClass3.default)(TxlTitleEditable, [{
    key: '___handleCancel__REACT_HOT_LOADER__',
    value: function ___handleCancel__REACT_HOT_LOADER__() {
      return this.___handleCancel__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleCancel__REACT_HOT_LOADER__',
    value: function ___handleCancel__REACT_HOT_LOADER__() {
      this.setState(function (state) {
        return {
          editing: false,
          focused: false,
          value: state.cachedValue
        };
      });
    }
  }, {
    key: '___handleChange__REACT_HOT_LOADER__',


    // Enter edit mode. Set current value aside, in case user wants to cancel editing.
    value: function ___handleChange__REACT_HOT_LOADER__() {
      return this.___handleChange__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleChange__REACT_HOT_LOADER__',
    value: function ___handleChange__REACT_HOT_LOADER__(_ref) {
      var value = _ref.value;

      this.setState({ value: value });
    }
  }, {
    key: '___handleEdit__REACT_HOT_LOADER__',


    // Make sure internal state is ok, then trigger onSave callback
    value: function ___handleEdit__REACT_HOT_LOADER__() {
      return this.___handleEdit__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleEdit__REACT_HOT_LOADER__',
    value: function ___handleEdit__REACT_HOT_LOADER__() {
      this.setState(function (state) {
        return {
          cachedValue: state.value,
          editing: true,
          focused: true
        };
      });
    }
  }, {
    key: '___handleSave__REACT_HOT_LOADER__',
    value: function ___handleSave__REACT_HOT_LOADER__() {
      return this.___handleSave__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleSave__REACT_HOT_LOADER__',
    value: function ___handleSave__REACT_HOT_LOADER__(_ref2) {
      var _this2 = this;

      var value = _ref2.value;

      this.setState({
        cachedValue: value,
        editing: false,
        focused: false,
        value: value
      }, function () {
        return _this2.props.onSave({ value: value });
      });
    }
  }, {
    key: '___handleBlur__REACT_HOT_LOADER__',
    value: function ___handleBlur__REACT_HOT_LOADER__() {
      return this.___handleBlur__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleBlur__REACT_HOT_LOADER__',
    value: function ___handleBlur__REACT_HOT_LOADER__(obj) {
      this.props.onBlur(obj);
    }
  }, {
    key: '___handleFocus__REACT_HOT_LOADER__',
    value: function ___handleFocus__REACT_HOT_LOADER__() {
      return this.___handleFocus__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleFocus__REACT_HOT_LOADER__',
    value: function ___handleFocus__REACT_HOT_LOADER__(obj) {
      this.props.onFocus(obj);
    }
  }, {
    key: 'render',
    value: function render() {
      var _state = this.state,
          editing = _state.editing,
          focused = _state.focused,
          value = _state.value;
      var _props = this.props,
          placeholder = _props.placeholder,
          tabIndex = _props.tabIndex,
          type = _props.type;


      return _react2.default.createElement(_TitleEditableControlled2.default, {
        editing: editing,
        focused: focused,
        onBlur: this._handleBlur,
        onCancel: this._handleCancel,
        onChange: this._handleChange,
        onEdit: this._handleEdit,
        onFocus: this._handleFocus,
        onSave: this._handleSave,
        placeholder: placeholder,
        tabindex: tabIndex,
        type: type,
        value: value
      });
    }
  }]);
  return TxlTitleEditable;
}(_react2.default.Component), _class.displayName = 'TitleEditable', _class.defaultProps = {
  defaultValue: '',
  onBlur: _Base.NOOP,
  onFocus: _Base.NOOP,
  onSave: _Base.NOOP
}, _temp);
TxlTitleEditable.propTypes = {
  defaultValue: require('react').PropTypes.string.isRequired,
  onBlur: require('react').PropTypes.func.isRequired,
  onFocus: require('react').PropTypes.func.isRequired,
  onSave: require('react').PropTypes.func.isRequired,
  placeholder: require('react').PropTypes.string,
  tabIndex: require('react').PropTypes.number,
  type: require('react').PropTypes.oneOf(['h1', 'h2'])
};
var _default = TxlTitleEditable;
exports.default = _default;
;

var _temp2 = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlTitleEditable, 'TxlTitleEditable', 'src/input-fields/TitleEditable.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/input-fields/TitleEditable.jsx');
}();

;
module.exports = exports['default'];
