// txl/input-fields/ToggleSwitchGroup.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _ToggleSwitch = require('./ToggleSwitch');

var _ToggleSwitch2 = _interopRequireDefault(_ToggleSwitch);

var _InputStyleHelpers = require('./_InputStyleHelpers');

var _helpers = require('../styles/helpers');

var _list = require('../styles/mixins/list');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ITEM_STYLES = (0, _extends3.default)({}, _list.UNSTYLED_LIST_ITEM, {
  marginBottom: (0, _helpers.gridUnits)(5)
});

var LABEL_CONTAINER_STYLES = {
  marginBottom: (0, _helpers.gridUnits)(2)
};

var TxlToggleSwitchGroup = function TxlToggleSwitchGroup(props) {
  var labelContent = void 0;
  if (props.labelText) {
    var secondaryTextContent = void 0;
    if (props.secondaryText) {
      secondaryTextContent = _react2.default.createElement(
        'div',
        {
          style: (0, _InputStyleHelpers.getSecondaryTextStyle)(props.validationState),
          'data-test-group-secondary-text': true
        },
        props.secondaryText
      );
    }
    labelContent = _react2.default.createElement(
      'header',
      { style: LABEL_CONTAINER_STYLES },
      _react2.default.createElement(
        'div',
        {
          style: _InputStyleHelpers.INPUT_GROUP_LABEL_STYLES,
          'data-test-group-label-text': true
        },
        props.labelText
      ),
      secondaryTextContent
    );
  }

  return _react2.default.createElement(
    'section',
    { 'data-component': 'TxlToggleSwitchGroup' },
    labelContent,
    _react2.default.createElement(
      'ul',
      { style: _list.UNSTYLED_LIST },
      props.items.map(function (item) {
        return _react2.default.createElement(
          'li',
          { style: ITEM_STYLES, key: item.name },
          _react2.default.createElement(_ToggleSwitch2.default, {
            disabled: item.disabled,
            id: item.id,
            labelPosition: props.labelPosition,
            labelText: item.labelText,
            loading: item.loading,
            name: item.name,
            on: item.on,
            onChange: props.onChange,
            tabIndex: props.tabIndex,
            secondaryText: item.secondaryText,
            validationState: item.validationState
          })
        );
      })
    )
  );
};

TxlToggleSwitchGroup.propTypes = {
  items: _react2.default.PropTypes.arrayOf(_react2.default.PropTypes.shape({
    // We will be passing these props to ToggleSwitch so use whatever it expects
    disabled: _ToggleSwitch2.default.propTypes.disabled,
    labelText: _ToggleSwitch2.default.propTypes.labelText,
    name: _ToggleSwitch2.default.propTypes.name,
    on: _ToggleSwitch2.default.propTypes.on,
    secondaryText: _ToggleSwitch2.default.propTypes.secondaryText,
    validationState: _ToggleSwitch2.default.propTypes.validationState
  })).isRequired,
  labelPosition: _react2.default.PropTypes.oneOf(['left', 'right']),
  labelText: _react2.default.PropTypes.string,
  onChange: _react2.default.PropTypes.func.isRequired,
  secondaryText: _react2.default.PropTypes.string,
  tabIndex: _react2.default.PropTypes.number,
  validationState: _react2.default.PropTypes.oneOf(['base', 'error', 'warning', 'success'])
};

TxlToggleSwitchGroup.defaultProps = {
  labelPosition: 'left',
  tabIndex: 0,
  validationState: 'base'
};

var _default = (0, _radium2.default)(TxlToggleSwitchGroup);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ITEM_STYLES, 'ITEM_STYLES', 'src/input-fields/ToggleSwitchGroup.jsx');

  __REACT_HOT_LOADER__.register(LABEL_CONTAINER_STYLES, 'LABEL_CONTAINER_STYLES', 'src/input-fields/ToggleSwitchGroup.jsx');

  __REACT_HOT_LOADER__.register(TxlToggleSwitchGroup, 'TxlToggleSwitchGroup', 'src/input-fields/ToggleSwitchGroup.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/input-fields/ToggleSwitchGroup.jsx');
}();

;
module.exports = exports['default'];
