// txl/input-fields/UncontrolledTextField.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _TextField = require('./TextField');

var _TextField2 = _interopRequireDefault(_TextField);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlUncontrolledTextField = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlUncontrolledTextField, _React$Component);

  function TxlUncontrolledTextField() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlUncontrolledTextField);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlUncontrolledTextField.__proto__ || (0, _getPrototypeOf2.default)(TxlUncontrolledTextField)).call.apply(_ref, [this].concat(args)));

    _this.state = {
      inputValue: _this.props.defaultValue
    };
    return _this;
  }

  (0, _createClass3.default)(TxlUncontrolledTextField, [{
    key: '_updateInputState',
    value: function _updateInputState(_ref2) {
      var value = _ref2.value;

      this.props.onChange({ value: value });
      this.setState({ inputValue: value });
    }
  }, {
    key: 'focus',
    value: function focus() {
      this.refs.textField.focus();
    }
  }, {
    key: 'select',
    value: function select() {
      this.refs.textField.select();
    }
  }, {
    key: 'render',
    value: function render() {
      /* eslint no-unused-vars: off */
      // TODO: Fix prop pass through.
      var _props = this.props,
          defaultValue = _props.defaultValue,
          restProps = (0, _objectWithoutProperties3.default)(_props, ['defaultValue']);


      return _react2.default.createElement(_TextField2.default, (0, _extends3.default)({}, restProps, {
        onChange: this._updateInputState,
        placeholder: this.props.placeholder,
        ref: 'textField',
        value: this.state.inputValue
      }));
    }
  }]);
  return TxlUncontrolledTextField;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_updateInputState', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_updateInputState'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, 'focus', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, 'focus'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, 'select', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, 'select'), _class2.prototype)), _class2)) || _class;

exports.default = TxlUncontrolledTextField;


TxlUncontrolledTextField.propTypes = (0, _extends3.default)({}, _TextField2.default.propTypes, {
  defaultValue: _TextField2.default.propTypes.value
});
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlUncontrolledTextField, 'TxlUncontrolledTextField', 'src/input-fields/UncontrolledTextField.jsx');
}();

;
module.exports = exports['default'];
