// txl/input-fields/UncontrolledSearchBox.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _SearchBox = require('./SearchBox');

var _SearchBox2 = _interopRequireDefault(_SearchBox);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlUncontrolledSearchBox = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlUncontrolledSearchBox, _React$Component);

  function TxlUncontrolledSearchBox() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlUncontrolledSearchBox);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlUncontrolledSearchBox.__proto__ || (0, _getPrototypeOf2.default)(TxlUncontrolledSearchBox)).call.apply(_ref, [this].concat(args)));

    _this.state = { searchValue: _this.props.defaultValue };
    return _this;
  }

  (0, _createClass3.default)(TxlUncontrolledSearchBox, [{
    key: 'clear',
    value: function clear() {
      this._handleClear();
    }
  }, {
    key: '_handleClear',
    value: function _handleClear() {
      this._handleChange({ value: '' });
    }
  }, {
    key: '_handleChange',
    value: function _handleChange(_ref2) {
      var value = _ref2.value;

      this.props.onChange({ value: value });
      this.setState({ searchValue: value });
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(_SearchBox2.default, {
        id: this.props.id,
        onClear: this._handleClear,
        onSubmit: this.props.onSubmit,
        onChange: this._handleChange,
        placeholder: this.props.placeholder,
        value: this.state.searchValue
      });
    }
  }]);
  return TxlUncontrolledSearchBox;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, 'clear', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, 'clear'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleClear', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleClear'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleChange', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleChange'), _class2.prototype)), _class2)) || _class;

exports.default = TxlUncontrolledSearchBox;


TxlUncontrolledSearchBox.defaultProps = {
  defaultValue: '',
  onChange: _Base.NOOP,
  onSubmit: _Base.NOOP
};

TxlUncontrolledSearchBox.propTypes = {
  defaultValue: _react.PropTypes.string,
  // don't use SearchBox.propTypes.onChange since it is required
  id: _SearchBox2.default.propTypes.id,
  onChange: _react.PropTypes.func,
  onSubmit: _SearchBox2.default.propTypes.onSubmit,
  placeholder: _SearchBox2.default.propTypes.placeholder
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlUncontrolledSearchBox, 'TxlUncontrolledSearchBox', 'src/input-fields/UncontrolledSearchBox.jsx');
}();

;
module.exports = exports['default'];
