// txl/_date-time-range-picker/_TimeRangePicker.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.DATE_QUERY_FORMAT = exports.TIME_QUERY_FORMAT = exports.TIME_DISPLAY_FORMAT = exports.TIME_RANGE_FORMAT = exports.TIME_DISPLAY_FORMAT_SIMPLIFIED = exports.TIME_RANGE_FORMAT_SIMPLIFIED = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _Base = require('../base/Base');

var _DateTimeRangePicker = require('./DateTimeRangePicker.style');

var _theme = require('../styles/theme');

var _EditOutline = require('../icons/EditOutline');

var _EditOutline2 = _interopRequireDefault(_EditOutline);

var _Modal = require('../modals/Modal');

var _Modal2 = _interopRequireDefault(_Modal);

var _Button = require('../buttons/Button');

var _Button2 = _interopRequireDefault(_Button);

var _SingleSelect = require('../dropdowns/_SingleSelect');

var _SingleSelect2 = _interopRequireDefault(_SingleSelect);

var _Checkbox = require('../input-fields/Checkbox');

var _Checkbox2 = _interopRequireDefault(_Checkbox);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TIME_RANGE_FORMAT_SIMPLIFIED = 'h:mma';
var TIME_DISPLAY_FORMAT_SIMPLIFIED = 'h:mma (MMM D)';
var TIME_RANGE_FORMAT = 'h:mm:ssa';
var TIME_DISPLAY_FORMAT = 'h:mm:ssa (MMM D)';
var TIME_QUERY_FORMAT = 'HH:mm:ss';
var DATE_QUERY_FORMAT = 'YYYY-MM-DD';

exports.TIME_RANGE_FORMAT_SIMPLIFIED = TIME_RANGE_FORMAT_SIMPLIFIED;
exports.TIME_DISPLAY_FORMAT_SIMPLIFIED = TIME_DISPLAY_FORMAT_SIMPLIFIED;
exports.TIME_RANGE_FORMAT = TIME_RANGE_FORMAT;
exports.TIME_DISPLAY_FORMAT = TIME_DISPLAY_FORMAT;
exports.TIME_QUERY_FORMAT = TIME_QUERY_FORMAT;
exports.DATE_QUERY_FORMAT = DATE_QUERY_FORMAT;

var TxlTimeRangePicker = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlTimeRangePicker, _React$Component);

  function TxlTimeRangePicker() {
    var _ref;

    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, TxlTimeRangePicker);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlTimeRangePicker.__proto__ || (0, _getPrototypeOf2.default)(TxlTimeRangePicker)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
      includeSeconds: false,
      openEndTime: false,
      openStartTime: false,
      showSeconds: false,
      timeRange: {
        endTime: {
          key: (0, _moment2.default)(_this.props.items.endDateTime).format(TIME_QUERY_FORMAT),
          value: (0, _moment2.default)(_this.props.items.endDateTime).format(TIME_RANGE_FORMAT_SIMPLIFIED)
        },
        startTime: {
          key: (0, _moment2.default)(_this.props.items.startDateTime).format(TIME_QUERY_FORMAT),
          value: (0, _moment2.default)(_this.props.items.startDateTime).format(TIME_RANGE_FORMAT_SIMPLIFIED)
        }
      }
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  (0, _createClass3.default)(TxlTimeRangePicker, [{
    key: '_toggleModal',
    value: function _toggleModal() {
      this.props.updateDropdown({ open: !this.props.open });
    }
  }, {
    key: '_handleSaveModal',
    value: function _handleSaveModal() {
      var _state = this.state,
          includeSeconds = _state.includeSeconds,
          timeRange = _state.timeRange;
      var _props = this.props,
          items = _props.items,
          onChange = _props.onChange;

      var startDate = (0, _moment2.default)(items.startDateTime).format(DATE_QUERY_FORMAT);
      var endDate = (0, _moment2.default)(items.endDateTime).format(DATE_QUERY_FORMAT);
      items.startDateTime = startDate + ' ' + timeRange.startTime.key;
      items.endDateTime = endDate + ' ' + timeRange.endTime.key;
      onChange(items);
      this._toggleModal();
      this.setState({ showSeconds: includeSeconds });
    }
  }, {
    key: '_handleCancelModal',
    value: function _handleCancelModal() {
      var items = this.props.items;
      var showSeconds = this.state.showSeconds;

      var rangeFormat = showSeconds ? TIME_RANGE_FORMAT : TIME_RANGE_FORMAT_SIMPLIFIED;

      var timeRange = {
        endTime: {
          key: (0, _moment2.default)(items.endDateTime).format(TIME_QUERY_FORMAT),
          value: (0, _moment2.default)(items.endDateTime).format(rangeFormat)
        },
        startTime: {
          key: (0, _moment2.default)(items.startDateTime).format(TIME_QUERY_FORMAT),
          value: (0, _moment2.default)(items.startDateTime).format(rangeFormat)
        }
      };
      this.setState({ timeRange: timeRange });
      this._toggleModal();
    }
  }, {
    key: '_handleTimeRangeChange',
    value: function _handleTimeRangeChange(type) {
      var _this2 = this;

      var timeRange = this.state.timeRange;


      return function (e) {
        _this2.setState({
          timeRange: {
            endTime: type === 'end' ? e.value : timeRange.endTime,
            startTime: type === 'start' ? e.value : timeRange.startTime
          }
        });
      };
    }
  }, {
    key: '_handleCustomTimeChange',
    value: function _handleCustomTimeChange(type) {
      var _this3 = this;

      var _state2 = this.state,
          includeSeconds = _state2.includeSeconds,
          timeRange = _state2.timeRange;

      var rangeFormat = includeSeconds ? TIME_RANGE_FORMAT : TIME_RANGE_FORMAT_SIMPLIFIED;

      return function (e) {
        var timeObj = {
          key: (0, _moment2.default)(e.value, rangeFormat).format(TIME_QUERY_FORMAT),
          value: (0, _moment2.default)(e.value, rangeFormat).format(rangeFormat)
        };
        _this3.setState({
          timeRange: {
            endTime: type === 'end' ? timeObj : timeRange.endTime,
            startTime: type === 'start' ? timeObj : timeRange.startTime
          }
        });
      };
    }
  }, {
    key: '_generateTimes',
    value: function _generateTimes() {
      var timeArray = [];
      var includeSeconds = this.state.includeSeconds;

      var startTime = (0, _moment2.default)().startOf('day');
      var endTime = (0, _moment2.default)().endOf('day');
      var rangeFormat = includeSeconds ? TIME_RANGE_FORMAT : TIME_RANGE_FORMAT_SIMPLIFIED;

      while ((0, _moment2.default)(startTime).isSameOrBefore(endTime)) {
        timeArray.push({
          key: (0, _moment2.default)(startTime).format(TIME_QUERY_FORMAT),
          value: (0, _moment2.default)(startTime).format(rangeFormat)
        });
        startTime = (0, _moment2.default)(startTime).add(30, 'minutes');
      }

      return timeArray;
    }
  }, {
    key: '_openDropdown',
    value: function _openDropdown(type) {
      var _this4 = this;

      return function () {
        _this4.setState({
          openEndTime: type === 'end',
          openStartTime: type === 'start'
        });
      };
    }
  }, {
    key: '_closeDropdown',
    value: function _closeDropdown(type) {
      var _this5 = this;

      return function () {
        _this5.setState({
          openEndTime: !type === 'end',
          openStartTime: !type === 'start'
        });
      };
    }
  }, {
    key: '_getTimeDisplayFormat',
    value: function _getTimeDisplayFormat() {
      var items = this.props.items;
      var showSeconds = this.state.showSeconds;

      var isSameDate = (0, _moment2.default)(items.startDateTime).isSame(items.endDateTime, 'date');
      var timeFormat = void 0;
      if (showSeconds) {
        timeFormat = isSameDate ? TIME_RANGE_FORMAT : TIME_DISPLAY_FORMAT;
      } else {
        timeFormat = isSameDate ? TIME_RANGE_FORMAT_SIMPLIFIED : TIME_DISPLAY_FORMAT_SIMPLIFIED;
      }

      return timeFormat;
    }
  }, {
    key: '_getTimeDisplay',
    value: function _getTimeDisplay() {
      var items = this.props.items;

      var timeFormat = this._getTimeDisplayFormat();
      var startTime = (0, _moment2.default)(items.startDateTime).format(timeFormat);
      var endTime = (0, _moment2.default)(items.endDateTime).format(timeFormat);
      var timeDisplay = startTime + ' to ' + endTime;
      return timeDisplay;
    }
  }, {
    key: '_handleSecondsCheck',
    value: function _handleSecondsCheck(obj) {
      this.setState({ includeSeconds: obj.checked });
    }
  }, {
    key: 'renderModal',
    value: function renderModal() {
      var _props2 = this.props,
          items = _props2.items,
          title = _props2.title;
      var _state3 = this.state,
          openEndTime = _state3.openEndTime,
          openStartTime = _state3.openStartTime,
          includeSeconds = _state3.includeSeconds,
          timeRange = _state3.timeRange;


      var modalContent = _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(
          'div',
          { style: _DateTimeRangePicker.CHECKBOX_STYLE },
          _react2.default.createElement(_Checkbox2.default, {
            checked: includeSeconds,
            labelText: 'Include seconds',
            name: 'includeSeconds',
            onChange: this._handleSecondsCheck,
            value: 'includeSeconds'
          })
        ),
        _react2.default.createElement(
          'div',
          { style: _DateTimeRangePicker.TIME_WRAPPER_STYLE },
          _react2.default.createElement(
            'div',
            { style: _DateTimeRangePicker.PICKER_WRAPPER_STYLE },
            _react2.default.createElement(_SingleSelect2.default, {
              closeDropdown: this._closeDropdown('start'),
              items: this._generateTimes(),
              label: 'Time range',
              onChange: this._handleTimeRangeChange('start'),
              onItemCreate: this._handleCustomTimeChange('start'),
              open: openStartTime,
              openDropdown: this._openDropdown('start'),
              secondaryText: (0, _moment2.default)(items.startDateTime).format('ll'),
              showCreateNew: true,
              value: timeRange.startTime
            })
          ),
          _react2.default.createElement(
            'div',
            { style: _DateTimeRangePicker.INPUT_SEPARATOR_STYLE },
            'to'
          ),
          _react2.default.createElement(
            'div',
            { style: _DateTimeRangePicker.PICKER_WRAPPER_STYLE },
            _react2.default.createElement(_SingleSelect2.default, {
              closeDropdown: this._closeDropdown('end'),
              items: this._generateTimes(),
              label: ' ',
              onChange: this._handleTimeRangeChange('end'),
              onItemCreate: this._handleCustomTimeChange('end'),
              open: openEndTime,
              openDropdown: this._openDropdown('end'),
              secondaryText: (0, _moment2.default)(items.endDateTime).format('ll'),
              showCreateNew: true,
              value: timeRange.endTime
            })
          )
        )
      );

      var modalFooter = _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(
          _Button2.default,
          {
            size: 'large',
            variant: 'accent',
            onClick: this._handleSaveModal
          },
          'Apply'
        ),
        _react2.default.createElement(
          _Button2.default,
          {
            size: 'large',
            variant: 'muted',
            onClick: this._handleCancelModal,
            _style: _DateTimeRangePicker.BUTTON
          },
          'Cancel'
        )
      );

      return _react2.default.createElement(
        _Modal2.default,
        {
          header: title,
          onDismiss: this._toggleModal,
          footer: modalFooter
        },
        modalContent
      );
    }
  }, {
    key: 'render',
    value: function render() {
      var open = this.props.open;


      return _react2.default.createElement(
        'div',
        { 'data-component': 'TxlTimeRangePicker' },
        _react2.default.createElement(
          'div',
          { style: _DateTimeRangePicker.BUTTON_STYLE },
          _react2.default.createElement(
            'div',
            {
              id: 'timeRangeToggle',
              onClick: this._toggleModal,
              style: _DateTimeRangePicker.RANGE
            },
            this._getTimeDisplay(),
            _react2.default.createElement(
              'span',
              { style: _DateTimeRangePicker.BUTTON },
              _react2.default.createElement(_EditOutline2.default, { color: _theme.COLOR_NEUTRAL[400] })
            )
          )
        ),
        open && this.renderModal()
      );
    }
  }]);
  return TxlTimeRangePicker;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_toggleModal', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_toggleModal'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleSaveModal', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleSaveModal'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleCancelModal', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleCancelModal'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleTimeRangeChange', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleTimeRangeChange'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleCustomTimeChange', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleCustomTimeChange'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_generateTimes', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_generateTimes'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_openDropdown', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_openDropdown'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_closeDropdown', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_closeDropdown'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_getTimeDisplayFormat', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_getTimeDisplayFormat'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_getTimeDisplay', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_getTimeDisplay'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleSecondsCheck', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleSecondsCheck'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, 'renderModal', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, 'renderModal'), _class2.prototype)), _class2)) || _class;

TxlTimeRangePicker.propTypes = {
  items: require('react').PropTypes.object.isRequired,
  onChange: require('react').PropTypes.func.isRequired,
  open: require('react').PropTypes.bool,
  title: require('react').PropTypes.any,
  updateDropdown: require('react').PropTypes.func.isRequired
};
exports.default = TxlTimeRangePicker;


TxlTimeRangePicker.defaultProps = {
  items: {},
  onChange: _Base.NOOP,
  open: false,
  title: 'Time range',
  updateDropdown: _Base.NOOP
};
;

var _temp2 = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TIME_RANGE_FORMAT_SIMPLIFIED, 'TIME_RANGE_FORMAT_SIMPLIFIED', 'src/_date-time-range-picker/_TimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(TIME_DISPLAY_FORMAT_SIMPLIFIED, 'TIME_DISPLAY_FORMAT_SIMPLIFIED', 'src/_date-time-range-picker/_TimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(TIME_RANGE_FORMAT, 'TIME_RANGE_FORMAT', 'src/_date-time-range-picker/_TimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(TIME_DISPLAY_FORMAT, 'TIME_DISPLAY_FORMAT', 'src/_date-time-range-picker/_TimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(TIME_QUERY_FORMAT, 'TIME_QUERY_FORMAT', 'src/_date-time-range-picker/_TimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(DATE_QUERY_FORMAT, 'DATE_QUERY_FORMAT', 'src/_date-time-range-picker/_TimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(TxlTimeRangePicker, 'TxlTimeRangePicker', 'src/_date-time-range-picker/_TimeRangePicker.jsx');
}();

;
