// txl/_date-time-range-picker/DateTimeRangePicker.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.DEFAULT_ITEMS = exports.DEFAULT_END_DATE = exports.DEFAULT_START_DATE = exports.DATE_QUERY_FORMAT = exports.DATE_RANGE_FORMAT = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _Base = require('../base/Base');

var _DateTimeRangePicker = require('./DateTimeRangePicker.style');

var _TimezonePicker = require('./_TimezonePicker');

var _TimezonePicker2 = _interopRequireDefault(_TimezonePicker);

var _DateRangePicker = require('./_DateRangePicker');

var _DateRangePicker2 = _interopRequireDefault(_DateRangePicker);

var _TimeRangePicker = require('./_TimeRangePicker');

var _TimeRangePicker2 = _interopRequireDefault(_TimeRangePicker);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var DATE_RANGE_FORMAT = 'll';
var DATE_QUERY_FORMAT = 'YYYY-MM-DD HH:mm:ss';

var DEFAULT_START_DATE = (0, _moment2.default)().startOf('day');
var DEFAULT_END_DATE = (0, _moment2.default)().endOf('day');

var DEFAULT_ITEMS = {
  endDateTime: DEFAULT_END_DATE.format(DATE_QUERY_FORMAT),
  locale: {
    label: '(UTC -05:00) America/New_York',
    timeZone: 'America/New_York',
    value: '(UTC -05:00) America/New_York'
  },
  preset: {
    endDate: DEFAULT_END_DATE,
    label: 'Today',
    range: (0, _moment2.default)().format(DATE_RANGE_FORMAT) + ' - ' + (0, _moment2.default)().format(DATE_RANGE_FORMAT),
    startDate: DEFAULT_START_DATE,
    value: 'today'
  },
  startDateTime: DEFAULT_START_DATE.format(DATE_QUERY_FORMAT)
};

exports.DATE_RANGE_FORMAT = DATE_RANGE_FORMAT;
exports.DATE_QUERY_FORMAT = DATE_QUERY_FORMAT;
exports.DEFAULT_START_DATE = DEFAULT_START_DATE;
exports.DEFAULT_END_DATE = DEFAULT_END_DATE;
exports.DEFAULT_ITEMS = DEFAULT_ITEMS;

var TxlDateTimeRangePicker = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlDateTimeRangePicker, _React$Component);

  function TxlDateTimeRangePicker() {
    var _ref;

    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, TxlDateTimeRangePicker);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlDateTimeRangePicker.__proto__ || (0, _getPrototypeOf2.default)(TxlDateTimeRangePicker)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
      items: _this.props.items || DEFAULT_ITEMS,
      presetPickerOpen: false,
      timeRangePickerOpen: false,
      timezonePickerOpen: false
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  (0, _createClass3.default)(TxlDateTimeRangePicker, [{
    key: '_handleChange',
    value: function _handleChange(items) {
      this.setState({ items: items });
      this.props.onChange(items);
    }
  }, {
    key: '_handleDropdown',
    value: function _handleDropdown(type) {
      var _this2 = this;

      return function (value) {
        _this2.setState({
          presetPickerOpen: type === 'preset' ? value.open : false,
          timeRangePickerOpen: type === 'time_range' ? value.open : false,
          timezonePickerOpen: type === 'timezone' ? value.open : false
        });
      };
    }
  }, {
    key: 'render',
    value: function render() {
      var _state = this.state,
          items = _state.items,
          presetPickerOpen = _state.presetPickerOpen,
          timeRangePickerOpen = _state.timeRangePickerOpen,
          timezonePickerOpen = _state.timezonePickerOpen;
      var _props = this.props,
          buttonItems = _props.buttonItems,
          presets = _props.presets,
          timeRangeTitle = _props.timeRangeTitle,
          withTimeRange = _props.withTimeRange;


      return _react2.default.createElement(
        'div',
        { 'data-component': 'TxlDateTimeRangePicker' },
        _react2.default.createElement(
          'div',
          { style: _DateTimeRangePicker.BUTTON_ROW },
          _react2.default.createElement(_DateRangePicker2.default, {
            buttonItems: buttonItems,
            items: items,
            onChange: this._handleChange,
            open: presetPickerOpen,
            presets: presets,
            updateDropdown: this._handleDropdown('preset')
          }),
          _react2.default.createElement(
            'div',
            { style: _DateTimeRangePicker.CUSTOM_DATE_RANGE },
            withTimeRange && _react2.default.createElement(_TimeRangePicker2.default, {
              items: items,
              onChange: this._handleChange,
              open: timeRangePickerOpen,
              title: timeRangeTitle,
              updateDropdown: this._handleDropdown('time_range')
            }),
            _react2.default.createElement(_TimezonePicker2.default, {
              items: items,
              onChange: this._handleChange,
              open: timezonePickerOpen,
              updateDropdown: this._handleDropdown('timezone')
            })
          )
        )
      );
    }
  }]);
  return TxlDateTimeRangePicker;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleChange', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleChange'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDropdown', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDropdown'), _class2.prototype)), _class2)) || _class;

TxlDateTimeRangePicker.propTypes = {
  buttonItems: require('react').PropTypes.arrayOf(require('react').PropTypes.any),
  items: require('react').PropTypes.object,
  onChange: require('react').PropTypes.func.isRequired,
  presets: require('react').PropTypes.arrayOf(require('react').PropTypes.object),
  timeRangeTitle: require('react').PropTypes.any,
  withTimeRange: require('react').PropTypes.bool
};
exports.default = TxlDateTimeRangePicker;


TxlDateTimeRangePicker.defaultProps = {
  onChange: _Base.NOOP
};
;

var _temp2 = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(DATE_RANGE_FORMAT, 'DATE_RANGE_FORMAT', 'src/_date-time-range-picker/DateTimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(DATE_QUERY_FORMAT, 'DATE_QUERY_FORMAT', 'src/_date-time-range-picker/DateTimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(DEFAULT_START_DATE, 'DEFAULT_START_DATE', 'src/_date-time-range-picker/DateTimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(DEFAULT_END_DATE, 'DEFAULT_END_DATE', 'src/_date-time-range-picker/DateTimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(DEFAULT_ITEMS, 'DEFAULT_ITEMS', 'src/_date-time-range-picker/DateTimeRangePicker.jsx');

  __REACT_HOT_LOADER__.register(TxlDateTimeRangePicker, 'TxlDateTimeRangePicker', 'src/_date-time-range-picker/DateTimeRangePicker.jsx');
}();

;
