// txl/_date-time-range-picker/DateTimeRangePicker.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TIME_WRAPPER_STYLE = exports.PICKER_WRAPPER_STYLE = exports.LABEL_STYLE = exports.INPUT_SEPARATOR_STYLE = exports.BUTTON = exports.CUSTOM_DATE_CONTAINER = exports.PRESET_CONTAINER = exports.VALIDATION = exports.TIMEZONE_SELECT = exports.TIMEZONE = exports.RANGE = exports.CUSTOM_DATE_RANGE = exports.CUSTOM_DATE_RANGE_TEXT = exports.TIMEZONE_WRAPPER = exports.CHECKBOX_STYLE = exports.BUTTON_STYLE = exports.BUTTON_ROW = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _type = require('../styles/mixins/type');

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BUTTON_ROW = exports.BUTTON_ROW = {
  alignItems: 'flex-start',
  display: 'flex',
  height: (0, _helpers.gridUnits)(12),
  justifyContent: 'space-between',
  minWidth: (0, _helpers.gridUnits)(160),
  position: 'relative'
};

var BUTTON_STYLE = exports.BUTTON_STYLE = {
  alignItems: 'center',
  display: 'flex',
  height: (0, _helpers.gridUnits)(7)
};

var CHECKBOX_STYLE = exports.CHECKBOX_STYLE = {
  paddingBottom: (0, _helpers.gridUnits)(4)
};

var TIMEZONE_WRAPPER = exports.TIMEZONE_WRAPPER = (0, _extends3.default)({}, BUTTON_STYLE, {
  marginLeft: (0, _helpers.gridUnits)(12)
});

var CUSTOM_DATE_RANGE_TEXT = exports.CUSTOM_DATE_RANGE_TEXT = (0, _extends3.default)({}, _type.TYPE_SMALL, {
  color: _theme.COLOR_NEUTRAL['400'],
  paddingLeft: (0, _helpers.gridUnits)(2),
  paddingRight: (0, _helpers.gridUnits)(2)
});

var CUSTOM_DATE_RANGE = exports.CUSTOM_DATE_RANGE = {
  alignItems: 'center',
  display: 'flex'
};

var RANGE = exports.RANGE = (0, _extends3.default)({}, _type.TYPE_SMALL, {
  alignItems: 'center',
  color: _theme.COLOR_NEUTRAL[400],
  cursor: 'pointer',
  display: 'flex',
  height: (0, _helpers.gridUnits)(5)
});

var TIMEZONE = exports.TIMEZONE = (0, _extends3.default)({}, RANGE, {
  position: 'absolute',
  right: 0
});

var TIMEZONE_SELECT = exports.TIMEZONE_SELECT = {
  height: 0,
  opacity: 0
};

var VALIDATION = exports.VALIDATION = (0, _extends3.default)({}, _type.TYPE_SMALL, {
  alignItems: 'center',
  color: _theme.COLOR_ERROR[400],
  display: 'flex',
  height: (0, _helpers.gridUnits)(5)
});

var PRESET_CONTAINER = exports.PRESET_CONTAINER = {
  alignItems: 'flex-start',
  display: 'flex',
  flexDirection: 'column',
  height: (0, _helpers.gridUnits)(12)
};

var CUSTOM_DATE_CONTAINER = exports.CUSTOM_DATE_CONTAINER = {
  height: (0, _helpers.gridUnits)(7),
  width: (0, _helpers.gridUnits)(26)
};

var BUTTON = exports.BUTTON = {
  marginLeft: (0, _helpers.gridUnits)(2)
};

var INPUT_SEPARATOR_STYLE = exports.INPUT_SEPARATOR_STYLE = (0, _extends3.default)({}, _type.LABEL, {
  alignSelf: 'center',
  color: _theme.COLOR_NEUTRAL['400'],
  paddingLeft: (0, _helpers.gridUnits)(2),
  paddingRight: (0, _helpers.gridUnits)(2)
});

var LABEL_STYLE = exports.LABEL_STYLE = (0, _extends3.default)({}, _type.LABEL, {
  color: _theme.COLOR_NEUTRAL['400']
});

var PICKER_WRAPPER_STYLE = exports.PICKER_WRAPPER_STYLE = {
  width: (0, _helpers.gridUnits)(24)
};

var TIME_WRAPPER_STYLE = exports.TIME_WRAPPER_STYLE = {
  alignItems: 'stretch',
  display: 'flex',
  flexWrap: 'nowrap'
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(BUTTON_ROW, 'BUTTON_ROW', 'src/_date-time-range-picker/DateTimeRangePicker.style.js');

  __REACT_HOT_LOADER__.register(BUTTON_STYLE, 'BUTTON_STYLE', 'src/_date-time-range-picker/DateTimeRangePicker.style.js');

  __REACT_HOT_LOADER__.register(CHECKBOX_STYLE, 'CHECKBOX_STYLE', 'src/_date-time-range-picker/DateTimeRangePicker.style.js');

  __REACT_HOT_LOADER__.register(TIMEZONE_WRAPPER, 'TIMEZONE_WRAPPER', 'src/_date-time-range-picker/DateTimeRangePicker.style.js');

  __REACT_HOT_LOADER__.register(CUSTOM_DATE_RANGE_TEXT, 'CUSTOM_DATE_RANGE_TEXT', 'src/_date-time-range-picker/DateTimeRangePicker.style.js');

  __REACT_HOT_LOADER__.register(CUSTOM_DATE_RANGE, 'CUSTOM_DATE_RANGE', 'src/_date-time-range-picker/DateTimeRangePicker.style.js');

  __REACT_HOT_LOADER__.register(RANGE, 'RANGE', 'src/_date-time-range-picker/DateTimeRangePicker.style.js');

  __REACT_HOT_LOADER__.register(TIMEZONE, 'TIMEZONE', 'src/_date-time-range-picker/DateTimeRangePicker.style.js');

  __REACT_HOT_LOADER__.register(TIMEZONE_SELECT, 'TIMEZONE_SELECT', 'src/_date-time-range-picker/DateTimeRangePicker.style.js');

  __REACT_HOT_LOADER__.register(VALIDATION, 'VALIDATION', 'src/_date-time-range-picker/DateTimeRangePicker.style.js');

  __REACT_HOT_LOADER__.register(PRESET_CONTAINER, 'PRESET_CONTAINER', 'src/_date-time-range-picker/DateTimeRangePicker.style.js');

  __REACT_HOT_LOADER__.register(CUSTOM_DATE_CONTAINER, 'CUSTOM_DATE_CONTAINER', 'src/_date-time-range-picker/DateTimeRangePicker.style.js');

  __REACT_HOT_LOADER__.register(BUTTON, 'BUTTON', 'src/_date-time-range-picker/DateTimeRangePicker.style.js');

  __REACT_HOT_LOADER__.register(INPUT_SEPARATOR_STYLE, 'INPUT_SEPARATOR_STYLE', 'src/_date-time-range-picker/DateTimeRangePicker.style.js');

  __REACT_HOT_LOADER__.register(LABEL_STYLE, 'LABEL_STYLE', 'src/_date-time-range-picker/DateTimeRangePicker.style.js');

  __REACT_HOT_LOADER__.register(PICKER_WRAPPER_STYLE, 'PICKER_WRAPPER_STYLE', 'src/_date-time-range-picker/DateTimeRangePicker.style.js');

  __REACT_HOT_LOADER__.register(TIME_WRAPPER_STYLE, 'TIME_WRAPPER_STYLE', 'src/_date-time-range-picker/DateTimeRangePicker.style.js');
}();

;
