// txl/_tooltips/_Positioner.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _desc, _value, _class;

var _Base = require('../base/Base');

var _constants = require('../styles/constants');

var _PositionerHelpers = require('./_PositionerHelpers');

var _jsStyleHelpers = require('../styles/_helpers/jsStyleHelpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var Positioner = (_class = function () {
  function Positioner(overlay) {
    (0, _classCallCheck3.default)(this, Positioner);

    this._overlay = overlay;
    this._initialize();
  }

  (0, _createClass3.default)(Positioner, [{
    key: '_initialize',
    value: function _initialize() {
      this._initializeOverlayStyles();
      this.updatePositioningData();
      this._initialized = true;
    }
  }, {
    key: '_getTranslateX',
    value: function _getTranslateX(xCoord) {
      return xCoord - (this._width / 2 + this._left);
    }
  }, {
    key: '_getTranslateY',
    value: function _getTranslateY(yCoord) {
      return yCoord - (this._height + this._top);
    }
  }, {
    key: '_getTransformStyleFromCoords',
    value: function _getTransformStyleFromCoords(xCoord, yCoord) {
      var translateY = this._getTranslateY(yCoord);
      var translateX = this._getTranslateX(xCoord);

      translateX = (0, _PositionerHelpers.correctOverlappingTranslateX)(this._left, this._right, translateX);

      return (0, _PositionerHelpers.getPrefixedTransformStyleObject)(translateX, translateY);
    }
  }, {
    key: '_initializeOverlayStyles',
    value: function _initializeOverlayStyles() {
      var overlayStyles = {
        display: 'inline-flex',
        left: '-9999px',
        position: 'absolute',
        top: 0,
        zIndex: _constants.Z_INDEXES.popover
      };

      (0, _jsStyleHelpers.setStylesForNode)(this._overlay, overlayStyles);
    }
  }, {
    key: '_resetOverlayTransform',
    value: function _resetOverlayTransform() {
      var resetStyleObject = (0, _PositionerHelpers.getPrefixedTransformStyleObject)(0, 0);
      (0, _jsStyleHelpers.setStylesForNode)(this._overlay, resetStyleObject);
    }
  }, {
    key: 'updatePositioningData',
    value: function updatePositioningData() {
      if (this._initialized) {
        // Since we're reading from getBoundingClientRect, we need to reset the transform
        // applied to the overlay before reading the value so we know the perspective of
        // subsequent transforms.
        this._resetOverlayTransform();
      }

      var boundingRect = this._overlay.getBoundingClientRect();

      this._left = boundingRect.left;
      this._right = boundingRect.right;
      this._top = boundingRect.top;
      this._width = boundingRect.width;
      this._height = boundingRect.height;
    }
  }, {
    key: 'positionOverlayToCoords',
    value: function positionOverlayToCoords(xCoord, yCoord) {
      var transformStyleObject = this._getTransformStyleFromCoords(xCoord, yCoord);
      (0, _jsStyleHelpers.setStylesForNode)(this._overlay, transformStyleObject);
    }
  }]);
  return Positioner;
}(), (_applyDecoratedDescriptor(_class.prototype, '_initialize', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_initialize'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_getTranslateX', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_getTranslateX'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_getTranslateY', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_getTranslateY'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_getTransformStyleFromCoords', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_getTransformStyleFromCoords'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_initializeOverlayStyles', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_initializeOverlayStyles'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_resetOverlayTransform', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_resetOverlayTransform'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, 'updatePositioningData', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, 'updatePositioningData'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, 'positionOverlayToCoords', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, 'positionOverlayToCoords'), _class.prototype)), _class);
exports.default = Positioner;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(Positioner, 'Positioner', 'src/_tooltips/_Positioner.js');
}();

;
module.exports = exports['default'];
