// txl/_tooltips/_TooltipTargetBase.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _helpers = require('../styles/helpers');

var _TooltipContent = require('./_TooltipContent');

var _TooltipContent2 = _interopRequireDefault(_TooltipContent);

var _TetheredPopoverTarget = require('../_popover/_TetheredPopoverTarget');

var _TetheredPopoverTarget2 = _interopRequireDefault(_TetheredPopoverTarget);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TETHER_CONFIGURATION = {
  attachment: 'bottom center',
  constraints: [{
    attachment: 'both',
    pin: true,
    to: 'window'
  }],
  offset: (0, _helpers.gridUnits)(1) + ' 0',
  optimizations: {
    moveElement: false
  },
  targetAttachment: 'top center',
  targetOffset: '0 0'
};

var TxlTooltipTargetBase = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlTooltipTargetBase, _React$Component);

  function TxlTooltipTargetBase() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlTooltipTargetBase);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlTooltipTargetBase.__proto__ || (0, _getPrototypeOf2.default)(TxlTooltipTargetBase)).call.apply(_ref, [this].concat(args)));

    _this._timeout = null;
    _this.state = { visible: false };
    _this._mounted = false;
    return _this;
  }

  (0, _createClass3.default)(TxlTooltipTargetBase, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      this._mounted = true;
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      this._mounted = false;
    }
  }, {
    key: '_showTooltip',
    value: function _showTooltip() {
      var _this2 = this;

      this._timeout = setTimeout(function () {
        if (_this2._mounted) {
          _this2.setState({ visible: true });
        }
      }, this.props.delay);
    }
  }, {
    key: '_hideTooltip',
    value: function _hideTooltip() {
      clearTimeout(this._timeout);
      this.setState({ visible: false });
    }
  }, {
    key: '_handleMouseMove',
    value: function _handleMouseMove() {
      if (this.state.visible || this._timeout) return;

      this._showTooltip();
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          children = _props.children,
          message = _props.message,
          otherProps = (0, _objectWithoutProperties3.default)(_props, ['children', 'message']);


      var mouseProps = {
        onMouseEnter: this._showTooltip,
        onMouseLeave: this._hideTooltip,
        onMouseMove: this._handleMouseMove
      };

      var tooltipContent = _react2.default.createElement(_TooltipContent2.default, { message: message });

      return _react2.default.createElement(
        _TetheredPopoverTarget2.default,
        (0, _extends3.default)({}, mouseProps, otherProps, {
          configuration: TETHER_CONFIGURATION,
          visible: this.state.visible,
          content: tooltipContent,
          popoverZIndex: this.props.popoverZIndex
        }),
        children
      );
    }
  }]);
  return TxlTooltipTargetBase;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_showTooltip', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_showTooltip'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_hideTooltip', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_hideTooltip'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleMouseMove', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseMove'), _class2.prototype)), _class2)) || _class;

exports.default = TxlTooltipTargetBase;


TxlTooltipTargetBase.propTypes = {
  children: _TetheredPopoverTarget2.default.propTypes.children,
  componentName: _TetheredPopoverTarget2.default.propTypes.componentName,
  delay: _react.PropTypes.number,
  message: _TooltipContent2.default.propTypes.message,
  popoverZIndex: _TetheredPopoverTarget2.default.propTypes.popoverZIndex,
  style: _TetheredPopoverTarget2.default.propTypes.style
};

TxlTooltipTargetBase.defaultProps = {
  delay: 0
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TETHER_CONFIGURATION, 'TETHER_CONFIGURATION', 'src/_tooltips/_TooltipTargetBase.jsx');

  __REACT_HOT_LOADER__.register(TxlTooltipTargetBase, 'TxlTooltipTargetBase', 'src/_tooltips/_TooltipTargetBase.jsx');
}();

;
module.exports = exports['default'];
