// txl/_tooltips/_TooltipContent.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _theme = require('../styles/theme');

var _layers = require('../styles/mixins/layers');

var _type = require('../styles/mixins/type');

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var STYLES = (0, _extends3.default)({}, _layers.LAYER_2, _type.TYPE_BASE, _type.DEFAULT_FONT_FAMILY, (0, _helpers.allBorderRadius)(2), {
  alignItems: 'center',
  backgroundColor: _theme.COLOR_NEUTRAL['900'],
  color: _theme.COLOR_NEUTRAL['900c'],
  display: 'inline-flex',
  maxWidth: (0, _helpers.gridUnits)(60),
  minHeight: (0, _helpers.gridUnits)(5),
  padding: '2.5px ' + (0, _helpers.gridUnits)(2),
  wordBreak: 'break-word'
});

var TxlTooltipContent = function TxlTooltipContent(props) {
  return _react2.default.createElement(
    'div',
    {
      'data-component': 'TxlTooltipContent',
      style: STYLES,
      id: props.id
    },
    props.message
  );
};

TxlTooltipContent.propTypes = {
  id: _react.PropTypes.string,
  message: _react.PropTypes.node.isRequired
};

var _default = (0, _radium2.default)(TxlTooltipContent);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(STYLES, 'STYLES', 'src/_tooltips/_TooltipContent.jsx');

  __REACT_HOT_LOADER__.register(TxlTooltipContent, 'TxlTooltipContent', 'src/_tooltips/_TooltipContent.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/_tooltips/_TooltipContent.jsx');
}();

;
module.exports = exports['default'];
