// txl/_tooltips/CursorTooltipTarget.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _ReactUtils = require('../ReactUtils');

var _Base = require('../base/Base');

var _StyleRoot = require('../styles/StyleRoot');

var _StyleRoot2 = _interopRequireDefault(_StyleRoot);

var _TooltipContent = require('./_TooltipContent');

var _TooltipContent2 = _interopRequireDefault(_TooltipContent);

var _Positioner = require('./_Positioner');

var _Positioner2 = _interopRequireDefault(_Positioner);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

// @TODO: Import window.requestAnimationFrame instead of referencing it inline

// @TODO Maybe:
// If you happen to scroll the window while still hovering over a thing
// it will maintain the older incorrect positioning since this wont fire
// a second time, consider adding resize/scroll handlers to recalculate

// @TODO: Animate in and out of existence

var TxlCursorTooltipTarget = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlCursorTooltipTarget, _React$Component);

  function TxlCursorTooltipTarget() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlCursorTooltipTarget);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlCursorTooltipTarget.__proto__ || (0, _getPrototypeOf2.default)(TxlCursorTooltipTarget)).call.apply(_ref, [this].concat(args)));

    _this._yCoordMargin = 15;
    _this._yCoordCache = null;
    _this._xCoordCache = null;
    _this._positionableTooltip = null;
    _this._tooltip = null;
    _this._bootstrapped = false;
    _this._renderRequired = false;
    return _this;
  }

  (0, _createClass3.default)(TxlCursorTooltipTarget, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      if (this.props.message !== nextProps.message) {
        this._renderRequired = true;
      }
    }
  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate() {
      if (!this._renderRequired || !this._bootstrapped) {
        return;
      }

      this._renderTooltip();
      window.requestAnimationFrame(this._handleTooltipContentUpdate);
      this._positionToCoordinates(this._yCoordCache, this._xCoordCache);
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      this._teardownTooltip();
    }
  }, {
    key: '_bootstrapTooltip',
    value: function _bootstrapTooltip() {
      this._bootstrapped = true;
      this._createTooltip();
      this._renderTooltip();
      this._positionableTooltip = new _Positioner2.default(this._tooltip);
    }
  }, {
    key: '_teardownTooltip',
    value: function _teardownTooltip() {
      if (this._bootstrapped) {
        (0, _ReactUtils.unmountComponentAtNode)(this._tooltip);
        document.querySelector('body').removeChild(this._tooltip);
      }

      this._bootstrapped = false;
    }
  }, {
    key: '_createTooltip',
    value: function _createTooltip() {
      var tooltipContainer = document.createElement('div');
      tooltipContainer.setAttribute('data-component', 'TxlCursorTooltip');

      if (this.props.id) {
        tooltipContainer.id = this.props.id;
      }

      this._tooltip = tooltipContainer;
      // @NOTE: Body could be cached as an optimization
      document.querySelector('body').appendChild(this._tooltip);
    }
  }, {
    key: '_renderTooltip',
    value: function _renderTooltip() {
      var tooltipContents = _react2.default.createElement(
        _StyleRoot2.default,
        null,
        _react2.default.createElement(_TooltipContent2.default, { message: this.props.message })
      );

      (0, _ReactUtils.render)(tooltipContents, this._tooltip);
    }
  }, {
    key: '_handleTooltipContentUpdate',
    value: function _handleTooltipContentUpdate() {
      this._positionableTooltip.updatePositioningData();
    }
  }, {
    key: '_handleMouseEnter',
    value: function _handleMouseEnter() {
      if (this._bootstrapped) {
        return;
      }

      this._bootstrapTooltip();
    }
  }, {
    key: '_handleMouseLeave',
    value: function _handleMouseLeave() {
      this._teardownTooltip();
    }
  }, {
    key: '_handleMouseMove',
    value: function _handleMouseMove(e) {
      if (!this._bootstrapped) {
        this._bootstrapTooltip();
      }

      this._yCoordCache = e.clientY - this._yCoordMargin;
      this._xCoordCache = e.clientX;

      this._positionToCoordinates(this._yCoordCache, this._xCoordCache);
    }
  }, {
    key: '_positionToCoordinates',
    value: function _positionToCoordinates(y, x) {
      var _this2 = this;

      window.requestAnimationFrame(function () {
        _this2._positionableTooltip.positionOverlayToCoords(x, y);
      });
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlCursorTooltipTarget',
          style: this.props.style,
          onMouseMoveCapture: this._handleMouseMove,
          onMouseEnter: this._handleMouseEnter,
          onMouseLeave: this._handleMouseLeave
        },
        this.props.children
      );
    }
  }]);
  return TxlCursorTooltipTarget;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_bootstrapTooltip', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_bootstrapTooltip'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_teardownTooltip', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_teardownTooltip'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_createTooltip', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_createTooltip'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_renderTooltip', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_renderTooltip'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleTooltipContentUpdate', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleTooltipContentUpdate'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleMouseEnter', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseEnter'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleMouseLeave', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseLeave'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleMouseMove', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseMove'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_positionToCoordinates', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_positionToCoordinates'), _class2.prototype)), _class2)) || _class;

exports.default = TxlCursorTooltipTarget;


TxlCursorTooltipTarget.propTypes = {
  children: _react.PropTypes.node,
  id: _react.PropTypes.string,
  message: _react.PropTypes.node.isRequired,
  style: _react.PropTypes.object
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlCursorTooltipTarget, 'TxlCursorTooltipTarget', 'src/_tooltips/CursorTooltipTarget.jsx');
}();

;
module.exports = exports['default'];
