// txl/tag/Tag.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

var _type = require('../styles/mixins/type');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TAG_HEIGHT = (0, _helpers.gridUnitsRaw)(3.4); // 17px

var TAG_BASE_STYLE = (0, _extends3.default)({}, _type.DEFAULT_FONT_FAMILY, _type.TYPE_SMALL, {
  borderRadius: TAG_HEIGHT * 0.588 + 'px', // 10:17 padding/border-radius:height ratio
  boxSizing: 'border-box',
  display: 'inline-flex',
  height: TAG_HEIGHT,
  lineHeight: TAG_HEIGHT + 'px',
  padding: '0 ' + TAG_HEIGHT * 0.588 + 'px'
});

var TxlTag = function TxlTag(props) {
  var TAG_COLOR_STYLE = {
    backgroundColor: props.backgroundColor,
    color: props.textColor
  };

  var TAG_STYLE = (0, _helpers.combineStyles)(TAG_BASE_STYLE, TAG_COLOR_STYLE);

  return _react2.default.createElement(
    'div',
    {
      'data-component': 'TxlTag',
      id: props.id,
      style: TAG_STYLE
    },
    _react2.default.createElement(
      'div',
      null,
      props.children
    )
  );
};

TxlTag.propTypes = {
  backgroundColor: _react.PropTypes.string,
  children: _react.PropTypes.node,
  id: _react.PropTypes.string,
  textColor: _react.PropTypes.string
};

TxlTag.defaultProps = {
  backgroundColor: _theme.COLOR_NEUTRAL['100'],
  textColor: _theme.COLOR_NEUTRAL['100c']
};

var _default = (0, _radium2.default)(TxlTag);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TAG_HEIGHT, 'TAG_HEIGHT', 'src/tag/Tag.jsx');

  __REACT_HOT_LOADER__.register(TAG_BASE_STYLE, 'TAG_BASE_STYLE', 'src/tag/Tag.jsx');

  __REACT_HOT_LOADER__.register(TxlTag, 'TxlTag', 'src/tag/Tag.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/tag/Tag.jsx');
}();

;
module.exports = exports['default'];
