// txl/calendar-picker/_Calendar.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _helpers = require('../styles/helpers');

var _type = require('../styles/mixins/type');

var _theme = require('../styles/theme');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CELL_SIZE = (0, _helpers.gridUnitsRaw)(1);
var CAL_SIZE = 7 * CELL_SIZE;

var _default = {
  calendar: (0, _extends3.default)({}, _type.DEFAULT_FONT_FAMILY, {
    userSelect: 'none',
    width: (0, _helpers.gridUnits)(CAL_SIZE)
  }),
  cell: {
    base: (0, _extends3.default)({}, (0, _helpers.square)(CELL_SIZE), {
      boxSizing: 'border-box',
      padding: 0,
      textAlign: 'center'
    }),
    current: (0, _extends3.default)({}, (0, _helpers.allBorderRadius)('2px'), {
      backgroundColor: _theme.COLOR_ACCENT['100'],
      color: _theme.COLOR_ACCENT['100c']
    }),
    hover: (0, _extends3.default)({}, (0, _helpers.allBorderRadius)('50%'), {
      backgroundColor: _theme.COLOR_NEUTRAL['100'],
      color: _theme.COLOR_NEUTRAL['100c']
    })
  },
  container: {
    display: 'flex',
    justifyContent: 'space-between',
    position: 'absolute',
    width: (0, _helpers.gridUnits)(35)
  },
  date: {
    base: (0, _extends3.default)({}, (0, _helpers.square)((0, _helpers.gridUnits)(CELL_SIZE)), {
      display: 'block',
      fontSize: 12,
      lineHeight: (0, _helpers.gridUnits)(CELL_SIZE)
    }),
    disabled: {
      color: _theme.COLOR_NEUTRAL['200'],
      cursor: 'not-allowed'
    },
    inRange: {
      backgroundColor: _theme.COLOR_ACCENT['500'],
      color: _theme.COLOR_ACCENT['500c']
    },
    rangeEnd: {
      borderBottomRightRadius: '50%',
      borderTopRightRadius: '50%'
    },
    rangeStart: {
      borderBottomLeftRadius: '50%',
      borderTopLeftRadius: '50%'
    }
  },
  dayHeader: (0, _extends3.default)({}, (0, _helpers.square)((0, _helpers.gridUnits)(CELL_SIZE)), _type.TYPE_SMALL, {
    color: _theme.COLOR_NEUTRAL['800'],
    textAlign: 'center'
  }),
  monthHeader: {
    fontSize: 18,
    fontWeight: 400,
    textAlign: 'center'
  },
  table: {
    borderCollapse: 'collapse',
    boxSizing: 'border-box',
    marginBottom: 0,
    tableLayout: 'fixed',
    width: (0, _helpers.gridUnits)(CAL_SIZE)
  }
};
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(CELL_SIZE, 'CELL_SIZE', 'src/calendar-picker/_Calendar.style.js');

  __REACT_HOT_LOADER__.register(CAL_SIZE, 'CAL_SIZE', 'src/calendar-picker/_Calendar.style.js');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/calendar-picker/_Calendar.style.js');
}();

;
module.exports = exports['default'];
