// txl/calendar-picker/_CalendarPresets.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

var _default = {
  icon: {
    flex: '0 0 13'
  },
  label: {
    flex: '0 1 auto'
  },
  list: {
    listStyleType: 'none',
    margin: 0,
    padding: 0
  },
  listItem: {
    base: {
      alignItems: 'center',
      cursor: 'pointer',
      display: 'flex',
      fontSize: 14,
      justifyContent: 'space-between',
      lineHeight: (0, _helpers.gridUnits)(7),
      padding: '0 ' + (0, _helpers.gridUnits)(2) + ' 0 ' + (0, _helpers.gridUnits)(4)
    },
    selected: {
      backgroundColor: _theme.COLOR_ACCENT['500'],
      color: _theme.COLOR_ACCENT['500c']
    }
  }
};
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/calendar-picker/_CalendarPresets.style.js');
}();

;
module.exports = exports['default'];
