// txl/calendar-picker/_CalendarPresets.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _clone = require('lodash/clone');

var _clone2 = _interopRequireDefault(_clone);

var _Base = require('../base/Base');

var _propTypes = require('../utils/propTypes');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

var _CalendarPresets = require('./_CalendarPresets.style');

var _CalendarPresets2 = _interopRequireDefault(_CalendarPresets);

var _CalendarUtils = require('./_CalendarUtils');

var _Check = require('../icons/Check');

var _Check2 = _interopRequireDefault(_Check);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlCalendarPresets = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlCalendarPresets, _React$Component);

  function TxlCalendarPresets() {
    (0, _classCallCheck3.default)(this, TxlCalendarPresets);
    return (0, _possibleConstructorReturn3.default)(this, (TxlCalendarPresets.__proto__ || (0, _getPrototypeOf2.default)(TxlCalendarPresets)).apply(this, arguments));
  }

  (0, _createClass3.default)(TxlCalendarPresets, [{
    key: '_handlePresetClick',
    value: function _handlePresetClick(event, preset, key) {
      this.props.onSelect({
        event: event,
        key: key,
        preset: preset
      });
    }
  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      var _props = this.props,
          presets = _props.presets,
          selectedKey = _props.selectedKey;


      return _react2.default.createElement(
        'div',
        { 'data-component': 'TxlCalendarPresets' },
        _react2.default.createElement(
          'ul',
          { style: _CalendarPresets2.default.list },
          (0, _clone2.default)(presets).map(function (preset) {
            var isSelected = selectedKey === preset.key;
            var liStyles = (0, _helpers.combineStyles)(_CalendarPresets2.default.listItem.base, isSelected && _CalendarPresets2.default.listItem.selected);

            return _react2.default.createElement(
              'li',
              {
                style: liStyles,
                key: preset.key,
                onClick: function onClick(event) {
                  return _this2._handlePresetClick(event, preset, preset.key);
                }
              },
              _react2.default.createElement(
                'div',
                { style: _CalendarPresets2.default.label },
                preset.label
              ),
              isSelected && _react2.default.createElement(
                'div',
                { style: _CalendarPresets2.default.icon },
                _react2.default.createElement(_Check2.default, {
                  ref: 'icon',
                  color: _theme.COLOR_NEUTRAL['50']
                })
              )
            );
          })
        )
      );
    }
  }]);
  return TxlCalendarPresets;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handlePresetClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handlePresetClick'), _class2.prototype)), _class2)) || _class;

exports.default = TxlCalendarPresets;


TxlCalendarPresets.propTypes = {
  onSelect: _react.PropTypes.func.isRequired,
  presets: _react.PropTypes.arrayOf(_propTypes2.default.dateRangePreset),
  selectedKey: _react.PropTypes.oneOf(_CalendarUtils.DATE_RANGE_PRESETS.map(function (preset) {
    return preset.key;
  }))
};

TxlCalendarPresets.defaultProps = {
  presets: _CalendarUtils.DATE_RANGE_PRESETS
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlCalendarPresets, 'TxlCalendarPresets', 'src/calendar-picker/_CalendarPresets.jsx');
}();

;
module.exports = exports['default'];
