// txl/calendar-picker/_SingleCalendarPicker.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _propTypes = require('../utils/propTypes');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Base = require('../base/Base');

var _IconButton = require('../buttons/IconButton');

var _IconButton2 = _interopRequireDefault(_IconButton);

var _Calendar = require('./_Calendar');

var _Calendar2 = _interopRequireDefault(_Calendar);

var _ArrowLeft = require('../icons/ArrowLeft');

var _ArrowLeft2 = _interopRequireDefault(_ArrowLeft);

var _ArrowRight = require('../icons/ArrowRight');

var _ArrowRight2 = _interopRequireDefault(_ArrowRight);

var _Calendar3 = require('./_Calendar.style');

var _Calendar4 = _interopRequireDefault(_Calendar3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlSingleCalendarPicker = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlSingleCalendarPicker, _React$Component);

  function TxlSingleCalendarPicker() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlSingleCalendarPicker);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlSingleCalendarPicker.__proto__ || (0, _getPrototypeOf2.default)(TxlSingleCalendarPicker)).call.apply(_ref, [this].concat(args)));

    _this.state = { visibleDate: _this.props.date };
    return _this;
  }

  (0, _createClass3.default)(TxlSingleCalendarPicker, [{
    key: '_pagePrev',
    value: function _pagePrev() {
      this.setState({
        visibleDate: (0, _moment2.default)(this.state.visibleDate).subtract(1, 'month')
      });
    }
  }, {
    key: '_pageNext',
    value: function _pageNext() {
      this.setState({
        visibleDate: (0, _moment2.default)(this.state.visibleDate).add(1, 'month')
      });
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          date = _props.date,
          onDateClick = _props.onDateClick,
          onHoverUpdate = _props.onHoverUpdate,
          rangeEndBoundary = _props.rangeEndBoundary,
          rangeStartBoundary = _props.rangeStartBoundary;


      var commonProps = {
        date: this.state.visibleDate,
        onDateClick: onDateClick,
        onHoverUpdate: onHoverUpdate,
        rangeEnd: date,
        rangeEndBoundary: rangeEndBoundary,
        rangeStart: date,
        rangeStartBoundary: rangeStartBoundary
      };

      return _react2.default.createElement(
        'div',
        { 'data-component': 'TxlSingleCalendarPicker' },
        _react2.default.createElement(
          'div',
          { style: _Calendar4.default.container },
          _react2.default.createElement(_IconButton2.default, {
            icon: _ArrowLeft2.default,
            variant: 'plain',
            onClick: this._pagePrev
          }),
          _react2.default.createElement(_IconButton2.default, {
            icon: _ArrowRight2.default,
            variant: 'plain',
            onClick: this._pageNext
          })
        ),
        _react2.default.createElement(_Calendar2.default, commonProps)
      );
    }
  }]);
  return TxlSingleCalendarPicker;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_pagePrev', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_pagePrev'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_pageNext', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_pageNext'), _class2.prototype)), _class2)) || _class;

exports.default = TxlSingleCalendarPicker;


TxlSingleCalendarPicker.propTypes = {
  date: _propTypes2.default.date,
  onDateClick: _react.PropTypes.func,
  onHoverUpdate: _react.PropTypes.func,
  rangeEndBoundary: _Calendar2.default.propTypes.rangeEndBoundary,
  rangeStartBoundary: _Calendar2.default.propTypes.rangeStartBoundary
};

TxlSingleCalendarPicker.defaultProps = {
  date: (0, _moment2.default)(),
  onDateClick: _Base.NOOP,
  onHoverUpdate: _Base.NOOP
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlSingleCalendarPicker, 'TxlSingleCalendarPicker', 'src/calendar-picker/_SingleCalendarPicker.jsx');
}();

;
module.exports = exports['default'];
