// txl/calendar-picker/CalendarPicker.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _slicedToArray2 = require('babel-runtime/helpers/slicedToArray');

var _slicedToArray3 = _interopRequireDefault(_slicedToArray2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _propTypes = require('../utils/propTypes');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _CalendarUtils = require('./_CalendarUtils');

var _CalendarPicker = require('./CalendarPicker.style');

var _ArrowLeft = require('../icons/ArrowLeft');

var _ArrowLeft2 = _interopRequireDefault(_ArrowLeft);

var _ArrowRight = require('../icons/ArrowRight');

var _ArrowRight2 = _interopRequireDefault(_ArrowRight);

var _Calendar = require('./_Calendar');

var _Calendar2 = _interopRequireDefault(_Calendar);

var _IconButton = require('../buttons/IconButton');

var _IconButton2 = _interopRequireDefault(_IconButton);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlCalendarPicker = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlCalendarPicker, _React$Component);

  function TxlCalendarPicker() {
    (0, _classCallCheck3.default)(this, TxlCalendarPicker);
    return (0, _possibleConstructorReturn3.default)(this, (TxlCalendarPicker.__proto__ || (0, _getPrototypeOf2.default)(TxlCalendarPicker)).apply(this, arguments));
  }

  (0, _createClass3.default)(TxlCalendarPicker, [{
    key: '_onNextClick',
    value: function _onNextClick() {
      this.props.onNextClick();
    }
  }, {
    key: '_onPrevClick',
    value: function _onPrevClick() {
      this.props.onPrevClick();
    }
  }, {
    key: '_handleDateClick',
    value: function _handleDateClick(date) {
      this.props.onDateClick(date);
    }
  }, {
    key: '_handleHoverUpdate',
    value: function _handleHoverUpdate(date) {
      this.props.onHoverUpdate(date);
    }
  }, {
    key: 'render',
    value: function render() {
      var date = (0, _moment2.default)(this.props.date).startOf('month');

      var _props = this.props,
          outOfRangeTooltip = _props.outOfRangeTooltip,
          _props$range = (0, _slicedToArray3.default)(_props.range, 2),
          rangeStart = _props$range[0],
          rangeEnd = _props$range[1],
          rangeEndBoundary = _props.rangeEndBoundary,
          rangeStartBoundary = _props.rangeStartBoundary;

      var commonProps = {
        onDateClick: this._handleDateClick,
        onHoverUpdate: this._handleHoverUpdate,
        outOfRangeTooltip: outOfRangeTooltip,
        rangeEnd: rangeEnd,
        rangeEndBoundary: rangeEndBoundary,
        rangeStart: rangeStart,
        rangeStartBoundary: rangeStartBoundary
      };

      var isLeftArrowDisabled = !(0, _CalendarUtils.isMonthPartOfRange)(date, rangeStartBoundary, rangeEndBoundary);
      var isRightArrowDisabled = !(0, _CalendarUtils.isMonthPartOfRange)((0, _moment2.default)(date).add(1, 'month'), rangeStartBoundary, rangeEndBoundary);

      return _react2.default.createElement(
        'div',
        { 'data-component': 'TxlCalendarPicker' },
        _react2.default.createElement(
          'div',
          { style: _CalendarPicker.WRAPPER_STYLE },
          _react2.default.createElement(_IconButton2.default, {
            disabled: isLeftArrowDisabled,
            icon: _ArrowLeft2.default,
            onClick: this._onPrevClick,
            variant: 'plain'
          }),
          [0, 1].map(function (cal) {
            return _react2.default.createElement(
              'div',
              {
                key: cal,
                style: _CalendarPicker.CALENDAR_STYLE
              },
              _react2.default.createElement(_Calendar2.default, (0, _extends3.default)({
                date: (0, _moment2.default)(date).add(cal, 'month')
              }, commonProps))
            );
          }),
          _react2.default.createElement(_IconButton2.default, {
            disabled: isRightArrowDisabled,
            icon: _ArrowRight2.default,
            onClick: this._onNextClick,
            variant: 'plain'
          })
        )
      );
    }
  }]);
  return TxlCalendarPicker;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_onNextClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_onNextClick'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_onPrevClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_onPrevClick'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDateClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDateClick'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleHoverUpdate', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleHoverUpdate'), _class2.prototype)), _class2)) || _class;

exports.default = TxlCalendarPicker;


TxlCalendarPicker.propTypes = {
  date: _propTypes2.default.date,
  onDateClick: _react.PropTypes.func,
  onHoverUpdate: _react.PropTypes.func,
  onNextClick: _react.PropTypes.func,
  onPrevClick: _react.PropTypes.func,
  outOfRangeTooltip: _Calendar2.default.propTypes.outOfRangeTooltip,
  range: _propTypes2.default.dateRange.isRequired,
  rangeEndBoundary: _Calendar2.default.propTypes.rangeEndBoundary,
  rangeStartBoundary: _Calendar2.default.propTypes.rangeStartBoundary
};

TxlCalendarPicker.defaultProps = {
  date: (0, _moment2.default)(),
  onDateClick: _Base.NOOP,
  onHoverUpdate: _Base.NOOP,
  onNextClick: _Base.NOOP,
  onPrevClick: _Base.NOOP,
  range: []
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlCalendarPicker, 'TxlCalendarPicker', 'src/calendar-picker/CalendarPicker.jsx');
}();

;
module.exports = exports['default'];
