// txl/calendar-picker/CalendarPickerWithPresets.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _theme = require('../styles/theme');

var _helpers = require('../styles/helpers');

var _type = require('../styles/mixins/type');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  container: (0, _extends3.default)({}, _type.DEFAULT_FONT_FAMILY, {
    display: 'flex',
    flexWrap: 'wrap',
    width: (0, _helpers.gridUnits)(136)
  }),
  picker: {
    display: 'inline-flex',
    flex: '0 0 ' + (0, _helpers.gridUnits)(104),
    flexDirection: 'column',
    justifyContent: 'center'
  },
  presets: {
    borderRight: '1px solid ' + _theme.COLOR_NEUTRAL['100'],
    boxSizing: 'border-box',
    width: '' + (0, _helpers.gridUnits)(32)
  }
};
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/calendar-picker/CalendarPickerWithPresets.style.js');
}();

;
module.exports = exports['default'];
