// txl/_selects/FilteredItemsWrapper.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _toConsumableArray2 = require('babel-runtime/helpers/toConsumableArray');

var _toConsumableArray3 = _interopRequireDefault(_toConsumableArray2);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _Actions = require('../constants/Actions');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function defaultPredicate() {
  return true;
}

var FilteredItemsWrapper = function () {
  function FilteredItemsWrapper() {
    var items = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : [];
    var predicate = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultPredicate;
    var searchTerm = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : '';
    var showCreateNew = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : false;
    (0, _classCallCheck3.default)(this, FilteredItemsWrapper);

    this.getItems = this.getItems.bind(this);
    this.getItem = this.getItem.bind(this);
    this.getSize = this.getSize.bind(this);
    this.getOriginalIndex = this.getOriginalIndex.bind(this);

    this._showCreateNew = showCreateNew;
    this._items = [].concat((0, _toConsumableArray3.default)(items));
    this._predicate = predicate;
    this._filteredIndexes = [];
    this._searchTerm = searchTerm;
    this._init();
  }

  (0, _createClass3.default)(FilteredItemsWrapper, [{
    key: '_init',
    value: function _init() {
      var filteredIndexes = [];
      var filteredItems = [];
      for (var itemIndex = 0; itemIndex < this._items.length; itemIndex++) {
        var item = this._items[itemIndex];
        if (this._predicate(item)) {
          filteredIndexes.push(itemIndex);
          filteredItems.push(item);
        }
      }

      if (this._showCreateNew && this._searchTerm) {
        filteredIndexes.push(this._items.length);
        filteredItems.push({
          groupKey: 'New Item',
          id: _Actions.CREATE_NEW,
          value: this._searchTerm
        });
      }

      this._filteredIndexes = filteredIndexes;
      this._filteredItems = filteredItems;
    }
  }, {
    key: 'getOriginalIndex',
    value: function getOriginalIndex(index) {
      return this._filteredIndexes[index];
    }
  }, {
    key: 'getSize',
    value: function getSize() {
      return this._filteredItems.length;
    }
  }, {
    key: 'getItem',
    value: function getItem(index) {
      return this._filteredItems[index];
    }
  }, {
    key: 'getItems',
    value: function getItems() {
      return this._filteredItems;
    }
  }]);
  return FilteredItemsWrapper;
}();

var _default = FilteredItemsWrapper;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(defaultPredicate, 'defaultPredicate', 'src/_selects/FilteredItemsWrapper.js');

  __REACT_HOT_LOADER__.register(FilteredItemsWrapper, 'FilteredItemsWrapper', 'src/_selects/FilteredItemsWrapper.js');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/_selects/FilteredItemsWrapper.js');
}();

;
module.exports = exports['default'];
