// txl/_selects/SearchToViewMoreItem.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _theme = require('../styles/theme');

var _type = require('../styles/mixins/type');

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ITEM_STYLE = (0, _extends3.default)({}, _type.TYPE_BASE, {
  color: _theme.COLOR_WARNING['500'],
  marginLeft: (0, _helpers.gridUnits)(2),
  paddingBottom: (0, _helpers.gridUnits)(3),
  paddingTop: (0, _helpers.gridUnits)(3),
  width: 'auto',
  wordWrap: 'break-word'
});

// TODO Translate
var SEARCH_TO_VIEW_MORE_TEXT = 'Use the search box above to view more results';
var TxlSearchToViewMoreItem = function TxlSearchToViewMoreItem() {
  return _react2.default.createElement(
    'div',
    { style: ITEM_STYLE },
    SEARCH_TO_VIEW_MORE_TEXT
  );
};

TxlSearchToViewMoreItem.displayName = 'TxlSearchToViewMoreItem';

var _default = TxlSearchToViewMoreItem;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ITEM_STYLE, 'ITEM_STYLE', 'src/_selects/SearchToViewMoreItem.jsx');

  __REACT_HOT_LOADER__.register(SEARCH_TO_VIEW_MORE_TEXT, 'SEARCH_TO_VIEW_MORE_TEXT', 'src/_selects/SearchToViewMoreItem.jsx');

  __REACT_HOT_LOADER__.register(TxlSearchToViewMoreItem, 'TxlSearchToViewMoreItem', 'src/_selects/SearchToViewMoreItem.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/_selects/SearchToViewMoreItem.jsx');
}();

;
module.exports = exports['default'];
