// txl/_selects/SelectItemBase.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _TruncatedText = require('../text/TruncatedText');

var _TruncatedText2 = _interopRequireDefault(_TruncatedText);

var _helpers = require('../styles/helpers');

var _lists = require('../lists/lists.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var INDENT_OVERRIDE_STYLE = {
  paddingLeft: 5,
  paddingRight: 5
};

function getItemStyle(props) {
  var active = props.active,
      disabled = props.disabled,
      height = props.height,
      selected = props.selected;


  return (0, _helpers.combineStyles)({ height: height }, _lists.LIST_ITEM_STYLE, INDENT_OVERRIDE_STYLE, active && _lists.LIST_ITEM_HOVER_STYLE, selected && _lists.LIST_ITEM_SELECTED_STYLE, disabled && _lists.LIST_ITEM_DISABLED_STYLE);
}

var TxlSelectItemBase = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlSelectItemBase, _React$Component);

  function TxlSelectItemBase() {
    (0, _classCallCheck3.default)(this, TxlSelectItemBase);
    return (0, _possibleConstructorReturn3.default)(this, (TxlSelectItemBase.__proto__ || (0, _getPrototypeOf2.default)(TxlSelectItemBase)).apply(this, arguments));
  }

  (0, _createClass3.default)(TxlSelectItemBase, [{
    key: '_handleMouseEnter',
    value: function _handleMouseEnter() {
      this.props.onItemActivate({ index: this.props.index });
    }
  }, {
    key: '_handleClick',
    value: function _handleClick(event) {
      var _props = this.props,
          disabled = _props.disabled,
          item = _props.item,
          onClick = _props.onClick;


      if (!disabled) {
        onClick && onClick({ event: event, item: item });
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var _props2 = this.props,
          children = _props2.children,
          id = _props2.id;


      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlSelectItemBase',
          id: id,
          onMouseEnter: this._handleMouseEnter,
          onClick: this._handleClick,
          style: getItemStyle(this.props)
        },
        (0, _react.isValidElement)(children) ? children : _react2.default.createElement(_TruncatedText2.default, { text: children || '' })
      );
    }
  }]);
  return TxlSelectItemBase;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleMouseEnter', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseEnter'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleClick'), _class2.prototype)), _class2)) || _class;

TxlSelectItemBase.propTypes = {
  active: _react.PropTypes.bool,
  children: _react.PropTypes.any,
  disabled: _react.PropTypes.bool,
  height: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.number]),
  id: _react.PropTypes.string,
  index: _react.PropTypes.number.isRequired,
  item: _react.PropTypes.any,
  onClick: _react.PropTypes.func,
  onItemActivate: _react.PropTypes.func.isRequired,
  selected: _react.PropTypes.bool
};

TxlSelectItemBase.defaultProps = {
  active: false,
  disabled: false,
  height: (0, _helpers.gridUnitsRaw)(7),
  onClick: _Base.NOOP,
  selected: false
};

var _default = TxlSelectItemBase;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(INDENT_OVERRIDE_STYLE, 'INDENT_OVERRIDE_STYLE', 'src/_selects/SelectItemBase.jsx');

  __REACT_HOT_LOADER__.register(getItemStyle, 'getItemStyle', 'src/_selects/SelectItemBase.jsx');

  __REACT_HOT_LOADER__.register(TxlSelectItemBase, 'TxlSelectItemBase', 'src/_selects/SelectItemBase.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/_selects/SelectItemBase.jsx');
}();

;
module.exports = exports['default'];
