// txl/_selects/SingleSelectItem.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _desc, _value, _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Base = require('../base/Base');

var _SelectItemBase = require('./SelectItemBase');

var _SelectItemBase2 = _interopRequireDefault(_SelectItemBase);

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

var _Actions = require('../constants/Actions');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var ITEM_STYLE = {
  alignItems: 'center',
  display: 'flex',
  paddingLeft: (0, _helpers.gridUnits)(3),
  paddingRight: (0, _helpers.gridUnits)(3),
  whiteSpace: 'nowrap',
  width: 'auto'
};

var TxlSingleSelectItem = (_class = function (_React$Component) {
  (0, _inherits3.default)(TxlSingleSelectItem, _React$Component);

  function TxlSingleSelectItem() {
    (0, _classCallCheck3.default)(this, TxlSingleSelectItem);
    return (0, _possibleConstructorReturn3.default)(this, (TxlSingleSelectItem.__proto__ || (0, _getPrototypeOf2.default)(TxlSingleSelectItem)).apply(this, arguments));
  }

  (0, _createClass3.default)(TxlSingleSelectItem, [{
    key: '_handleClick',
    value: function _handleClick() {
      var _props = this.props,
          item = _props.item,
          selectItem = _props.selectItem;

      selectItem({ item: item });
    }
  }, {
    key: 'render',
    value: function render() {
      var _props2 = this.props,
          height = _props2.height,
          item = _props2.item,
          valueMapper = _props2.valueMapper;


      var style = (0, _extends3.default)({}, ITEM_STYLE, {
        height: height
      });

      if (item.id === _Actions.SELECT_NONE) {
        style.color = _theme.COLOR_NEUTRAL['200'];
      }

      return _react2.default.createElement(
        _SelectItemBase2.default,
        (0, _extends3.default)({}, this.props, {
          onClick: this._handleClick
        }),
        _react2.default.createElement(
          'div',
          {
            style: style
          },
          valueMapper(item)
        )
      );
    }
  }]);
  return TxlSingleSelectItem;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class.prototype, '_handleClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleClick'), _class.prototype)), _class);


TxlSingleSelectItem.propTypes = {
  active: _SelectItemBase2.default.propTypes.active,
  children: _SelectItemBase2.default.propTypes.children,
  disabled: _SelectItemBase2.default.propTypes.disabled,
  height: _SelectItemBase2.default.propTypes.height,
  index: _SelectItemBase2.default.propTypes.index,
  item: _react.PropTypes.any.isRequired,
  onClick: _SelectItemBase2.default.propTypes.onClick,
  onItemActivate: _SelectItemBase2.default.propTypes.onItemActivate,
  selectItem: _react.PropTypes.func.isRequired,
  valueMapper: _react.PropTypes.func.isRequired
};

TxlSingleSelectItem.defaultProps = {
  height: (0, _helpers.gridUnitsRaw)(7)
};

var _default = TxlSingleSelectItem;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ITEM_STYLE, 'ITEM_STYLE', 'src/_selects/SingleSelectItem.jsx');

  __REACT_HOT_LOADER__.register(TxlSingleSelectItem, 'TxlSingleSelectItem', 'src/_selects/SingleSelectItem.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/_selects/SingleSelectItem.jsx');
}();

;
module.exports = exports['default'];
