// txl/chip/Chip.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _Close = require('../icons/Close');

var _Close2 = _interopRequireDefault(_Close);

var _KeyCodes = require('../constants/KeyCodes');

var _Chip = require('./Chip.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TxlChip = (0, _radium2.default)(_class = function (_React$Component) {
  (0, _inherits3.default)(TxlChip, _React$Component);

  function TxlChip() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlChip);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlChip.__proto__ || (0, _getPrototypeOf2.default)(TxlChip)).call.apply(_ref, [this].concat(args)));

    _this.focus = function () {
      return _this.__focus__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this.hasFocus = function () {
      return _this.__hasFocus__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleButtonMouseEnter = function () {
      return _this.___handleButtonMouseEnter__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleButtonMouseLeave = function () {
      return _this.___handleButtonMouseLeave__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleChipBlur = function () {
      return _this.___handleChipBlur__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleChipFocus = function () {
      return _this.___handleChipFocus__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleDismissClick = function () {
      return _this.___handleDismissClick__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleKeyDown = function () {
      return _this.___handleKeyDown__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._handleRef = function () {
      return _this.___handleRef__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this.state = {
      buttonHovered: false,
      chipFocused: false
    };
    return _this;
  }

  (0, _createClass3.default)(TxlChip, [{
    key: '__focus__REACT_HOT_LOADER__',
    value: function __focus__REACT_HOT_LOADER__() {
      return this.__focus__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__focus__REACT_HOT_LOADER__',
    value: function __focus__REACT_HOT_LOADER__() {
      return this.__focus__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__focus__REACT_HOT_LOADER__',
    value: function __focus__REACT_HOT_LOADER__() {
      var _this2 = this;

      if (!this.props.disabled) {
        this.setState({ chipFocused: true }, function () {
          return _this2._chip.focus();
        });
      }
    }
  }, {
    key: '__hasFocus__REACT_HOT_LOADER__',
    value: function __hasFocus__REACT_HOT_LOADER__() {
      return this.__hasFocus__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__hasFocus__REACT_HOT_LOADER__',
    value: function __hasFocus__REACT_HOT_LOADER__() {
      return this.__hasFocus__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__hasFocus__REACT_HOT_LOADER__',
    value: function __hasFocus__REACT_HOT_LOADER__() {
      return this.state.chipFocused;
    }
  }, {
    key: '___handleButtonMouseEnter__REACT_HOT_LOADER__',
    value: function ___handleButtonMouseEnter__REACT_HOT_LOADER__() {
      return this.___handleButtonMouseEnter__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleButtonMouseEnter__REACT_HOT_LOADER__',
    value: function ___handleButtonMouseEnter__REACT_HOT_LOADER__() {
      return this.___handleButtonMouseEnter__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleButtonMouseEnter__REACT_HOT_LOADER__',
    value: function ___handleButtonMouseEnter__REACT_HOT_LOADER__() {
      this.setState({ buttonHovered: true });
    }
  }, {
    key: '___handleButtonMouseLeave__REACT_HOT_LOADER__',
    value: function ___handleButtonMouseLeave__REACT_HOT_LOADER__() {
      return this.___handleButtonMouseLeave__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleButtonMouseLeave__REACT_HOT_LOADER__',
    value: function ___handleButtonMouseLeave__REACT_HOT_LOADER__() {
      return this.___handleButtonMouseLeave__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleButtonMouseLeave__REACT_HOT_LOADER__',
    value: function ___handleButtonMouseLeave__REACT_HOT_LOADER__() {
      this.setState({ buttonHovered: false });
    }
  }, {
    key: '___handleChipBlur__REACT_HOT_LOADER__',
    value: function ___handleChipBlur__REACT_HOT_LOADER__() {
      return this.___handleChipBlur__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleChipBlur__REACT_HOT_LOADER__',
    value: function ___handleChipBlur__REACT_HOT_LOADER__() {
      return this.___handleChipBlur__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleChipBlur__REACT_HOT_LOADER__',
    value: function ___handleChipBlur__REACT_HOT_LOADER__() {
      this.setState({ chipFocused: false });
    }
  }, {
    key: '___handleChipFocus__REACT_HOT_LOADER__',
    value: function ___handleChipFocus__REACT_HOT_LOADER__() {
      return this.___handleChipFocus__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleChipFocus__REACT_HOT_LOADER__',
    value: function ___handleChipFocus__REACT_HOT_LOADER__() {
      return this.___handleChipFocus__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleChipFocus__REACT_HOT_LOADER__',
    value: function ___handleChipFocus__REACT_HOT_LOADER__() {
      this.setState({ chipFocused: true });
    }
  }, {
    key: '___handleDismissClick__REACT_HOT_LOADER__',
    value: function ___handleDismissClick__REACT_HOT_LOADER__() {
      return this.___handleDismissClick__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleDismissClick__REACT_HOT_LOADER__',
    value: function ___handleDismissClick__REACT_HOT_LOADER__() {
      return this.___handleDismissClick__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleDismissClick__REACT_HOT_LOADER__',
    value: function ___handleDismissClick__REACT_HOT_LOADER__(e) {
      this.props.onDismiss(e);
    }
  }, {
    key: '___handleKeyDown__REACT_HOT_LOADER__',
    value: function ___handleKeyDown__REACT_HOT_LOADER__() {
      return this.___handleKeyDown__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleKeyDown__REACT_HOT_LOADER__',
    value: function ___handleKeyDown__REACT_HOT_LOADER__() {
      return this.___handleKeyDown__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleKeyDown__REACT_HOT_LOADER__',
    value: function ___handleKeyDown__REACT_HOT_LOADER__(_ref2) {
      var keyCode = _ref2.keyCode;

      if (keyCode === _KeyCodes.BACKSPACE) {
        this.props.onDismiss();
      }
    }
  }, {
    key: '___handleRef__REACT_HOT_LOADER__',
    value: function ___handleRef__REACT_HOT_LOADER__() {
      return this.___handleRef__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleRef__REACT_HOT_LOADER__',
    value: function ___handleRef__REACT_HOT_LOADER__() {
      return this.___handleRef__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___handleRef__REACT_HOT_LOADER__',
    value: function ___handleRef__REACT_HOT_LOADER__(ref) {
      this._chip = ref;
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          children = _props.children,
          disabled = _props.disabled,
          id = _props.id;
      var _state = this.state,
          buttonHovered = _state.buttonHovered,
          chipFocused = _state.chipFocused;


      var chipStyles = chipFocused && !disabled ? _Chip.CHIP_HOVER_STYLES : _Chip.CHIP_STYLES;

      var closeIconStyles = _Chip.CLOSE_ICON_STYLES;
      if (disabled) {
        closeIconStyles = _Chip.CLOSE_ICON_DISABLED_STYLES;
      } else if (buttonHovered) {
        closeIconStyles = _Chip.CLOSE_ICON_HOVER_STYLES;
      }

      var dismissHandler = !disabled ? this._handleDismissClick : _Base.NOOP;
      var keyDownHandler = !disabled ? this._handleKeyDown : _Base.NOOP;

      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlChip',
          id: id,
          onBlur: this._handleChipBlur,
          onFocus: this._handleChipFocus,
          onKeyDown: keyDownHandler,
          ref: this._handleRef,
          style: chipStyles,
          tabIndex: 1
        },
        _react2.default.createElement(
          'div',
          null,
          children
        ),
        _react2.default.createElement(
          'span',
          {
            'data-test-chip-x-wrapper': true,
            onClick: dismissHandler,
            onMouseEnter: this._handleButtonMouseEnter,
            onMouseLeave: this._handleButtonMouseLeave,
            style: closeIconStyles
          },
          _react2.default.createElement(_Close2.default, {
            color: closeIconStyles.color
          })
        )
      );
    }
  }]);
  return TxlChip;
}(_react2.default.Component)) || _class;

exports.default = TxlChip;


TxlChip.displayName = 'TxlChip';

TxlChip.propTypes = {
  children: _react.PropTypes.node,
  disabled: _react.PropTypes.bool,
  id: _react.PropTypes.string,
  onDismiss: _react.PropTypes.func.isRequired
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlChip, 'TxlChip', 'src/chip/Chip.jsx');
}();

;
module.exports = exports['default'];
