// txl/chip/Chip.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CLOSE_ICON_HOVER_STYLES = exports.CLOSE_ICON_DISABLED_STYLES = exports.CLOSE_ICON_STYLES = exports.CHIP_HOVER_STYLES = exports.CHIP_STYLES = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _helpers = require('../styles/helpers');

var _transitions = require('../styles/mixins/transitions');

var _theme = require('../styles/theme');

var _type = require('../styles/mixins/type');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CHIP_STYLES = exports.CHIP_STYLES = (0, _extends3.default)({}, _type.DEFAULT_FONT_FAMILY, _type.TYPE_SMALL, {
  alignItems: 'center',
  backgroundColor: _theme.COLOR_NEUTRAL['100'],
  borderRadius: (0, _helpers.gridUnits)(5),
  boxSizing: 'border-box',
  color: _theme.COLOR_NEUTRAL['100c'],
  display: 'inline-flex',
  height: (0, _helpers.gridUnits)(5),
  marginRight: (0, _helpers.gridUnits)(1),
  padding: '0 ' + (0, _helpers.gridUnits)(1) + ' 0 ' + (0, _helpers.gridUnits)(2),
  transition: (0, _transitions.cubicTransition)(['background-color'])
});

var CHIP_HOVER_STYLES = exports.CHIP_HOVER_STYLES = (0, _extends3.default)({}, CHIP_STYLES, {
  backgroundColor: _theme.COLOR_NEUTRAL['200']
});

var CLOSE_ICON_STYLES = exports.CLOSE_ICON_STYLES = {
  alignItems: 'center',
  color: _theme.COLOR_NEUTRAL['400'],
  display: 'inline-flex',
  height: 'inherit',
  padding: '0 ' + (0, _helpers.gridUnits)(1.2) };

var CLOSE_ICON_DISABLED_STYLES = exports.CLOSE_ICON_DISABLED_STYLES = (0, _extends3.default)({}, CLOSE_ICON_STYLES, {
  cursor: 'not-allowed'
});

var CLOSE_ICON_HOVER_STYLES = exports.CLOSE_ICON_HOVER_STYLES = (0, _extends3.default)({}, CLOSE_ICON_STYLES, {
  color: _theme.COLOR_NEUTRAL['800'],
  cursor: 'pointer'
});
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(CHIP_STYLES, 'CHIP_STYLES', 'src/chip/Chip.style.js');

  __REACT_HOT_LOADER__.register(CHIP_HOVER_STYLES, 'CHIP_HOVER_STYLES', 'src/chip/Chip.style.js');

  __REACT_HOT_LOADER__.register(CLOSE_ICON_STYLES, 'CLOSE_ICON_STYLES', 'src/chip/Chip.style.js');

  __REACT_HOT_LOADER__.register(CLOSE_ICON_DISABLED_STYLES, 'CLOSE_ICON_DISABLED_STYLES', 'src/chip/Chip.style.js');

  __REACT_HOT_LOADER__.register(CLOSE_ICON_HOVER_STYLES, 'CLOSE_ICON_HOVER_STYLES', 'src/chip/Chip.style.js');
}();

;
