// txl/layout/_InputUnderline.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Underline = require('./_Underline');

var _Underline2 = _interopRequireDefault(_Underline);

var _InputFieldsPropTypes = require('../input-fields/_InputFieldsPropTypes');

var _theme = require('../styles/theme');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var UNDERLINE_STYLES = {
  base: _theme.COLOR_NEUTRAL['300'],
  disabled: 'transparent',
  error: _theme.COLOR_ERROR['400'],
  focused: _theme.COLOR_ACCENT['500'],
  readOnly: 'transparent',
  success: _theme.COLOR_SUCCESS['400'],
  warning: _theme.COLOR_WARNING['400']
};

function getUnderlineColor(props) {
  var validationState = props.validationState,
      focused = props.focused,
      disabled = props.disabled,
      readOnly = props.readOnly;

  var underlineVariant = void 0;
  if (focused && validationState === 'base') {
    underlineVariant = 'focused';
  } else if (disabled) {
    underlineVariant = 'disabled';
  } else if (readOnly) {
    underlineVariant = 'readOnly';
  } else {
    underlineVariant = validationState;
  }

  return UNDERLINE_STYLES[underlineVariant];
}

/**
 * TxlInputUnderline abstracts InputFields underline style requirements using
 * various input-fields states (disabled, focused, validationState....)
 */
var TxlInputUnderline = function TxlInputUnderline(props) {
  return _react2.default.createElement(_Underline2.default, {
    'data-component': 'TxlInputUnderline',
    style: props.style,
    color: getUnderlineColor(props)
  });
};

TxlInputUnderline.propTypes = {
  disabled: _react.PropTypes.bool,
  focused: _react.PropTypes.bool,
  readOnly: _react.PropTypes.bool,
  style: _react.PropTypes.object,
  validationState: _InputFieldsPropTypes.validationStateType
};

TxlInputUnderline.defaultProps = {
  disabled: false,
  focused: false,
  validationState: 'base'
};

var _default = TxlInputUnderline;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(UNDERLINE_STYLES, 'UNDERLINE_STYLES', 'src/layout/_InputUnderline.jsx');

  __REACT_HOT_LOADER__.register(getUnderlineColor, 'getUnderlineColor', 'src/layout/_InputUnderline.jsx');

  __REACT_HOT_LOADER__.register(TxlInputUnderline, 'TxlInputUnderline', 'src/layout/_InputUnderline.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/layout/_InputUnderline.jsx');
}();

;
module.exports = exports['default'];
