// txl/layout/_Underline.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _theme = require('../styles/theme');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var UnderLineStyle = {
  borderBottomStyle: 'solid',
  borderBottomWidth: '1px',
  width: '100%'
};

function getUnderlineStyle(props) {
  var style = props.style || {};
  return (0, _extends3.default)({}, UnderLineStyle, {
    borderColor: props.color
  }, style);
}

var TxlUnderline = function TxlUnderline(props) {
  return _react2.default.createElement('div', { style: getUnderlineStyle(props) });
};

TxlUnderline.propTypes = {
  color: _react.PropTypes.string
};

TxlUnderline.defaultProps = {
  color: _theme.COLOR_NEUTRAL['300']
};

var _default = TxlUnderline;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(UnderLineStyle, 'UnderLineStyle', 'src/layout/_Underline.jsx');

  __REACT_HOT_LOADER__.register(getUnderlineStyle, 'getUnderlineStyle', 'src/layout/_Underline.jsx');

  __REACT_HOT_LOADER__.register(TxlUnderline, 'TxlUnderline', 'src/layout/_Underline.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/layout/_Underline.jsx');
}();

;
module.exports = exports['default'];
