// txl/layout/FlexContainer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _FlexBox = require('./FlexBox.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * FlexContainer is layout component that can be only flex-container.
 *
 * **For layout that requires to be both container and item,
 * use FlexBox.
 */
var FlexContainer = function FlexContainer(props) {
  var alignContent = props.alignContent,
      alignItems = props.alignItems,
      flexDirection = props.flexDirection,
      flexWrap = props.flexWrap,
      justifyContent = props.justifyContent,
      tagName = props.tagName,
      children = props.children,
      others = (0, _objectWithoutProperties3.default)(props, ['alignContent', 'alignItems', 'flexDirection', 'flexWrap', 'justifyContent', 'tagName', 'children']);


  var otherStyle = others.style;
  var style = (0, _extends3.default)({}, props.style || {}, (0, _FlexBox.getContainerStyle)({
    alignContent: alignContent,
    alignItems: alignItems,
    flexDirection: flexDirection,
    flexWrap: flexWrap,
    justifyContent: justifyContent
  }), otherStyle);

  var containerProps = (0, _extends3.default)({}, others, {
    'data-component': 'TxlFlexContainer',
    style: style
  });

  return _react2.default.createElement(tagName, containerProps, children);
};

FlexContainer.displayName = 'FlexContainer';
FlexContainer.propTypes = _FlexBox.CONTAINER_PROP_TYPES;
FlexContainer.defaultProps = _FlexBox.CONTAINER_DEFAULT_PROPS;

var _default = (0, _radium2.default)(FlexContainer);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(FlexContainer, 'FlexContainer', 'src/layout/FlexContainer.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/layout/FlexContainer.jsx');
}();

;
module.exports = exports['default'];
