// txl/layout/FlexItem.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _FlexBox = require('./FlexBox.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * FlexItem is layout component that can be used as
 * flex-item only.
 *
 * **For layout that requires to be both container and item,
 * use FlexBox.
 */
var FlexItem = function FlexItem(props) {
  var flexGrow = props.flexGrow,
      flexShrink = props.flexShrink,
      flexBasis = props.flexBasis,
      alignSelf = props.alignSelf,
      tagName = props.tagName,
      children = props.children,
      propsStyle = props.style,
      others = (0, _objectWithoutProperties3.default)(props, ['flexGrow', 'flexShrink', 'flexBasis', 'alignSelf', 'tagName', 'children', 'style']);


  var style = (0, _extends3.default)({}, propsStyle || {}, (0, _FlexBox.getItemStyle)({
    alignSelf: alignSelf,
    flexBasis: flexBasis,
    flexGrow: flexGrow,
    flexShrink: flexShrink
  }));

  var containerProps = (0, _extends3.default)({
    'data-component': 'FlexItem'
  }, others, {
    style: style
  });

  return _react2.default.createElement(tagName, containerProps, children);
};

FlexItem.displayName = 'FlexItem';
FlexItem.propTypes = _FlexBox.ITEM_PROP_TYPES;
FlexItem.defaultProps = _FlexBox.ITEM_DEFAULT_PROPS;

var _default = (0, _radium2.default)(FlexItem);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(FlexItem, 'FlexItem', 'src/layout/FlexItem.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/layout/FlexItem.jsx');
}();

;
module.exports = exports['default'];
