// txl/layout/FlexPropTypes.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getFlexItemStyle = exports.getFlexContainerStyle = exports.FLEX_ITEM_TYPES = exports.FLEX_ITEM_DEFAULT_PROPS = exports.FLEX_CONTAINER_TYPES = exports.FLEX_CONTAINER_DEFAULT_PROPS = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FLEX_START = 'flex-start';
var FLEX_END = 'flex-end';
var CENTER = 'center';
var BASE_LINE = 'baseline';
var STRETCH = 'stretch';
var SPACE_BETWEEN = 'space-between';
var SPACE_AROUND = 'space-around';
var NOWRAP = 'nowrap';
var WRAP = 'wrap';
var WRAP_REVERSE = 'wrap-reverse';
var ROW = 'row';
var ROW_REVERSE = 'row-reverse';
var COLUMN = 'column';
var COLUMN_REVERSE = 'column-reverse';

// PropTypes for FlexContainer
var AlignContentType = _react.PropTypes.oneOf([FLEX_START, FLEX_END, CENTER, SPACE_BETWEEN, SPACE_AROUND, STRETCH]);

var AlignItemsType = _react.PropTypes.oneOf([FLEX_START, FLEX_END, CENTER, BASE_LINE, STRETCH]);

var FlexDirectionType = _react.PropTypes.oneOf([ROW, ROW_REVERSE, COLUMN, COLUMN_REVERSE]);
var FlexWrapType = _react.PropTypes.oneOf([NOWRAP, WRAP, WRAP_REVERSE]);
var JustifyContentType = _react.PropTypes.oneOf([FLEX_START, FLEX_END, CENTER, SPACE_BETWEEN, SPACE_AROUND]);

function getFlexContainerStyle(props) {
  var inline = props.inline,
      flexWrap = props.flexWrap,
      flexDirection = props.flexDirection,
      justifyContent = props.justifyContent,
      alignItems = props.alignItems,
      alignContent = props.alignContent,
      _props$style = props.style,
      style = _props$style === undefined ? {} : _props$style;


  return (0, _extends3.default)({
    alignContent: alignContent,
    alignItems: alignItems,
    display: inline ? 'inline-flex' : 'flex',
    flexDirection: flexDirection,
    flexWrap: flexWrap,
    justifyContent: justifyContent
  }, style);
}

function getFlexItemStyle(props) {
  var style = props.style || {};
  return (0, _extends3.default)({}, style, {
    flex: props.flexGrow + ' ' + props.flexShrink + ' ' + props.flexBasis
  });
}

var FLEX_CONTAINER_TYPES = {
  alignContent: AlignContentType,
  alignItems: AlignItemsType,
  flexDirection: FlexDirectionType,
  flexWrap: FlexWrapType,
  inline: _react.PropTypes.bool,
  justifyContent: JustifyContentType
};

var FLEX_CONTAINER_DEFAULT_PROPS = {
  alignContent: STRETCH,
  alignItems: FLEX_START,
  flexDirection: ROW,
  flexWrap: NOWRAP,
  justifyContent: FLEX_START
};

var FLEX_ITEM_TYPES = {
  flexBasis: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.number]),
  flexGrow: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.number]),
  flexShrink: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.number])
};

var FLEX_ITEM_DEFAULT_PROPS = {
  flexBasis: 'auto',
  flexGrow: 0,
  flexShrink: 0
};

exports.FLEX_CONTAINER_DEFAULT_PROPS = FLEX_CONTAINER_DEFAULT_PROPS;
exports.FLEX_CONTAINER_TYPES = FLEX_CONTAINER_TYPES;
exports.FLEX_ITEM_DEFAULT_PROPS = FLEX_ITEM_DEFAULT_PROPS;
exports.FLEX_ITEM_TYPES = FLEX_ITEM_TYPES;
exports.getFlexContainerStyle = getFlexContainerStyle;
exports.getFlexItemStyle = getFlexItemStyle;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(FLEX_START, 'FLEX_START', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(FLEX_END, 'FLEX_END', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(CENTER, 'CENTER', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(BASE_LINE, 'BASE_LINE', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(STRETCH, 'STRETCH', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(SPACE_BETWEEN, 'SPACE_BETWEEN', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(SPACE_AROUND, 'SPACE_AROUND', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(NOWRAP, 'NOWRAP', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(WRAP, 'WRAP', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(WRAP_REVERSE, 'WRAP_REVERSE', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(ROW, 'ROW', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(ROW_REVERSE, 'ROW_REVERSE', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(COLUMN, 'COLUMN', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(COLUMN_REVERSE, 'COLUMN_REVERSE', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(AlignContentType, 'AlignContentType', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(AlignItemsType, 'AlignItemsType', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(FlexDirectionType, 'FlexDirectionType', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(FlexWrapType, 'FlexWrapType', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(JustifyContentType, 'JustifyContentType', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(getFlexContainerStyle, 'getFlexContainerStyle', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(getFlexItemStyle, 'getFlexItemStyle', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(FLEX_CONTAINER_TYPES, 'FLEX_CONTAINER_TYPES', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(FLEX_CONTAINER_DEFAULT_PROPS, 'FLEX_CONTAINER_DEFAULT_PROPS', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(FLEX_ITEM_TYPES, 'FLEX_ITEM_TYPES', 'src/layout/FlexPropTypes.js');

  __REACT_HOT_LOADER__.register(FLEX_ITEM_DEFAULT_PROPS, 'FLEX_ITEM_DEFAULT_PROPS', 'src/layout/FlexPropTypes.js');
}();

;
