// txl/modals/_BaseModal.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _helpers = require('../styles/helpers');

var _BaseOverlay = require('./_BaseOverlay');

var _BaseOverlay2 = _interopRequireDefault(_BaseOverlay);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var MODAL_SIZES = {
  default: {
    maxWidth: (0, _helpers.gridUnits)(164),
    minWidth: (0, _helpers.gridUnits)(104)
  },
  large: {
    width: (0, _helpers.gridUnits)(164)
  },
  small: {
    width: (0, _helpers.gridUnits)(104)
  }
};

var MODAL_STYLES = {
  borderRadius: '2px',
  maxHeight: '95%',
  minHeight: (0, _helpers.gridUnits)(50)
};

var getModalWidth = function getModalWidth(size) {
  return typeof size === 'number' ? { width: size } : MODAL_SIZES[size];
};

var TxlBaseModal = function TxlBaseModal(props) {
  var styles = (0, _extends3.default)({}, MODAL_STYLES, getModalWidth(props.size));

  return _react2.default.createElement(_BaseOverlay2.default, (0, _extends3.default)({}, props, {
    style: styles
  }));
};

TxlBaseModal.displayName = 'TxlBaseModal';

TxlBaseModal.propTypes = _BaseOverlay2.default.propTypes;

TxlBaseModal.defaultProps = {
  componentName: 'TxlBaseModal',
  size: _BaseOverlay2.default.defaultProps.size
};

var _default = (0, _radium2.default)(TxlBaseModal);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(MODAL_SIZES, 'MODAL_SIZES', 'src/modals/_BaseModal.jsx');

  __REACT_HOT_LOADER__.register(MODAL_STYLES, 'MODAL_STYLES', 'src/modals/_BaseModal.jsx');

  __REACT_HOT_LOADER__.register(getModalWidth, 'getModalWidth', 'src/modals/_BaseModal.jsx');

  __REACT_HOT_LOADER__.register(TxlBaseModal, 'TxlBaseModal', 'src/modals/_BaseModal.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/modals/_BaseModal.jsx');
}();

;
module.exports = exports['default'];
