// txl/modals/_BaseDrawer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _BaseOverlay = require('./_BaseOverlay');

var _BaseOverlay2 = _interopRequireDefault(_BaseOverlay);

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DRAWER_SIZES = {
  large: {
    width: (0, _helpers.gridUnits)(152)
  },
  small: {
    width: (0, _helpers.gridUnits)(75)
  }
};
// BaseModal uses a flexible 'default' size, but drawer should
// just default to the larger size
DRAWER_SIZES.default = DRAWER_SIZES.large;

var DRAWER_STYLES = (0, _extends3.default)({}, _helpers.CORNERS, {
  left: 'auto', // override left from corners
  margin: 0,
  position: 'absolute'
});

var getDrawerWidth = function getDrawerWidth(size) {
  return {
    width: typeof size === 'number' ? size : DRAWER_SIZES[size].width
  };
};

var TxlBaseDrawer = function TxlBaseDrawer(props) {
  var styles = (0, _extends3.default)({}, DRAWER_STYLES, getDrawerWidth(props.size));

  return _react2.default.createElement(_BaseOverlay2.default, (0, _extends3.default)({}, props, {
    style: styles
  }));
};

TxlBaseDrawer.displayName = 'TxlBaseDrawer';

TxlBaseDrawer.propTypes = _BaseOverlay2.default.propTypes;

TxlBaseDrawer.defaultProps = {
  componentName: 'TxlBaseDrawer',
  size: _BaseOverlay2.default.defaultProps.size
};

var _default = (0, _radium2.default)(TxlBaseDrawer);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(DRAWER_SIZES, 'DRAWER_SIZES', 'src/modals/_BaseDrawer.jsx');

  __REACT_HOT_LOADER__.register(DRAWER_STYLES, 'DRAWER_STYLES', 'src/modals/_BaseDrawer.jsx');

  __REACT_HOT_LOADER__.register(getDrawerWidth, 'getDrawerWidth', 'src/modals/_BaseDrawer.jsx');

  __REACT_HOT_LOADER__.register(TxlBaseDrawer, 'TxlBaseDrawer', 'src/modals/_BaseDrawer.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/modals/_BaseDrawer.jsx');
}();

;
module.exports = exports['default'];
