// txl/modals/_ControlledTabDrawer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _BaseDrawer = require('./_BaseDrawer');

var _BaseDrawer2 = _interopRequireDefault(_BaseDrawer);

var _TabList = require('../tabs/TabList');

var _TabList2 = _interopRequireDefault(_TabList);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TxlTabbedDrawer = function TxlTabbedDrawer(props) {
  return _react2.default.createElement(
    _BaseDrawer2.default,
    (0, _extends3.default)({}, props, {
      componentName: 'TxlTabbedDrawer',
      tabs: props.items,
      selectedTabIndex: props.selectedIndex
    }),
    props.content
  );
};

TxlTabbedDrawer.propTypes = {
  content: _react.PropTypes.node,
  items: _BaseDrawer2.default.propTypes.tabs,
  selectedIndex: _TabList2.default.propTypes.selectedIndex,
  tabs: _TabList2.default.propTypes.items
};

var _default = TxlTabbedDrawer;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlTabbedDrawer, 'TxlTabbedDrawer', 'src/modals/_ControlledTabDrawer.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/modals/_ControlledTabDrawer.jsx');
}();

;
module.exports = exports['default'];
