// txl/modals/_ControlledTabbedModal.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _BaseModal = require('./_BaseModal');

var _BaseModal2 = _interopRequireDefault(_BaseModal);

var _TabList = require('../tabs/TabList');

var _TabList2 = _interopRequireDefault(_TabList);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TxlTabbedModal = function TxlTabbedModal(props) {
  return _react2.default.createElement(
    _BaseModal2.default,
    (0, _extends3.default)({}, props, {
      componentName: 'TxlTabbedModal',
      tabs: props.items,
      selectedTabIndex: props.selectedIndex
    }),
    props.content
  );
};

TxlTabbedModal.propTypes = {
  content: _react.PropTypes.node,
  items: _BaseModal2.default.propTypes.items,
  selectedIndex: _TabList2.default.propTypes.selectedIndex,
  tabs: _TabList2.default.propTypes.items
};

var _default = TxlTabbedModal;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlTabbedModal, 'TxlTabbedModal', 'src/modals/_ControlledTabbedModal.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/modals/_ControlledTabbedModal.jsx');
}();

;
module.exports = exports['default'];
