// txl/modals/_OverlayBackdrop.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _colorHelpers = require('../styles/_helpers/colorHelpers');

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

var _constants = require('../styles/constants');

var _ReactUtils = require('../ReactUtils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var BACKDROP_STYLES = (0, _extends3.default)({}, _helpers.CORNERS, {
  alignItems: 'center',
  backgroundColor: (0, _colorHelpers.getColorWithOpacity)(_theme.COLOR_NEUTRAL['900'], 0.5),
  display: 'flex',
  justifyContent: 'center',
  position: 'fixed',
  zIndex: _constants.Z_INDEXES.overlayBackdrop
});

var overflowCache = void 0;

var TxlOverlayBackdrop = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlOverlayBackdrop, _React$Component);

  function TxlOverlayBackdrop() {
    (0, _classCallCheck3.default)(this, TxlOverlayBackdrop);
    return (0, _possibleConstructorReturn3.default)(this, (TxlOverlayBackdrop.__proto__ || (0, _getPrototypeOf2.default)(TxlOverlayBackdrop)).apply(this, arguments));
  }

  (0, _createClass3.default)(TxlOverlayBackdrop, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      window.addEventListener('keydown', this._handleKeyDown);
      this._disablePageScrolling();
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      window.removeEventListener('keydown', this._handleKeyDown);
      this._enablePageScrolling();
    }
  }, {
    key: '_handleClick',
    value: function _handleClick(_ref) {
      var target = _ref.target;

      // only bubble up click event when user clicks on backdrop, not on children
      if (target === this._backdropNode) {
        this.props.onClick({});
      }
    }
  }, {
    key: '_disablePageScrolling',
    value: function _disablePageScrolling() {
      var htmlElement = document.querySelector('html');
      overflowCache = {
        body: window.getComputedStyle(document.body, null).overflow,
        html: window.getComputedStyle(htmlElement, null).overflow
      };
      document.body.style.overflow = 'hidden';
      htmlElement.style.overflow = 'hidden';
    }
  }, {
    key: '_enablePageScrolling',
    value: function _enablePageScrolling() {
      var htmlElement = document.querySelector('html');
      document.body.style.overflow = overflowCache.body || 'visible';
      htmlElement.style.overflow = overflowCache.html || 'visible';
    }
  }, {
    key: '_handleKeyDown',
    value: function _handleKeyDown(e) {
      if (e.keyCode === 27) {
        this.props.onDismiss({});
      }
    }
  }, {
    key: '_setBackdropRef',
    value: function _setBackdropRef(ref) {
      this._backdropNode = (0, _ReactUtils.findDOMNode)(ref);
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        {
          'data-test-backdrop-container': true,
          onClick: this._handleClick,
          ref: this._setBackdropRef,
          style: BACKDROP_STYLES
        },
        this.props.children
      );
    }
  }]);
  return TxlOverlayBackdrop;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleClick'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_disablePageScrolling', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_disablePageScrolling'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_enablePageScrolling', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_enablePageScrolling'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleKeyDown', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleKeyDown'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_setBackdropRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_setBackdropRef'), _class2.prototype)), _class2)) || _class;

exports.default = TxlOverlayBackdrop;


TxlOverlayBackdrop.propTypes = {
  children: _react.PropTypes.node.isRequired,
  onClick: _react.PropTypes.func,
  onDismiss: _react.PropTypes.func.isRequired
};

TxlOverlayBackdrop.defaultProps = {
  onClick: _Base.NOOP,
  onDismiss: _Base.NOOP
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(BACKDROP_STYLES, 'BACKDROP_STYLES', 'src/modals/_OverlayBackdrop.jsx');

  __REACT_HOT_LOADER__.register(overflowCache, 'overflowCache', 'src/modals/_OverlayBackdrop.jsx');

  __REACT_HOT_LOADER__.register(TxlOverlayBackdrop, 'TxlOverlayBackdrop', 'src/modals/_OverlayBackdrop.jsx');
}();

;
module.exports = exports['default'];
