// txl/scrolling-table/constants.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SCROLL_DIRECTIONS = exports.HEADER_CELL_MIN_WIDTH = exports.SCROLLING_REFRESH_RATE = exports.RESIZE_REFRESH_RATE = exports.ON_RESIZE_ROW_COUNT = exports.ROW_INITIAL_HEIGHT = exports.INITIAL_ROW_MAX = undefined;

var _immutable = require('immutable');

var _ = require('./');

var INITIAL_ROW_MAX = exports.INITIAL_ROW_MAX = 50;
var ROW_INITIAL_HEIGHT = exports.ROW_INITIAL_HEIGHT = 22;
var ON_RESIZE_ROW_COUNT = exports.ON_RESIZE_ROW_COUNT = 50;
var RESIZE_REFRESH_RATE = exports.RESIZE_REFRESH_RATE = 33;
var SCROLLING_REFRESH_RATE = exports.SCROLLING_REFRESH_RATE = 16;
var HEADER_CELL_MIN_WIDTH = exports.HEADER_CELL_MIN_WIDTH = 30;

// Scroll Directions
var SCROLL_DIRECTIONS = exports.SCROLL_DIRECTIONS = {
  DOWN: 'down',
  UP: 'up'
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(INITIAL_ROW_MAX, 'INITIAL_ROW_MAX', 'src/scrolling-table/constants.js');

  __REACT_HOT_LOADER__.register(ROW_INITIAL_HEIGHT, 'ROW_INITIAL_HEIGHT', 'src/scrolling-table/constants.js');

  __REACT_HOT_LOADER__.register(ON_RESIZE_ROW_COUNT, 'ON_RESIZE_ROW_COUNT', 'src/scrolling-table/constants.js');

  __REACT_HOT_LOADER__.register(RESIZE_REFRESH_RATE, 'RESIZE_REFRESH_RATE', 'src/scrolling-table/constants.js');

  __REACT_HOT_LOADER__.register(SCROLLING_REFRESH_RATE, 'SCROLLING_REFRESH_RATE', 'src/scrolling-table/constants.js');

  __REACT_HOT_LOADER__.register(HEADER_CELL_MIN_WIDTH, 'HEADER_CELL_MIN_WIDTH', 'src/scrolling-table/constants.js');

  __REACT_HOT_LOADER__.register(SCROLL_DIRECTIONS, 'SCROLL_DIRECTIONS', 'src/scrolling-table/constants.js');
}();

;
