// txl/scrolling-table/DefaultHeaderTemplate.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var babelPluginFlowReactPropTypes_proptype_Header = require('.').babelPluginFlowReactPropTypes_proptype_Header || require('react').PropTypes.any;

var ArrowUP = function ArrowUP() {
  return _react2.default.createElement(
    'svg',
    { style: { display: 'block' }, fill: '#fff', viewBox: '0 0 13 13' },
    _react2.default.createElement(
      'g',
      null,
      _react2.default.createElement('path', { d: 'M6.5 13l3.9-4.3H2.6z' })
    )
  );
};
var ArrowDown = function ArrowDown() {
  return _react2.default.createElement(
    'svg',
    { style: { display: 'block' }, fill: '#fff', viewBox: '0 0 13 13' },
    _react2.default.createElement(
      'g',
      null,
      _react2.default.createElement('path', { d: 'M6.5 0L2.6 4.3h7.8z' })
    )
  );
};
var Arrows = function Arrows() {
  return _react2.default.createElement(
    'svg',
    { style: { display: 'block' }, fill: '#4D5e77', viewBox: '0 0 13 13' },
    _react2.default.createElement(
      'g',
      null,
      _react2.default.createElement('path', { d: 'M6.5 0L2.6 4.3h7.8z' }),
      _react2.default.createElement('path', { d: 'M6.5 13l3.9-4.3H2.6z' })
    )
  );
};

var DefaultHeaderTemplate = function (_Component) {
  (0, _inherits3.default)(DefaultHeaderTemplate, _Component);

  function DefaultHeaderTemplate() {
    (0, _classCallCheck3.default)(this, DefaultHeaderTemplate);
    return (0, _possibleConstructorReturn3.default)(this, (DefaultHeaderTemplate.__proto__ || (0, _getPrototypeOf2.default)(DefaultHeaderTemplate)).apply(this, arguments));
  }

  (0, _createClass3.default)(DefaultHeaderTemplate, [{
    key: '_renderArrows',
    value: function _renderArrows() {
      var sortDirection = this.props.sortDirection;


      switch (sortDirection) {
        case 'asc':
          return ArrowUP();
        case 'desc':
          return ArrowDown();
        default:
          return Arrows();
      }
    }
  }, {
    key: '_renderSortArrow',
    value: function _renderSortArrow() {
      var _props = this.props,
          header = _props.header,
          onSort = _props.onSort,
          sortable = _props.sortable;


      if (!sortable) {
        return null;
      }

      return _react2.default.createElement(
        'i',
        {
          style: {
            cursor: 'pointer',
            display: 'inline-block',
            height: 13,
            position: 'absolute',
            right: 5,
            top: '50%',
            transform: 'translate(0, -50%)',
            width: 13
          },
          onClick: function onClick() {
            return onSort && onSort(header);
          }
        },
        this._renderArrows()
      );
    }
  }, {
    key: 'render',
    value: function render() {
      var _props2 = this.props,
          title = _props2.title,
          sortable = _props2.sortable;


      var style = {};

      if (sortable) {
        style.paddingRight = 10;
      }

      return _react2.default.createElement(
        'div',
        { style: style },
        title,
        this._renderSortArrow()
      );
    }
  }]);
  return DefaultHeaderTemplate;
}(_react.Component);

DefaultHeaderTemplate.propTypes = {
  header: babelPluginFlowReactPropTypes_proptype_Header,
  onSort: require('react').PropTypes.func,
  sortable: require('react').PropTypes.bool.isRequired,
  title: require('react').PropTypes.oneOfType([require('react').PropTypes.string, require('react').PropTypes.any]).isRequired,
  sortDirection: require('react').PropTypes.oneOf(['asc', 'desc', 'none']).isRequired
};
exports.default = DefaultHeaderTemplate;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ArrowUP, 'ArrowUP', 'src/scrolling-table/DefaultHeaderTemplate.jsx');

  __REACT_HOT_LOADER__.register(ArrowDown, 'ArrowDown', 'src/scrolling-table/DefaultHeaderTemplate.jsx');

  __REACT_HOT_LOADER__.register(Arrows, 'Arrows', 'src/scrolling-table/DefaultHeaderTemplate.jsx');

  __REACT_HOT_LOADER__.register(DefaultHeaderTemplate, 'DefaultHeaderTemplate', 'src/scrolling-table/DefaultHeaderTemplate.jsx');
}();

;
module.exports = exports['default'];
