// txl/scrolling-table/ScrollingTableHeader.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _ = require('./');

var _ScrollingTable = require('./ScrollingTable.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var babelPluginFlowReactPropTypes_proptype_SyntheticMouseEvent = require('react').babelPluginFlowReactPropTypes_proptype_SyntheticMouseEvent || require('react').PropTypes.any;

var babelPluginFlowReactPropTypes_proptype_SyntheticDragEvent = require('react').babelPluginFlowReactPropTypes_proptype_SyntheticDragEvent || require('react').PropTypes.any;

var babelPluginFlowReactPropTypes_proptype_Headers = require('./').babelPluginFlowReactPropTypes_proptype_Headers || require('react').PropTypes.any;

function clearDragOptions(event) {
  var dataTransfer = event.dataTransfer;

  dataTransfer.effectAllowed = 'none';
  dataTransfer.dropEffect = 'none';
}

var ScrollingTableHeader = (0, _radium2.default)(_class = function (_Component) {
  (0, _inherits3.default)(ScrollingTableHeader, _Component);

  function ScrollingTableHeader(props, context) {
    (0, _classCallCheck3.default)(this, ScrollingTableHeader);

    var _this = (0, _possibleConstructorReturn3.default)(this, (ScrollingTableHeader.__proto__ || (0, _getPrototypeOf2.default)(ScrollingTableHeader)).call(this, props, context));

    _this._mouseMoveHandler = function () {
      return _this.___mouseMoveHandler__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._resizingHandler = function () {
      return _this.___resizingHandler__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._startResizing = function () {
      return _this.___startResizing__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._stopResizing = function () {
      return _this.___stopResizing__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._onDragOver = function () {
      return _this.___onDragOver__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._startDragging = function () {
      return _this.___startDragging__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._stopDragging = function () {
      return _this.___stopDragging__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this.state = {
      dragging: false,
      resizing: false
    };

    _this._ticking = false;
    return _this;
  }

  (0, _createClass3.default)(ScrollingTableHeader, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      window.addEventListener('mouseup', this._stopResizing, false);
      window.addEventListener('mousemove', this._mouseMoveHandler, false);
    }
  }, {
    key: 'shouldComponentUpdate',
    value: function shouldComponentUpdate(nextProps, nextState) {
      if (this.props.header !== nextProps.header) {
        return true;
      }

      if (this.state.dragging !== nextState.dragging) {
        return true;
      }

      return this.state.resizing !== nextState.resizing;
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      window.removeEventListener('mouseup', this._stopResizing);
      window.removeEventListener('mousemove', this._mouseMoveHandler);
    }
  }, {
    key: '___mouseMoveHandler__REACT_HOT_LOADER__',
    value: function ___mouseMoveHandler__REACT_HOT_LOADER__() {
      return this.___mouseMoveHandler__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___mouseMoveHandler__REACT_HOT_LOADER__',
    value: function ___mouseMoveHandler__REACT_HOT_LOADER__() {
      return this.___mouseMoveHandler__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___mouseMoveHandler__REACT_HOT_LOADER__',
    value: function ___mouseMoveHandler__REACT_HOT_LOADER__(event) {
      if (this.state.resizing && this.props.header.resizable) {
        this._resizingHandler(event);
      }
    }
  }, {
    key: '___resizingHandler__REACT_HOT_LOADER__',
    value: function ___resizingHandler__REACT_HOT_LOADER__() {
      return this.___resizingHandler__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___resizingHandler__REACT_HOT_LOADER__',
    value: function ___resizingHandler__REACT_HOT_LOADER__() {
      return this.___resizingHandler__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___resizingHandler__REACT_HOT_LOADER__',
    value: function ___resizingHandler__REACT_HOT_LOADER__(event) {
      var _this2 = this;

      if (this._resizeHandle != null && !this._ticking) {
        (function () {
          var mouseX = event.clientX;
          /**
           * Find the offsetLeft relative to the resizeHandle
           * then add half the width so that the delta is calculated
           * from the middle of the resize handle
           */
          var offsetCenter = _this2._resizeHandle.offsetWidth / 2;
          var offsetLeft = _this2._resizeHandle.getBoundingClientRect().left;
          var resizeFn = function resizeFn() {
            _this2.props.onResize(offsetLeft + offsetCenter - mouseX);

            _this2._ticking = false;
          };
          window.requestAnimationFrame(function () {
            resizeFn();
          });
        })();
      }

      this._ticking = true;
    }
  }, {
    key: '___startResizing__REACT_HOT_LOADER__',
    value: function ___startResizing__REACT_HOT_LOADER__() {
      return this.___startResizing__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___startResizing__REACT_HOT_LOADER__',
    value: function ___startResizing__REACT_HOT_LOADER__() {
      return this.___startResizing__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___startResizing__REACT_HOT_LOADER__',
    value: function ___startResizing__REACT_HOT_LOADER__(event) {
      // only start resizing if left button mouse down
      if (event.button === 0 && this.props.header.resizable) {
        event.preventDefault();
        event.stopPropagation();
        document.body.style.cursor = 'col-resize';
        this.setState({ resizing: true });
      }
    }
  }, {
    key: '___stopResizing__REACT_HOT_LOADER__',
    value: function ___stopResizing__REACT_HOT_LOADER__() {
      return this.___stopResizing__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___stopResizing__REACT_HOT_LOADER__',
    value: function ___stopResizing__REACT_HOT_LOADER__() {
      return this.___stopResizing__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___stopResizing__REACT_HOT_LOADER__',
    value: function ___stopResizing__REACT_HOT_LOADER__() {
      if (this.props.header.resizable) {
        document.body.style.cursor = null;
        this.setState({ resizing: false });
      }
    }
  }, {
    key: '___onDragOver__REACT_HOT_LOADER__',
    value: function ___onDragOver__REACT_HOT_LOADER__() {
      return this.___onDragOver__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___onDragOver__REACT_HOT_LOADER__',
    value: function ___onDragOver__REACT_HOT_LOADER__() {
      return this.___onDragOver__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___onDragOver__REACT_HOT_LOADER__',
    value: function ___onDragOver__REACT_HOT_LOADER__(event) {
      if (this.props.header.draggable) {
        // By default elements cannot be dropped in other elements.
        // To allow a drop we must prevent the default handling of the element.
        event.preventDefault();
        this.props.onDragOver(this.props.header);
      }
    }
  }, {
    key: '___startDragging__REACT_HOT_LOADER__',
    value: function ___startDragging__REACT_HOT_LOADER__() {
      return this.___startDragging__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___startDragging__REACT_HOT_LOADER__',
    value: function ___startDragging__REACT_HOT_LOADER__() {
      return this.___startDragging__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___startDragging__REACT_HOT_LOADER__',
    value: function ___startDragging__REACT_HOT_LOADER__(event) {
      if (this.props.header.draggable) {
        clearDragOptions(event);
        this.props.onDragStart(this.props.header);
        this.setState({ dragging: true });
      }
    }
  }, {
    key: '___stopDragging__REACT_HOT_LOADER__',
    value: function ___stopDragging__REACT_HOT_LOADER__() {
      return this.___stopDragging__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___stopDragging__REACT_HOT_LOADER__',
    value: function ___stopDragging__REACT_HOT_LOADER__() {
      return this.___stopDragging__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___stopDragging__REACT_HOT_LOADER__',
    value: function ___stopDragging__REACT_HOT_LOADER__() {
      if (this.props.header.draggable) {
        this.props.onDragEnd(this.props.header);
        this.setState({ dragging: false });
      }
    }
  }, {
    key: '_isLastHeader',
    value: function _isLastHeader() {
      return this.props.headers.size - 1 === this.props.header.order;
    }
  }, {
    key: '_renderResizeHandle',
    value: function _renderResizeHandle() {
      var _this3 = this;

      if (this._isLastHeader()) {
        return null;
      }

      return _react2.default.createElement(
        'div',
        {
          ref: function ref(resizeHandle) {
            _this3._resizeHandle = resizeHandle;
          },
          onMouseDown: this._startResizing,
          style: _ScrollingTable.HEADER_RESIZING_HANDLE_STYLES
        },
        _react2.default.createElement('div', { style: _ScrollingTable.HEADER_RESIZING_HANDLE_BAR_STYLES })
      );
    }
  }, {
    key: 'render',
    value: function render() {
      var _this4 = this;

      var header = this.props.header;

      var HeaderTemplate = header.template;
      var width = header.width;

      var style = (0, _extends3.default)({}, _ScrollingTable.HEADER_CELL_CONTAINER_STYLES);

      style.minWidth = header.minWidth;
      if (width != null) {
        style.width = width;
      }

      return _react2.default.createElement(
        'div',
        {
          draggable: 'true',
          onDragStart: this._startDragging,
          onDragEnd: this._stopDragging,
          onDragEnter: clearDragOptions,
          onDragOver: this._onDragOver,
          ref: function ref(root) {
            _this4._root = root;
          },
          style: _ScrollingTable.HEADER_CELL_STYLES
        },
        _react2.default.createElement(
          'div',
          { style: style },
          _react2.default.createElement(HeaderTemplate, (0, _extends3.default)({ onSort: function onSort() {
              return _this4.props.onSort(header);
            } }, header.toJS())),
          this._renderResizeHandle()
        )
      );
    }
  }]);
  return ScrollingTableHeader;
}(_react.Component)) || _class;

ScrollingTableHeader.propTypes = {
  header: require('react').PropTypes.any.isRequired,
  headers: babelPluginFlowReactPropTypes_proptype_Headers,
  onSort: require('react').PropTypes.func.isRequired,
  onDragStart: require('react').PropTypes.func.isRequired,
  onDragOver: require('react').PropTypes.func.isRequired,
  onDragEnd: require('react').PropTypes.func.isRequired,
  onResize: require('react').PropTypes.func.isRequired
};
exports.default = ScrollingTableHeader;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(clearDragOptions, 'clearDragOptions', 'src/scrolling-table/ScrollingTableHeader.jsx');

  __REACT_HOT_LOADER__.register(ScrollingTableHeader, 'ScrollingTableHeader', 'src/scrolling-table/ScrollingTableHeader.jsx');
}();

;
module.exports = exports['default'];
