// txl/scrolling-table/ScrollingTableRow.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _invariant = require('invariant');

var _invariant2 = _interopRequireDefault(_invariant);

var _ScrollingTableCell = require('./ScrollingTableCell');

var _ScrollingTableCell2 = _interopRequireDefault(_ScrollingTableCell);

var _ScrollingTable = require('./ScrollingTable.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var babelPluginFlowReactPropTypes_proptype_Header = require('./').babelPluginFlowReactPropTypes_proptype_Header || require('react').PropTypes.any;

var babelPluginFlowReactPropTypes_proptype_Headers = require('./').babelPluginFlowReactPropTypes_proptype_Headers || require('react').PropTypes.any;

/**
 * ScrollingTableRow represents one row item in the table.
 */
var ScrollingTableRow = function (_Component) {
  (0, _inherits3.default)(ScrollingTableRow, _Component);

  function ScrollingTableRow(props, context) {
    (0, _classCallCheck3.default)(this, ScrollingTableRow);

    var _this = (0, _possibleConstructorReturn3.default)(this, (ScrollingTableRow.__proto__ || (0, _getPrototypeOf2.default)(ScrollingTableRow)).call(this, props, context));

    _this._onMouseEnter = function () {
      return _this.___onMouseEnter__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this._onMouseLeave = function () {
      return _this.___onMouseLeave__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    if (props.height != null) {
      (0, _invariant2.default)(props.height >= 0, 'Height must be positive');
    }

    _this.state = {
      hovering: false
    };

    (0, _invariant2.default)(props.rowNumber >= 0, 'The starting row number must be positive');
    return _this;
  }

  (0, _createClass3.default)(ScrollingTableRow, [{
    key: 'shouldComponentUpdate',
    value: function shouldComponentUpdate(nextProps, nextState) {
      if (this.props.data !== nextProps.data) {
        return true;
      }

      if (this.props.height !== nextProps.height) {
        return true;
      }

      if (this.props.rowNumber !== nextProps.rowNumber) {
        return true;
      }

      if (this.state.hovering !== nextState.hovering) {
        return true;
      }

      return !this.props.headers.equals(nextProps.headers);
    }
  }, {
    key: '___onMouseEnter__REACT_HOT_LOADER__',
    value: function ___onMouseEnter__REACT_HOT_LOADER__() {
      return this.___onMouseEnter__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___onMouseEnter__REACT_HOT_LOADER__',
    value: function ___onMouseEnter__REACT_HOT_LOADER__() {
      this.setState({ hovering: true });
    }
  }, {
    key: '___onMouseLeave__REACT_HOT_LOADER__',
    value: function ___onMouseLeave__REACT_HOT_LOADER__() {
      return this.___onMouseLeave__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___onMouseLeave__REACT_HOT_LOADER__',
    value: function ___onMouseLeave__REACT_HOT_LOADER__() {
      this.setState({ hovering: false });
    }
  }, {
    key: '_renderCells',
    value: function _renderCells() {
      var _props = this.props,
          data = _props.data,
          headers = _props.headers;


      return headers.sort(function (a, b) {
        return a.order - b.order;
      }).toArray().map(function (header) {
        return _react2.default.createElement(_ScrollingTableCell2.default, {
          key: header.order,
          value: data.get(header.key),
          data: data,
          header: header
        });
      });
    }
  }, {
    key: 'render',
    value: function render() {
      var _props2 = this.props,
          rowNumber = _props2.rowNumber,
          data = _props2.data,
          height = _props2.height;


      var style = (0, _extends3.default)({
        verticalAlign: 'top'
      }, data.size === 0 ? _ScrollingTable.ROW_LOADING_STYLES : {}, rowNumber % 2 === 1 ? _ScrollingTable.STRIPED_ROW_STYLES : {}, this.state.hovering ? { background: 'rgb(245, 246, 247)' } : {});

      if (data.size === 0 && height != null) {
        style.height = height;
        return _react2.default.createElement(
          'tr',
          { style: style },
          _react2.default.createElement('td', { colSpan: '100%', style: _ScrollingTable.ROW_LOADING_BACKGROUND_STYLES })
        );
      }

      return _react2.default.createElement(
        'tr',
        {
          onMouseEnter: this._onMouseEnter,
          onMouseLeave: this._onMouseLeave,
          style: style
        },
        this._renderCells()
      );
    }
  }]);
  return ScrollingTableRow;
}(_react.Component);

ScrollingTableRow.propTypes = {
  data: require('react').PropTypes.any.isRequired,
  headers: babelPluginFlowReactPropTypes_proptype_Headers,
  height: require('react').PropTypes.number,
  rowNumber: require('react').PropTypes.number.isRequired
};
exports.default = ScrollingTableRow;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ScrollingTableRow, 'ScrollingTableRow', 'src/scrolling-table/ScrollingTableRow.jsx');
}();

;
module.exports = exports['default'];
