// txl/scrolling-table/ScrollingTableContainer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _immutable = require('immutable');

var _LoadingIndicator = require('../progress-indicators/LoadingIndicator');

var _LoadingIndicator2 = _interopRequireDefault(_LoadingIndicator);

var _ScrollingTableRow = require('./ScrollingTableRow');

var _ScrollingTableRow2 = _interopRequireDefault(_ScrollingTableRow);

var _ScrollState = require('./records/ScrollState');

var _ScrollState2 = _interopRequireDefault(_ScrollState);

var _ = require('./');

var _ScrollingTable = require('./ScrollingTable.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var babelPluginFlowReactPropTypes_proptype_Headers = require('./constants').babelPluginFlowReactPropTypes_proptype_Headers || require('react').PropTypes.any;

var ScrollingTableContainer = function (_Component) {
  (0, _inherits3.default)(ScrollingTableContainer, _Component);

  function ScrollingTableContainer() {
    (0, _classCallCheck3.default)(this, ScrollingTableContainer);
    return (0, _possibleConstructorReturn3.default)(this, (ScrollingTableContainer.__proto__ || (0, _getPrototypeOf2.default)(ScrollingTableContainer)).apply(this, arguments));
  }

  (0, _createClass3.default)(ScrollingTableContainer, [{
    key: '_getStart',
    value: function _getStart() {
      return Math.max(this.props.scrollState.get('offset') - 1, 0);
    }
  }, {
    key: '_getEnd',
    value: function _getEnd() {
      var _props = this.props,
          scrollState = _props.scrollState,
          viewSizeInRows = _props.viewSizeInRows,
          totalRows = _props.totalRows;

      return Math.min(scrollState.get('offset') + viewSizeInRows, totalRows);
    }
  }, {
    key: '_renderTopOffset',
    value: function _renderTopOffset() {
      var start = this._getStart();
      var rowHeight = this.props.rowHeight;


      if (start === 0) {
        return null;
      }

      return _react2.default.createElement('tr', { key: 'start', style: { height: start * rowHeight } });
    }
  }, {
    key: '_renderBottomOffset',
    value: function _renderBottomOffset() {
      var _props2 = this.props,
          totalRows = _props2.totalRows,
          rowHeight = _props2.rowHeight;

      var end = this._getEnd();
      if (end === totalRows) {
        return null;
      }

      return _react2.default.createElement('tr', { key: 'end', style: { height: (totalRows - end) * rowHeight } });
    }
  }, {
    key: '_renderRows',
    value: function _renderRows() {
      var rows = [];
      var _props3 = this.props,
          rowHeight = _props3.rowHeight,
          viewSizeInRows = _props3.viewSizeInRows;

      var start = this._getStart();
      var end = this._getEnd();

      for (var i = start; i < end; i++) {
        var data = this.props.rows.getIn(['' + i, 'data']);
        /**
         * This key function is important, by taking the modulo of
         * the current row number by the view port size we can be
         * assured that ScrollingTableRow containers are recycled,
         * improving render perf.
         */
        var key = i % (viewSizeInRows + 1);

        rows.push(_react2.default.createElement(_ScrollingTableRow2.default, {
          key: key,
          headers: this.props.headers,
          data: data == null ? new _immutable.List() : data,
          height: rowHeight,
          rowNumber: i
        }));
      }

      return rows;
    }
  }, {
    key: '_renderHeaders',
    value: function _renderHeaders() {
      return this.props.headers.valueSeq().toArray().map(function (header) {
        return _react2.default.createElement(
          'th',
          {
            key: header.id,
            style: (0, _extends3.default)({
              minWidth: header.minWidth,
              padding: 0
            }, _ScrollingTable.HEADER_CELL_PLACEHOLDER_STYLES)
          },
          _react2.default.createElement('div', { style: { width: header.width } })
        );
      });
    }
  }, {
    key: 'render',
    value: function render() {
      if (this.props.loading) {
        return _react2.default.createElement(
          'div',
          { style: _ScrollingTable.TABLE_STYLES },
          _react2.default.createElement(
            'div',
            { style: _ScrollingTable.LOADING_INDICATOR_STYLES },
            _react2.default.createElement(_LoadingIndicator2.default, { size: 'large' })
          )
        );
      }

      return _react2.default.createElement(
        'table',
        {
          style: _ScrollingTable.TABLE_STYLES,
          cellPadding: 0,
          cellSpacing: 0
        },
        _react2.default.createElement(
          'thead',
          null,
          _react2.default.createElement(
            'tr',
            null,
            this._renderHeaders()
          )
        ),
        _react2.default.createElement(
          'tbody',
          null,
          this._renderTopOffset(),
          this._renderRows(),
          this._renderBottomOffset()
        )
      );
    }
  }]);
  return ScrollingTableContainer;
}(_react.Component);

ScrollingTableContainer.propTypes = {
  headers: babelPluginFlowReactPropTypes_proptype_Headers,
  loading: require('react').PropTypes.bool.isRequired,
  rowHeight: require('react').PropTypes.number.isRequired,
  rows: require('react').PropTypes.any.isRequired,
  scrollState: require('react').PropTypes.any.isRequired,
  totalRows: require('react').PropTypes.number.isRequired,
  viewSizeInRows: require('react').PropTypes.number.isRequired
};
exports.default = ScrollingTableContainer;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ScrollingTableContainer, 'ScrollingTableContainer', 'src/scrolling-table/ScrollingTableContainer.jsx');
}();

;
module.exports = exports['default'];
