// txl/scrolling-table/ScrollingTable.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LOADING_INDICATOR_STYLES = exports.HEADER_RESIZING_HANDLE_BAR_STYLES = exports.HEADER_RESIZING_HANDLE_STYLES = exports.ROW_LOADING_STYLES = exports.ROW_LOADING_BACKGROUND_STYLES = exports.CELL_CONTENT_STYLES = exports.STRIPED_ROW_STYLES = exports.CELL_STYLES = exports.HEADER_CELL_PLACEHOLDER_STYLES = exports.HEADER_CELL_STYLES = exports.HEADER_CELL_SIZING_STYLES = exports.HEADER_CELL_CONTAINER_STYLES = exports.HEADER_STYLES = exports.SIZING_HEADER_STYLES = exports.TABLE_STYLES = exports.CONTAINER_STYLES = exports.SIZING_CONTAINER_STYLES = exports.SIZING_STYLES = exports.CONSTANTS = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CONSTANTS = exports.CONSTANTS = {
  background: '#fff',
  borderColor: '#edeff0',
  boxShadow: '0 4px 4px rgba(0, 0, 0, 0.2)',
  cellBorder: '1px solid #edeff0',
  cellPadding: (0, _helpers.gridUnits)(1) + ' ' + (0, _helpers.gridUnits)(2),
  headerBackground: '#283346',
  headerColor: '#fff',
  headerPadding: (0, _helpers.gridUnits)(1) + ' 0',
  loadingBackground: '#9cb4d8',
  rowHeight: (0, _helpers.gridUnits)(4),
  rowPadding: 1,
  striped: '#fcfcfc'
};

var SIZING_STYLES = exports.SIZING_STYLES = {
  minWidth: '100%'
};

var SIZING_CONTAINER_STYLES = exports.SIZING_CONTAINER_STYLES = {
  height: '100%',
  overflowY: 'auto',
  position: 'absolute',
  top: 0,
  visibility: 'hidden',
  width: '100%',
  zIndex: -1
};

var CONTAINER_STYLES = exports.CONTAINER_STYLES = {
  background: CONSTANTS.background,
  boxShadow: CONSTANTS.boxShadow,
  display: 'block',
  margin: '0 auto',
  overflow: 'auto',
  position: 'relative'
};

var TABLE_STYLES = exports.TABLE_STYLES = {
  borderCollapse: 'separate',
  borderSpacing: 0,
  height: '100%',
  minWidth: '100%'
};

var SIZING_HEADER_STYLES = exports.SIZING_HEADER_STYLES = {
  background: CONSTANTS.headerBackground,
  boxSizing: 'border-box',
  color: CONSTANTS.headerColor,
  lineHeight: (0, _helpers.gridUnits)(6),
  minWidth: '100%'
};

var HEADER_STYLES = exports.HEADER_STYLES = (0, _extends3.default)({}, SIZING_HEADER_STYLES, {
  display: 'table',
  paddingRight: 8
});

var HEADER_CELL_CONTAINER_STYLES = exports.HEADER_CELL_CONTAINER_STYLES = {
  boxSizing: 'border-box',
  padding: '0 ' + (0, _helpers.gridUnits)(2)
};

var HEADER_CELL_SIZING_STYLES = exports.HEADER_CELL_SIZING_STYLES = {
  ':active': {
    cursor: 'grabbing'
  },
  cursor: 'grab',
  display: 'table-cell',
  fontWeight: 'normal',
  padding: CONSTANTS.headerPadding,
  position: 'relative',
  textAlign: 'left',
  whiteSpace: 'nowrap'
};

var HEADER_CELL_STYLES = exports.HEADER_CELL_STYLES = (0, _extends3.default)({
  borderRight: '1px solid transparent'
}, HEADER_CELL_SIZING_STYLES);

var HEADER_CELL_PLACEHOLDER_STYLES = exports.HEADER_CELL_PLACEHOLDER_STYLES = {
  borderRight: '1px solid transparent',
  borderTop: CONSTANTS.cellBorder,
  boxSizing: 'border-box'
};

var CELL_STYLES = exports.CELL_STYLES = (0, _extends3.default)({}, HEADER_CELL_PLACEHOLDER_STYLES, {
  display: 'table-cell'
});

var STRIPED_ROW_STYLES = exports.STRIPED_ROW_STYLES = {
  background: CONSTANTS.striped
};

var CELL_CONTENT_STYLES = exports.CELL_CONTENT_STYLES = {
  boxSizing: 'border-box',
  overflow: 'hidden',
  padding: CONSTANTS.cellPadding
};

var ROW_LOADING_BACKGROUND_STYLES = exports.ROW_LOADING_BACKGROUND_STYLES = {
  background: CONSTANTS.loadingBackground,
  border: (0, _helpers.gridUnits)(1) + ' solid ' + CONSTANTS.background,
  boxSizing: 'border-box'
};

var ROW_LOADING_STYLES = exports.ROW_LOADING_STYLES = {
  height: CONSTANTS.rowHeight,
  minWidth: '100%',
  padding: CONSTANTS.rowPadding
};

var HEADER_RESIZING_HANDLE_STYLES = exports.HEADER_RESIZING_HANDLE_STYLES = {
  bottom: 0,
  boxSizing: 'border-box',
  cursor: 'col-resize',
  padding: '0 4px',
  position: 'absolute',
  right: '-' + (0, _helpers.gridUnits)(1),
  top: 0,
  width: (0, _helpers.gridUnits)(2),
  zIndex: 5
};

var HEADER_RESIZING_HANDLE_BAR_STYLES = exports.HEADER_RESIZING_HANDLE_BAR_STYLES = {
  background: CONSTANTS.borderColor,
  height: '100%',
  width: '1px'
};

var LOADING_INDICATOR_STYLES = exports.LOADING_INDICATOR_STYLES = {
  left: '50%',
  position: 'absolute',
  top: '50%',
  transform: 'translate(-50%, -50%)'
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(CONSTANTS, 'CONSTANTS', 'src/scrolling-table/ScrollingTable.style.js');

  __REACT_HOT_LOADER__.register(SIZING_STYLES, 'SIZING_STYLES', 'src/scrolling-table/ScrollingTable.style.js');

  __REACT_HOT_LOADER__.register(SIZING_CONTAINER_STYLES, 'SIZING_CONTAINER_STYLES', 'src/scrolling-table/ScrollingTable.style.js');

  __REACT_HOT_LOADER__.register(CONTAINER_STYLES, 'CONTAINER_STYLES', 'src/scrolling-table/ScrollingTable.style.js');

  __REACT_HOT_LOADER__.register(TABLE_STYLES, 'TABLE_STYLES', 'src/scrolling-table/ScrollingTable.style.js');

  __REACT_HOT_LOADER__.register(SIZING_HEADER_STYLES, 'SIZING_HEADER_STYLES', 'src/scrolling-table/ScrollingTable.style.js');

  __REACT_HOT_LOADER__.register(HEADER_STYLES, 'HEADER_STYLES', 'src/scrolling-table/ScrollingTable.style.js');

  __REACT_HOT_LOADER__.register(HEADER_CELL_CONTAINER_STYLES, 'HEADER_CELL_CONTAINER_STYLES', 'src/scrolling-table/ScrollingTable.style.js');

  __REACT_HOT_LOADER__.register(HEADER_CELL_SIZING_STYLES, 'HEADER_CELL_SIZING_STYLES', 'src/scrolling-table/ScrollingTable.style.js');

  __REACT_HOT_LOADER__.register(HEADER_CELL_STYLES, 'HEADER_CELL_STYLES', 'src/scrolling-table/ScrollingTable.style.js');

  __REACT_HOT_LOADER__.register(HEADER_CELL_PLACEHOLDER_STYLES, 'HEADER_CELL_PLACEHOLDER_STYLES', 'src/scrolling-table/ScrollingTable.style.js');

  __REACT_HOT_LOADER__.register(CELL_STYLES, 'CELL_STYLES', 'src/scrolling-table/ScrollingTable.style.js');

  __REACT_HOT_LOADER__.register(STRIPED_ROW_STYLES, 'STRIPED_ROW_STYLES', 'src/scrolling-table/ScrollingTable.style.js');

  __REACT_HOT_LOADER__.register(CELL_CONTENT_STYLES, 'CELL_CONTENT_STYLES', 'src/scrolling-table/ScrollingTable.style.js');

  __REACT_HOT_LOADER__.register(ROW_LOADING_BACKGROUND_STYLES, 'ROW_LOADING_BACKGROUND_STYLES', 'src/scrolling-table/ScrollingTable.style.js');

  __REACT_HOT_LOADER__.register(ROW_LOADING_STYLES, 'ROW_LOADING_STYLES', 'src/scrolling-table/ScrollingTable.style.js');

  __REACT_HOT_LOADER__.register(HEADER_RESIZING_HANDLE_STYLES, 'HEADER_RESIZING_HANDLE_STYLES', 'src/scrolling-table/ScrollingTable.style.js');

  __REACT_HOT_LOADER__.register(HEADER_RESIZING_HANDLE_BAR_STYLES, 'HEADER_RESIZING_HANDLE_BAR_STYLES', 'src/scrolling-table/ScrollingTable.style.js');

  __REACT_HOT_LOADER__.register(LOADING_INDICATOR_STYLES, 'LOADING_INDICATOR_STYLES', 'src/scrolling-table/ScrollingTable.style.js');
}();

;
