// txl/scrolling-table/ScrollingTableCell.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _isFunction2 = require('lodash/isFunction');

var _isFunction3 = _interopRequireDefault(_isFunction2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactAddonsShallowCompare = require('react-addons-shallow-compare');

var _reactAddonsShallowCompare2 = _interopRequireDefault(_reactAddonsShallowCompare);

var _immutable = require('immutable');

var _formatters = require('./formatters');

var _ = require('./');

var _ScrollingTable = require('./ScrollingTable.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var convert = function convert(cell, type) {
  switch (type) {
    case 'currency':
      return (0, _formatters.currencyFormatter)(cell);
    case 'percent':
      return (0, _formatters.percentFormatter)(cell);
    case 'number':
      return (0, _formatters.numberFormatter)(cell);
    default:
      return cell;
  }
};

var ScrollingTableCell = function (_Component) {
  (0, _inherits3.default)(ScrollingTableCell, _Component);

  function ScrollingTableCell() {
    (0, _classCallCheck3.default)(this, ScrollingTableCell);
    return (0, _possibleConstructorReturn3.default)(this, (ScrollingTableCell.__proto__ || (0, _getPrototypeOf2.default)(ScrollingTableCell)).apply(this, arguments));
  }

  (0, _createClass3.default)(ScrollingTableCell, [{
    key: 'shouldComponentUpdate',
    value: function shouldComponentUpdate(nextProps, nextState) {
      return (0, _reactAddonsShallowCompare2.default)(this, nextProps, nextState);
    }
  }, {
    key: '_formatData',
    value: function _formatData() {
      var _props = this.props,
          header = _props.header,
          value = _props.value,
          data = _props.data;


      if ((0, _isFunction3.default)(header.formatCell)) {
        return header.formatCell(value, data);
      }

      return value;
    }
  }, {
    key: 'render',
    value: function render() {
      var header = this.props.header;

      var cell = convert(this._formatData(), header.type);

      var styles = (0, _extends3.default)({}, _ScrollingTable.CELL_CONTENT_STYLES, {
        width: header.width
      });

      switch (header.textOverflow) {
        case 'hidden':
          styles.overflow = 'hidden';
          break;
        case 'truncate':
          styles.overflow = 'hidden';
          styles.whiteSpace = 'nowrap';
          styles.wordBreak = 'break-all';
          styles.textOverflow = 'ellipsis';
          break;
        default:
          break;
      }

      switch (header.type) {
        case 'date':
          styles.textAlign = 'middle';
          break;
        case 'currency':
        case 'number':
        case 'percent':
          styles.textAlign = 'right';
          break;
        default:
          break;
      }

      return _react2.default.createElement(
        'td',
        {
          style: _ScrollingTable.CELL_STYLES
        },
        _react2.default.createElement(
          'div',
          { style: styles },
          cell
        )
      );
    }
  }]);
  return ScrollingTableCell;
}(_react.Component);

ScrollingTableCell.propTypes = {
  data: require('react').PropTypes.any.isRequired,
  header: require('react').PropTypes.any.isRequired,
  value: require('react').PropTypes.any.isRequired
};
exports.default = ScrollingTableCell;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(convert, 'convert', 'src/scrolling-table/ScrollingTableCell.jsx');

  __REACT_HOT_LOADER__.register(ScrollingTableCell, 'ScrollingTableCell', 'src/scrolling-table/ScrollingTableCell.jsx');
}();

;
module.exports = exports['default'];
