// txl/two-panel/TwoPanel.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RIGHT_TAB = exports.TAB_STYLES = exports.CONTAINER_STYLES = exports.LOADING_RIGHT_STYLES = exports.LOADING_LEFT_STYLES = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _theme = require('../styles/theme');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var LOADING_LEFT_STYLES = exports.LOADING_LEFT_STYLES = {
  alignItems: 'center',
  display: 'flex',
  justifyContent: 'center'
};

var LOADING_RIGHT_STYLES = exports.LOADING_RIGHT_STYLES = (0, _extends3.default)({}, LOADING_LEFT_STYLES, {
  flex: '1'
});

var CONTAINER_STYLES = exports.CONTAINER_STYLES = {
  display: 'flex',
  overflow: 'hidden'
};

var TAB_STYLES = exports.TAB_STYLES = {
  overflow: 'hidden'
};

var RIGHT_TAB = exports.RIGHT_TAB = {
  background: _theme.COLOR_NEUTRAL['50'],
  borderLeft: '1px solid ' + _theme.COLOR_NEUTRAL['100'],
  flex: '1',
  height: '100%',
  overflow: 'scroll'
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(LOADING_LEFT_STYLES, 'LOADING_LEFT_STYLES', 'src/two-panel/TwoPanel.style.js');

  __REACT_HOT_LOADER__.register(LOADING_RIGHT_STYLES, 'LOADING_RIGHT_STYLES', 'src/two-panel/TwoPanel.style.js');

  __REACT_HOT_LOADER__.register(CONTAINER_STYLES, 'CONTAINER_STYLES', 'src/two-panel/TwoPanel.style.js');

  __REACT_HOT_LOADER__.register(TAB_STYLES, 'TAB_STYLES', 'src/two-panel/TwoPanel.style.js');

  __REACT_HOT_LOADER__.register(RIGHT_TAB, 'RIGHT_TAB', 'src/two-panel/TwoPanel.style.js');
}();

;
