// txl/two-panel/TwoPanelSelector.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _SelectableItemList = require('../lists/_SelectableItemList');

var _SelectableItemList2 = _interopRequireDefault(_SelectableItemList);

var _SearchBox = require('../input-fields/SearchBox');

var _SearchBox2 = _interopRequireDefault(_SearchBox);

var _LoadingIndicator = require('../progress-indicators/LoadingIndicator');

var _LoadingIndicator2 = _interopRequireDefault(_LoadingIndicator);

var _lists = require('../lists/lists.style');

var _TwoPanel = require('./TwoPanel.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TwoPanelSelector = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TwoPanelSelector, _React$Component);

  function TwoPanelSelector() {
    (0, _classCallCheck3.default)(this, TwoPanelSelector);
    return (0, _possibleConstructorReturn3.default)(this, (TwoPanelSelector.__proto__ || (0, _getPrototypeOf2.default)(TwoPanelSelector)).apply(this, arguments));
  }

  (0, _createClass3.default)(TwoPanelSelector, [{
    key: '_handleSearchBoxRef',
    value: function _handleSearchBoxRef(ref) {
      this._searchBox = ref;
    }
  }, {
    key: '_handleItemListRef',
    value: function _handleItemListRef(ref) {
      this._itemList = ref;
    }
  }, {
    key: '_handleItemsLoadingIndicatorRef',
    value: function _handleItemsLoadingIndicatorRef(ref) {
      this._itemsLoadingIndicator = ref;
    }
  }, {
    key: '_handleContentLoadingIndicatorRef',
    value: function _handleContentLoadingIndicatorRef(ref) {
      this._contentLoadingIndicator = ref;
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          id = _props.id,
          content = _props.content,
          contentLoading = _props.contentLoading,
          itemListWidth = _props.itemListWidth,
          itemsLoading = _props.itemsLoading,
          items = _props.items,
          header = _props.header,
          selectedValue = _props.selectedValue,
          onItemClick = _props.onItemClick,
          onSearchChange = _props.onSearchChange,
          onSearchClear = _props.onSearchClear,
          searchPlaceholder = _props.searchPlaceholder,
          searchValue = _props.searchValue,
          visibleItemsCount = _props.visibleItemsCount,
          headerHeight = _props.headerHeight,
          shouldShowSearch = _props.shouldShowSearch;


      var selectableItems = items.map(function (item, index) {
        return {
          label: item.label || item,
          value: item.value ? '' + item.value : '' + index
        };
      });

      var height = _lists.ITEM_HEIGHT * (visibleItemsCount + (shouldShowSearch ? 1 : 0)) + headerHeight;

      var ITEM_LIST_STYLE = (0, _extends3.default)({}, _TwoPanel.TAB_STYLES, { width: itemListWidth });

      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlTwoPanelSelector',
          style: (0, _extends3.default)({}, _TwoPanel.CONTAINER_STYLES, {
            height: height
          })
        },
        _react2.default.createElement(
          'div',
          { style: ITEM_LIST_STYLE },
          header,
          shouldShowSearch && _react2.default.createElement(
            'div',
            null,
            _react2.default.createElement(_SearchBox2.default, {
              id: id,
              onClear: onSearchClear,
              onChange: onSearchChange,
              placeholder: searchPlaceholder,
              ref: this._handleSearchBoxRef,
              value: searchValue || ''
            })
          ),
          itemsLoading ? _react2.default.createElement(
            'div',
            { style: _TwoPanel.LOADING_LEFT_STYLES },
            _react2.default.createElement(_LoadingIndicator2.default, { ref: this._handleItemsLoadingIndicatorRef })
          ) : _react2.default.createElement(_SelectableItemList2.default, {
            items: selectableItems,
            onClick: onItemClick,
            ref: this._handleItemListRef,
            selectedValues: ['' + selectedValue],
            visibleItemCount: visibleItemsCount
          })
        ),
        contentLoading ? _react2.default.createElement(
          'div',
          { style: _TwoPanel.LOADING_RIGHT_STYLES },
          _react2.default.createElement(_LoadingIndicator2.default, { ref: this._handleContentLoadingIndicatorRef })
        ) : _react2.default.createElement(
          'div',
          { 'data-tabs-content': true, style: _TwoPanel.RIGHT_TAB },
          content
        )
      );
    }
  }]);
  return TwoPanelSelector;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleSearchBoxRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleSearchBoxRef'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleItemListRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleItemListRef'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleItemsLoadingIndicatorRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleItemsLoadingIndicatorRef'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleContentLoadingIndicatorRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleContentLoadingIndicatorRef'), _class2.prototype)), _class2)) || _class;

exports.default = TwoPanelSelector;


TwoPanelSelector.defaultProps = {
  headerHeight: 0,
  itemListWidth: '50%',
  onItemClick: _Base.NOOP,
  onSearchChange: _Base.NOOP,
  onSearchClear: _Base.NOOP,
  onSearchSubmit: _Base.NOOP,
  shouldShowSearch: false,
  visibleItemsCount: 10
};

TwoPanelSelector.propTypes = {
  content: _react.PropTypes.node,
  contentLoading: _react.PropTypes.bool,
  header: _react.PropTypes.node,
  headerHeight: _react.PropTypes.number,
  id: _SearchBox2.default.propTypes.id,
  itemListWidth: _react.PropTypes.string,
  items: _react.PropTypes.arrayOf(_react.PropTypes.shape({
    label: _react.PropTypes.oneOfType([_react.PropTypes.node, _react.PropTypes.string]).isRequired,
    value: _react.PropTypes.oneOfType([_react.PropTypes.number, _react.PropTypes.string])
  })).isRequired,
  itemsLoading: _react.PropTypes.bool,
  onItemClick: _react.PropTypes.func,
  onSearchChange: _SearchBox2.default.propTypes.onChange,
  onSearchClear: _SearchBox2.default.propTypes.onClear,
  onSearchSubmit: _SearchBox2.default.propTypes.onSubmit,
  searchPlaceholder: _SearchBox2.default.propTypes.placeholder,
  searchValue: _SearchBox2.default.propTypes.value,
  selectedValue: _react.PropTypes.string,
  shouldShowSearch: _react.PropTypes.bool,
  visibleItemsCount: _react.PropTypes.number
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TwoPanelSelector, 'TwoPanelSelector', 'src/two-panel/TwoPanelSelector.jsx');
}();

;
module.exports = exports['default'];
