// txl/two-panel/UncontrolledTwoPanelSelector.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _desc, _value, _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _debounce = require('lodash/debounce');

var _debounce2 = _interopRequireDefault(_debounce);

var _Base = require('../base/Base');

var _propTypes = require('../utils/propTypes');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _TwoPanelSelector = require('./TwoPanelSelector');

var _TwoPanelSelector2 = _interopRequireDefault(_TwoPanelSelector);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var UncontrolledTwoPanelSelector = (_class = function (_React$Component) {
  (0, _inherits3.default)(UncontrolledTwoPanelSelector, _React$Component);

  function UncontrolledTwoPanelSelector() {
    var _ref;

    (0, _classCallCheck3.default)(this, UncontrolledTwoPanelSelector);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = UncontrolledTwoPanelSelector.__proto__ || (0, _getPrototypeOf2.default)(UncontrolledTwoPanelSelector)).call.apply(_ref, [this].concat(args)));

    _this.state = {
      filteredItems: _this.props.items,
      searchValue: null
    };
    _this._filterItems = (0, _debounce2.default)(_this._filterItems, 300).bind(_this);
    return _this;
  }

  (0, _createClass3.default)(UncontrolledTwoPanelSelector, [{
    key: '_filterItems',
    value: function _filterItems(value) {
      var _this2 = this;

      var items = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : this.props.items;

      var newItems = void 0;

      if (!value) {
        newItems = items;
      } else {
        newItems = items.filter(function (item) {
          return _this2.props.filterFunction(item.searchContent, value);
        });
      }

      this.setState({
        filteredItems: newItems
      });

      return newItems;
    }
  }, {
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      if (this.props.items !== nextProps.items) {
        this._filterItems(this.state.searchValue);
      }
    }
  }, {
    key: '_handleSearchChange',
    value: function _handleSearchChange(_ref2) {
      var value = _ref2.value;

      if (!value) {
        this._handleSearchClear();
      } else {
        this._filterItems(value);

        this.setState({
          searchValue: value
        });

        this.props.onSearchChange({ value: value });
      }
    }
  }, {
    key: '_handleSearchClear',
    value: function _handleSearchClear() {
      this.setState({
        filteredItems: this.props.items,
        searchValue: null
      });
    }
  }, {
    key: '_handleItemSelect',
    value: function _handleItemSelect(obj) {
      this.setState({
        selectedValue: obj.value
      });
      this.props.onItemClick(obj);
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          header = _props.header,
          contentLoading = _props.contentLoading,
          content = _props.content,
          itemListWidth = _props.itemListWidth,
          items = _props.items,
          visibleItems = _props.visibleItems,
          props = (0, _objectWithoutProperties3.default)(_props, ['header', 'contentLoading', 'content', 'itemListWidth', 'items', 'visibleItems']);
      var _state = this.state,
          selectedValue = _state.selectedValue,
          searchValue = _state.searchValue,
          filteredItems = _state.filteredItems;


      var shouldShowSearch = items.length > visibleItems;

      return _react2.default.createElement(_TwoPanelSelector2.default, (0, _extends3.default)({}, props, {
        contentLoading: contentLoading || false,
        content: content,
        header: header,
        itemListWidth: itemListWidth,
        items: filteredItems,
        onItemClick: this._handleItemSelect,
        onSearchChange: this._handleSearchChange,
        onSearchClear: this._handleSearchClear,
        onSearchSubmit: _Base.NOOP,
        searchValue: searchValue,
        selectedValue: selectedValue,
        shouldShowSearch: shouldShowSearch,
        visibleItemsCount: visibleItems
      }));
    }
  }]);
  return UncontrolledTwoPanelSelector;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class.prototype, '_handleSearchChange', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleSearchChange'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_handleSearchClear', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleSearchClear'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_handleItemSelect', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleItemSelect'), _class.prototype)), _class);


UncontrolledTwoPanelSelector.defaultProps = {
  contentLoading: false,
  filterFunction: function filterFunction(value, searchValue) {
    return new RegExp(searchValue, 'i').test(value);
  },
  itemListWidth: '50%',
  itemsLoading: false,
  onSearchChange: _Base.NOOP,
  placeholderContent: null,
  visibleItems: 12
};

UncontrolledTwoPanelSelector.propTypes = {
  content: _TwoPanelSelector2.default.propTypes.content,
  contentLoading: _TwoPanelSelector2.default.propTypes.contentLoading,
  filterFunction: _propTypes2.default.filterFunction,
  header: _TwoPanelSelector2.default.propTypes.header,
  itemListWidth: _react.PropTypes.string,
  items: _react.PropTypes.arrayOf(_react.PropTypes.shape({
    label: _react.PropTypes.oneOfType([_react.PropTypes.node, _react.PropTypes.string]).isRequired,
    searchContent: _react.PropTypes.string.isRequired,
    value: _react.PropTypes.oneOfType([_react.PropTypes.number, _react.PropTypes.string])
  })).isRequired,
  itemsLoading: _TwoPanelSelector2.default.propTypes.itemsLoading,
  onItemClick: _TwoPanelSelector2.default.propTypes.onItemClick,
  onSearchChange: _TwoPanelSelector2.default.propTypes.onSearchChange,
  placeholderContent: _react.PropTypes.node,
  visibleItems: _react.PropTypes.number
};

var _default = UncontrolledTwoPanelSelector;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(UncontrolledTwoPanelSelector, 'UncontrolledTwoPanelSelector', 'src/two-panel/UncontrolledTwoPanelSelector.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/two-panel/UncontrolledTwoPanelSelector.jsx');
}();

;
module.exports = exports['default'];
