// txl/dropdowns/_containers/AbsoluteContainer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _helpers = require('../../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DEFAULT_MARGIN = (0, _helpers.gridUnits)(1);
var DEFAULT_WIDTH = '100%';
var ContentStyle = {
  left: 0,
  position: 'absolute',
  zIndex: 20
};

var AbsoluteContainer = function AbsoluteContainer(props) {
  if (props.open) {
    return _react2.default.createElement(
      'div',
      {
        style: (0, _extends3.default)({}, ContentStyle, {
          marginTop: -props.yOffset || DEFAULT_MARGIN,
          width: props.width || DEFAULT_WIDTH
        }),
        onClick: props.onClick
      },
      props.children
    );
  }

  return _react2.default.createElement('noscript', null);
};

AbsoluteContainer.propTypes = {
  children: _react.PropTypes.node,
  onClick: _react.PropTypes.func,
  open: _react.PropTypes.bool,
  width: _react.PropTypes.oneOfType([_react.PropTypes.number, _react.PropTypes.string]),
  yOffset: _react.PropTypes.number
};

var _default = AbsoluteContainer;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(DEFAULT_MARGIN, 'DEFAULT_MARGIN', 'src/dropdowns/_containers/AbsoluteContainer.jsx');

  __REACT_HOT_LOADER__.register(DEFAULT_WIDTH, 'DEFAULT_WIDTH', 'src/dropdowns/_containers/AbsoluteContainer.jsx');

  __REACT_HOT_LOADER__.register(ContentStyle, 'ContentStyle', 'src/dropdowns/_containers/AbsoluteContainer.jsx');

  __REACT_HOT_LOADER__.register(AbsoluteContainer, 'AbsoluteContainer', 'src/dropdowns/_containers/AbsoluteContainer.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/dropdowns/_containers/AbsoluteContainer.jsx');
}();

;
module.exports = exports['default'];
