// txl/dropdowns/_containers/TetheredContainer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _DropdownTargetBase = require('../_DropdownTargetBase');

var _DropdownTargetBase2 = _interopRequireDefault(_DropdownTargetBase);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TetheredContainer = function TetheredContainer(props) {
  var children = props.children,
      open = props.open,
      width = props.width,
      yOffset = props.yOffset,
      onClick = props.onClick;


  return _react2.default.createElement(
    _DropdownTargetBase2.default,
    {
      componentName: 'TxlTetheredContainer',
      content: children,
      onContentClick: onClick,
      visible: open,
      width: width,
      yOffset: yOffset
    },
    _react2.default.createElement('noscript', null)
  );
};

TetheredContainer.propTypes = {
  children: _react.PropTypes.any,
  onClick: _react.PropTypes.func,
  open: _react.PropTypes.bool,
  width: _DropdownTargetBase2.default.propTypes.width,
  yOffset: _DropdownTargetBase2.default.propTypes.yOffset
};

var _default = TetheredContainer;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TetheredContainer, 'TetheredContainer', 'src/dropdowns/_containers/TetheredContainer.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/dropdowns/_containers/TetheredContainer.jsx');
}();

;
module.exports = exports['default'];
