// txl/styles/mixins/animations.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PULSE_DURATION = exports.SPIN_DURATION = exports.PLACEHOLDER_ANIMATION_NAME = undefined;
exports.getSpinStyles = getSpinStyles;
exports.getPulseStyles = getPulseStyles;

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * @NOTE: See https://github.com/FormidableLabs/radium/tree/v0.18.1/docs/api#keyframes
 * Radium needs a default shorthand animation property, which it will
 * then go through and replace with a newly generated name that maps to the
 * keyframes generated and inserted into the style block generated by StyleRoot.
 */
var PLACEHOLDER_ANIMATION_NAME = exports.PLACEHOLDER_ANIMATION_NAME = 'placeholder';

var SPIN_DURATION = exports.SPIN_DURATION = 650;

var SPIN_KEYFRAME = _radium2.default.keyframes({
  '0%': { transform: 'rotate(0deg)' },
  '100%': { transform: 'rotate(360deg)' }
}, 'spin');

var PULSE_DURATION = exports.PULSE_DURATION = 1300;

var PULSE_KEYFRAME = _radium2.default.keyframes({
  '0%': {
    opacity: 1,
    transform: 'scale(1)'
  },
  '100%': {
    opacity: 0,
    transform: 'scale(.3)'
  }
}, 'pulse');

function getSpinStyles() {
  var duration = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : SPIN_DURATION;

  return {
    animation: PLACEHOLDER_ANIMATION_NAME,
    animationDirection: 'normal',
    animationDuration: duration + 'ms',
    animationIterationCount: 'infinite',
    animationName: SPIN_KEYFRAME,
    animationTimingFunction: 'linear'
  };
}

function getPulseStyles() {
  var duration = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : PULSE_DURATION;

  return {
    animation: PLACEHOLDER_ANIMATION_NAME,
    animationDirection: 'normal',
    animationDuration: duration + 'ms',
    animationIterationCount: 'infinite',
    animationName: PULSE_KEYFRAME
  };
}
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(PLACEHOLDER_ANIMATION_NAME, 'PLACEHOLDER_ANIMATION_NAME', 'src/styles/mixins/animations.js');

  __REACT_HOT_LOADER__.register(SPIN_DURATION, 'SPIN_DURATION', 'src/styles/mixins/animations.js');

  __REACT_HOT_LOADER__.register(SPIN_KEYFRAME, 'SPIN_KEYFRAME', 'src/styles/mixins/animations.js');

  __REACT_HOT_LOADER__.register(PULSE_DURATION, 'PULSE_DURATION', 'src/styles/mixins/animations.js');

  __REACT_HOT_LOADER__.register(PULSE_KEYFRAME, 'PULSE_KEYFRAME', 'src/styles/mixins/animations.js');

  __REACT_HOT_LOADER__.register(getSpinStyles, 'getSpinStyles', 'src/styles/mixins/animations.js');

  __REACT_HOT_LOADER__.register(getPulseStyles, 'getPulseStyles', 'src/styles/mixins/animations.js');
}();

;
