// txl/styles/mixins/type.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HEADING_LARGE = exports.HEADING_MEDIUM = exports.HEADING_SMALL = exports.PARAGRAPH = exports.LABEL = exports.TYPE_BOLD = exports.TYPE_SMALL = exports.TYPE_BASE = exports.DEFAULT_FONT_FAMILY = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _helpers = require('../helpers');

var _resets = require('./resets');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DEFAULT_FONT_FAMILY = exports.DEFAULT_FONT_FAMILY = {
  fontFamily: '\'Roboto\', helvetica, arial, verdana, sans-serif'
};

var TYPE_BASE = exports.TYPE_BASE = {
  fontSize: 14,
  fontWeight: 400,
  lineHeight: (0, _helpers.gridUnits)(4)
};

var TYPE_SMALL = exports.TYPE_SMALL = (0, _extends3.default)({}, TYPE_BASE, {
  fontSize: 12
});

var TYPE_BOLD = exports.TYPE_BOLD = (0, _extends3.default)({}, TYPE_BASE, {
  fontWeight: 500
});

var LABEL = exports.LABEL = (0, _extends3.default)({}, TYPE_SMALL, _resets.RESET_MARGIN, {
  cursor: 'default',
  display: 'block'
});

var PARAGRAPH = exports.PARAGRAPH = (0, _extends3.default)({}, TYPE_BASE, _resets.RESET_MARGIN, {
  marginBottom: (0, _helpers.gridUnits)(3)
});

var HEADING_SMALL = exports.HEADING_SMALL = (0, _extends3.default)({}, _resets.RESET_MARGIN, _resets.RESET_PADDING, {
  fontSize: 18,
  fontWeight: 400,
  lineHeight: (0, _helpers.gridUnits)(6)
});

var HEADING_MEDIUM = exports.HEADING_MEDIUM = (0, _extends3.default)({}, _resets.RESET_MARGIN, _resets.RESET_PADDING, {
  fontSize: 22,
  fontWeight: 100,
  lineHeight: (0, _helpers.gridUnits)(7)
});

var HEADING_LARGE = exports.HEADING_LARGE = (0, _extends3.default)({}, _resets.RESET_MARGIN, _resets.RESET_PADDING, {
  fontSize: 28,
  fontWeight: 100,
  lineHeight: (0, _helpers.gridUnits)(8)
});
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(DEFAULT_FONT_FAMILY, 'DEFAULT_FONT_FAMILY', 'src/styles/mixins/type.js');

  __REACT_HOT_LOADER__.register(TYPE_BASE, 'TYPE_BASE', 'src/styles/mixins/type.js');

  __REACT_HOT_LOADER__.register(TYPE_SMALL, 'TYPE_SMALL', 'src/styles/mixins/type.js');

  __REACT_HOT_LOADER__.register(TYPE_BOLD, 'TYPE_BOLD', 'src/styles/mixins/type.js');

  __REACT_HOT_LOADER__.register(LABEL, 'LABEL', 'src/styles/mixins/type.js');

  __REACT_HOT_LOADER__.register(PARAGRAPH, 'PARAGRAPH', 'src/styles/mixins/type.js');

  __REACT_HOT_LOADER__.register(HEADING_SMALL, 'HEADING_SMALL', 'src/styles/mixins/type.js');

  __REACT_HOT_LOADER__.register(HEADING_MEDIUM, 'HEADING_MEDIUM', 'src/styles/mixins/type.js');

  __REACT_HOT_LOADER__.register(HEADING_LARGE, 'HEADING_LARGE', 'src/styles/mixins/type.js');
}();

;
