// txl/styles/helpers.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TRUNCATE = exports.CORNERS = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

exports.gridUnits = gridUnits;
exports.gridUnitsRaw = gridUnitsRaw;
exports.combineStyles = combineStyles;
exports.square = square;
exports.allBorderRadius = allBorderRadius;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function gridUnits(units) {
  return units * 5 + 'px';
}

function gridUnitsRaw(units) {
  return units * 5;
}

function combineStyles() {
  for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
    args[_key] = arguments[_key];
  }

  var validArgs = args.filter(function (arg) {
    return arg && arg.constructor === Object;
  });
  var ret = {};
  while (validArgs.length > 0) {
    ret = (0, _extends3.default)({}, ret, validArgs.shift());
  }
  return ret;
}

function square(size) {
  return {
    height: size,
    width: size
  };
}

function allBorderRadius(radius) {
  return {
    borderBottomLeftRadius: radius,
    borderBottomRightRadius: radius,
    borderTopLeftRadius: radius,
    borderTopRightRadius: radius
  };
}

var CORNERS = exports.CORNERS = {
  bottom: 0,
  left: 0,
  right: 0,
  top: 0
};

var TRUNCATE = exports.TRUNCATE = {
  overflow: 'hidden',
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap'
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(gridUnits, 'gridUnits', 'src/styles/helpers.js');

  __REACT_HOT_LOADER__.register(gridUnitsRaw, 'gridUnitsRaw', 'src/styles/helpers.js');

  __REACT_HOT_LOADER__.register(combineStyles, 'combineStyles', 'src/styles/helpers.js');

  __REACT_HOT_LOADER__.register(square, 'square', 'src/styles/helpers.js');

  __REACT_HOT_LOADER__.register(allBorderRadius, 'allBorderRadius', 'src/styles/helpers.js');

  __REACT_HOT_LOADER__.register(CORNERS, 'CORNERS', 'src/styles/helpers.js');

  __REACT_HOT_LOADER__.register(TRUNCATE, 'TRUNCATE', 'src/styles/helpers.js');
}();

;
