// txl/styles/palettes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
/**
  * This file is automatically generated by running "npm run generatePalettes".
  * If you wish to modify this file, make your changes in _paletteDefinitions.
  * If you are modifying it directly, your changes will be lost.
*/

var PACIFIC_STORM = exports.PACIFIC_STORM = {
  "50": "#F5F6F7",
  "100": "#D9DEE2",
  "200": "#C3C9D1",
  "300": "#97A2B2",
  "400": "#707E91",
  "500": "#4D5e77",
  "600": "#37455b",
  "700": "#283346",
  "800": "#1B222E",
  "900": "#0D1017",
  "50c": "#1B222E",
  "100c": "#1B222E",
  "200c": "#1B222E",
  "300c": "#1B222E",
  "400c": "#1B222E",
  "500c": "#FFFFFF",
  "600c": "#FFFFFF",
  "700c": "#FFFFFF",
  "800c": "#FFFFFF",
  "900c": "#FFFFFF"
};

var GRAY_SKIES = exports.GRAY_SKIES = {
  "50": "#F6F6F6",
  "100": "#EDEDED",
  "200": "#d0d0d0",
  "300": "#A2A2A2",
  "400": "#737373",
  "500": "#535353",
  "600": "#434343",
  "700": "#323232",
  "800": "#212121",
  "900": "#101010",
  "50c": "#1B222E",
  "100c": "#1B222E",
  "200c": "#1B222E",
  "300c": "#1B222E",
  "400c": "#1B222E",
  "500c": "#FFFFFF",
  "600c": "#FFFFFF",
  "700c": "#FFFFFF",
  "800c": "#FFFFFF",
  "900c": "#FFFFFF"
};

var TUNE_BLUE = exports.TUNE_BLUE = {
  "50": "#E6F2FF",
  "100": "#D4EDFF",
  "200": "#B6E0FF",
  "300": "#94CCFF",
  "400": "#55AEFF",
  "500": "#007AFF",
  "600": "#0252C6",
  "700": "#083E7F",
  "800": "#002951",
  "900": "#001C3D",
  "50c": "#1B222E",
  "100c": "#1B222E",
  "200c": "#1B222E",
  "300c": "#1B222E",
  "400c": "#1B222E",
  "500c": "#FFFFFF",
  "600c": "#FFFFFF",
  "700c": "#FFFFFF",
  "800c": "#FFFFFF",
  "900c": "#FFFFFF"
};

var BELLTOWN_BLUE = exports.BELLTOWN_BLUE = {
  "50": "#eefcff",
  "100": "#d0edf6",
  "200": "#aedbe8",
  "300": "#78c0d1",
  "400": "#3ea0b7",
  "500": "#0c7c96",
  "600": "#065972",
  "700": "#0b485b",
  "800": "#023949",
  "900": "#022c37",
  "50c": "#1B222E",
  "100c": "#1B222E",
  "200c": "#1B222E",
  "300c": "#1B222E",
  "400c": "#1B222E",
  "500c": "#FFFFFF",
  "600c": "#FFFFFF",
  "700c": "#FFFFFF",
  "800c": "#FFFFFF",
  "900c": "#FFFFFF"
};

var BAJA_TEAL = exports.BAJA_TEAL = {
  "50": "#e5f9f5",
  "100": "#d5faf3",
  "200": "#bdf1e8",
  "300": "#94eddc",
  "400": "#3fc7b0",
  "500": "#00a888",
  "600": "#00846b",
  "700": "#226659",
  "800": "#184940",
  "900": "#0f332c",
  "50c": "#1B222E",
  "100c": "#1B222E",
  "200c": "#1B222E",
  "300c": "#1B222E",
  "400c": "#1B222E",
  "500c": "#FFFFFF",
  "600c": "#FFFFFF",
  "700c": "#FFFFFF",
  "800c": "#FFFFFF",
  "900c": "#FFFFFF"
};

var EMERALD_CITY = exports.EMERALD_CITY = {
  "50": "#f8fff0",
  "100": "#e6f2d7",
  "200": "#daefc4",
  "300": "#b5df8b",
  "400": "#8ecf4e",
  "500": "#5caf00",
  "600": "#549105",
  "700": "#39700a",
  "800": "#205100",
  "900": "#12310e",
  "50c": "#1B222E",
  "100c": "#1B222E",
  "200c": "#1B222E",
  "300c": "#1B222E",
  "400c": "#1B222E",
  "500c": "#FFFFFF",
  "600c": "#FFFFFF",
  "700c": "#FFFFFF",
  "800c": "#FFFFFF",
  "900c": "#FFFFFF"
};

var CHEETO = exports.CHEETO = {
  "50": "#fff0cc",
  "100": "#f4e0a6",
  "200": "#f2ce86",
  "300": "#f5bd63",
  "400": "#ffaa45",
  "500": "#ff7c18",
  "600": "#f45a16",
  "700": "#c94517",
  "800": "#a32f00",
  "900": "#7a1b02",
  "50c": "#1B222E",
  "100c": "#1B222E",
  "200c": "#1B222E",
  "300c": "#1B222E",
  "400c": "#1B222E",
  "500c": "#FFFFFF",
  "600c": "#FFFFFF",
  "700c": "#FFFFFF",
  "800c": "#FFFFFF",
  "900c": "#FFFFFF"
};

var POPPY = exports.POPPY = {
  "50": "#ffeeee",
  "100": "#ffdbd9",
  "200": "#ffb3b3",
  "300": "#ff8482",
  "400": "#ff5957",
  "500": "#ed3a3a",
  "600": "#d32a2a",
  "700": "#b9001d",
  "800": "#890e1b",
  "900": "#61090e",
  "50c": "#1B222E",
  "100c": "#1B222E",
  "200c": "#1B222E",
  "300c": "#1B222E",
  "400c": "#1B222E",
  "500c": "#FFFFFF",
  "600c": "#FFFFFF",
  "700c": "#FFFFFF",
  "800c": "#FFFFFF",
  "900c": "#FFFFFF"
};

var PLUM = exports.PLUM = {
  "50": "#fcf5ff",
  "100": "#edddf2",
  "200": "#eabbef",
  "300": "#e89cf2",
  "400": "#cf7cd8",
  "500": "#b440c1",
  "600": "#930e9e",
  "700": "#5b0d6b",
  "800": "#3b0947",
  "900": "#260630",
  "50c": "#1B222E",
  "100c": "#1B222E",
  "200c": "#1B222E",
  "300c": "#1B222E",
  "400c": "#1B222E",
  "500c": "#FFFFFF",
  "600c": "#FFFFFF",
  "700c": "#FFFFFF",
  "800c": "#FFFFFF",
  "900c": "#FFFFFF"
};

var VIOLET = exports.VIOLET = {
  "50": "#f5fbff",
  "100": "#dde9f2",
  "200": "#b7caf2",
  "300": "#9fa1ef",
  "400": "#8777ec",
  "500": "#6e51e5",
  "600": "#5841b7",
  "700": "#413296",
  "800": "#262684",
  "900": "#202054",
  "50c": "#1B222E",
  "100c": "#1B222E",
  "200c": "#1B222E",
  "300c": "#1B222E",
  "400c": "#1B222E",
  "500c": "#FFFFFF",
  "600c": "#FFFFFF",
  "700c": "#FFFFFF",
  "800c": "#FFFFFF",
  "900c": "#FFFFFF"
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(PACIFIC_STORM, "PACIFIC_STORM", "src/styles/palettes.js");

  __REACT_HOT_LOADER__.register(GRAY_SKIES, "GRAY_SKIES", "src/styles/palettes.js");

  __REACT_HOT_LOADER__.register(TUNE_BLUE, "TUNE_BLUE", "src/styles/palettes.js");

  __REACT_HOT_LOADER__.register(BELLTOWN_BLUE, "BELLTOWN_BLUE", "src/styles/palettes.js");

  __REACT_HOT_LOADER__.register(BAJA_TEAL, "BAJA_TEAL", "src/styles/palettes.js");

  __REACT_HOT_LOADER__.register(EMERALD_CITY, "EMERALD_CITY", "src/styles/palettes.js");

  __REACT_HOT_LOADER__.register(CHEETO, "CHEETO", "src/styles/palettes.js");

  __REACT_HOT_LOADER__.register(POPPY, "POPPY", "src/styles/palettes.js");

  __REACT_HOT_LOADER__.register(PLUM, "PLUM", "src/styles/palettes.js");

  __REACT_HOT_LOADER__.register(VIOLET, "VIOLET", "src/styles/palettes.js");
}();

;
