// txl/styles/theme.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.determineColorAppTheme = exports.contextThemePropType = exports.COLOR_WARNING = exports.COLOR_SUCCESS = exports.COLOR_NEUTRAL = exports.COLOR_ERROR = exports.COLOR_APP_THEME = exports.COLOR_ACCENT = undefined;

var _react = require('react');

var _palettes = require('./palettes');

var palettes = _interopRequireWildcard(_palettes);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

// COLOR_APP_THEME is imported directly by external projects (such as Multiverse).
// Removing it will constitute a major version change.
/* global __APP_THEME__ */
var themeColor = typeof __APP_THEME__ === 'undefined' ? 'TUNE_BLUE' : __APP_THEME__;
var COLOR_APP_THEME = palettes[themeColor];

var COLOR_ACCENT = palettes.TUNE_BLUE;
var COLOR_ERROR = palettes.POPPY;
var COLOR_NEUTRAL = palettes.PACIFIC_STORM;
var COLOR_SUCCESS = palettes.EMERALD_CITY;
var COLOR_WARNING = palettes.CHEETO;

var determineColorAppTheme = function determineColorAppTheme(palette) {
  // Provide backwards compatibility for defined webpack plugins and gives them priority
  // over custom palettes passed in
  if (typeof __APP_THEME__ !== 'undefined') {
    return palettes[__APP_THEME__];
  } else if (palette !== undefined) {
    return palette;
  }
  return COLOR_ACCENT;
};

// Expected shape of custom palette passed into StyleRoot
var contextThemePropType = _react.PropTypes.shape({
  50: _react.PropTypes.string,
  100: _react.PropTypes.string, // eslint-disable-line sort-keys
  200: _react.PropTypes.string,
  300: _react.PropTypes.string,
  400: _react.PropTypes.string,
  500: _react.PropTypes.string,
  600: _react.PropTypes.string,
  700: _react.PropTypes.string,
  800: _react.PropTypes.string,
  900: _react.PropTypes.string,
  '50c': _react.PropTypes.string, // eslint-disable-line sort-keys
  '100c': _react.PropTypes.string, // eslint-disable-line sort-keys
  '200c': _react.PropTypes.string,
  '300c': _react.PropTypes.string,
  '400c': _react.PropTypes.string,
  '500c': _react.PropTypes.string,
  '600c': _react.PropTypes.string,
  '700c': _react.PropTypes.string,
  '800c': _react.PropTypes.string,
  '900c': _react.PropTypes.string
});

exports.COLOR_ACCENT = COLOR_ACCENT;
exports.COLOR_APP_THEME = COLOR_APP_THEME;
exports.COLOR_ERROR = COLOR_ERROR;
exports.COLOR_NEUTRAL = COLOR_NEUTRAL;
exports.COLOR_SUCCESS = COLOR_SUCCESS;
exports.COLOR_WARNING = COLOR_WARNING;
exports.contextThemePropType = contextThemePropType;
exports.determineColorAppTheme = determineColorAppTheme;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(themeColor, 'themeColor', 'src/styles/theme.js');

  __REACT_HOT_LOADER__.register(COLOR_APP_THEME, 'COLOR_APP_THEME', 'src/styles/theme.js');

  __REACT_HOT_LOADER__.register(COLOR_ACCENT, 'COLOR_ACCENT', 'src/styles/theme.js');

  __REACT_HOT_LOADER__.register(COLOR_ERROR, 'COLOR_ERROR', 'src/styles/theme.js');

  __REACT_HOT_LOADER__.register(COLOR_NEUTRAL, 'COLOR_NEUTRAL', 'src/styles/theme.js');

  __REACT_HOT_LOADER__.register(COLOR_SUCCESS, 'COLOR_SUCCESS', 'src/styles/theme.js');

  __REACT_HOT_LOADER__.register(COLOR_WARNING, 'COLOR_WARNING', 'src/styles/theme.js');

  __REACT_HOT_LOADER__.register(determineColorAppTheme, 'determineColorAppTheme', 'src/styles/theme.js');

  __REACT_HOT_LOADER__.register(contextThemePropType, 'contextThemePropType', 'src/styles/theme.js');
}();

;
