// txl/logos/AppStoreAnalyticsLogo.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Logos = require('./Logos.style');

var LogoStyles = _interopRequireWildcard(_Logos);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TxlAppStoreAnalyticsLogo = function TxlAppStoreAnalyticsLogo(props) {
  var outlineSvgLarge = _react2.default.createElement(
    'svg',
    { xmlns: 'http://www.w3.org/2000/svg', viewBox: '0 0 40 40' },
    _react2.default.createElement('path', { fill: '#FFF', d: 'M32.9 33.9C29.5 37.1 25 39 20 39 9.5 39 1 30.5 1 20S9.5 1 20 1v1h-1c-9.5.5-17 8.4-17 18 0 9.9 8.1 18 18 18 4.2 0 8.2-1.5 11.4-4.1l.8-.7.7.7z' }),
    _react2.default.createElement('path', { fill: '#FFF', d: 'M36 24c0-1.7 1.3-3 3-3 .3 0 .6.1.9.2 0-.4.1-.8.1-1.2C40 9.3 32 .6 21 .1v8c1.9.2 3.7.8 5.3 1.7C26.7 8.7 27.8 8 29 8c1.7 0 3 1.3 3 3 0 1.2-.7 2.3-1.8 2.7 1.1 1.8 1.8 4 1.8 6.3 0 3-1.1 5.8-3 7.9l5.7 5.6c1.7-1.9 3.1-4.1 4-6.5-1.5-.2-2.7-1.4-2.7-3zM20 9C13.9 9 9 13.9 9 20s4.9 11 11 11c2.8 0 5.3-1 7.2-2.7l.7.7c-2.1 1.9-4.9 3-7.9 3-6.6 0-12-5.4-12-12S13.4 8 20 8v1z' }),
    _react2.default.createElement('path', { fill: '#FFF', d: 'M29 14c-1.7 0-3-1.3-3-3v-.2c-1.5-1-3.2-1.6-5-1.8v11l7.3 7.2C30 25.3 31 22.8 31 20c0-2.2-.7-4.3-1.8-6H29z' })
  );
  var colorSvgLarge = _react2.default.createElement(
    'svg',
    { xmlns: 'http://www.w3.org/2000/svg', viewBox: '0 0 40 40' },
    _react2.default.createElement(
      'g',
      { fill: '#00A888' },
      _react2.default.createElement('path', { d: 'M28 29c-2.1 1.9-4.9 3-8 3-6.6 0-12-5.4-12-12S13.4 8 20 8V1C9.5 1 1 9.5 1 20s8.5 19 19 19c5 0 9.5-1.9 12.9-5.1L28 29z' }),
      _react2.default.createElement('path', { d: 'M9 20c0 6.1 4.9 11 11 11 2.8 0 5.3-1 7.3-2.7L20 21V9C13.9 9 9 13.9 9 20z' })
    ),
    _react2.default.createElement('path', { fill: '#226659', d: 'M21 9.1V20l7.3 7.2C30 25.3 31 22.8 31 20c0-2.2-.7-4.3-1.8-6H29c-1.7 0-3-1.3-3-3v-.2c-1.4-1-3.2-1.6-5-1.7z' }),
    _react2.default.createElement('path', { fill: '#226659', d: 'M36 24c0-1.7 1.3-3 3-3 .3 0 .6.1.9.2 0-.4.1-.8.1-1.2C40 9.3 32 .6 21 .1v8c1.9.2 3.7.8 5.3 1.7C26.7 8.7 27.8 8 29 8c1.7 0 3 1.3 3 3 0 1.2-.7 2.3-1.8 2.7 1.1 1.8 1.8 4 1.8 6.3 0 3-1.1 5.8-3 7.9l5.7 5.6c1.7-1.9 3.1-4.1 4-6.5-1.5-.2-2.7-1.4-2.7-3z' })
  );

  var outlineSvgSmall = _react2.default.createElement(
    'svg',
    { xmlns: 'http://www.w3.org/2000/svg', viewBox: '0 0 30 30' },
    _react2.default.createElement('path', { fill: '#FFF', d: 'M21 10c-1.1 0-1.9-.9-2-1.9-1.2-.7-2.5-1.1-4-1.1v6.8l6.2 6.2c1.1-1.4 1.8-3.1 1.8-5s-.7-3.6-1.8-5H21z' }),
    _react2.default.createElement('path', { fill: '#FFF', d: 'M29 20c-1.1 0-2-.9-2-2s.9-2 2-2c.3 0 .7.1.9.2 0-.4.1-.8.1-1.2 0-8.3-6.7-15-15-15v6c1.5 0 3 .4 4.2 1.1.4-.7 1-1.1 1.8-1.1 1.1 0 2 .9 2 2 0 .7-.3 1.2-.8 1.6C23.3 11.1 24 13 24 15c0 2.2-.8 4.2-2.1 5.7l4.2 4.2c1.3-1.4 2.3-3.1 2.9-5 .1.1 0 .1 0 .1zM20.5 20.8C19.1 22.1 17.2 23 15 23c-4.4 0-8-3.6-8-8 0-4.1 3.1-7.4 7-7.9v-1c-4.5.5-8 4.3-8 8.9 0 5 4 9 9 9 2.4 0 4.6-1 6.3-2.5l-.8-.7z' }),
    _react2.default.createElement('path', { fill: '#FFF', d: 'M24.4 24.6l-.6.6C21.3 27.6 18 29 14.5 29 7.1 29 1 22.9 1 15.5 1 8.2 6.8 2.3 14 2V1C6.2 1.3 0 7.7 0 15.5 0 23.5 6.5 30 14.5 30c4.2 0 8-1.8 10.6-4.7l-.7-.7z' })
  );
  var colorSvgSmall = _react2.default.createElement(
    'svg',
    { xmlns: 'http://www.w3.org/2000/svg', viewBox: '0 0 30 30' },
    _react2.default.createElement(
      'g',
      { fill: '#00A888' },
      _react2.default.createElement('path', { d: 'M7 15c0 4.4 3.6 8 8 8 2 0 3.7-.7 5.1-1.9L14 15V7.1c-3.9.5-7 3.8-7 7.9z' }),
      _react2.default.createElement('path', { d: 'M20.8 21.8C19.3 23.2 17.2 24 15 24c-5 0-9-4-9-9 0-4.6 3.5-8.4 8-8.9v-5C6.2 1.3 0 7.7 0 15.5 0 23.5 6.5 30 14.5 30c4 0 7.6-1.6 10.2-4.3l-3.9-3.9z' })
    ),
    _react2.default.createElement('path', { fill: '#226659', d: 'M21 10c-1.1 0-1.9-.9-2-1.9-1.2-.7-2.5-1.1-4-1.1v6.8l6.2 6.2c1.1-1.4 1.8-3.1 1.8-5s-.7-3.6-1.8-5H21z' }),
    _react2.default.createElement('path', { fill: '#226659', d: 'M29 20c-1.1 0-2-.9-2-2s.9-2 2-2c.3 0 .7.1.9.2 0-.4.1-.8.1-1.2 0-8.3-6.7-15-15-15v6c1.5 0 3 .4 4.2 1.1.4-.7 1-1.1 1.8-1.1 1.1 0 2 .9 2 2 0 .7-.3 1.2-.8 1.6C23.3 11.1 24 13 24 15c0 2.2-.8 4.2-2.1 5.7l4.2 4.2c1.3-1.4 2.3-3.1 2.9-5 .1.1 0 .1 0 .1z' })
  );

  var svgs = {
    large: {
      color: colorSvgLarge,
      outline: outlineSvgLarge
    },
    small: {
      color: colorSvgSmall,
      outline: outlineSvgSmall
    }
  };

  return _react2.default.createElement(
    'div',
    {
      'data-component': 'TxlAppStoreAnalyticsLogo',
      style: LogoStyles['PRODUCT_LOGO_' + props.size.toUpperCase()]
    },
    svgs[props.size][props.type]
  );
};

TxlAppStoreAnalyticsLogo.propTypes = {
  size: _react.PropTypes.oneOf(['small', 'large']),
  type: _react.PropTypes.oneOf(['color', 'outline'])
};

TxlAppStoreAnalyticsLogo.defaultProps = {
  size: 'large',
  type: 'color'
};

var _default = TxlAppStoreAnalyticsLogo;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlAppStoreAnalyticsLogo, 'TxlAppStoreAnalyticsLogo', 'src/logos/AppStoreAnalyticsLogo.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/logos/AppStoreAnalyticsLogo.jsx');
}();

;
module.exports = exports['default'];
