// txl/logos/AttributionAnalyticsLogo.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Logos = require('./Logos.style');

var LogoStyles = _interopRequireWildcard(_Logos);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TxlAttributionAnalyticsLogo = function TxlAttributionAnalyticsLogo(props) {
  var outlineSvgLarge = _react2.default.createElement(
    'svg',
    { xmlns: 'http://www.w3.org/2000/svg', viewBox: '0 0 40 40' },
    _react2.default.createElement('path', { fill: 'none', d: 'M20 1C9.5 1 1 9.5 1 20s8.5 19 19 19c6 0 11.3-2.8 14.8-7.2-2.2-.5-3.8-2.4-3.8-4.8 0-1.1.4-2.2 1-3l-8.8-8.2c-.8.7-1.9 1.2-3.1 1.2l-.9 12.1c1.6.3 2.7 1.7 2.7 3.4 0 1.9-1.6 3.5-3.5 3.5S15 34.4 15 32.5c0-1.8 1.4-3.3 3.3-3.5l.9-12.1c-2.4-.4-4.2-2.4-4.2-4.9 0-2.8 2.2-5 5-5s5 2.2 5 5c0 1.1-.4 2.2-1 3l8.8 8.2c.9-.8 2-1.2 3.3-1.2 1 0 2 .3 2.8.8 0-.9.1-1.8.1-2.8C39 9.5 30.5 1 20 1z' }),
    _react2.default.createElement('path', { fill: '#FFF', d: 'M20 0C9 0 0 9 0 20s9 20 20 20c6.5 0 12.3-3.2 16-8 1.8-2.4 3.1-5.3 3.7-8.4.2-1.2.3-2.4.3-3.6C40 9 31 0 20 0zm16 22c-1.3 0-2.4.5-3.3 1.2L24 15c.6-.8 1-1.9 1-3 0-2.8-2.2-5-5-5s-5 2.2-5 5c0 2.5 1.8 4.5 4.1 4.9L18.3 29c-1.8.1-3.3 1.6-3.3 3.5s1.6 3.5 3.5 3.5 3.5-1.6 3.5-3.5c0-1.7-1.2-3.1-2.7-3.4l.8-12.1c1.2 0 2.3-.5 3.1-1.2L32 24c-.6.8-1 1.9-1 3 0 2.4 1.6 4.3 3.8 4.8C31.3 36.2 26 39 20 39 9.5 39 1 30.5 1 20S9.5 1 20 1s19 8.5 19 19c0 1-.1 1.9-.2 2.8-.8-.5-1.8-.8-2.8-.8z' })
  );
  var colorSvgLarge = _react2.default.createElement(
    'svg',
    { xmlns: 'http://www.w3.org/2000/svg', viewBox: '0 0 40 40' },
    _react2.default.createElement('path', { fill: '#0252C6', d: 'M30 27c0-1.4.5-2.7 1.3-3.7L24 16.4c-1 .9-2.4 1.5-4 1.5L19.3 28c2.1.4 3.7 2.2 3.7 4.4 0 2.5-2 4.5-4.5 4.5S14 35 14 32.5c0-2.4 1.9-4.4 4.3-4.5l.7-10.1c-2.8-.4-5-2.9-5-5.9 0-3.3 2.7-6 6-6s6 2.7 6 6c0 1.4-.5 2.7-1.3 3.7l7.3 6.8c1.1-1 2.5-1.6 4-1.6s2.8.5 3.8 1.4c.1-.8.2-1.6.2-2.4C40 9 31 0 20 0S0 9 0 20s9 20 20 20c6.1 0 11.6-2.7 15.2-7.1C32.3 32.6 30 30 30 27z' }),
    _react2.default.createElement('circle', { fill: '#55AEFF', cx: '20', cy: '12', r: '5' }),
    _react2.default.createElement('circle', { fill: '#007AFF', cx: '18.5', cy: '32.5', r: '3.5' }),
    _react2.default.createElement('path', { fill: '#007AFF', d: 'M36 22c-2.8 0-5 2.2-5 5s2.2 5 5 5c1.8-2.4 3.1-5.3 3.7-8.4-1-1-2.3-1.6-3.7-1.6z' })
  );

  var outlineSvgSmall = _react2.default.createElement(
    'svg',
    { xmlns: 'http://www.w3.org/2000/svg', viewBox: '0 0 30 30' },
    _react2.default.createElement('path', { fill: '#FFF', d: 'M15 0C6.7 0 0 6.7 0 15s6.7 15 15 15 15-6.7 15-15S23.3 0 15 0zm0 29C7.3 29 1 22.7 1 15S7.3 1 15 1s14 6.3 14 14c0 .5 0 1-.1 1.5-.6-.3-1.2-.5-1.9-.5-1 0-1.9.4-2.6 1l-6.2-5.7c.5-.6.8-1.4.8-2.3 0-2.2-1.8-4-4-4s-4 1.8-4 4c0 1.9 1.3 3.5 3.1 3.9l-.9 9.1c-1.2.2-2.2 1.2-2.2 2.5 0 1.4 1.1 2.5 2.5 2.5s2.5-1.1 2.5-2.5c0-1.1-.7-2.1-1.8-2.4l.9-9.1c.9 0 1.8-.4 2.5-.9l6.2 5.7c-.5.6-.8 1.3-.8 2.2 0 1.8 1.2 3.3 2.9 3.8C23.3 27 19.4 29 15 29z' })
  );
  var colorSvgSmall = _react2.default.createElement(
    'svg',
    { xmlns: 'http://www.w3.org/2000/svg', viewBox: '0 0 30 30' },
    _react2.default.createElement('path', { fill: '#0252C6', d: 'M22 20.2c0-1.1.4-2.2 1.1-3l-5.3-4.9c-.7.6-1.6 1-2.7 1.1l-.6 7.7c1.5.4 2.6 1.8 2.6 3.5 0 2-1.6 3.6-3.6 3.6S10 26.6 10 24.6s1.6-3.6 3.6-3.6l.6-7.6c-2.3-.3-4.2-2.3-4.2-4.7C10 6.1 12.1 4 14.8 4s4.8 2.1 4.8 4.8c0 1.1-.4 2.1-1 2.9l5.3 4.9c.8-.7 1.8-1.1 3-1.1s2.3.5 3.1 1.2V15c0-8.3-6.7-15-15-15S0 6.7 0 15s6.7 15 15 15c4.5 0 8.5-2 11.2-5.1-2.4-.2-4.2-2.2-4.2-4.7z' }),
    _react2.default.createElement('circle', { fill: '#55AEFF', cx: '14.8', cy: '8.8', r: '3.8' }),
    _react2.default.createElement('circle', { fill: '#007AFF', cx: '13.6', cy: '24.6', r: '2.6' }),
    _react2.default.createElement('path', { fill: '#007AFF', d: 'M26.8 16.5c-2.1 0-3.8 1.7-3.8 3.8s1.7 3.8 3.8 3.8h.2c1.3-1.8 2.3-3.8 2.7-6-.7-1-1.7-1.6-2.9-1.6z' })
  );

  var svgs = {
    large: {
      color: colorSvgLarge,
      outline: outlineSvgLarge
    },
    small: {
      color: colorSvgSmall,
      outline: outlineSvgSmall
    }
  };

  return _react2.default.createElement(
    'div',
    {
      'data-component': 'TxlAttributionAnalyticsLogo',
      style: LogoStyles['PRODUCT_LOGO_' + props.size.toUpperCase()]
    },
    svgs[props.size][props.type]
  );
};

TxlAttributionAnalyticsLogo.propTypes = {
  size: _react.PropTypes.oneOf(['small', 'large']),
  type: _react.PropTypes.oneOf(['color', 'outline'])
};

TxlAttributionAnalyticsLogo.defaultProps = {
  size: 'large',
  type: 'color'
};

var _default = TxlAttributionAnalyticsLogo;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlAttributionAnalyticsLogo, 'TxlAttributionAnalyticsLogo', 'src/logos/AttributionAnalyticsLogo.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/logos/AttributionAnalyticsLogo.jsx');
}();

;
module.exports = exports['default'];
