// txl/logos/InAppMarketingLogo.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Logos = require('./Logos.style');

var LogoStyles = _interopRequireWildcard(_Logos);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TxlInAppMarketingLogo = function TxlInAppMarketingLogo(props) {
  var outlineSvgLarge = _react2.default.createElement(
    'svg',
    { xmlns: 'http://www.w3.org/2000/svg', viewBox: '0 0 40 40' },
    _react2.default.createElement('path', { fill: '#FFF', d: 'M20 0C9 0 0 9 0 20c0 4.2 1.3 8 3.4 11.2 1.3 2.2 3.1 4.1 5.3 5.3C12 38.7 15.8 40 20 40c11 0 20-9 20-20S31 0 20 0zM2.5 24c0-7.4 6.1-13.5 13.5-13.5S29.5 16.6 29.5 24 23.4 37.5 16 37.5c-2.3 0-4.5-.6-6.4-1.6-.2-.1-.3-.2-.5-.4.7.3 1.6.5 2.4.5 3.6 0 6.5-2.9 6.5-6.5S15.1 23 11.5 23 5 25.9 5 29.5c0 1.1.3 2.1.7 3-.6-.7-1.1-1.4-1.6-2.1-1-1.9-1.6-4.1-1.6-6.4zM6 29.5c0-3 2.5-5.5 5.5-5.5s5.5 2.5 5.5 5.5-2.5 5.5-5.5 5.5S6 32.5 6 29.5zM20 39c-1.5 0-2.9-.2-4.3-.5h.3c8 0 14.5-6.5 14.5-14.5S24 9.5 16 9.5 1.5 16 1.5 24v.3C1.2 22.9 1 21.5 1 20 1 9.5 9.5 1 20 1s19 8.5 19 19-8.5 19-19 19z' })
  );
  var colorSvgLarge = _react2.default.createElement(
    'svg',
    { xmlns: 'http://www.w3.org/2000/svg', viewBox: '0 0 40 40' },
    _react2.default.createElement('path', { fill: '#E71336', d: 'M20 0C9 0 0 9 0 20c0 2.5.5 4.9 1.3 7.1-.2-1-.3-2-.3-3.1C1 15.7 7.7 9 16 9s15 6.7 15 15-6.7 15-15 15c-1.1 0-2.1-.1-3.1-.3 2.2.9 4.6 1.3 7.1 1.3 11 0 20-9 20-20S31 0 20 0z' }),
    _react2.default.createElement('path', { fill: '#FF5957', d: 'M16 10C8.3 10 2 16.3 2 24c0 2.7.8 5.3 2.1 7.4 0-.4-.1-.9-.1-1.4 0-3.9 3.1-7 7-7s7 3.1 7 7-3.1 7-7 7h-.2c1.6.6 3.4 1 5.2 1 7.7 0 14-6.3 14-14s-6.3-14-14-14z' }),
    _react2.default.createElement('circle', { fill: '#FF8482', cx: '11', cy: '30', r: '6' })
  );

  var outlineSvgSmall = _react2.default.createElement(
    'svg',
    { xmlns: 'http://www.w3.org/2000/svg', viewBox: '0 0 30 30' },
    _react2.default.createElement('path', { fill: '#FFF', d: 'M15 0C6.7 0 0 6.7 0 15s6.7 15 15 15 15-6.7 15-15S23.3 0 15 0zM4 21.5C4 19 6 17 8.5 17s4.5 2 4.5 4.5S11 26 8.5 26 4 24 4 21.5zm10 0c0-3-2.5-5.5-5.5-5.5-2.9 0-5.3 2.2-5.5 5.1-.3-1-.5-2-.5-3.1 0-5.2 4.3-9.5 9.5-9.5s9.5 4.3 9.5 9.5-4.3 9.5-9.5 9.5c-1.1 0-2.1-.2-3.1-.5 2.9-.2 5.1-2.6 5.1-5.5zm1 7.5c-1.3 0-2.6-.2-3.8-.5h.8c5.8 0 10.5-4.7 10.5-10.5S17.8 7.5 12 7.5 1.5 12.2 1.5 18v.8C1.2 17.6 1 16.3 1 15 1 7.3 7.3 1 15 1s14 6.3 14 14-6.3 14-14 14z' })
  );
  var colorSvgSmall = _react2.default.createElement(
    'svg',
    { xmlns: 'http://www.w3.org/2000/svg', viewBox: '0 0 30 30' },
    _react2.default.createElement('path', { fill: '#E71336', d: 'M15 0C6.7 0 0 6.7 0 15c0 .6.1 1.3.1 1.9C.9 11.3 5.7 7 11.5 7 17.8 7 23 12.2 23 18.5c0 5.8-4.3 10.6-9.9 11.4.6.1 1.2.1 1.9.1 8.3 0 15-6.7 15-15S23.3 0 15 0z' }),
    _react2.default.createElement('path', { fill: '#FF5957', d: 'M11.5 8C5.7 8 1 12.7 1 18.5c0 1.6.4 3.1 1 4.5v-.5c0-3 2.5-5.5 5.5-5.5s5.5 2.5 5.5 5.5S10.5 28 7.5 28H7c1.4.7 2.9 1 4.5 1C17.3 29 22 24.3 22 18.5S17.3 8 11.5 8z' }),
    _react2.default.createElement('circle', { fill: '#FF8482', cx: '7.5', cy: '22.5', r: '4.5' })
  );

  var svgs = {
    large: {
      color: colorSvgLarge,
      outline: outlineSvgLarge
    },
    small: {
      color: colorSvgSmall,
      outline: outlineSvgSmall
    }
  };

  return _react2.default.createElement(
    'div',
    {
      'data-component': 'TxlInAppMarketingLogo',
      style: LogoStyles['PRODUCT_LOGO_' + props.size.toUpperCase()]
    },
    svgs[props.size][props.type]
  );
};

TxlInAppMarketingLogo.propTypes = {
  size: _react.PropTypes.oneOf(['small', 'large']),
  type: _react.PropTypes.oneOf(['color', 'outline'])
};

TxlInAppMarketingLogo.defaultProps = {
  size: 'large',
  type: 'color'
};

var _default = TxlInAppMarketingLogo;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlInAppMarketingLogo, 'TxlInAppMarketingLogo', 'src/logos/InAppMarketingLogo.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/logos/InAppMarketingLogo.jsx');
}();

;
module.exports = exports['default'];
