// txl/logos/TuneBILogo.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Logos = require('./Logos.style');

var LogoStyles = _interopRequireWildcard(_Logos);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TxlTuneBILogo = function TxlTuneBILogo(props) {
  var outlineSvgLarge = _react2.default.createElement(
    'svg',
    { xmlns: 'http://www.w3.org/2000/svg', viewBox: '0 0 40 40' },
    _react2.default.createElement('defs', null),
    _react2.default.createElement('path', { fill: '#fff', d: 'M20 1A19 19 0 1 1 1 20 19 19 0 0 1 20 1m0-1a20 20 0 1 0 20 20A20 20 0 0 0 20 0z' }),
    _react2.default.createElement('rect', { fill: '#fff', x: '16', y: '33', width: '8', height: '1', rx: '.5', ry: '.5' }),
    _react2.default.createElement('rect', { fill: '#fff', x: '16', y: '35', width: '8', height: '1', rx: '.5', ry: '.5' }),
    _react2.default.createElement('path', { fill: '#fff', d: 'M28 19a8 8 0 0 1-3.25 6.43 1 1 0 0 0-.39.58l-1 4.22a1 1 0 0 1-1 .77h-4.77a1 1 0 0 1-1-.77l-1-4.22a1 1 0 0 0-.39-.58A8 8 0 0 1 20 11v-1a9 9 0 0 0-5.61 16 1 1 0 0 1 .35.55l1.08 4.65a1 1 0 0 0 1 .77h6.41a1 1 0 0 0 1-.77l1.08-4.65a1 1 0 0 1 .3-.55A9 9 0 0 0 29 19h-1z' }),
    _react2.default.createElement('path', { fill: '#fff', d: 'M21 8v10h10A10 10 0 0 0 21 8z' })
  );
  var colorSvgLarge = _react2.default.createElement(
    'svg',
    { xmlns: 'http://www.w3.org/2000/svg', viewBox: '0 0 40 40' },
    _react2.default.createElement('defs', null),
    _react2.default.createElement('path', { fill: '#ff7c18', d: 'M20 0a20 20 0 1 0 20 20A20 20 0 0 0 20 0zm11 17v1H21V8h1a9 9 0 0 1 9 9zm-7.5 19h-7a.5.5 0 0 1 0-1h7a.5.5 0 0 1 0 1zm0-2h-7a.5.5 0 0 1 0-1h7a.5.5 0 0 1 0 1zm2-7.85a1 1 0 0 0-.37.55L24 31.24a1 1 0 0 1-1 .76h-6a1 1 0 0 1-1-.76l-1.1-4.56a1 1 0 0 0-.37-.54A9 9 0 0 1 20 10v9h9a9 9 0 0 1-3.55 7.15z' }),
    _react2.default.createElement('path', { fill: '#ffaa45', d: 'M30 17a8 8 0 0 0-8-8v8h8z' }),
    _react2.default.createElement('path', { opacity: '0.49', fill: '#ffaa45', d: 'M30 17a8 8 0 0 0-8-8v8h8z' }),
    _react2.default.createElement('path', { fill: '#ffaa45', d: 'M20 19v-8h-.95a8 8 0 0 0-3.69 14.46 1 1 0 0 1 .4.57l1 4.16a1 1 0 0 0 1 .76h4.44a1 1 0 0 0 1-.76l1-4.14a1 1 0 0 1 .4-.57A8 8 0 0 0 28 19h-8z' })
  );

  var colorSvgSmall = _react2.default.createElement(
    'svg',
    { xmlns: 'http://www.w3.org/2000/svg', viewBox: '0 0 30 30' },
    _react2.default.createElement('defs', null),
    _react2.default.createElement('path', { fill: '#ffaa45', d: 'M15 14V8a6.26 6.26 0 0 0-5.85 6 6 6 0 0 0 2.73 5.08 1 1 0 0 1 .49.72l.35 2.35a1 1 0 0 0 1 .85h2.6a1 1 0 0 0 1-.84l.4-2.49a1 1 0 0 1 .48-.7A6.57 6.57 0 0 0 21 14h-6z' }),
    _react2.default.createElement('path', { fill: '#ffaa45', d: 'M17 6v6h6a6 6 0 0 0-6-6z' }),
    _react2.default.createElement('path', { fill: '#ff7c18', d: 'M15 0a15 15 0 1 0 15 15A15 15 0 0 0 15 0zm2.5 28h-5a.5.5 0 0 1 0-1h5a.5.5 0 0 1 0 1zm0-2h-5a.5.5 0 0 1 0-1h5a.5.5 0 0 1 0 1zm1.5-6.29a1 1 0 0 0-.43.66l-.47 2.79a1 1 0 0 1-1 .84h-4.25a1 1 0 0 1-1-.84l-.45-2.79a1 1 0 0 0-.4-.66A7 7 0 0 1 15 7v7h7a7 7 0 0 1-3 5.71zM24 13h-8V5h1a7 7 0 0 1 7 7v1z' })
  );
  var outlineSvgSmall = _react2.default.createElement(
    'svg',
    { xmlns: 'http://www.w3.org/2000/svg', viewBox: '0 0 30 30' },
    _react2.default.createElement('defs', null),
    _react2.default.createElement('path', { fill: '#fff', d: 'M15 1A14 14 0 1 1 1 15 14 14 0 0 1 15 1m0-1a15 15 0 1 0 15 15A15 15 0 0 0 15 0z' }),
    _react2.default.createElement('rect', { fill: '#fff', x: '12', y: '25', width: '6', height: '1', rx: '.5', ry: '.5' }),
    _react2.default.createElement('rect', { fill: '#fff', x: '12', y: '27', width: '6', height: '1', rx: '.5', ry: '.5' }),
    _react2.default.createElement('path', { fill: '#fff', d: 'M15 8a6 6 0 0 0-3.22 11.06 1 1 0 0 1 .48.68l.43 2.43a1 1 0 0 0 1 .83h2.65a1 1 0 0 0 1-.83l.43-2.44a1 1 0 0 1 .47-.68A6.42 6.42 0 0 0 21 14h1a7.49 7.49 0 0 1-3 6 1 1 0 0 0-.42.64l-.46 2.56a1 1 0 0 1-1 .83h-4.28a1 1 0 0 1-1-.83l-.44-2.59A1 1 0 0 0 11 20a7.49 7.49 0 0 1-3-6 7 7 0 0 1 7-7v1z' }),
    _react2.default.createElement('path', { fill: '#fff', d: 'M24 13a8 8 0 0 0-8-8v8h8z' })
  );

  var svgs = {
    large: {
      color: colorSvgLarge,
      outline: outlineSvgLarge
    },
    small: {
      color: colorSvgSmall,
      outline: outlineSvgSmall
    }
  };

  return _react2.default.createElement(
    'div',
    {
      'data-component': 'TxlTuneBILogo',
      style: LogoStyles['PRODUCT_LOGO_' + props.size.toUpperCase()]
    },
    svgs[props.size][props.type]
  );
};

TxlTuneBILogo.propTypes = {
  size: _react.PropTypes.oneOf(['small', 'large']),
  type: _react.PropTypes.oneOf(['color', 'outline'])
};

TxlTuneBILogo.defaultProps = {
  size: 'large',
  type: 'color'
};

var _default = TxlTuneBILogo;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlTuneBILogo, 'TxlTuneBILogo', 'src/logos/TuneBILogo.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/logos/TuneBILogo.jsx');
}();

;
module.exports = exports['default'];
