// txl/navigation/BackTemplate.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _theme = require('../styles/theme');

var _helpers = require('../styles/helpers');

var _ArrowLeft = require('../icons/ArrowLeft');

var _ArrowLeft2 = _interopRequireDefault(_ArrowLeft);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ICON_STYLE = {
  alignItems: 'center',
  display: 'flex',
  justifyContent: 'space-around',
  left: (0, _helpers.gridUnits)(1),
  position: 'absolute',
  top: (0, _helpers.gridUnits)(2),
  width: (0, _helpers.gridUnits)(5)
};

var ITEM_STYLE = {
  backgroundColor: 'white',
  borderBottom: '1px solid ' + _theme.COLOR_NEUTRAL['100'],
  display: 'flex',
  position: 'relative',
  width: '100%'
};

var BackTemplate = function BackTemplate(props) {
  var children = props.children,
      hovered = props.hovered,
      style = props.style;

  var color = _theme.COLOR_NEUTRAL[hovered ? '800' : '400'];
  var itemContentStyle = (0, _extends3.default)({}, style, {
    color: color
  });

  return _react2.default.createElement(
    'div',
    { style: ITEM_STYLE },
    _react2.default.createElement(
      'div',
      { style: ICON_STYLE },
      _react2.default.createElement(_ArrowLeft2.default, { color: color })
    ),
    _react2.default.createElement(
      'div',
      { style: itemContentStyle },
      children
    )
  );
};

BackTemplate.propTypes = {
  children: _react.PropTypes.node,
  hovered: _react.PropTypes.bool,
  style: _react.PropTypes.object
};

var _default = (0, _radium2.default)(BackTemplate);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ICON_STYLE, 'ICON_STYLE', 'src/navigation/BackTemplate.jsx');

  __REACT_HOT_LOADER__.register(ITEM_STYLE, 'ITEM_STYLE', 'src/navigation/BackTemplate.jsx');

  __REACT_HOT_LOADER__.register(BackTemplate, 'BackTemplate', 'src/navigation/BackTemplate.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/navigation/BackTemplate.jsx');
}();

;
module.exports = exports['default'];
