// txl/navigation/Navigation.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.STYLES = exports.NAV_ITEM_GU_HEIGHT = exports.FLIP_ARROW = exports.LINK_COLOR = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

var _transitions = require('../styles/mixins/transitions');

var _layers = require('../styles/mixins/layers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var LINK_COLOR = _theme.COLOR_NEUTRAL['800'];

var NAV_ITEM_GU_HEIGHT = 7;

var ITEM_COMMON = {
  cursor: 'pointer',
  height: (0, _helpers.gridUnits)(NAV_ITEM_GU_HEIGHT),
  lineHeight: (0, _helpers.gridUnits)(NAV_ITEM_GU_HEIGHT), // valign tweak
  position: 'relative',
  transition: (0, _transitions.cubicTransition)('background')
};

var FLIP_ARROW = {
  color: _theme.COLOR_NEUTRAL['800'],
  transform: 'rotateZ(180deg)'
};

var ITEM_HOVERS = {
  backgroundColor: _theme.COLOR_NEUTRAL['100'],
  color: _theme.COLOR_NEUTRAL['100c']
};

var ICON_WRAP_COMMON = {
  alignItems: 'center',
  display: 'flex',
  height: (0, _helpers.gridUnits)(5),
  justifyContent: 'space-around',
  position: 'absolute',
  top: (0, _helpers.gridUnits)(1),
  width: (0, _helpers.gridUnits)(5)
};

var STYLES = {
  arrowWrap: (0, _extends3.default)({}, ICON_WRAP_COMMON, {
    right: (0, _helpers.gridUnits)(1),
    transition: (0, _transitions.cubicTransition)('transform')
  }),
  container: {
    base: (0, _extends3.default)({}, _layers.LAYER_1, {
      backgroundColor: 'white',
      fontFamily: 'Roboto, sans-serif', // probably can remove at some point
      minHeight: 'calc(100vh - ' + (0, _helpers.gridUnits)(13) + ')', // gu(13) is header height
      transition: (0, _transitions.cubicTransition)('background'),
      userSelect: 'none',
      width: (0, _helpers.gridUnits)(40)
    })
  },
  iconWrap: (0, _extends3.default)({}, ICON_WRAP_COMMON, {
    left: (0, _helpers.gridUnits)(1)
  }),
  itemMain: {
    base: (0, _extends3.default)({}, ITEM_COMMON, {
      fontSize: 14
    }),
    hover: ITEM_HOVERS
  },
  itemSub: {
    base: (0, _extends3.default)({}, ITEM_COMMON, {
      fontSize: 12
    }),
    hover: ITEM_HOVERS
  },
  itemSubContainer: {
    height: '0px',
    overflow: 'hidden',
    transition: (0, _transitions.cubicTransition)('height')
  },
  itemTemplate: {
    base: {
      boxSizing: 'border-box',
      color: LINK_COLOR,
      display: 'block',
      height: ITEM_COMMON.height,
      lineHeight: (0, _helpers.gridUnits)(NAV_ITEM_GU_HEIGHT),
      overflow: 'hidden',
      padding: '0 ' + (0, _helpers.gridUnits)(7),
      textDecoration: 'none',
      whitespace: 'nowrap',
      width: '100%'
    }
  },
  navSectionExpanded: {
    backgroundColor: _theme.COLOR_NEUTRAL['50']
  }
};

exports.LINK_COLOR = LINK_COLOR;
exports.FLIP_ARROW = FLIP_ARROW;
exports.NAV_ITEM_GU_HEIGHT = NAV_ITEM_GU_HEIGHT;
exports.STYLES = STYLES;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(LINK_COLOR, 'LINK_COLOR', 'src/navigation/Navigation.style.js');

  __REACT_HOT_LOADER__.register(NAV_ITEM_GU_HEIGHT, 'NAV_ITEM_GU_HEIGHT', 'src/navigation/Navigation.style.js');

  __REACT_HOT_LOADER__.register(ITEM_COMMON, 'ITEM_COMMON', 'src/navigation/Navigation.style.js');

  __REACT_HOT_LOADER__.register(FLIP_ARROW, 'FLIP_ARROW', 'src/navigation/Navigation.style.js');

  __REACT_HOT_LOADER__.register(ITEM_HOVERS, 'ITEM_HOVERS', 'src/navigation/Navigation.style.js');

  __REACT_HOT_LOADER__.register(ICON_WRAP_COMMON, 'ICON_WRAP_COMMON', 'src/navigation/Navigation.style.js');

  __REACT_HOT_LOADER__.register(STYLES, 'STYLES', 'src/navigation/Navigation.style.js');
}();

;
