// txl/navigation/NavigationItem.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Navigation = require('./Navigation.style');

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

var _Item = require('./records/Item');

var _Item2 = _interopRequireDefault(_Item);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var NavigationItem = function (_Component) {
  (0, _inherits3.default)(NavigationItem, _Component);

  function NavigationItem() {
    var _ref;

    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, NavigationItem);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = NavigationItem.__proto__ || (0, _getPrototypeOf2.default)(NavigationItem)).call.apply(_ref, [this].concat(args))), _this), _this._onMouseEnter = function () {
      var _this2;

      return (_this2 = _this).___onMouseEnter__REACT_HOT_LOADER__.apply(_this2, arguments);
    }, _this._onMouseLeave = function () {
      var _this3;

      return (_this3 = _this).___onMouseLeave__REACT_HOT_LOADER__.apply(_this3, arguments);
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  (0, _createClass3.default)(NavigationItem, [{
    key: 'shouldComponentUpdate',
    value: function shouldComponentUpdate(nextProps) {
      return this.props.data !== nextProps.data;
    }
  }, {
    key: '___onMouseEnter__REACT_HOT_LOADER__',
    value: function ___onMouseEnter__REACT_HOT_LOADER__() {
      return this.___onMouseEnter__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___onMouseEnter__REACT_HOT_LOADER__',
    value: function ___onMouseEnter__REACT_HOT_LOADER__() {
      var _props = this.props,
          onUpdate = _props.onUpdate,
          data = _props.data;

      onUpdate(function (item) {
        return item.id === data.id && !item.hovered;
      }, function (item) {
        return item.set('hovered', true);
      });
    }
  }, {
    key: '___onMouseLeave__REACT_HOT_LOADER__',
    value: function ___onMouseLeave__REACT_HOT_LOADER__() {
      return this.___onMouseLeave__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___onMouseLeave__REACT_HOT_LOADER__',
    value: function ___onMouseLeave__REACT_HOT_LOADER__() {
      var _props2 = this.props,
          onUpdate = _props2.onUpdate,
          data = _props2.data;

      onUpdate(function (item) {
        return item.id === data.id && item.hovered;
      }, function (item) {
        return item.set('hovered', false);
      });
    }
  }, {
    key: '_renderTemplate',
    value: function _renderTemplate(itemActive) {
      var _props$data = this.props.data,
          active = _props$data.active,
          display = _props$data.display,
          hovered = _props$data.hovered,
          meta = _props$data.meta,
          name = _props$data.name,
          Template = _props$data.template,
          templateProps = _props$data.templateProps;


      var style = (0, _helpers.combineStyles)(_Navigation.STYLES.itemTemplate.base, active && itemActive);

      var props = (0, _extends3.default)({}, templateProps, {
        active: active,
        display: display,
        hovered: hovered,
        meta: meta,
        name: name,
        style: style
      });

      return _react2.default.createElement(
        Template,
        props,
        display
      );
    }
  }, {
    key: 'render',
    value: function render() {
      var COLOR_APP_THEME = (0, _theme.determineColorAppTheme)(this.context.txlAppTheme);
      var itemActive = {
        backgroundColor: COLOR_APP_THEME['500'],
        color: COLOR_APP_THEME['500c']
      };
      var styles = (0, _helpers.combineStyles)(_Navigation.STYLES.itemSub.base, this.props.data.active && itemActive, this.props.data.hovered && _Navigation.STYLES.itemSub.hover);

      return _react2.default.createElement(
        'div',
        {
          onMouseEnter: this._onMouseEnter,
          onMouseLeave: this._onMouseLeave,
          style: styles
        },
        this._renderTemplate(itemActive)
      );
    }
  }]);
  return NavigationItem;
}(_react.Component);

NavigationItem.propTypes = {
  data: require('react').PropTypes.any.isRequired,
  onUpdate: require('react').PropTypes.func.isRequired
};


NavigationItem.contextTypes = {
  txlAppTheme: _theme.contextThemePropType
};

var _default = NavigationItem;
exports.default = _default;
;

var _temp2 = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(NavigationItem, 'NavigationItem', 'src/navigation/NavigationItem.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/navigation/NavigationItem.jsx');
}();

;
module.exports = exports['default'];
