// txl/navigation/NavigationGroup.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Group = require('./records/Group');

var _Group2 = _interopRequireDefault(_Group);

var _Item = require('./records/Item');

var _Item2 = _interopRequireDefault(_Item);

var _Navigation = require('./Navigation.style');

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

var _ArrowDown = require('../icons/ArrowDown');

var _ArrowDown2 = _interopRequireDefault(_ArrowDown);

var _NavigationItem = require('./NavigationItem');

var _NavigationItem2 = _interopRequireDefault(_NavigationItem);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ITEM_HOVER = _Navigation.STYLES.itemMain.hover;

var NavigationGroup = function (_Component) {
  (0, _inherits3.default)(NavigationGroup, _Component);

  function NavigationGroup() {
    var _ref;

    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, NavigationGroup);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = NavigationGroup.__proto__ || (0, _getPrototypeOf2.default)(NavigationGroup)).call.apply(_ref, [this].concat(args))), _this), _this._onMouseEnter = function () {
      var _this2;

      return (_this2 = _this).___onMouseEnter__REACT_HOT_LOADER__.apply(_this2, arguments);
    }, _this._onMouseLeave = function () {
      var _this3;

      return (_this3 = _this).___onMouseLeave__REACT_HOT_LOADER__.apply(_this3, arguments);
    }, _this._onClick = function () {
      var _this4;

      return (_this4 = _this).___onClick__REACT_HOT_LOADER__.apply(_this4, arguments);
    }, _this._onUpdate = function () {
      var _this5;

      return (_this5 = _this).___onUpdate__REACT_HOT_LOADER__.apply(_this5, arguments);
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  (0, _createClass3.default)(NavigationGroup, [{
    key: '___onMouseEnter__REACT_HOT_LOADER__',
    value: function ___onMouseEnter__REACT_HOT_LOADER__() {
      return this.___onMouseEnter__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___onMouseEnter__REACT_HOT_LOADER__',
    value: function ___onMouseEnter__REACT_HOT_LOADER__() {
      var _props = this.props,
          onUpdate = _props.onUpdate,
          data = _props.data;

      onUpdate(function (group) {
        return !group.hovered && group.id === data.id;
      }, function (group) {
        return group.set('hovered', true);
      });
    }
  }, {
    key: '___onMouseLeave__REACT_HOT_LOADER__',
    value: function ___onMouseLeave__REACT_HOT_LOADER__() {
      return this.___onMouseLeave__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___onMouseLeave__REACT_HOT_LOADER__',
    value: function ___onMouseLeave__REACT_HOT_LOADER__() {
      var _props2 = this.props,
          onUpdate = _props2.onUpdate,
          data = _props2.data;

      onUpdate(function (group) {
        return group.hovered && group.id === data.id;
      }, function (group) {
        return group.set('hovered', false);
      });
    }
  }, {
    key: '___onClick__REACT_HOT_LOADER__',
    value: function ___onClick__REACT_HOT_LOADER__() {
      return this.___onClick__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___onClick__REACT_HOT_LOADER__',
    value: function ___onClick__REACT_HOT_LOADER__() {
      var _props3 = this.props,
          onUpdate = _props3.onUpdate,
          data = _props3.data;

      onUpdate(function (group) {
        return group.id === data.id;
      }, function (group) {
        return group.set('expanded', !group.expanded);
      });
    }
  }, {
    key: '_renderArrow',
    value: function _renderArrow(activeLinkColor) {
      var _props$data = this.props.data,
          items = _props$data.items,
          expanded = _props$data.expanded,
          active = _props$data.active;


      if (!items.size) {
        return null;
      }

      var arrowStyle = (0, _helpers.combineStyles)(_Navigation.STYLES.arrowWrap, expanded && _Navigation.FLIP_ARROW);

      return _react2.default.createElement(
        'div',
        { style: arrowStyle },
        _react2.default.createElement(_ArrowDown2.default, { color: active ? activeLinkColor : _Navigation.LINK_COLOR })
      );
    }
  }, {
    key: '_renderIcon',
    value: function _renderIcon(activeLinkColor) {
      var _props$data2 = this.props.data,
          active = _props$data2.active,
          Icon = _props$data2.icon;


      if (Icon == null) {
        return null;
      }

      return _react2.default.createElement(
        'div',
        { style: _Navigation.STYLES.iconWrap },
        _react2.default.createElement(Icon, { color: active ? activeLinkColor : _Navigation.LINK_COLOR })
      );
    }
  }, {
    key: '___onUpdate__REACT_HOT_LOADER__',
    value: function ___onUpdate__REACT_HOT_LOADER__() {
      return this.___onUpdate__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___onUpdate__REACT_HOT_LOADER__',
    value: function ___onUpdate__REACT_HOT_LOADER__(predicate, update) {
      var _props4 = this.props,
          data = _props4.data,
          onUpdate = _props4.onUpdate;

      onUpdate(function (group) {
        return group.id === data.id;
      }, function (group) {
        return group.update(predicate, update);
      });
    }
  }, {
    key: '_renderItems',
    value: function _renderItems() {
      var _this6 = this;

      var data = this.props.data;

      if (!data.items.size || !data.expanded) {
        return null;
      }

      var style = (0, _helpers.combineStyles)(_Navigation.STYLES.itemSubContainer, data.expanded && { height: (0, _helpers.gridUnits)(data.items.size * _Navigation.NAV_ITEM_GU_HEIGHT) });

      var items = data.items.toArray().map(function (item) {
        return _react2.default.createElement(_NavigationItem2.default, {
          key: item.id,
          data: item,
          onUpdate: _this6._onUpdate
        });
      });

      return _react2.default.createElement(
        'div',
        { style: style },
        items
      );
    }
  }, {
    key: '_renderTemplate',
    value: function _renderTemplate(itemActive) {
      var _props$data3 = this.props.data,
          active = _props$data3.active,
          display = _props$data3.display,
          hovered = _props$data3.hovered,
          meta = _props$data3.meta,
          name = _props$data3.name,
          Template = _props$data3.template,
          templateProps = _props$data3.templateProps;


      var style = (0, _helpers.combineStyles)(_Navigation.STYLES.itemTemplate.base, active && itemActive);

      var props = (0, _extends3.default)({}, templateProps, {
        active: active,
        display: display,
        hovered: hovered,
        meta: meta,
        name: name,
        style: style
      });

      return _react2.default.createElement(
        Template,
        props,
        display
      );
    }
  }, {
    key: 'render',
    value: function render() {
      var data = this.props.data;

      var COLOR_APP_THEME = (0, _theme.determineColorAppTheme)(this.context.txlAppTheme);
      var activeLinkColor = COLOR_APP_THEME['500c'];
      var itemActive = {
        backgroundColor: COLOR_APP_THEME['500'],
        color: activeLinkColor
      };
      var style = (0, _helpers.combineStyles)(_Navigation.STYLES.itemMain.base, data.active && itemActive, data.hovered && !data.expanded && ITEM_HOVER);

      return _react2.default.createElement(
        'div',
        {
          style: data.expanded ? _Navigation.STYLES.navSectionExpanded : {},
          onMouseEnter: this._onMouseEnter,
          onMouseLeave: this._onMouseLeave,
          onClick: this._onClick
        },
        _react2.default.createElement(
          'div',
          {
            style: style
          },
          this._renderIcon(activeLinkColor),
          this._renderTemplate(itemActive),
          this._renderArrow(activeLinkColor)
        ),
        this._renderItems()
      );
    }
  }]);
  return NavigationGroup;
}(_react.Component);

NavigationGroup.propTypes = {
  data: require('react').PropTypes.any.isRequired,
  onUpdate: require('react').PropTypes.func.isRequired
};


NavigationGroup.contextTypes = {
  txlAppTheme: _theme.contextThemePropType
};

var _default = NavigationGroup;
exports.default = _default;
;

var _temp2 = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ITEM_HOVER, 'ITEM_HOVER', 'src/navigation/NavigationGroup.jsx');

  __REACT_HOT_LOADER__.register(NavigationGroup, 'NavigationGroup', 'src/navigation/NavigationGroup.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/navigation/NavigationGroup.jsx');
}();

;
module.exports = exports['default'];
