// txl/navigation/Navigation.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DefaultTemplate = exports.default = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _DefaultTemplate = require('./DefaultTemplate');

var _DefaultTemplate2 = _interopRequireDefault(_DefaultTemplate);

var _NavigationItems = require('./NavigationItems');

var _NavigationItems2 = _interopRequireDefault(_NavigationItems);

var _NavigationBase = require('./NavigationBase');

var _NavigationBase2 = _interopRequireDefault(_NavigationBase);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var checkArray = function checkArray(input) {
  return Array.isArray(input) ? input : [input];
};

var TxlNavigation = (0, _radium2.default)(_class = function (_React$Component) {
  (0, _inherits3.default)(TxlNavigation, _React$Component);

  function TxlNavigation(props, context) {
    (0, _classCallCheck3.default)(this, TxlNavigation);

    var _this = (0, _possibleConstructorReturn3.default)(this, (TxlNavigation.__proto__ || (0, _getPrototypeOf2.default)(TxlNavigation)).call(this, props, context));

    _this._onUpdate = function () {
      return _this.___onUpdate__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    var items = props.items,
        defaultTemplate = props.template;

    var activeNames = checkArray(props.activeNames);
    var expandedNames = checkArray(props.expandedNames);

    _this.state = {
      groups: new _NavigationItems2.default({ activeNames: activeNames, defaultTemplate: defaultTemplate, expandedNames: expandedNames, items: items })
    };
    return _this;
  }

  (0, _createClass3.default)(TxlNavigation, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      var activeNames = checkArray(nextProps.activeNames);
      var expandedNames = checkArray(nextProps.expandedNames);

      this.setState({
        groups: this.state.groups.updateActive(activeNames).updateExpanded(expandedNames)
      });
    }
  }, {
    key: '___onUpdate__REACT_HOT_LOADER__',
    value: function ___onUpdate__REACT_HOT_LOADER__() {
      return this.___onUpdate__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___onUpdate__REACT_HOT_LOADER__',
    value: function ___onUpdate__REACT_HOT_LOADER__() {
      return this.___onUpdate__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___onUpdate__REACT_HOT_LOADER__',
    value: function ___onUpdate__REACT_HOT_LOADER__(predicate, update) {
      this.setState(function (prevState) {
        return { groups: prevState.groups.updateGroup(predicate, update) };
      });
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(_NavigationBase2.default, { onUpdate: this._onUpdate, items: this.state.groups });
    }
  }]);
  return TxlNavigation;
}(_react2.default.Component)) || _class;

TxlNavigation.propTypes = {
  activeNames: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.arrayOf(_react.PropTypes.string)]),
  expandedNames: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.arrayOf(_react.PropTypes.string)]),
  items: _react.PropTypes.arrayOf(_react.PropTypes.shape({
    display: _react.PropTypes.oneOfType([_react.PropTypes.node, _react.PropTypes.instanceOf(_react2.default.Component)]),
    expanded: _react.PropTypes.bool,
    icon: _Base.ReactClassPropCheck,
    items: _react.PropTypes.arrayOf(_react.PropTypes.shape({
      display: _react.PropTypes.oneOfType([_react.PropTypes.node, _react.PropTypes.instanceOf(_react2.default.Component)]),
      name: _react.PropTypes.string.isRequired,
      template: _react2.default.PropTypes.oneOfType([_Base.ReactClassPropCheck, _react.PropTypes.func])
    })),
    name: _react.PropTypes.string.isRequired,
    template: _react2.default.PropTypes.oneOfType([_Base.ReactClassPropCheck, _react.PropTypes.func])
  })),
  template: _react2.default.PropTypes.oneOfType([_Base.ReactClassPropCheck, _react.PropTypes.func])
};

TxlNavigation.defaultProps = {
  activeNames: [],
  expandedNames: [],
  template: _DefaultTemplate2.default
};

exports.default = TxlNavigation;
exports.DefaultTemplate = _DefaultTemplate2.default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(checkArray, 'checkArray', 'src/navigation/Navigation.jsx');

  __REACT_HOT_LOADER__.register(TxlNavigation, 'TxlNavigation', 'src/navigation/Navigation.jsx');
}();

;
