// txl/navigation/NavigationBase.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Navigation = require('./Navigation.style');

var _Group = require('./records/Group');

var _Group2 = _interopRequireDefault(_Group);

var _NavigationItems = require('./NavigationItems');

var _NavigationItems2 = _interopRequireDefault(_NavigationItems);

var _NavigationGroup = require('./NavigationGroup');

var _NavigationGroup2 = _interopRequireDefault(_NavigationGroup);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TxlNavigationBase = (0, _radium2.default)(_class = function (_React$Component) {
  (0, _inherits3.default)(TxlNavigationBase, _React$Component);

  function TxlNavigationBase() {
    var _ref;

    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, TxlNavigationBase);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlNavigationBase.__proto__ || (0, _getPrototypeOf2.default)(TxlNavigationBase)).call.apply(_ref, [this].concat(args))), _this), _this._onUpdate = function () {
      var _this2;

      return (_this2 = _this).___onUpdate__REACT_HOT_LOADER__.apply(_this2, arguments);
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  (0, _createClass3.default)(TxlNavigationBase, [{
    key: '___onUpdate__REACT_HOT_LOADER__',
    value: function ___onUpdate__REACT_HOT_LOADER__() {
      return this.___onUpdate__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___onUpdate__REACT_HOT_LOADER__',
    value: function ___onUpdate__REACT_HOT_LOADER__() {
      return this.___onUpdate__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '___onUpdate__REACT_HOT_LOADER__',
    value: function ___onUpdate__REACT_HOT_LOADER__(predicate, update) {
      var onUpdate = this.props.onUpdate;


      onUpdate && onUpdate(predicate, update);
    }
  }, {
    key: 'render',
    value: function render() {
      var _this3 = this;

      var groups = this.props.items.valueOf().toArray().map(function (group) {
        return _react2.default.createElement(_NavigationGroup2.default, {
          key: group.id,
          data: group,
          onUpdate: _this3._onUpdate
        });
      });

      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlNavigation',
          style: _Navigation.STYLES.container.base
        },
        groups
      );
    }
  }]);
  return TxlNavigationBase;
}(_react2.default.Component)) || _class;

TxlNavigationBase.propTypes = {
  items: _react.PropTypes.instanceOf(_NavigationItems2.default),
  onUpdate: _react.PropTypes.func
};

var _default = TxlNavigationBase;
exports.default = _default;
;

var _temp2 = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlNavigationBase, 'TxlNavigationBase', 'src/navigation/NavigationBase.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/navigation/NavigationBase.jsx');
}();

;
module.exports = exports['default'];
