// txl/dropdowns/_targets/ContentEditableTarget.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _desc, _value, _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Base = require('../../base/Base');

var _helpers = require('../../styles/helpers');

var _FlexBox = require('../../layout/FlexBox');

var _FlexBox2 = _interopRequireDefault(_FlexBox);

var _InputFieldBase = require('../../input-fields/_InputFieldBase');

var _InputFieldBase2 = _interopRequireDefault(_InputFieldBase);

var _ContentEditable = require('../../input-fields/ContentEditable');

var _ContentEditable2 = _interopRequireDefault(_ContentEditable);

var _ArrowUp = require('../../icons/ArrowUp');

var _ArrowUp2 = _interopRequireDefault(_ArrowUp);

var _ArrowDown = require('../../icons/ArrowDown');

var _ArrowDown2 = _interopRequireDefault(_ArrowDown);

var _ReactUtils = require('../../ReactUtils');

var _KeyCodes = require('../../constants/KeyCodes');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var CONTAINER_PROPS = {
  alignItems: 'flex-end',
  flexWrap: 'nowrap',
  justifyContent: 'space-between'
};

// used to fix container div width so that its children can act on overflow.
// 15px padding to allow for the arrow on dropdown target (13px wide).
var VALUE_INPUT_WRAPPER_STYLES = {
  width: 'calc(100% - ' + (0, _helpers.gridUnits)(3) + ')'
};

var MINIMUM_FIELD_HEIGHT = (0, _helpers.gridUnits)(7);

var TxlContentEditableTarget = (_class = function (_React$Component) {
  (0, _inherits3.default)(TxlContentEditableTarget, _React$Component);

  function TxlContentEditableTarget() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlContentEditableTarget);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlContentEditableTarget.__proto__ || (0, _getPrototypeOf2.default)(TxlContentEditableTarget)).call.apply(_ref, [this].concat(args)));

    _this.state = { focused: false };
    return _this;
  }

  (0, _createClass3.default)(TxlContentEditableTarget, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      var editable = this._contentEditable;
      if (nextProps.open && !this.props.open) {
        editable && typeof editable.focus === 'function' && editable.focus();
      } else if (!nextProps.open && this.props.open) {
        editable && typeof editable.blur === 'function' && editable.blur();
      }
    }
  }, {
    key: '_handleKeyDown',
    value: function _handleKeyDown(e) {
      var _props = this.props,
          onKeyDown = _props.onKeyDown,
          search = _props.search,
          selectItem = _props.selectItem,
          value = _props.value;


      onKeyDown && onKeyDown(e, search);
      if (e.keyCode === _KeyCodes.BACKSPACE && !search) {
        if (Array.isArray(value)) {
          selectItem && selectItem({ item: value.slice(-1).pop() });
        } else {
          selectItem && selectItem({ item: null });
        }
      }
    }
  }, {
    key: '_handleClick',
    value: function _handleClick(e) {
      e.stopPropagation();
      var open = this.props.open;
      var focused = this.state.focused;


      if ((0, _ReactUtils.findDOMNode)(this._arrowButton).contains(e.target)) {
        if (open) {
          this.props.closeDropdown();
        } else {
          this.props.openDropdown();
        }
      } else if (!focused && open || !open) {
        this.props.openDropdown();
      } else {
        this.props.closeDropdown();
      }
    }
  }, {
    key: '_handleOpen',
    value: function _handleOpen() {
      var editable = this._contentEditable;
      if (editable && typeof editable.focus === 'function') {
        editable.focus();
      }
    }
  }, {
    key: '_handleFocus',
    value: function _handleFocus() {
      this.setState({
        focused: true
      }, this.props.openDropdown);
    }
  }, {
    key: '_handleBlur',
    value: function _handleBlur() {
      this.setState({ focused: false });
    }
  }, {
    key: '_handleInputClick',
    value: function _handleInputClick(e) {
      var focused = this.state.focused;

      if (focused) {
        // Prevents dropdown toggle
        e.stopPropagation();
      }
    }
  }, {
    key: '_handleArrowRef',
    value: function _handleArrowRef(ref) {
      this._arrowButton = ref;
    }
  }, {
    key: '_handleContentEditableRef',
    value: function _handleContentEditableRef(ref) {
      this._contentEditable = ref;
    }
  }, {
    key: '_renderArrow',
    value: function _renderArrow() {
      var open = this.props.open;

      var ArrowIcon = open ? _ArrowUp2.default : _ArrowDown2.default;
      var arrowIconContainerStyle = {
        alignItems: 'center',
        display: 'flex',
        minHeight: MINIMUM_FIELD_HEIGHT
      };

      return _react2.default.createElement(
        'div',
        { style: arrowIconContainerStyle },
        _react2.default.createElement(ArrowIcon, { ref: this._handleArrowRef })
      );
    }
  }, {
    key: 'render',
    value: function render() {
      var _props2 = this.props,
          children = _props2.children,
          disabled = _props2.disabled,
          label = _props2.label,
          open = _props2.open,
          placeholder = _props2.placeholder,
          search = _props2.search,
          secondaryText = _props2.secondaryText,
          shouldInputWrap = _props2.shouldInputWrap,
          updateSearch = _props2.updateSearch,
          validationState = _props2.validationState;

      var targetLineWrapping = shouldInputWrap ? 'wrap' : 'nowrap';

      return _react2.default.createElement(
        _InputFieldBase2.default,
        {
          _afterContent: this._renderArrow,
          _containerProps: CONTAINER_PROPS,
          disabled: disabled,
          focused: open,
          label: label,
          onClick: this._handleClick,
          secondaryText: secondaryText,
          validationState: validationState
        },
        _react2.default.createElement(
          _FlexBox2.default,
          {
            alignItems: 'center',
            flexShrink: 1,
            flexWrap: targetLineWrapping,
            style: VALUE_INPUT_WRAPPER_STYLES
          },
          children,
          _react2.default.createElement(_ContentEditable2.default, {
            onBlur: this._handleBlur,
            onChange: updateSearch,
            onFocus: this._handleFocus,
            onClick: this._handleInputClick,
            onKeyDown: this._handleKeyDown,
            placeholder: placeholder,
            ref: this._handleContentEditableRef,
            spellCheck: false,
            value: search
          })
        )
      );
    }
  }]);
  return TxlContentEditableTarget;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class.prototype, 'componentWillReceiveProps', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, 'componentWillReceiveProps'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_handleKeyDown', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleKeyDown'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_handleClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleClick'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_handleOpen', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleOpen'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_handleFocus', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleFocus'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_handleBlur', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleBlur'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_handleInputClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleInputClick'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_handleArrowRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleArrowRef'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_handleContentEditableRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleContentEditableRef'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_renderArrow', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_renderArrow'), _class.prototype)), _class);


TxlContentEditableTarget.propTypes = {
  children: _react.PropTypes.any,
  clearSearch: _react.PropTypes.func.isRequired,
  closeDropdown: _react.PropTypes.func,
  disabled: _InputFieldBase2.default.propTypes.disabled,
  items: _react.PropTypes.array,
  label: _InputFieldBase2.default.propTypes.label,
  onKeyDown: _react.PropTypes.func,
  open: _react.PropTypes.bool,
  openDropdown: _react.PropTypes.func,
  placeholder: _react.PropTypes.string,
  search: _ContentEditable2.default.propTypes.value,
  secondaryText: _InputFieldBase2.default.propTypes.secondaryText,
  selectItem: _react.PropTypes.func,
  shouldInputWrap: _react.PropTypes.bool,
  updateSearch: _ContentEditable2.default.propTypes.onChange,
  validationState: _InputFieldBase2.default.propTypes.validationState,
  value: _react.PropTypes.oneOfType([_react.PropTypes.object, _react.PropTypes.array, _react.PropTypes.any]),
  valueMapper: _react.PropTypes.func
};
TxlContentEditableTarget.displayName = 'TxlContentEditableTarget';

var _default = TxlContentEditableTarget;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(CONTAINER_PROPS, 'CONTAINER_PROPS', 'src/dropdowns/_targets/ContentEditableTarget.jsx');

  __REACT_HOT_LOADER__.register(VALUE_INPUT_WRAPPER_STYLES, 'VALUE_INPUT_WRAPPER_STYLES', 'src/dropdowns/_targets/ContentEditableTarget.jsx');

  __REACT_HOT_LOADER__.register(MINIMUM_FIELD_HEIGHT, 'MINIMUM_FIELD_HEIGHT', 'src/dropdowns/_targets/ContentEditableTarget.jsx');

  __REACT_HOT_LOADER__.register(TxlContentEditableTarget, 'TxlContentEditableTarget', 'src/dropdowns/_targets/ContentEditableTarget.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/dropdowns/_targets/ContentEditableTarget.jsx');
}();

;
module.exports = exports['default'];
