// txl/dropdowns/_targets/SingleSelectTarget.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _theme = require('../../styles/theme');

var _helpers = require('../../styles/helpers');

var _ContentEditableTarget = require('./ContentEditableTarget');

var _ContentEditableTarget2 = _interopRequireDefault(_ContentEditableTarget);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var VALUE_STYLE = {
  lineHeight: 'normal',
  maxWidth: 'calc(100% - ' + (0, _helpers.gridUnits)(2) + ')',
  overflow: 'hidden',
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap'
};

var VALUE_STYLE_OPEN = (0, _extends3.default)({}, VALUE_STYLE, {
  backgroundColor: _theme.COLOR_ACCENT['200']
});

var getDisplayValue = function getDisplayValue(props) {
  return !props.search && props.value && props.valueMapper(props.value);
};

var TxlSingleSelectTarget = function TxlSingleSelectTarget(props) {
  var clearSearch = props.clearSearch,
      closeDropdown = props.closeDropdown,
      disabled = props.disabled,
      items = props.items,
      label = props.label,
      onKeyDown = props.onKeyDown,
      open = props.open,
      openDropdown = props.openDropdown,
      placeholder = props.placeholder,
      search = props.search,
      secondaryText = props.secondaryText,
      selectItem = props.selectItem,
      updateSearch = props.updateSearch,
      validationState = props.validationState,
      value = props.value,
      valueMapper = props.valueMapper,
      others = (0, _objectWithoutProperties3.default)(props, ['clearSearch', 'closeDropdown', 'disabled', 'items', 'label', 'onKeyDown', 'open', 'openDropdown', 'placeholder', 'search', 'secondaryText', 'selectItem', 'updateSearch', 'validationState', 'value', 'valueMapper']);

  var valueStyle = open ? VALUE_STYLE_OPEN : VALUE_STYLE;

  return _react2.default.createElement(
    _ContentEditableTarget2.default,
    (0, _extends3.default)({
      clearSearch: clearSearch,
      closeDropdown: closeDropdown,
      disabled: disabled,
      items: items,
      label: label,
      onKeyDown: onKeyDown,
      open: open,
      openDropdown: openDropdown,
      placeholder: !value && placeholder || undefined,
      search: search,
      secondaryText: secondaryText,
      selectItem: selectItem,
      updateSearch: updateSearch,
      validationState: validationState,
      value: search,
      valueMapper: valueMapper
    }, others),
    _react2.default.createElement(
      'div',
      { style: valueStyle },
      getDisplayValue(props)
    )
  );
};

TxlSingleSelectTarget.propTypes = {
  clearSearch: _ContentEditableTarget2.default.propTypes.clearSearch,
  closeDropdown: _react.PropTypes.func,
  disabled: _ContentEditableTarget2.default.propTypes.disabled,
  items: _ContentEditableTarget2.default.propTypes.items,
  label: _ContentEditableTarget2.default.propTypes.label,
  onKeyDown: _ContentEditableTarget2.default.propTypes.onKeyDown,
  open: _ContentEditableTarget2.default.propTypes.open,
  openDropdown: _ContentEditableTarget2.default.propTypes.openDropdown,
  placeholder: _ContentEditableTarget2.default.propTypes.placeholder,
  search: _ContentEditableTarget2.default.propTypes.search,
  secondaryText: _ContentEditableTarget2.default.propTypes.secondaryText,
  selectItem: _ContentEditableTarget2.default.propTypes.selectItem,
  updateSearch: _ContentEditableTarget2.default.propTypes.updateSearch,
  validationState: _ContentEditableTarget2.default.propTypes.validationState,
  value: _react.PropTypes.oneOfType([_react.PropTypes.object, _react.PropTypes.string]),
  valueMapper: _ContentEditableTarget2.default.propTypes.valueMapper
};

TxlSingleSelectTarget.displayName = 'TxlSingleSelectTarget';

var _default = TxlSingleSelectTarget;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(VALUE_STYLE, 'VALUE_STYLE', 'src/dropdowns/_targets/SingleSelectTarget.jsx');

  __REACT_HOT_LOADER__.register(VALUE_STYLE_OPEN, 'VALUE_STYLE_OPEN', 'src/dropdowns/_targets/SingleSelectTarget.jsx');

  __REACT_HOT_LOADER__.register(getDisplayValue, 'getDisplayValue', 'src/dropdowns/_targets/SingleSelectTarget.jsx');

  __REACT_HOT_LOADER__.register(TxlSingleSelectTarget, 'TxlSingleSelectTarget', 'src/dropdowns/_targets/SingleSelectTarget.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/dropdowns/_targets/SingleSelectTarget.jsx');
}();

;
module.exports = exports['default'];
