// txl/containers/_TourCardContainer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Container = require('../containers/Container');

var _Container2 = _interopRequireDefault(_Container);

var _BullhornOutline = require('../icons/BullhornOutline');

var _BullhornOutline2 = _interopRequireDefault(_BullhornOutline);

var _TourCardContainer = require('./_TourCardContainer.style');

var _type = require('../styles/mixins/type');

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ICON_SCALE = 1.3;

var TxlTourCardContainer = function TxlTourCardContainer(props) {
  var content = props.content,
      footer = props.footer,
      header = props.header,
      Icon = props.icon,
      onDismiss = props.onDismiss,
      width = props.width;

  return _react2.default.createElement(
    'div',
    {
      'data-component': 'TxlTourCardContainer',
      style: (0, _extends3.default)({
        width: width
      }, _type.DEFAULT_FONT_FAMILY)
    },
    _react2.default.createElement(
      _Container2.default,
      null,
      _react2.default.createElement(
        'header',
        {
          style: (0, _extends3.default)({}, _TourCardContainer.HEADER_STYLE, _TourCardContainer.BORDER)
        },
        _react2.default.createElement(
          'div',
          { style: _TourCardContainer.INNER_CONTENT },
          _react2.default.createElement(Icon, {
            scale: ICON_SCALE,
            color: _TourCardContainer.FONT_COLOR
          }),
          _react2.default.createElement(
            'span',
            { style: _TourCardContainer.HEADER_TEXT },
            header
          )
        )
      ),
      _react2.default.createElement(
        'main',
        null,
        _react2.default.createElement(
          'div',
          { style: _TourCardContainer.INNER_CONTENT },
          content
        )
      ),
      _react2.default.createElement(
        'footer',
        {
          style: (0, _extends3.default)({}, _TourCardContainer.INNER_CONTENT, _TourCardContainer.FOOTER_STYLE, {
            justifyContent: footer ? 'space-between' : 'flex-end'
          })
        },
        footer && _react2.default.createElement(
          'div',
          null,
          footer
        ),
        _react2.default.createElement(
          'a',
          {
            onClick: onDismiss,
            style: _TourCardContainer.DISMISS_TEXT,
            'data-test-dismiss': true
          },
          'Dismiss'
        )
      )
    )
  );
};
TxlTourCardContainer.defaultProps = {
  icon: _BullhornOutline2.default,
  onDismiss: _react.PropTypes.func.isRequired,
  width: (0, _helpers.gridUnits)(60)
};

var requiredStringOrNode = _react.PropTypes.oneOfType([_react.PropTypes.node, _react.PropTypes.string]).isRequired;

TxlTourCardContainer.propTypes = {
  content: requiredStringOrNode,
  footer: _react.PropTypes.oneOfType([_react.PropTypes.node, _react.PropTypes.string]),
  header: requiredStringOrNode,
  icon: _react.PropTypes.func,
  onDismiss: _react.PropTypes.func.isRequired,
  width: _react.PropTypes.oneOfType([_react.PropTypes.number, _react.PropTypes.string]).isRequired
};

var _default = (0, _radium2.default)(TxlTourCardContainer);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ICON_SCALE, 'ICON_SCALE', 'src/containers/_TourCardContainer.jsx');

  __REACT_HOT_LOADER__.register(TxlTourCardContainer, 'TxlTourCardContainer', 'src/containers/_TourCardContainer.jsx');

  __REACT_HOT_LOADER__.register(requiredStringOrNode, 'requiredStringOrNode', 'src/containers/_TourCardContainer.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/containers/_TourCardContainer.jsx');
}();

;
module.exports = exports['default'];
