// txl/containers/_TourCardContainer.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FOOTER_STYLE = exports.DISMISS_TEXT = exports.HEADER_TEXT = exports.BORDER = exports.INNER_CONTENT = exports.HEADER_STYLE = exports.FONT_COLOR = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _type = require('../styles/mixins/type');

var _theme = require('../styles/theme');

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FONT_COLOR = exports.FONT_COLOR = _theme.COLOR_NEUTRAL['400'];

var HEADER_STYLE = exports.HEADER_STYLE = (0, _extends3.default)({}, _type.HEADING_MEDIUM, _type.DEFAULT_FONT_FAMILY);

var INNER_CONTENT = exports.INNER_CONTENT = {
  color: FONT_COLOR,
  padding: (0, _helpers.gridUnits)(2) + ' ' + (0, _helpers.gridUnits)(4)
};

var BORDER = exports.BORDER = {
  borderBottom: '1px solid ' + _theme.COLOR_NEUTRAL['200']
};

var HEADER_TEXT = exports.HEADER_TEXT = {
  marginLeft: (0, _helpers.gridUnits)(2)
};

var DISMISS_TEXT = exports.DISMISS_TEXT = {
  color: _theme.COLOR_ACCENT['500'],
  cursor: 'pointer'
};

var FOOTER_STYLE = exports.FOOTER_STYLE = {
  alignItems: 'center',
  display: 'flex'
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(FONT_COLOR, 'FONT_COLOR', 'src/containers/_TourCardContainer.style.js');

  __REACT_HOT_LOADER__.register(HEADER_STYLE, 'HEADER_STYLE', 'src/containers/_TourCardContainer.style.js');

  __REACT_HOT_LOADER__.register(INNER_CONTENT, 'INNER_CONTENT', 'src/containers/_TourCardContainer.style.js');

  __REACT_HOT_LOADER__.register(BORDER, 'BORDER', 'src/containers/_TourCardContainer.style.js');

  __REACT_HOT_LOADER__.register(HEADER_TEXT, 'HEADER_TEXT', 'src/containers/_TourCardContainer.style.js');

  __REACT_HOT_LOADER__.register(DISMISS_TEXT, 'DISMISS_TEXT', 'src/containers/_TourCardContainer.style.js');

  __REACT_HOT_LOADER__.register(FOOTER_STYLE, 'FOOTER_STYLE', 'src/containers/_TourCardContainer.style.js');
}();

;
