// txl/containers/ExpandableContainer.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HEADER_STYLES = exports.CONTAINER_STYLES = exports.HEADER_TITLE_STYLES = exports.DESCRIPTION_STYLES = exports.ACTION_CONTAINER_STYLES = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

var _type = require('../styles/mixins/type');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ACTION_CONTAINER_STYLES = exports.ACTION_CONTAINER_STYLES = {
  display: 'flex',
  justifyContent: 'flex-end',
  minWidth: (0, _helpers.gridUnits)(26),
  paddingTop: (0, _helpers.gridUnits)(1)
};

var DESCRIPTION_STYLES = exports.DESCRIPTION_STYLES = (0, _extends3.default)({}, _type.PARAGRAPH, {
  color: _theme.COLOR_NEUTRAL['400']
});

var HEADER_TITLE_STYLES = exports.HEADER_TITLE_STYLES = (0, _extends3.default)({}, _type.HEADING_MEDIUM, {
  marginBottom: (0, _helpers.gridUnits)(2),
  paddingTop: (0, _helpers.gridUnits)(1)
});

var CONTAINER_STYLES = exports.CONTAINER_STYLES = (0, _extends3.default)({}, _type.DEFAULT_FONT_FAMILY, {
  boxSizing: 'border-box',
  paddingBottom: (0, _helpers.gridUnits)(4),
  paddingLeft: (0, _helpers.gridUnits)(4),
  paddingRight: (0, _helpers.gridUnits)(4),
  paddingTop: (0, _helpers.gridUnits)(4),
  width: '100%'
});

var HEADER_STYLES = exports.HEADER_STYLES = {
  display: 'flex',
  justifyContent: 'space-between'
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ACTION_CONTAINER_STYLES, 'ACTION_CONTAINER_STYLES', 'src/containers/ExpandableContainer.style.js');

  __REACT_HOT_LOADER__.register(DESCRIPTION_STYLES, 'DESCRIPTION_STYLES', 'src/containers/ExpandableContainer.style.js');

  __REACT_HOT_LOADER__.register(HEADER_TITLE_STYLES, 'HEADER_TITLE_STYLES', 'src/containers/ExpandableContainer.style.js');

  __REACT_HOT_LOADER__.register(CONTAINER_STYLES, 'CONTAINER_STYLES', 'src/containers/ExpandableContainer.style.js');

  __REACT_HOT_LOADER__.register(HEADER_STYLES, 'HEADER_STYLES', 'src/containers/ExpandableContainer.style.js');
}();

;
