// txl/containers/ExpandableContainer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _Button = require('../buttons/Button');

var _Button2 = _interopRequireDefault(_Button);

var _Container = require('../containers/Container');

var _Container2 = _interopRequireDefault(_Container);

var _ExpandableContainer = require('./ExpandableContainer.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlExpandableContainer = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlExpandableContainer, _React$Component);

  function TxlExpandableContainer() {
    (0, _classCallCheck3.default)(this, TxlExpandableContainer);
    return (0, _possibleConstructorReturn3.default)(this, (TxlExpandableContainer.__proto__ || (0, _getPrototypeOf2.default)(TxlExpandableContainer)).apply(this, arguments));
  }

  (0, _createClass3.default)(TxlExpandableContainer, [{
    key: 'componentWillMount',
    value: function componentWillMount() {
      this._setExpandedState(this.props.expanded || false);
    }
  }, {
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      if (nextProps.expanded === undefined) {
        return;
      }

      this._setExpandedState(nextProps.expanded);
    }
  }, {
    key: '_setExpandedState',
    value: function _setExpandedState(expanded) {
      this.setState({ expanded: !!expanded });
    }
  }, {
    key: '_handleExpandableContainerToggle',
    value: function _handleExpandableContainerToggle() {
      this.props.onToggle.apply(this, [{ expanded: !this.state.expanded }]);
    }
  }, {
    key: 'render',
    value: function render() {
      var buttonText = void 0;
      var content = void 0;

      if (this.state.expanded) {
        buttonText = 'Done';
        content = this.props.children;
      } else {
        buttonText = 'Edit';
        content = this.props.summary;
      }

      return _react2.default.createElement(
        _Container2.default,
        null,
        _react2.default.createElement(
          'div',
          {
            'data-component': 'TxlExpandableContainer',
            style: _ExpandableContainer.CONTAINER_STYLES
          },
          _react2.default.createElement(
            'div',
            { style: _ExpandableContainer.HEADER_STYLES },
            _react2.default.createElement(
              'div',
              null,
              _react2.default.createElement(
                'h2',
                {
                  'data-test-xcontainer-title': true,
                  style: _ExpandableContainer.HEADER_TITLE_STYLES
                },
                this.props.title
              ),
              _react2.default.createElement(
                'div',
                {
                  'data-test-xcontainer-description': true,
                  style: _ExpandableContainer.DESCRIPTION_STYLES
                },
                this.props.description
              )
            ),
            _react2.default.createElement(
              'div',
              { style: _ExpandableContainer.ACTION_CONTAINER_STYLES },
              _react2.default.createElement(
                _Button2.default,
                {
                  variant: 'muted',
                  size: 'large',
                  onClick: this._handleExpandableContainerToggle
                },
                buttonText
              )
            )
          ),
          _react2.default.createElement(
            'div',
            { 'data-test-xcontainer-content': true },
            content
          )
        )
      );
    }
  }]);
  return TxlExpandableContainer;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_setExpandedState', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_setExpandedState'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleExpandableContainerToggle', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleExpandableContainerToggle'), _class2.prototype)), _class2)) || _class;

exports.default = TxlExpandableContainer;


TxlExpandableContainer.defaultProps = {
  onToggle: function onToggle(obj) {
    this._setExpandedState(obj.expanded);
  }
};

TxlExpandableContainer.propTypes = {
  children: _react2.default.PropTypes.node,
  description: _react2.default.PropTypes.node,
  expanded: _react2.default.PropTypes.bool,
  onToggle: _react2.default.PropTypes.func,
  summary: _react2.default.PropTypes.node,
  title: _react2.default.PropTypes.node.isRequired
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlExpandableContainer, 'TxlExpandableContainer', 'src/containers/ExpandableContainer.jsx');
}();

;
module.exports = exports['default'];
