// txl/containers/SummaryContainer.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ITEM_CONTAINER_LAST = exports.ITEM_CONTAINER = exports.VALUE_STYLE = exports.HEADING_STYLE = exports.ITEMS_WRAPPER = exports.LOADING_WRAPPER = exports.MAIN_HEADING = exports.CONTAINER_STYLES = exports.CONTAINER_PADDING = exports.ITEM_GUTTER = exports.ITEM_GUTTER_RAW = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _type = require('../styles/mixins/type');

var _layers = require('../styles/mixins/layers');

var _theme = require('../styles/theme');

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ITEM_GUTTER_RAW = exports.ITEM_GUTTER_RAW = (0, _helpers.gridUnitsRaw)(6);
var ITEM_GUTTER = exports.ITEM_GUTTER = ITEM_GUTTER_RAW + 'px';
var CONTAINER_PADDING = exports.CONTAINER_PADDING = (0, _helpers.gridUnits)(4);

var CONTAINER_STYLES = exports.CONTAINER_STYLES = (0, _extends3.default)({}, _type.DEFAULT_FONT_FAMILY, _layers.LAYER_1, {
  backgroundColor: 'white',
  minHeight: (0, _helpers.gridUnits)(27),
  paddingBottom: 0,
  paddingLeft: 0,
  paddingRight: 0,
  paddingTop: CONTAINER_PADDING
});

var MAIN_HEADING = exports.MAIN_HEADING = (0, _extends3.default)({}, _type.HEADING_MEDIUM, {
  color: _theme.COLOR_NEUTRAL['800'],
  paddingBottom: 0,
  paddingLeft: CONTAINER_PADDING,
  paddingRight: CONTAINER_PADDING,
  paddingTop: 0
});

var LOADING_WRAPPER = exports.LOADING_WRAPPER = {
  margin: '0 auto'
};

var ITEMS_WRAPPER = exports.ITEMS_WRAPPER = {
  display: 'flex',
  overflow: 'auto',
  paddingBottom: CONTAINER_PADDING,
  paddingLeft: CONTAINER_PADDING,
  paddingRight: CONTAINER_PADDING,
  paddingTop: 0
};

var HEADING_STYLE = exports.HEADING_STYLE = (0, _extends3.default)({}, _type.TYPE_BASE, {
  borderBottom: '1px solid ' + _theme.COLOR_NEUTRAL['100'],
  color: _theme.COLOR_NEUTRAL['400'],
  lineHeight: (0, _helpers.gridUnits)(5), // overrides TYPE_BASE, as line-height was specified in spec
  textTransform: 'capitalize'
});

var VALUE_STYLE = exports.VALUE_STYLE = (0, _extends3.default)({}, _type.HEADING_MEDIUM, {
  color: _theme.COLOR_NEUTRAL['800']
});

var ITEM_CONTAINER = exports.ITEM_CONTAINER = {
  flex: '1 0 auto',
  marginRight: ITEM_GUTTER
};

var ITEM_CONTAINER_LAST = exports.ITEM_CONTAINER_LAST = {
  paddingRight: CONTAINER_PADDING
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ITEM_GUTTER_RAW, 'ITEM_GUTTER_RAW', 'src/containers/SummaryContainer.style.js');

  __REACT_HOT_LOADER__.register(ITEM_GUTTER, 'ITEM_GUTTER', 'src/containers/SummaryContainer.style.js');

  __REACT_HOT_LOADER__.register(CONTAINER_PADDING, 'CONTAINER_PADDING', 'src/containers/SummaryContainer.style.js');

  __REACT_HOT_LOADER__.register(CONTAINER_STYLES, 'CONTAINER_STYLES', 'src/containers/SummaryContainer.style.js');

  __REACT_HOT_LOADER__.register(MAIN_HEADING, 'MAIN_HEADING', 'src/containers/SummaryContainer.style.js');

  __REACT_HOT_LOADER__.register(LOADING_WRAPPER, 'LOADING_WRAPPER', 'src/containers/SummaryContainer.style.js');

  __REACT_HOT_LOADER__.register(ITEMS_WRAPPER, 'ITEMS_WRAPPER', 'src/containers/SummaryContainer.style.js');

  __REACT_HOT_LOADER__.register(HEADING_STYLE, 'HEADING_STYLE', 'src/containers/SummaryContainer.style.js');

  __REACT_HOT_LOADER__.register(VALUE_STYLE, 'VALUE_STYLE', 'src/containers/SummaryContainer.style.js');

  __REACT_HOT_LOADER__.register(ITEM_CONTAINER, 'ITEM_CONTAINER', 'src/containers/SummaryContainer.style.js');

  __REACT_HOT_LOADER__.register(ITEM_CONTAINER_LAST, 'ITEM_CONTAINER_LAST', 'src/containers/SummaryContainer.style.js');
}();

;
