// txl/containers/TileControlled.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TxlTileControlled = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Container = require('./Container');

var _Container2 = _interopRequireDefault(_Container);

var _Tile = require('./Tile.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var NOOP = function NOOP() {};

var TxlTileControlled = (_temp = _class = function (_React$Component) {
  (0, _inherits3.default)(TxlTileControlled, _React$Component);

  function TxlTileControlled(props) {
    (0, _classCallCheck3.default)(this, TxlTileControlled);

    var _this = (0, _possibleConstructorReturn3.default)(this, (TxlTileControlled.__proto__ || (0, _getPrototypeOf2.default)(TxlTileControlled)).call(this, props));

    _this.setHoverHeight = function () {
      return _this.__setHoverHeight__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this.handleSecondaryContentRef = function () {
      return _this.__handleSecondaryContentRef__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this.state = {
      hoverHeight: 0
    };
    return _this;
  }

  (0, _createClass3.default)(TxlTileControlled, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      this.setHoverHeight();
    }

    // if secondaryHoverContent changed, we need to update measured height

  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate(props) {
      if (props.secondaryHoverContent !== this.props.secondaryHoverContent) {
        this.setHoverHeight();
      }
    }
  }, {
    key: '__setHoverHeight__REACT_HOT_LOADER__',
    value: function __setHoverHeight__REACT_HOT_LOADER__() {
      return this.__setHoverHeight__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__setHoverHeight__REACT_HOT_LOADER__',
    value: function __setHoverHeight__REACT_HOT_LOADER__() {
      var h = this.secondaryContentRef.offsetHeight;
      this.setState({ hoverHeight: h });
    }
  }, {
    key: '__handleSecondaryContentRef__REACT_HOT_LOADER__',
    value: function __handleSecondaryContentRef__REACT_HOT_LOADER__() {
      return this.__handleSecondaryContentRef__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__handleSecondaryContentRef__REACT_HOT_LOADER__',
    value: function __handleSecondaryContentRef__REACT_HOT_LOADER__(node) {
      if (node) this.secondaryContentRef = node;
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          secondaryContent = _props.secondaryContent,
          secondaryHeight = _props.secondaryHeight,
          secondaryHoverContent = _props.secondaryHoverContent,
          children = _props.children,
          hovered = _props.hovered,
          imageURL = _props.imageURL,
          imageStyle = _props.imageStyle,
          containerStyle = _props.containerStyle,
          onMouseLeave = _props.onMouseLeave,
          onMouseOver = _props.onMouseOver;
      var hoverHeight = this.state.hoverHeight;


      var containerStyles = (0, _extends3.default)({}, _Tile.DEFAULT_CONTAINER, containerStyle);

      var imageWrapStyles = (0, _extends3.default)({}, _Tile.IMAGE_WRAP, {
        height: 'calc(100% - ' + secondaryHeight + 'px)' });

      var imageStyles = (0, _extends3.default)({}, _Tile.IMAGE, imageStyle);

      var overlayStyles = hovered ? _Tile.OVERLAY_SHOWN : _Tile.OVERLAY_HIDDEN;

      var transformBy = (hoverHeight - secondaryHeight) * -1;

      var secondaryContentWrapStyles = (0, _extends3.default)({}, _Tile.SECONDARY_CONTENT_WRAP, {
        maxHeight: hovered ? '100%' : secondaryHeight + 'px',
        transform: hovered && hoverHeight ? 'translateY(' + transformBy + 'px)' : 'none'
      });

      return _react2.default.createElement(
        _Container2.default,
        {
          onClick: NOOP // here only for iOS Safari to recognize this as interactive
          , onMouseLeave: onMouseLeave,
          onMouseOver: onMouseOver,
          style: containerStyles
        },
        _react2.default.createElement(
          'div',
          { style: imageWrapStyles },
          imageURL && _react2.default.createElement('img', {
            style: imageStyles,
            src: imageURL
          })
        ),
        _react2.default.createElement('div', { style: overlayStyles }),
        hovered && _react2.default.createElement(
          'div',
          { style: _Tile.ACTION_WRAP },
          children
        ),
        _react2.default.createElement(
          'div',
          { style: secondaryContentWrapStyles },
          _react2.default.createElement('div', { style: _Tile.SECONDARY_BG_FILL }),
          _react2.default.createElement(
            'div',
            { style: _Tile.SECONDARY_CONTENT },
            hovered && (secondaryHoverContent || secondaryContent),
            !hovered && secondaryContent
          ),
          _react2.default.createElement(
            'div',
            {
              style: _Tile.SECONDARY_MEASURING,
              ref: this.handleSecondaryContentRef
            },
            secondaryHoverContent
          )
        )
      );
    }
  }]);
  return TxlTileControlled;
}(_react2.default.Component), _class.defaultProps = {
  containerStyle: {},
  imageStyle: {},
  secondaryHeight: _Tile.DEFAULT_SECONDARY_HEIGHT
}, _temp);
TxlTileControlled.propTypes = {
  hovered: require('react').PropTypes.bool.isRequired,
  children: require('react').PropTypes.any,
  containerStyle: require('react').PropTypes.object,
  imageStyle: require('react').PropTypes.object,
  imageURL: require('react').PropTypes.string,
  onMouseLeave: require('react').PropTypes.func.isRequired,
  onMouseOver: require('react').PropTypes.func.isRequired,
  secondaryContent: require('react').PropTypes.any.isRequired,
  secondaryHeight: require('react').PropTypes.number,
  secondaryHoverContent: require('react').PropTypes.any.isRequired
};
exports.TxlTileControlled = TxlTileControlled;

var _default = (0, _radium2.default)(TxlTileControlled);

exports.default = _default;
;

var _temp2 = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(NOOP, 'NOOP', 'src/containers/TileControlled.jsx');

  __REACT_HOT_LOADER__.register(TxlTileControlled, 'TxlTileControlled', 'src/containers/TileControlled.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/containers/TileControlled.jsx');
}();

;
