// txl/containers/Tile.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ACTION_WRAP = exports.OVERLAY_SHOWN = exports.OVERLAY_HIDDEN = exports.SECONDARY_CONTENT = exports.SECONDARY_MEASURING = exports.DEFAULT_SECONDARY_HEIGHT = exports.SECONDARY_BG_FILL = exports.SECONDARY_CONTENT_WRAP = exports.IMAGE_WRAP = exports.IMAGE = exports.DEFAULT_CONTAINER = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DEFAULT_CONTAINER = exports.DEFAULT_CONTAINER = {
  height: (0, _helpers.gridUnits)(50),
  margin: (0, _helpers.gridUnits)(2),
  maxWidth: (0, _helpers.gridUnits)(60),
  minHeight: (0, _helpers.gridUnits)(50),
  minWidth: (0, _helpers.gridUnits)(60),
  overflow: 'hidden',
  position: 'relative'
};

var IMAGE = exports.IMAGE = {
  boxSizing: 'border-box',
  display: 'block',
  objectFit: 'contain',
  width: '100%'
};

var IMAGE_WRAP = exports.IMAGE_WRAP = {
  alignItems: 'center',
  display: 'flex',
  justifyContent: 'center'
};

var SECONDARY_CONTENT_WRAP = exports.SECONDARY_CONTENT_WRAP = {
  borderTop: '1px solid ' + _theme.COLOR_NEUTRAL['100'],
  position: 'relative',
  transition: 'transform 0.3s ease-in-out',
  zIndex: '3'
};

var SECONDARY_BG_FILL = exports.SECONDARY_BG_FILL = {
  backgroundColor: 'white',
  height: DEFAULT_CONTAINER.minHeight,
  position: 'absolute',
  top: '0',
  width: '100%',
  zIndex: '1'
};

var DEFAULT_SECONDARY_HEIGHT = exports.DEFAULT_SECONDARY_HEIGHT = (0, _helpers.gridUnitsRaw)(12); // 60px

var SECONDARY_SIZING = {
  boxSizing: 'border-box',
  padding: (0, _helpers.gridUnits)(4),
  width: '100%'
};

var SECONDARY_MEASURING = exports.SECONDARY_MEASURING = (0, _extends3.default)({}, SECONDARY_SIZING, {
  opacity: 0
});

var SECONDARY_CONTENT = exports.SECONDARY_CONTENT = (0, _extends3.default)({}, SECONDARY_SIZING, {
  alignItems: 'center',
  backgroundColor: 'white',
  display: 'flex',
  justifyContent: 'center',
  position: 'relative',
  zIndex: '2'
});

var OVERLAY_HIDDEN = exports.OVERLAY_HIDDEN = {
  backgroundColor: _theme.COLOR_NEUTRAL[900],
  height: '100%',
  opacity: '0',
  position: 'absolute',
  top: '0',
  transition: 'opacity 0.3s ease-in-out',
  width: '100%',
  zIndex: '2'
};

var OVERLAY_SHOWN = exports.OVERLAY_SHOWN = (0, _extends3.default)({}, OVERLAY_HIDDEN, {
  opacity: '0.8'
});

var ACTION_WRAP = exports.ACTION_WRAP = {
  alignItems: 'center',
  display: 'flex',
  justifyContent: 'center',
  position: 'absolute',
  top: '0',
  width: '100%',
  zIndex: '3'
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(DEFAULT_CONTAINER, 'DEFAULT_CONTAINER', 'src/containers/Tile.style.js');

  __REACT_HOT_LOADER__.register(IMAGE, 'IMAGE', 'src/containers/Tile.style.js');

  __REACT_HOT_LOADER__.register(IMAGE_WRAP, 'IMAGE_WRAP', 'src/containers/Tile.style.js');

  __REACT_HOT_LOADER__.register(SECONDARY_CONTENT_WRAP, 'SECONDARY_CONTENT_WRAP', 'src/containers/Tile.style.js');

  __REACT_HOT_LOADER__.register(SECONDARY_BG_FILL, 'SECONDARY_BG_FILL', 'src/containers/Tile.style.js');

  __REACT_HOT_LOADER__.register(DEFAULT_SECONDARY_HEIGHT, 'DEFAULT_SECONDARY_HEIGHT', 'src/containers/Tile.style.js');

  __REACT_HOT_LOADER__.register(SECONDARY_SIZING, 'SECONDARY_SIZING', 'src/containers/Tile.style.js');

  __REACT_HOT_LOADER__.register(SECONDARY_MEASURING, 'SECONDARY_MEASURING', 'src/containers/Tile.style.js');

  __REACT_HOT_LOADER__.register(SECONDARY_CONTENT, 'SECONDARY_CONTENT', 'src/containers/Tile.style.js');

  __REACT_HOT_LOADER__.register(OVERLAY_HIDDEN, 'OVERLAY_HIDDEN', 'src/containers/Tile.style.js');

  __REACT_HOT_LOADER__.register(OVERLAY_SHOWN, 'OVERLAY_SHOWN', 'src/containers/Tile.style.js');

  __REACT_HOT_LOADER__.register(ACTION_WRAP, 'ACTION_WRAP', 'src/containers/Tile.style.js');
}();

;
