// txl/containers/Tile.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _TileControlled = require('./TileControlled');

var _TileControlled2 = _interopRequireDefault(_TileControlled);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TxlTile = function (_React$Component) {
  (0, _inherits3.default)(TxlTile, _React$Component);

  function TxlTile(props) {
    (0, _classCallCheck3.default)(this, TxlTile);

    var _this = (0, _possibleConstructorReturn3.default)(this, (TxlTile.__proto__ || (0, _getPrototypeOf2.default)(TxlTile)).call(this, props));

    _this.onMouseOver = function () {
      return _this.__onMouseOver__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this.onMouseLeave = function () {
      return _this.__onMouseLeave__REACT_HOT_LOADER__.apply(_this, arguments);
    };

    _this.state = {
      hovered: false
    };
    return _this;
  }

  (0, _createClass3.default)(TxlTile, [{
    key: '__onMouseOver__REACT_HOT_LOADER__',
    value: function __onMouseOver__REACT_HOT_LOADER__() {
      return this.__onMouseOver__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__onMouseOver__REACT_HOT_LOADER__',
    value: function __onMouseOver__REACT_HOT_LOADER__() {
      this.setState({ hovered: true });
    }
  }, {
    key: '__onMouseLeave__REACT_HOT_LOADER__',
    value: function __onMouseLeave__REACT_HOT_LOADER__() {
      return this.__onMouseLeave__REACT_HOT_LOADER__.apply(this, arguments);
    }
  }, {
    key: '__onMouseLeave__REACT_HOT_LOADER__',
    value: function __onMouseLeave__REACT_HOT_LOADER__() {
      this.setState({ hovered: false });
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          children = _props.children,
          containerStyle = _props.containerStyle,
          imageStyle = _props.imageStyle,
          imageURL = _props.imageURL,
          secondaryContent = _props.secondaryContent,
          secondaryHeight = _props.secondaryHeight,
          secondaryHoverContent = _props.secondaryHoverContent;
      var hovered = this.state.hovered;


      return _react2.default.createElement(
        _TileControlled2.default,
        {
          hovered: hovered,
          containerStyle: containerStyle,
          imageStyle: imageStyle,
          imageURL: imageURL,
          onMouseOver: this.onMouseOver,
          onMouseLeave: this.onMouseLeave,
          secondaryHeight: secondaryHeight,
          secondaryContent: secondaryContent,
          secondaryHoverContent: secondaryHoverContent
        },
        children
      );
    }
  }]);
  return TxlTile;
}(_react2.default.Component);

TxlTile.propTypes = {
  children: require('react').PropTypes.any,
  containerStyle: require('react').PropTypes.object,
  imageStyle: require('react').PropTypes.object,
  imageURL: require('react').PropTypes.string,
  secondaryContent: require('react').PropTypes.any.isRequired,
  secondaryHeight: require('react').PropTypes.number,
  secondaryHoverContent: require('react').PropTypes.any.isRequired
};
var _default = TxlTile;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlTile, 'TxlTile', 'src/containers/Tile.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/containers/Tile.jsx');
}();

;
module.exports = exports['default'];
