// txl/containers/TitleContainer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Container = require('./Container');

var _Container2 = _interopRequireDefault(_Container);

var _helpers = require('../styles/helpers');

var _type = require('../styles/mixins/type');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var OUTER_CONTAINER_STYLES = {
  padding: (0, _helpers.gridUnits)(4)
};

// Total gutter width is 75px, account for the 20px of padding
var CONTAINER_WITH_MARGIN_STYLES = {
  marginLeft: (0, _helpers.gridUnits)(11)
};

var LEFT_MARGIN_CONTENT_STYLES = {
  alignItems: 'center',
  display: 'flex',
  justifyContent: 'center',
  // Offset the left margin content so it aligns with the left edge of the contianer.
  // This makes it easier to center the content in the 75px width
  marginLeft: '-' + (0, _helpers.gridUnits)(15),
  width: (0, _helpers.gridUnits)(15)
};

var RIGHT_MARGIN_CONTENT_STYLES = {
  alignItems: 'center',
  display: 'flex'
};

var TITLE_CONTAINER_STYLES = {
  display: 'flex',
  height: (0, _helpers.gridUnits)(7),
  justifyContent: 'center'
};

var TITLE_STYLES = (0, _extends3.default)({}, _type.HEADING_MEDIUM, {
  flexGrow: 1
});

var CONTENT_CONTAINER_STYLES = {
  marginTop: (0, _helpers.gridUnits)(2)
};

var FOOTER_STYLES = {
  marginTop: (0, _helpers.gridUnits)(2)
};

var TxlTitleContainer = function TxlTitleContainer(props) {
  var title = props.title,
      footer = props.footer,
      leftMarginContent = props.leftMarginContent,
      rightMarginContent = props.rightMarginContent,
      style = props.style;


  var containerStyles = (0, _extends3.default)({}, !!leftMarginContent && CONTAINER_WITH_MARGIN_STYLES);

  var outerContainerStyles = (0, _extends3.default)({}, OUTER_CONTAINER_STYLES, style);

  return _react2.default.createElement(
    _Container2.default,
    { style: outerContainerStyles },
    _react2.default.createElement(
      'div',
      {
        style: containerStyles,
        'data-component': 'TxlTitleContainer'
      },
      _react2.default.createElement(
        'div',
        {
          style: TITLE_CONTAINER_STYLES,
          'data-test-title-container-title': true
        },
        !!leftMarginContent && _react2.default.createElement(
          'div',
          {
            'data-test-title-container-left-margin': true,
            style: LEFT_MARGIN_CONTENT_STYLES
          },
          leftMarginContent
        ),
        _react2.default.createElement(
          'div',
          {
            'data-test-title-container-title-content': true,
            style: TITLE_STYLES
          },
          title
        ),
        !!rightMarginContent && _react2.default.createElement(
          'div',
          {
            'data-test-title-container-right-margin': true,
            style: RIGHT_MARGIN_CONTENT_STYLES
          },
          rightMarginContent
        )
      ),
      _react2.default.createElement(
        'div',
        {
          'data-test-title-container-content': true,
          style: CONTENT_CONTAINER_STYLES
        },
        props.children
      ),
      !!footer && _react2.default.createElement(
        'div',
        {
          'data-test-title-container-footer': true,
          style: FOOTER_STYLES
        },
        footer
      )
    )
  );
};

TxlTitleContainer.displayName = 'TxlTitleContainer';

TxlTitleContainer.propTypes = {
  children: _react.PropTypes.node,
  footer: _react.PropTypes.node,
  leftMarginContent: _react.PropTypes.node,
  rightMarginContent: _react.PropTypes.node,
  style: _react.PropTypes.object,
  title: _react.PropTypes.node
};

var _default = (0, _radium2.default)(TxlTitleContainer);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(OUTER_CONTAINER_STYLES, 'OUTER_CONTAINER_STYLES', 'src/containers/TitleContainer.jsx');

  __REACT_HOT_LOADER__.register(CONTAINER_WITH_MARGIN_STYLES, 'CONTAINER_WITH_MARGIN_STYLES', 'src/containers/TitleContainer.jsx');

  __REACT_HOT_LOADER__.register(LEFT_MARGIN_CONTENT_STYLES, 'LEFT_MARGIN_CONTENT_STYLES', 'src/containers/TitleContainer.jsx');

  __REACT_HOT_LOADER__.register(RIGHT_MARGIN_CONTENT_STYLES, 'RIGHT_MARGIN_CONTENT_STYLES', 'src/containers/TitleContainer.jsx');

  __REACT_HOT_LOADER__.register(TITLE_CONTAINER_STYLES, 'TITLE_CONTAINER_STYLES', 'src/containers/TitleContainer.jsx');

  __REACT_HOT_LOADER__.register(TITLE_STYLES, 'TITLE_STYLES', 'src/containers/TitleContainer.jsx');

  __REACT_HOT_LOADER__.register(CONTENT_CONTAINER_STYLES, 'CONTENT_CONTAINER_STYLES', 'src/containers/TitleContainer.jsx');

  __REACT_HOT_LOADER__.register(FOOTER_STYLES, 'FOOTER_STYLES', 'src/containers/TitleContainer.jsx');

  __REACT_HOT_LOADER__.register(TxlTitleContainer, 'TxlTitleContainer', 'src/containers/TitleContainer.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/containers/TitleContainer.jsx');
}();

;
module.exports = exports['default'];
