// txl/_data-grid/_ColumnModelPropType.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var propTypes = _react.PropTypes.shape({
  cell: _react.PropTypes.func.isRequired,
  cellStyle: _react.PropTypes.oneOfType([_react.PropTypes.func, _react.PropTypes.object]),
  growRatio: _react.PropTypes.number,
  header: _react.PropTypes.oneOfType([_react.PropTypes.node, _react.PropTypes.func]).isRequired,
  headerCentered: _react.PropTypes.bool,
  headerSimple: _react.PropTypes.string, // FIXME: unused?
  key: _react.PropTypes.string.isRequired,
  maxWidth: _react.PropTypes.number,
  minWidth: _react.PropTypes.number,
  pinned: _react.PropTypes.bool,
  resizeable: _react.PropTypes.bool, // FIXME: unused?
  sortType: _react.PropTypes.string, // required when passing sortable
  width: _react.PropTypes.number
});

// Flow types for DataGrid

/* global React$Element:false, React$Component:false */

Object.defineProperty(module.exports, 'babelPluginFlowReactPropTypes_proptype_HeaderProps', require('react').PropTypes.shape({
  model: require('react').PropTypes.any.isRequired
}));
Object.defineProperty(module.exports, 'babelPluginFlowReactPropTypes_proptype_RowProps', require('react').PropTypes.shape({
  row: require('react').PropTypes.object.isRequired
}));
Object.defineProperty(module.exports, 'babelPluginFlowReactPropTypes_proptype_Column', require('react').PropTypes.shape({
  cell: require('react').PropTypes.func.isRequired,
  cellStyle: require('react').PropTypes.oneOfType([require('react').PropTypes.object, require('react').PropTypes.func]),
  growRatio: require('react').PropTypes.number,
  header: require('react').PropTypes.any.isRequired,
  headerCentered: require('react').PropTypes.bool,
  headerSimple: require('react').PropTypes.string,
  key: require('react').PropTypes.string.isRequired,
  maxWidth: require('react').PropTypes.number,
  minWidth: require('react').PropTypes.number,
  pinned: require('react').PropTypes.bool,
  resizeable: require('react').PropTypes.bool,
  sortType: require('react').PropTypes.string,
  width: require('react').PropTypes.number
}));
Object.defineProperty(module.exports, 'babelPluginFlowReactPropTypes_proptype_ColumnGroup', require('react').PropTypes.shape({
  columns: require('react').PropTypes.arrayOf(require('react').PropTypes.shape({
    cell: require('react').PropTypes.func.isRequired,
    cellStyle: require('react').PropTypes.oneOfType([require('react').PropTypes.object, require('react').PropTypes.func]),
    growRatio: require('react').PropTypes.number,
    header: require('react').PropTypes.any.isRequired,
    headerCentered: require('react').PropTypes.bool,
    headerSimple: require('react').PropTypes.string,
    key: require('react').PropTypes.string.isRequired,
    maxWidth: require('react').PropTypes.number,
    minWidth: require('react').PropTypes.number,
    pinned: require('react').PropTypes.bool,
    resizeable: require('react').PropTypes.bool,
    sortType: require('react').PropTypes.string,
    width: require('react').PropTypes.number
  })).isRequired,
  header: require('react').PropTypes.any,
  pinned: require('react').PropTypes.bool
}));
var _default = propTypes;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(propTypes, 'propTypes', 'src/_data-grid/_ColumnModelPropType.js');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/_data-grid/_ColumnModelPropType.js');
}();

;
module.exports = exports['default'];
