// txl/_data-grid/_DataGridRowContainer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _defineProperty2 = require('babel-runtime/helpers/defineProperty');

var _defineProperty3 = _interopRequireDefault(_defineProperty2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _ReactUtils = require('../ReactUtils');

var _DataGridStyles = require('./DataGridStyles');

var _attributes = require('./_attributes');

var _Base = require('../base/Base');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlDataGridRowContainer = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlDataGridRowContainer, _React$Component);

  function TxlDataGridRowContainer() {
    (0, _classCallCheck3.default)(this, TxlDataGridRowContainer);
    return (0, _possibleConstructorReturn3.default)(this, (TxlDataGridRowContainer.__proto__ || (0, _getPrototypeOf2.default)(TxlDataGridRowContainer)).apply(this, arguments));
  }

  (0, _createClass3.default)(TxlDataGridRowContainer, [{
    key: '_updateHeights',
    value: function _updateHeights() {
      if (!this.refs.pinnedColumns) return;
      var unpinnedContainer = (0, _ReactUtils.findDOMNode)(this.refs.unpinnedColumns);
      var pinnedContainer = (0, _ReactUtils.findDOMNode)(this.refs.pinnedColumns);
      var pinnedContainerHeight = pinnedContainer.getBoundingClientRect().height;
      var unpinnedContainerHeight = unpinnedContainer.getBoundingClientRect().height;
      var maxHeight = Math.max(pinnedContainerHeight, unpinnedContainerHeight);
      var changedNode = void 0;

      // only set the height of the shortest container
      if (pinnedContainerHeight > unpinnedContainerHeight) {
        changedNode = unpinnedContainer;
      } else {
        changedNode = pinnedContainer;
      }
      changedNode.style.height = maxHeight + 'px';
    }
  }, {
    key: 'componentDidMount',
    value: function componentDidMount() {
      this._updateHeights();
    }
  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate() {
      this._updateHeights();
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          index = _props.index,
          pinnedCells = _props.pinnedCells,
          unpinnedCells = _props.unpinnedCells,
          style = _props.style,
          componentName = _props.componentName;

      var PINNED_ATTR = (0, _defineProperty3.default)({}, _attributes.PINNED_WRAPPER_ATTR, true);
      var UNPINNED_ATTR = (0, _defineProperty3.default)({}, _attributes.UNPINNED_WRAPPER_ATTR, true);

      return _react2.default.createElement(
        'div',
        {
          'data-component': componentName,
          'data-row': index,
          style: style,
          key: index
        },
        !!pinnedCells.length && _react2.default.createElement(
          'div',
          (0, _extends3.default)({}, PINNED_ATTR, {
            style: (0, _extends3.default)({}, _DataGridStyles.CELL_WRAPPER, {
              left: 0,
              position: 'absolute'
            }),
            ref: 'pinnedColumns'
          }),
          pinnedCells
        ),
        _react2.default.createElement(
          'div',
          (0, _extends3.default)({}, UNPINNED_ATTR, {
            style: _DataGridStyles.CELL_WRAPPER,
            ref: 'unpinnedColumns'
          }),
          unpinnedCells
        )
      );
    }
  }]);
  return TxlDataGridRowContainer;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_updateHeights', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_updateHeights'), _class2.prototype)), _class2)) || _class;

exports.default = TxlDataGridRowContainer;


TxlDataGridRowContainer.propTypes = {
  componentName: _react.PropTypes.string.isRequired,
  index: _react.PropTypes.number,
  pinnedCells: _react.PropTypes.arrayOf(_react.PropTypes.node),
  style: _react.PropTypes.obj,
  unpinnedCells: _react.PropTypes.arrayOf(_react.PropTypes.node)
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlDataGridRowContainer, 'TxlDataGridRowContainer', 'src/_data-grid/_DataGridRowContainer.jsx');
}();

;
module.exports = exports['default'];
