// txl/_data-grid/_DataGridRowSizer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _debounce = require('lodash/debounce');

var _debounce2 = _interopRequireDefault(_debounce);

var _ReactUtils = require('../ReactUtils');

var _Base = require('../base/Base');

var _DataGridStyles = require('./DataGridStyles');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var DEFAULT_COLUMN_WIDTH = 100;
var DEFAULT_GROW_RATIO = 0;
var DEFAULT_SHRINK_RATIO = 0;
var DEFAULT_MIN_WIDTH = 30;
var DEFAULT_MAX_WIDTH = 'none';

var DATA_GRID_ROW_SIZER_STYLES = (0, _extends3.default)({}, _DataGridStyles.BODY_ROW_STYLES, { height: '0px' });

var TxlDataGridRowSizer = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlDataGridRowSizer, _React$Component);

  function TxlDataGridRowSizer() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlDataGridRowSizer);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlDataGridRowSizer.__proto__ || (0, _getPrototypeOf2.default)(TxlDataGridRowSizer)).call.apply(_ref, [this].concat(args)));

    _this._debouncedOnSizeUpdate = (0, _debounce2.default)(_this._onSizeUpdate, 70);
    return _this;
  }

  (0, _createClass3.default)(TxlDataGridRowSizer, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      this._onSizeUpdate();
      window.addEventListener('resize', this._debouncedOnSizeUpdate);
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      window.removeEventListener('resize', this._debouncedOnSizeUpdate);
    }
  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate() {
      this._onSizeUpdate();
    }
  }, {
    key: '_onSizeUpdate',
    value: function _onSizeUpdate() {
      var _this2 = this;

      var widths = this.props.columnModel.map(function (model, index) {
        return (0, _ReactUtils.findDOMNode)(_this2.refs['column-' + index]).offsetWidth;
      });
      this.props.onRowSizeChange(widths);
    }
  }, {
    key: 'render',
    value: function render() {
      var columnModel = this.props.columnModel;


      var cells = columnModel.map(function (model, index) {
        var styles = {
          flexBasis: model.width || DEFAULT_COLUMN_WIDTH,
          flexGrow: model.growRatio || DEFAULT_GROW_RATIO,
          flexShrink: DEFAULT_SHRINK_RATIO,
          maxWidth: model.maxWidth || DEFAULT_MAX_WIDTH,
          minWidth: model.minWidth || DEFAULT_MIN_WIDTH
        };

        return _react2.default.createElement('div', {
          style: styles,
          key: model.key,
          ref: 'column-' + index
        });
      });

      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlDataGridRowSizer',
          style: DATA_GRID_ROW_SIZER_STYLES
        },
        cells
      );
    }
  }]);
  return TxlDataGridRowSizer;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_onSizeUpdate', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_onSizeUpdate'), _class2.prototype)), _class2)) || _class;

exports.default = TxlDataGridRowSizer;


TxlDataGridRowSizer.propTypes = {
  columnModel: _react.PropTypes.any.isRequired, // is validated higher up
  onRowSizeChange: _react.PropTypes.func.isRequired
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(DEFAULT_COLUMN_WIDTH, 'DEFAULT_COLUMN_WIDTH', 'src/_data-grid/_DataGridRowSizer.jsx');

  __REACT_HOT_LOADER__.register(DEFAULT_GROW_RATIO, 'DEFAULT_GROW_RATIO', 'src/_data-grid/_DataGridRowSizer.jsx');

  __REACT_HOT_LOADER__.register(DEFAULT_SHRINK_RATIO, 'DEFAULT_SHRINK_RATIO', 'src/_data-grid/_DataGridRowSizer.jsx');

  __REACT_HOT_LOADER__.register(DEFAULT_MIN_WIDTH, 'DEFAULT_MIN_WIDTH', 'src/_data-grid/_DataGridRowSizer.jsx');

  __REACT_HOT_LOADER__.register(DEFAULT_MAX_WIDTH, 'DEFAULT_MAX_WIDTH', 'src/_data-grid/_DataGridRowSizer.jsx');

  __REACT_HOT_LOADER__.register(DATA_GRID_ROW_SIZER_STYLES, 'DATA_GRID_ROW_SIZER_STYLES', 'src/_data-grid/_DataGridRowSizer.jsx');

  __REACT_HOT_LOADER__.register(TxlDataGridRowSizer, 'TxlDataGridRowSizer', 'src/_data-grid/_DataGridRowSizer.jsx');
}();

;
module.exports = exports['default'];
