// txl/_data-grid/_DataGridHeader.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _defineProperty2 = require('babel-runtime/helpers/defineProperty');

var _defineProperty3 = _interopRequireDefault(_defineProperty2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _helpers = require('../styles/helpers');

var _DataGridStyles = require('./DataGridStyles');

var _attributes = require('./_attributes');

var _DataGridRowContainer = require('./_DataGridRowContainer');

var _DataGridRowContainer2 = _interopRequireDefault(_DataGridRowContainer);

var _SortDropdown = require('./_SortDropdown');

var _SortDropdown2 = _interopRequireDefault(_SortDropdown);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function getCell(_ref) {
  var last = _ref.last,
      model = _ref.model,
      pinned = _ref.pinned,
      props = _ref.props;

  var header = typeof model.header === 'function' ? model.header({ model: model }) : model.header;
  var dropdownHeader = model.simpleHeader || header;
  var _onSortRequested = props.onSortRequested,
      sortedDirection = props.sortedDirection,
      sortedKey = props.sortedKey;


  var cellStyles = (0, _helpers.combineStyles)(_DataGridStyles.HEADER_CELL_STYLES, pinned && _DataGridStyles.CELL_PINNED_STYLES, last && _DataGridStyles.CELL_PINNED_LAST);

  var PINNED_ATTR = (0, _defineProperty3.default)({}, _attributes.PINNED_COLUMN_ATTR, pinned);

  return _react2.default.createElement(
    'div',
    (0, _extends3.default)({}, PINNED_ATTR, {
      key: model.key,
      'data-column': model.key,
      style: cellStyles
    }),
    _react2.default.createElement(
      'div',
      { style: _DataGridStyles.HEADER_TEXT_STYLES },
      header
    ),
    !!model.sortType && _react2.default.createElement(
      'span',
      null,
      _react2.default.createElement(_SortDropdown2.default, {
        sortByName: dropdownHeader,
        type: model.sortType,
        direction: model.key === sortedKey ? sortedDirection : undefined,
        onSortRequested: function onSortRequested(obj) {
          return _onSortRequested({ direction: obj.direction, key: model.key });
        }
      })
    )
  );
} /* eslint react/prop-types: "off" */

var TxlDataGridRow = function TxlDataGridRow(props) {
  var columnModel = props.columnModel;


  var pinnedCells = columnModel.filter(function (model) {
    return model.pinned;
  }).map(function (model, index, arr) {
    return getCell({
      last: index === arr.length - 1,
      model: model,
      pinned: true,
      props: undefined.props
    });
  });

  var unpinnedCells = columnModel.filter(function (model) {
    return !model.pinned;
  }).map(function (model, index, arr) {
    return getCell({
      last: index === arr.length - 1,
      model: model,
      props: undefined.props
    });
  });

  return _react2.default.createElement(_DataGridRowContainer2.default, {
    componentName: 'TxlDataGridHeader',
    style: _DataGridStyles.HEADER_ROW_STYLES,
    pinnedCells: pinnedCells,
    unpinnedCells: unpinnedCells,
    index: -1
  });
};

TxlDataGridRow.propTypes = {
  columnModel: _react.PropTypes.any.isRequired, // is validated higher up
  onSortRequested: _SortDropdown2.default.propTypes.onSortRequested,
  sortedDirection: _SortDropdown2.default.propTypes.direction,
  sortedKey: _react.PropTypes.string
};

var _default = (0, _radium2.default)(TxlDataGridRow);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(getCell, 'getCell', 'src/_data-grid/_DataGridHeader.jsx');

  __REACT_HOT_LOADER__.register(TxlDataGridRow, 'TxlDataGridRow', 'src/_data-grid/_DataGridHeader.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/_data-grid/_DataGridHeader.jsx');
}();

;
module.exports = exports['default'];
