// txl/_data-grid/_DataGridHeaderCell.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _defineProperty2 = require('babel-runtime/helpers/defineProperty');

var _defineProperty3 = _interopRequireDefault(_defineProperty2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _helpers = require('../styles/helpers');

var _DataGridStyles = require('./DataGridStyles');

var _attributes = require('./_attributes');

var _SortDropdown = require('./_SortDropdown');

var _SortDropdown2 = _interopRequireDefault(_SortDropdown);

var _ColumnModelPropType = require('./_ColumnModelPropType');

var _ColumnModelPropType2 = _interopRequireDefault(_ColumnModelPropType);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DataGridHeaderCell = function DataGridHeaderCell(props) {
  var sortedKey = props.sortedKey,
      sortedDirection = props.sortedDirection,
      _onSortRequested = props.onSortRequested,
      model = props.model,
      last = props.last;

  // model.header can be either be a function that returns a renderer or a string

  var header = typeof model.header === 'function' ? model.header({ model: model }) : model.header;

  var cellStyles = (0, _helpers.combineStyles)(_DataGridStyles.HEADER_CELL_STYLES, model.headerCentered && _DataGridStyles.HEADER_CELL_CENTERED_STYLES, model.pinned && _DataGridStyles.CELL_PINNED_STYLES, last && _DataGridStyles.CELL_PINNED_LAST);

  var PINNED_ATTR = (0, _defineProperty3.default)({}, _attributes.PINNED_COLUMN_ATTR, model.pinned);

  return _react2.default.createElement(
    'div',
    (0, _extends3.default)({}, PINNED_ATTR, {
      key: model.key,
      'data-column': model.key,
      style: cellStyles
    }),
    _react2.default.createElement(
      'div',
      { style: _DataGridStyles.HEADER_TEXT_STYLES },
      header
    ),
    !!model.sortType && _react2.default.createElement(
      'span',
      null,
      _react2.default.createElement(_SortDropdown2.default, {
        type: model.sortType,
        direction: model.key === sortedKey ? sortedDirection : undefined,
        onSortRequested: function onSortRequested(obj) {
          return _onSortRequested({ direction: obj.direction, key: model.key });
        }
      })
    )
  );
};

DataGridHeaderCell.propTypes = {
  last: _react.PropTypes.bool,
  model: _ColumnModelPropType2.default,
  onSortRequested: _react.PropTypes.func,
  sortedDirection: _SortDropdown2.default.propTypes.direction,
  sortedKey: _react.PropTypes.string
};

var _default = (0, _radium2.default)(DataGridHeaderCell);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(DataGridHeaderCell, 'DataGridHeaderCell', 'src/_data-grid/_DataGridHeaderCell.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/_data-grid/_DataGridHeaderCell.jsx');
}();

;
module.exports = exports['default'];
