// txl/_data-grid/_DataGridBodyCell.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _defineProperty2 = require('babel-runtime/helpers/defineProperty');

var _defineProperty3 = _interopRequireDefault(_defineProperty2);

var _typeof2 = require('babel-runtime/helpers/typeof');

var _typeof3 = _interopRequireDefault(_typeof2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _cloneDeep = require('lodash/cloneDeep');

var _cloneDeep2 = _interopRequireDefault(_cloneDeep);

var _helpers = require('../styles/helpers');

var _DataGridStyles = require('./DataGridStyles');

var _attributes = require('./_attributes');

var _ColumnModelPropType = require('./_ColumnModelPropType');

var _ColumnModelPropType2 = _interopRequireDefault(_ColumnModelPropType);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var getHoverStyle = function getHoverStyle(customBackgroundColor) {
  if (customBackgroundColor) {
    return (0, _extends3.default)({}, _DataGridStyles.BODY_CELL_HOVER_STYLES, {
      backgroundColor: customBackgroundColor,
      filter: 'brightness(95%)'
    });
  }

  return _DataGridStyles.BODY_CELL_HOVER_STYLES;
};

var TxlDataGridBodyCell = function TxlDataGridBodyCell(props) {
  var model = props.model,
      record = props.record,
      index = props.index,
      last = props.last,
      lastRow = props.lastRow,
      isHovering = props.isHovering;
  var cell = model.cell,
      cellStyle = model.cellStyle,
      key = model.key;


  var individualCellStyle = {};
  if (typeof cellStyle === 'function') {
    individualCellStyle = cellStyle({ index: index, model: model, row: record });
  } else if ((typeof cellStyle === 'undefined' ? 'undefined' : (0, _typeof3.default)(cellStyle)) === 'object') {
    individualCellStyle = cellStyle;
  }

  var cellStyles = (0, _helpers.combineStyles)(_DataGridStyles.BODY_CELL_STYLES, model.pinned && _DataGridStyles.CELL_PINNED_STYLES, lastRow && _DataGridStyles.BODY_CELL_LAST_ROW_STYLES, last && _DataGridStyles.CELL_PINNED_LAST, individualCellStyle, isHovering && getHoverStyle(individualCellStyle.backgroundColor));

  var args = (0, _cloneDeep2.default)({
    index: index,
    model: model,
    row: record
  });

  var PINNED_ATTR = (0, _defineProperty3.default)({}, _attributes.PINNED_COLUMN_ATTR, model.pinned);

  return _react2.default.createElement(
    'div',
    (0, _extends3.default)({}, PINNED_ATTR, {
      key: key,
      'data-column': key,
      style: cellStyles
    }),
    cell(args)
  );
};

TxlDataGridBodyCell.propTypes = {
  index: _react.PropTypes.number,
  isHovering: _react.PropTypes.bool,
  last: _react.PropTypes.bool,
  lastRow: _react.PropTypes.oneOfType([_react.PropTypes.object, _react.PropTypes.bool]),
  model: _ColumnModelPropType2.default,
  record: _react.PropTypes.any.isRequired
};

var _default = (0, _radium2.default)(TxlDataGridBodyCell);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(getHoverStyle, 'getHoverStyle', 'src/_data-grid/_DataGridBodyCell.jsx');

  __REACT_HOT_LOADER__.register(TxlDataGridBodyCell, 'TxlDataGridBodyCell', 'src/_data-grid/_DataGridBodyCell.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/_data-grid/_DataGridBodyCell.jsx');
}();

;
module.exports = exports['default'];
