// txl/_data-grid/_DataGridBodyRow.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _DataGridStyles = require('./DataGridStyles');

var _DataGridRow = require('./_DataGridRow');

var _DataGridRow2 = _interopRequireDefault(_DataGridRow);

var _DataGridBodyCell = require('./_DataGridBodyCell');

var _DataGridBodyCell2 = _interopRequireDefault(_DataGridBodyCell);

var _ColumnModelPropType = require('./_ColumnModelPropType');

var _ColumnModelPropType2 = _interopRequireDefault(_ColumnModelPropType);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TxlDataGridBodyRow = function TxlDataGridBodyRow(props) {
  var columnModel = props.columnModel,
      index = props.index,
      lastRow = props.lastRow,
      record = props.record;


  return _react2.default.createElement(_DataGridRow2.default, {
    componentName: 'TxlDataGridBodyRow',
    style: _DataGridStyles.BODY_ROW_STYLES,
    columnModel: columnModel,
    cellRenderer: _DataGridBodyCell2.default,
    record: record,
    index: index,
    lastRow: lastRow
  });
};

TxlDataGridBodyRow.propTypes = {
  columnModel: _react.PropTypes.arrayOf(_ColumnModelPropType2.default).isRequired,
  index: _react.PropTypes.number,
  lastRow: _react.PropTypes.bool,
  record: _react.PropTypes.any.isRequired
};

var _default = (0, _radium2.default)(TxlDataGridBodyRow);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlDataGridBodyRow, 'TxlDataGridBodyRow', 'src/_data-grid/_DataGridBodyRow.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/_data-grid/_DataGridBodyRow.jsx');
}();

;
module.exports = exports['default'];
