// txl/_data-grid/_DataGridHeaderRow.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _DataGridStyles = require('./DataGridStyles');

var _DataGridRow = require('./_DataGridRow');

var _DataGridRow2 = _interopRequireDefault(_DataGridRow);

var _DataGridHeaderCell = require('./_DataGridHeaderCell');

var _DataGridHeaderCell2 = _interopRequireDefault(_DataGridHeaderCell);

var _SortDropdown = require('./_SortDropdown');

var _SortDropdown2 = _interopRequireDefault(_SortDropdown);

var _ColumnModelPropType = require('./_ColumnModelPropType');

var _ColumnModelPropType2 = _interopRequireDefault(_ColumnModelPropType);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TxlDataGridHeaderRow = function TxlDataGridHeaderRow(props) {
  var columnModel = props.columnModel,
      sortedKey = props.sortedKey,
      sortedDirection = props.sortedDirection,
      onHeightUpdated = props.onHeightUpdated,
      onSortRequested = props.onSortRequested;


  return _react2.default.createElement(_DataGridRow2.default, {
    cellRenderer: _DataGridHeaderCell2.default,
    columnModel: columnModel,
    componentName: 'TxlDataGridHeaderRow',
    index: -1,
    onHeightUpdated: onHeightUpdated,
    onSortRequested: onSortRequested,
    sortedDirection: sortedDirection,
    sortedKey: sortedKey,
    style: _DataGridStyles.HEADER_ROW_STYLES
  });
};

TxlDataGridHeaderRow.propTypes = {
  columnModel: _react.PropTypes.arrayOf(_ColumnModelPropType2.default).isRequired,
  onHeightUpdated: _react.PropTypes.func,
  onSortRequested: _react.PropTypes.func,
  sortedDirection: _SortDropdown2.default.propTypes.direction,
  sortedKey: _react.PropTypes.string
};

var _default = (0, _radium2.default)(TxlDataGridHeaderRow);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlDataGridHeaderRow, 'TxlDataGridHeaderRow', 'src/_data-grid/_DataGridHeaderRow.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/_data-grid/_DataGridHeaderRow.jsx');
}();

;
module.exports = exports['default'];
