// txl/_data-grid/_DataGridHeaderGroupCell.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _DataGridStyles = require('./DataGridStyles');

var _DataGridHeaderRow = require('./_DataGridHeaderRow');

var _DataGridHeaderRow2 = _interopRequireDefault(_DataGridHeaderRow);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function DataGridHeaderGroupCell(props) {
  var model = props.model,
      _props$model = props.model,
      columns = _props$model.columns,
      header = _props$model.header,
      key = _props$model.key;

  // model.header can be either be a function that returns a renderer or a string

  var headerContent = typeof header === 'function' ? header({ model: model }) : header;

  return _react2.default.createElement(
    'div',
    {
      'data-column': key,
      key: key,
      style: _DataGridStyles.HEADER_CELL_GROUP_STYLES
    },
    _react2.default.createElement(
      'div',
      { style: _DataGridStyles.HEADER_GROUP_NAME_STYLES },
      headerContent
    ),
    columns && _react2.default.createElement(
      'div',
      { style: _DataGridStyles.HEADER_GROUP_NESTED_ROW_WRAPPER },
      _react2.default.createElement(_DataGridHeaderRow2.default, {
        columnModel: columns,
        isHeaderGroup: true,
        sortedKey: props.sortedKey,
        sortedDirection: props.sortedDirection,
        onSortRequested: props.onSortRequested
      })
    )
  );
}

DataGridHeaderGroupCell.propTypes = {
  model: require('react').PropTypes.object.isRequired,
  onSortRequested: require('react').PropTypes.func.isRequired,
  sortedDirection: require('react').PropTypes.oneOf(['ASC', 'DESC']).isRequired,
  sortedKey: require('react').PropTypes.string.isRequired
};
var _default = DataGridHeaderGroupCell;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(DataGridHeaderGroupCell, 'DataGridHeaderGroupCell', 'src/_data-grid/_DataGridHeaderGroupCell.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/_data-grid/_DataGridHeaderGroupCell.jsx');
}();

;
module.exports = exports['default'];
