// txl/_data-grid/DataGridStyles.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NO_DATA_CONTENT_STYLES = exports.INTERMEDIATE_CONTENT_INNER_STYLES = exports.NO_RESULTS_CONTENT_STYLES = exports.INTERMEDIATE_CONTENT_STYLES = exports.LOADER_HEIGHT = exports.BODY_CELL_HOVER_STYLES = exports.BODY_CELL_LAST_ROW_STYLES = exports.BODY_CELL_STYLES = exports.HEADER_CELL_CENTERED_STYLES = exports.HEADER_TEXT_STYLES = exports.HEADER_CELL_STYLES = exports.BODY_ROW_STYLES = exports.CELL_WRAPPER = exports.HEADER_ROW_STYLES = exports.HEADER_WRAPPER_STYLES = exports.SCROLLER_STYLES = exports.WRAPPER_STYLES = exports.CELL_PINNED_LAST = exports.CELL_PINNED_STYLES = exports.CELL_COMMON_STYLES = exports.ROW_COMMON_STYLES = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

var _layers = require('../styles/mixins/layers');

var _type = require('../styles/mixins/type');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ROW_COMMON_STYLES = exports.ROW_COMMON_STYLES = {
  display: 'flex'
};

var CELL_COMMON_STYLES = exports.CELL_COMMON_STYLES = {
  borderRight: '1px solid white',
  boxSizing: 'border-box',
  minHeight: (0, _helpers.gridUnits)(7),
  overflow: 'hidden',
  padding: (0, _helpers.gridUnits)(1.4) + ' ' + (0, _helpers.gridUnits)(0) + ' ' + (0, _helpers.gridUnits)(1.6) + ' ' + (0, _helpers.gridUnits)(2),
  position: 'relative',
  zIndex: 1
};

var CELL_PINNED_STYLES = exports.CELL_PINNED_STYLES = {
  zIndex: 2
};

var CELL_PINNED_LAST = exports.CELL_PINNED_LAST = {
  borderRight: '1px solid ' + _theme.COLOR_NEUTRAL[100]
};

var WRAPPER_STYLES = exports.WRAPPER_STYLES = (0, _extends3.default)({}, _type.DEFAULT_FONT_FAMILY, _type.TYPE_BASE, {
  boxSizing: 'border-box',
  marginBottom: (0, _helpers.gridUnits)(3),
  position: 'relative',
  width: '100%'
});

var SCROLLER_STYLES = exports.SCROLLER_STYLES = (0, _extends3.default)({}, _layers.LAYER_2, {
  backgroundColor: 'white',
  overflowX: 'auto'
});

var HEADER_WRAPPER_STYLES = exports.HEADER_WRAPPER_STYLES = {
  backfaceVisibility: 'hidden',
  zIndex: 3
};

var HEADER_ROW_STYLES = exports.HEADER_ROW_STYLES = (0, _extends3.default)({}, ROW_COMMON_STYLES);

var CELL_WRAPPER = exports.CELL_WRAPPER = {
  display: 'flex',
  flexGrow: 1
};

var BODY_ROW_STYLES = exports.BODY_ROW_STYLES = (0, _extends3.default)({}, ROW_COMMON_STYLES);

var HEADER_CELL_STYLES = exports.HEADER_CELL_STYLES = (0, _extends3.default)({}, CELL_COMMON_STYLES, {
  backgroundColor: _theme.COLOR_NEUTRAL['700'],
  color: _theme.COLOR_NEUTRAL['700c'],
  display: 'flex',
  justifyContent: 'space-between',
  padding: 0
});

var HEADER_TEXT_STYLES = exports.HEADER_TEXT_STYLES = {
  padding: CELL_COMMON_STYLES.padding
};

var HEADER_CELL_CENTERED_STYLES = exports.HEADER_CELL_CENTERED_STYLES = {
  alignItems: 'center',
  display: 'flex',
  justifyContent: 'space-between'
};

var BODY_CELL_STYLES = exports.BODY_CELL_STYLES = (0, _extends3.default)({}, CELL_COMMON_STYLES, {
  backgroundColor: 'white',
  borderBottom: '1px solid ' + _theme.COLOR_NEUTRAL[100]
});

var BODY_CELL_LAST_ROW_STYLES = exports.BODY_CELL_LAST_ROW_STYLES = {
  borderBottom: 'none'
};

var BODY_CELL_HOVER_STYLES = exports.BODY_CELL_HOVER_STYLES = {
  backgroundColor: _theme.COLOR_NEUTRAL['50'],
  color: _theme.COLOR_NEUTRAL['50c'],
  cursor: 'pointer'
};

var LOADER_HEIGHT = exports.LOADER_HEIGHT = (0, _helpers.gridUnits)(22);

var INTERMEDIATE_CONTENT_STYLES = exports.INTERMEDIATE_CONTENT_STYLES = {
  color: _theme.COLOR_NEUTRAL['300'],
  height: LOADER_HEIGHT,
  padding: (0, _helpers.gridUnits)(4),
  position: 'relative',
  width: '100%'
};

var NO_RESULTS_CONTENT_STYLES = exports.NO_RESULTS_CONTENT_STYLES = (0, _extends3.default)({}, INTERMEDIATE_CONTENT_STYLES, {
  alignItems: 'center',
  height: 'auto',
  textAlign: 'center'
});

var INTERMEDIATE_CONTENT_INNER_STYLES = exports.INTERMEDIATE_CONTENT_INNER_STYLES = (0, _extends3.default)({}, _helpers.CORNERS, {
  display: 'flex',
  justifyContent: 'center',
  position: 'relative'
});

var NO_DATA_CONTENT_STYLES = exports.NO_DATA_CONTENT_STYLES = {
  alignSelf: 'center'
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ROW_COMMON_STYLES, 'ROW_COMMON_STYLES', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(CELL_COMMON_STYLES, 'CELL_COMMON_STYLES', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(CELL_PINNED_STYLES, 'CELL_PINNED_STYLES', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(CELL_PINNED_LAST, 'CELL_PINNED_LAST', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(WRAPPER_STYLES, 'WRAPPER_STYLES', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(SCROLLER_STYLES, 'SCROLLER_STYLES', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(HEADER_WRAPPER_STYLES, 'HEADER_WRAPPER_STYLES', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(HEADER_ROW_STYLES, 'HEADER_ROW_STYLES', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(CELL_WRAPPER, 'CELL_WRAPPER', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(BODY_ROW_STYLES, 'BODY_ROW_STYLES', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(HEADER_CELL_STYLES, 'HEADER_CELL_STYLES', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(HEADER_TEXT_STYLES, 'HEADER_TEXT_STYLES', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(HEADER_CELL_CENTERED_STYLES, 'HEADER_CELL_CENTERED_STYLES', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(BODY_CELL_STYLES, 'BODY_CELL_STYLES', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(BODY_CELL_LAST_ROW_STYLES, 'BODY_CELL_LAST_ROW_STYLES', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(BODY_CELL_HOVER_STYLES, 'BODY_CELL_HOVER_STYLES', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(LOADER_HEIGHT, 'LOADER_HEIGHT', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(INTERMEDIATE_CONTENT_STYLES, 'INTERMEDIATE_CONTENT_STYLES', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(NO_RESULTS_CONTENT_STYLES, 'NO_RESULTS_CONTENT_STYLES', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(INTERMEDIATE_CONTENT_INNER_STYLES, 'INTERMEDIATE_CONTENT_INNER_STYLES', 'src/_data-grid/DataGridStyles.js');

  __REACT_HOT_LOADER__.register(NO_DATA_CONTENT_STYLES, 'NO_DATA_CONTENT_STYLES', 'src/_data-grid/DataGridStyles.js');
}();

;
