// txl/_data-grid/Pager.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _ReactUtils = require('../ReactUtils');

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

var _type = require('../styles/mixins/type');

var _Base = require('../base/Base');

var _UncontrolledButtonSelect = require('../dropdowns/UncontrolledButtonSelect');

var _UncontrolledButtonSelect2 = _interopRequireDefault(_UncontrolledButtonSelect);

var _IconButton = require('../buttons/IconButton');

var _IconButton2 = _interopRequireDefault(_IconButton);

var _ArrowLeft = require('../icons/ArrowLeft');

var _ArrowLeft2 = _interopRequireDefault(_ArrowLeft);

var _ArrowRight = require('../icons/ArrowRight');

var _ArrowRight2 = _interopRequireDefault(_ArrowRight);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var PAGE_SIZES = [10, 25, 50, 100];

var PAGE_SIZE_OPTIONS = PAGE_SIZES.map(function (size) {
  return {
    label: 'View ' + size,
    pageSize: size,
    value: '' + size
  };
});

function getNumPages(results, resultsPerPage) {
  return results != null && Math.ceil(results / resultsPerPage);
}

function getStateFromProps(props) {
  var results = props.results,
      resultsPerPage = props.resultsPerPage,
      page = props.page;

  return {
    numPages: getNumPages(results, resultsPerPage),
    page: page || 1,
    resultsPerPage: resultsPerPage
  };
}

var WRAPPER = (0, _extends3.default)({}, _type.DEFAULT_FONT_FAMILY, _type.TYPE_BASE, {
  alignItems: 'center',
  display: 'flex',
  justifyContent: 'center',
  position: 'relative'
});

var BUTTON_SELECT_WRAPPER = {
  alignSelf: 'center',
  display: 'flex',
  left: '0',
  position: 'absolute',
  top: '0'
};

var ICON_BUTTONS_WRAP = {
  display: 'inline-flex'
};

var PAGE_SELECT_WRAPPER = {
  alignSelf: 'center',
  display: 'flex',
  margin: '0 ' + (0, _helpers.gridUnits)(2)
};

var PAGE_SELECT = {
  color: _theme.COLOR_NEUTRAL['400'],
  display: 'inline-flex',
  whiteSpace: 'nowrap'
};

var PAGE_INPUT = (0, _extends3.default)({}, _type.TYPE_BASE, {
  backgroundColor: 'transparent',
  borderRadius: 0,
  borderWidth: '0 0 1px',
  color: _theme.COLOR_NEUTRAL['400'],
  height: 'auto',
  lineHeight: 'auto',
  margin: '0 ' + (0, _helpers.gridUnits)(1),
  padding: 0,
  textAlign: 'center',
  textIndent: 0,
  verticalAlign: 'baseline',
  width: (0, _helpers.gridUnits)(6)
});

var TXLPager = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TXLPager, _React$Component);

  function TXLPager() {
    var _ref;

    (0, _classCallCheck3.default)(this, TXLPager);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TXLPager.__proto__ || (0, _getPrototypeOf2.default)(TXLPager)).call.apply(_ref, [this].concat(args)));

    _this.state = getStateFromProps(_this.props);
    return _this;
  }

  (0, _createClass3.default)(TXLPager, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      this._pageInput = (0, _ReactUtils.findDOMNode)(this.refs.pageInput);
    }
  }, {
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      this.setState(getStateFromProps(nextProps));
    }
  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate() {
      this._pageInput.value = this.state.page;
    }
  }, {
    key: '_updatePageNum',
    value: function _updatePageNum() {
      var newPageNum = parseInt(this._pageInput.value, 10);
      var pageInRange = newPageNum >= 1 && newPageNum <= this.state.numPages;
      if (!pageInRange) {
        // setting the value in the current page box back to current page number.
        this._pageInput.value = this.state.page;
      } else {
        this._triggerChange({ page: newPageNum });
      }
    }
  }, {
    key: '_handleFormSubmit',
    value: function _handleFormSubmit(e) {
      e.preventDefault();
      this._updatePageNum();
    }
  }, {
    key: '_triggerChange',
    value: function _triggerChange(newState) {
      this.props.onChange({
        page: newState.page || this.state.page,
        resultsPerPage: newState.resultsPerPage || this.state.resultsPerPage
      });
      this.setState(newState);
    }
  }, {
    key: '_pageNext',
    value: function _pageNext() {
      this._triggerChange({ page: this.state.page + 1 });
    }
  }, {
    key: '_pagePrev',
    value: function _pagePrev() {
      this._triggerChange({ page: this.state.page - 1 });
    }
  }, {
    key: '_handlePageSizeChange',
    value: function _handlePageSizeChange(obj) {
      var newState = {
        numPages: getNumPages(this.props.results, obj.pageSize),
        page: 1,
        resultsPerPage: obj.pageSize
      };
      this._triggerChange(newState);
    }
  }, {
    key: 'render',
    value: function render() {
      var _state = this.state,
          numPages = _state.numPages,
          page = _state.page,
          resultsPerPage = _state.resultsPerPage;
      var results = this.props.results;

      var validPageValues = PAGE_SIZE_OPTIONS;
      var disableNext = page >= numPages;
      var disablePrev = page <= 1;
      var value = PAGE_SIZE_OPTIONS.filter(function (item) {
        return item.pageSize === resultsPerPage;
      })[0].value;

      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlPager',
          style: WRAPPER
        },
        _react2.default.createElement(_radium.Style, {
          scopeSelector: '[data-component="TxlPager"]',
          rules: {
            /*
             This simply removes the input spinner which comes with numbers by default
             https://css-tricks.com/snippets/css/turn-off-number-input-spinners
             */
            '[data-page-input]::-webkit-inner-spin-button': {
              appearance: 'none',
              margin: 0
            },
            '[data-page-input]::-webkit-outer-spin-button': {
              appearance: 'none',
              margin: 0
            },
            'input[type="number"][data-page-input]': {
              appearance: 'textfield'
            }
          }
        }),
        _react2.default.createElement(
          'div',
          { style: BUTTON_SELECT_WRAPPER },
          _react2.default.createElement(_UncontrolledButtonSelect2.default, {
            items: validPageValues,
            onItemSelect: this._handlePageSizeChange,
            defaultValue: value
          })
        ),
        _react2.default.createElement(
          'div',
          { style: ICON_BUTTONS_WRAP },
          _react2.default.createElement(_IconButton2.default, {
            onClick: this._pagePrev,
            icon: _ArrowLeft2.default,
            variant: 'muted',
            disabled: disablePrev
          })
        ),
        _react2.default.createElement(
          'div',
          { style: PAGE_SELECT_WRAPPER },
          _react2.default.createElement(
            'div',
            { style: PAGE_SELECT },
            _react2.default.createElement(
              'span',
              null,
              'Page'
            ),
            _react2.default.createElement(
              'form',
              {
                onSubmit: this._handleFormSubmit,
                noValidate: true
              },
              _react2.default.createElement('input', {
                'data-page-input': true,
                style: PAGE_INPUT,
                type: 'number',
                min: 1,
                max: numPages,
                defaultValue: this.state.page,
                disabled: !results,
                ref: 'pageInput'
              })
            ),
            numPages && _react2.default.createElement(
              'span',
              null,
              'of ',
              ' ' + numPages
            )
          )
        ),
        _react2.default.createElement(
          'div',
          { style: ICON_BUTTONS_WRAP },
          _react2.default.createElement(_IconButton2.default, {
            onClick: this._pageNext,
            icon: _ArrowRight2.default,
            variant: 'muted',
            disabled: disableNext
          })
        )
      );
    }
  }]);
  return TXLPager;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_updatePageNum', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_updatePageNum'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleFormSubmit', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleFormSubmit'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_triggerChange', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_triggerChange'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_pageNext', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_pageNext'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_pagePrev', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_pagePrev'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handlePageSizeChange', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handlePageSizeChange'), _class2.prototype)), _class2)) || _class;

exports.default = TXLPager;


TXLPager.propTypes = {
  onChange: _react.PropTypes.func,
  page: _react.PropTypes.number,
  results: _react.PropTypes.number,
  resultsPerPage: _react.PropTypes.oneOf(PAGE_SIZES)
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(PAGE_SIZES, 'PAGE_SIZES', 'src/_data-grid/Pager.jsx');

  __REACT_HOT_LOADER__.register(PAGE_SIZE_OPTIONS, 'PAGE_SIZE_OPTIONS', 'src/_data-grid/Pager.jsx');

  __REACT_HOT_LOADER__.register(getNumPages, 'getNumPages', 'src/_data-grid/Pager.jsx');

  __REACT_HOT_LOADER__.register(getStateFromProps, 'getStateFromProps', 'src/_data-grid/Pager.jsx');

  __REACT_HOT_LOADER__.register(WRAPPER, 'WRAPPER', 'src/_data-grid/Pager.jsx');

  __REACT_HOT_LOADER__.register(BUTTON_SELECT_WRAPPER, 'BUTTON_SELECT_WRAPPER', 'src/_data-grid/Pager.jsx');

  __REACT_HOT_LOADER__.register(ICON_BUTTONS_WRAP, 'ICON_BUTTONS_WRAP', 'src/_data-grid/Pager.jsx');

  __REACT_HOT_LOADER__.register(PAGE_SELECT_WRAPPER, 'PAGE_SELECT_WRAPPER', 'src/_data-grid/Pager.jsx');

  __REACT_HOT_LOADER__.register(PAGE_SELECT, 'PAGE_SELECT', 'src/_data-grid/Pager.jsx');

  __REACT_HOT_LOADER__.register(PAGE_INPUT, 'PAGE_INPUT', 'src/_data-grid/Pager.jsx');

  __REACT_HOT_LOADER__.register(TXLPager, 'TXLPager', 'src/_data-grid/Pager.jsx');
}();

;
module.exports = exports['default'];
