// txl/tabs/_TabbedComponent.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

exports.default = generateTabbedComponent;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Base = require('../base/Base');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

function handleTabClickDefault(_ref) {
  var index = _ref.index;

  this._setActiveTab(index);
}

function generateTabbedComponent(Component) {
  var _desc, _value, _class;

  var TabbedComponent = (_class = function (_React$Component) {
    (0, _inherits3.default)(TabbedComponent, _React$Component);

    function TabbedComponent() {
      var _ref2;

      (0, _classCallCheck3.default)(this, TabbedComponent);

      for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      var _this = (0, _possibleConstructorReturn3.default)(this, (_ref2 = TabbedComponent.__proto__ || (0, _getPrototypeOf2.default)(TabbedComponent)).call.apply(_ref2, [this].concat(args)));

      _this.state = {
        selectedIndex: _this.props.selectedIndex || 0
      };
      return _this;
    }

    (0, _createClass3.default)(TabbedComponent, [{
      key: 'componentWillReceiveProps',
      value: function componentWillReceiveProps(nextProps) {
        this._setActiveTab(nextProps.selectedIndex);
      }
    }, {
      key: '_setActiveTab',
      value: function _setActiveTab(index) {
        var indexNum = Number(index);
        if (!(indexNum >= 0)) {
          return;
        }
        this.setState({
          selectedIndex: indexNum
        });
      }
    }, {
      key: '_handleTabClick',
      value: function _handleTabClick() {
        for (var _len2 = arguments.length, obj = Array(_len2), _key2 = 0; _key2 < _len2; _key2++) {
          obj[_key2] = arguments[_key2];
        }

        this.props.onTabClick.apply(this, obj);
      }
    }, {
      key: 'render',
      value: function render() {
        var items = this.props.items;
        var selectedIndex = this.state.selectedIndex;

        var selectedItemContent = items[selectedIndex].content;

        return _react2.default.createElement(Component, (0, _extends3.default)({}, this.props, this.state, {
          onTabClick: this._handleTabClick,
          content: selectedItemContent
        }));
      }
    }]);
    return TabbedComponent;
  }(_react2.default.Component), (_applyDecoratedDescriptor(_class.prototype, '_setActiveTab', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_setActiveTab'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_handleTabClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_handleTabClick'), _class.prototype)), _class);


  TabbedComponent.defaultProps = {
    onTabClick: handleTabClickDefault
  };

  TabbedComponent.propTypes = {
    items: _react.PropTypes.arrayOf(_react.PropTypes.shape({
      content: _react.PropTypes.node,
      heading: _react.PropTypes.node.isRequired,
      id: _react.PropTypes.string
    })).isRequired,
    onTabClick: _react.PropTypes.func,
    selectedIndex: _react.PropTypes.number
  };

  return TabbedComponent;
}
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(handleTabClickDefault, 'handleTabClickDefault', 'src/tabs/_TabbedComponent.jsx');

  __REACT_HOT_LOADER__.register(generateTabbedComponent, 'generateTabbedComponent', 'src/tabs/_TabbedComponent.jsx');
}();

;
module.exports = exports['default'];
