// txl/scrolling-table/records/Header.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _immutable = require('immutable');

var _constants = require('../constants');

var _DefaultHeaderTemplate = require('../DefaultHeaderTemplate');

var _DefaultHeaderTemplate2 = _interopRequireDefault(_DefaultHeaderTemplate);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var uid = 0;
var nextId = function nextId() {
  return uid++;
};

var Header = function (_Record) {
  (0, _inherits3.default)(Header, _Record);

  function Header(args) {
    (0, _classCallCheck3.default)(this, Header);

    var id = args.id || nextId();
    return (0, _possibleConstructorReturn3.default)(this, (Header.__proto__ || (0, _getPrototypeOf2.default)(Header)).call(this, (0, _extends3.default)({ id: id }, args)));
  }

  return Header;
}((0, _immutable.Record)({
  draggable: true,
  formatCell: null,
  id: 0,
  key: 0,
  minWidth: _constants.HEADER_CELL_MIN_WIDTH,
  order: 0,
  resizable: true,
  sortBy: null,
  sortDirection: 'none',
  sortable: false,
  template: _DefaultHeaderTemplate2.default,
  textOverflow: 'wrap',
  title: '',
  type: null,
  width: null
}));

exports.default = Header;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(uid, 'uid', 'src/scrolling-table/records/Header.js');

  __REACT_HOT_LOADER__.register(nextId, 'nextId', 'src/scrolling-table/records/Header.js');

  __REACT_HOT_LOADER__.register(Header, 'Header', 'src/scrolling-table/records/Header.js');
}();

;
module.exports = exports['default'];
