// txl/scrolling-table/records/Rows.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _immutable = require('immutable');

var _Row = require('./Row');

var _Row2 = _interopRequireDefault(_Row);

var _Header = require('./Header');

var _Header2 = _interopRequireDefault(_Header);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var rowMap = function rowMap(rows) {
  var output = {};

  for (var i = 0; i < rows.length; i++) {
    var row = new _Row2.default({ data: rows[i] });
    output[row.id] = row;
  }

  return (0, _immutable.Map)(output);
};

var Rows = function () {
  function Rows(rows) {
    (0, _classCallCheck3.default)(this, Rows);

    if (_immutable.Map.isMap(rows)) {
      this._rows = rows;
    } else {
      this._rows = (0, _immutable.Map)();
      return this.insert(rows);
    }
  }

  (0, _createClass3.default)(Rows, [{
    key: 'insert',
    value: function insert(rows) {
      var insert = Array.isArray(rows) ? rowMap(rows) : rows;

      return new Rows(this._rows.merge(insert));
    }
  }, {
    key: 'get',
    value: function get() {
      return this._rows;
    }
  }, {
    key: 'size',
    value: function size() {
      return this._rows.size;
    }
  }, {
    key: 'sortBy',
    value: function sortBy(headers) {
      var sortBy = function sortBy(field) {
        return function (rows) {
          var sorted = rows.sortBy(function (row) {
            return row.data.get(field.key);
          }, field.sortBy);

          if (field.sortDirection === 'asc') {
            sorted = sorted.reverse();
          }

          return sorted.groupBy(function (r) {
            return r.data.get(field.key);
          });
        };
      };

      var sort = function sort(fields, groups) {
        if (!fields.size) {
          return groups;
        }

        return groups.map(function (rows) {
          return sort(fields.rest(), sortBy(fields.first())(rows));
        });
      };

      var fields = (0, _immutable.List)(headers).sortBy(function (h) {
        return h.order;
      }).filter(function (h) {
        return h.sortable;
      }).filter(function (h) {
        return h.sortDirection !== 'none';
      });

      if (!fields.size) {
        return this._rows;
      }

      var toMap = function toMap(rows) {
        var output = {};
        for (var i = 0; i < rows.size; i++) {
          output[i] = rows.get(i);
        }

        return (0, _immutable.Map)(output);
      };

      var rows = this._rows.valueSeq();
      var groups = sort(fields, (0, _immutable.List)([rows]));
      return toMap(groups.flatten(fields.size + 1));
    }
  }]);
  return Rows;
}();

exports.default = Rows;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(rowMap, 'rowMap', 'src/scrolling-table/records/Rows.js');

  __REACT_HOT_LOADER__.register(Rows, 'Rows', 'src/scrolling-table/records/Rows.js');
}();

;
module.exports = exports['default'];
