// tune-react-utils/lib/ReactUtils.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createFragment = exports.ReactTestUtils = exports.unmountComponentAtNode = exports.renderToStaticMarkup = exports.renderToString = exports.findDOMNode = exports.render = exports.shallowCompare = undefined;

var _reactDom = require('react-dom');

var _reactDom2 = _interopRequireDefault(_reactDom);

var _server = require('react-dom/server');

var _reactAddonsShallowCompare = require('react-addons-shallow-compare');

var _reactAddonsShallowCompare2 = _interopRequireDefault(_reactAddonsShallowCompare);

var _reactAddonsCreateFragment = require('react-addons-create-fragment');

var _reactAddonsCreateFragment2 = _interopRequireDefault(_reactAddonsCreateFragment);

var _reactAddonsTestUtils = require('react-addons-test-utils');

var _reactAddonsTestUtils2 = _interopRequireDefault(_reactAddonsTestUtils);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var boundRender = _reactDom.render.bind(_reactDom2.default);
var boundFindDOMNode = _reactDom.findDOMNode.bind(_reactDom2.default);
var boundRenderToString = _server.renderToString.bind(_reactDom2.default);
var boundRenderToStaticMarkup = _server.renderToStaticMarkup.bind(_reactDom2.default);
var boundUnmountComponentAtNode = _reactDom.unmountComponentAtNode.bind(_reactDom2.default);

/**
 *
 * ReactUtils can work as bridge between 0.13 and 0.14
 * @module ReactUtils
 */
exports.shallowCompare = _reactAddonsShallowCompare2.default;
exports.render = boundRender;
exports.findDOMNode = boundFindDOMNode;
exports.renderToString = boundRenderToString;
exports.renderToStaticMarkup = boundRenderToStaticMarkup;
exports.unmountComponentAtNode = boundUnmountComponentAtNode;
exports.ReactTestUtils = _reactAddonsTestUtils2.default;
exports.createFragment = _reactAddonsCreateFragment2.default;

// Default module path

exports.default = {
  createFragment: _reactAddonsCreateFragment2.default,
  ReactTestUtils: _reactAddonsTestUtils2.default,
  shallowCompare: _reactAddonsShallowCompare2.default,
  findDOMNode: boundFindDOMNode,
  render: boundRender,
  renderToStaticMarkup: boundRenderToStaticMarkup,
  renderToString: boundRenderToString,
  unmountComponentAtNode: boundUnmountComponentAtNode
};
