// tune-react-utils/src/ReactUtils.js
import ReactDOM, {
  findDOMNode,
  render,
  unmountComponentAtNode,
} from 'react-dom';
import {
  renderToStaticMarkup,
  renderToString,
} from 'react-dom/server';

import shallowCompare from 'react-addons-shallow-compare';
import createFragment from 'react-addons-create-fragment';
import ReactTestUtils from 'react-addons-test-utils';

const boundRender = render.bind(ReactDOM);
const boundFindDOMNode = findDOMNode.bind(ReactDOM);
const boundRenderToString = renderToString.bind(ReactDOM);
const boundRenderToStaticMarkup = renderToStaticMarkup.bind(ReactDOM);
const boundUnmountComponentAtNode = unmountComponentAtNode.bind(ReactDOM);

/**
 *
 * ReactUtils can work as bridge between 0.13 and 0.14
 * @module ReactUtils
 */
export { shallowCompare };
export { boundRender as render };
export { boundFindDOMNode as findDOMNode };
export { boundRenderToString as renderToString };
export { boundRenderToStaticMarkup as renderToStaticMarkup };
export { boundUnmountComponentAtNode as unmountComponentAtNode };
export { ReactTestUtils };
export { createFragment };

// Default module path
export default {
  createFragment,
  ReactTestUtils,
  shallowCompare,
  findDOMNode: boundFindDOMNode,
  render: boundRender,
  renderToStaticMarkup: boundRenderToStaticMarkup,
  renderToString: boundRenderToString,
  unmountComponentAtNode: boundUnmountComponentAtNode,
};
