// txl/utils/currencyCodeToSymbolMap.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
// Source: https://github.com/smirzaei/currency-formatter/blob/master/currencies.js
var codeToSymbolMap = {
  AED: 'د.إ.‏',
  AFN: '؋',
  ALL: 'Lek',
  AMD: '֏',
  ANG: 'ƒ',
  AOA: 'Kz',
  ARS: '$',
  AUD: '$',
  AWG: 'ƒ',
  AZN: '₼',
  BAM: 'КМ',
  BBD: '$',
  BDT: '৳',
  BGN: 'лв.',
  BHD: 'د.ب.‏',
  BIF: 'FBu',
  BMD: '$',
  BND: '$',
  BOB: 'Bs',
  BRL: 'R$',
  BSD: '$',
  BTC: 'Ƀ',
  BTN: 'Nu.',
  BWP: 'P',
  BYR: 'р.',
  BZD: 'BZ$',
  CAD: '$',
  CDF: 'FC',
  CHF: 'CHF',
  CLP: '$',
  CNY: '¥',
  COP: '$',
  CRC: '₡',
  CUC: 'CUC',
  CUP: '$MN',
  CVE: '$',
  CZK: 'Kč',
  DJF: 'Fdj',
  DKK: 'kr.',
  DOP: 'RD$',
  DZD: 'د.ج.‏',
  EGP: 'ج.م.‏',
  ERN: 'Nfk',
  ETB: 'ETB',
  EUR: '€',
  FJD: '$',
  FKP: '£',
  GBP: '£',
  GEL: 'Lari',
  GHS: '₵',
  GIP: '£',
  GMD: 'D',
  GNF: 'FG',
  GTQ: 'Q',
  GYD: '$',
  HKD: 'HK$',
  HNL: 'L.',
  HRK: 'kn',
  HTG: 'G',
  HUF: 'Ft',
  IDR: 'Rp',
  ILS: '₪',
  INR: '₹',
  IQD: 'د.ع.‏',
  IRR: '﷼',
  ISK: 'kr.',
  JMD: 'J$',
  JOD: 'د.ا.‏',
  JPY: '¥',
  KES: 'S',
  KGS: 'сом',
  KHR: '៛',
  KMF: 'CF',
  KPW: '₩',
  KRW: '₩',
  KWD: 'د.ك.‏',
  KYD: '$',
  KZT: '₸',
  LAK: '₭',
  LBP: 'ل.ل.‏',
  LKR: '₨',
  LRD: '$',
  LSL: 'M',
  LYD: 'د.ل.‏',
  MAD: 'د.م.‏',
  MDL: 'lei',
  MGA: 'Ar',
  MKD: 'ден.',
  MMK: 'K',
  MNT: '₮',
  MOP: 'MOP$',
  MRO: 'UM',
  MTL: '₤',
  MUR: '₨',
  MVR: 'MVR',
  MWK: 'MK',
  MXN: '$',
  MYR: 'RM',
  MZN: 'MT',
  NAD: '$',
  NGN: '₦',
  NIO: 'C$',
  NOK: 'kr',
  NPR: '₨',
  NZD: '$',
  OMR: '﷼',
  PAB: 'B/.',
  PEN: 'S/.',
  PGK: 'K',
  PHP: '₱',
  PKR: '₨',
  PLN: 'zł',
  PYG: '₲',
  QAR: '﷼',
  RON: 'lei',
  RSD: 'Дин.',
  RUB: '₽',
  RWF: 'RWF',
  SAR: '﷼',
  SBD: '$',
  SCR: '₨',
  SDD: 'LSd',
  SDG: '£‏',
  SEK: 'kr',
  SGD: '$',
  SHP: '£',
  SLL: 'Le',
  SOS: 'S',
  SRD: '$',
  STD: 'Db',
  SVC: '₡',
  SYP: '£',
  SZL: 'E',
  THB: '฿',
  TJS: 'TJS',
  TMT: 'm',
  TND: 'د.ت.‏',
  TOP: 'T$',
  TRY: 'TL',
  TTD: 'TT$',
  TVD: '$',
  TWD: 'NT$',
  TZS: 'TSh',
  UAH: '₴',
  UGX: 'USh',
  USD: '$',
  UYU: '$U',
  UZS: 'сўм',
  VEB: 'Bs.',
  VEF: 'Bs. F.',
  VND: '₫',
  VUV: 'VT',
  WST: 'WS$',
  XAF: 'F',
  XCD: '$',
  XOF: 'F',
  XPF: 'F',
  YER: '﷼',
  ZAR: 'R',
  ZMW: 'ZK'
};

var _default = codeToSymbolMap;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(codeToSymbolMap, 'codeToSymbolMap', 'src/utils/currencyCodeToSymbolMap.js');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/utils/currencyCodeToSymbolMap.js');
}();

;
module.exports = exports['default'];
