// txl/utils/formatter.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getSeparatorsBasedOnLocale = getSeparatorsBasedOnLocale;
exports.convertCurrencyToDecimalValue = convertCurrencyToDecimalValue;
exports.formatCurrencyStringForLocale = formatCurrencyStringForLocale;
// Source: http://stackoverflow.com/q/15900485
function convertBytesToHumanReadableFormat(sizeInBytes) {
  if (sizeInBytes >= 1000000000) {
    return (sizeInBytes / 1000000000).toFixed() + ' GB';
  } else if (sizeInBytes >= 1000000) {
    return (sizeInBytes / 1000000).toFixed() + ' MB';
  } else if (sizeInBytes >= 1000) {
    return (sizeInBytes / 1000).toFixed() + ' KB';
  } else if (sizeInBytes > 1) {
    return sizeInBytes + ' bytes';
  } else if (sizeInBytes === 1) {
    return sizeInBytes + ' byte';
  }

  return '0 bytes';
}

function escapeRegularExpression(str) {
  return str.replace(/[-[\]/{}()*+?.\\^$|]/g, '\\$&');
}

exports.convertBytesToHumanReadableFormat = convertBytesToHumanReadableFormat;
exports.escapeRegularExpression = escapeRegularExpression;

// Chrome/FireFox/Safari: language, IE9+: browserLanguage

var currentLocale = navigator.language || navigator.browserLanguage;

function getSeparatorsBasedOnLocale() {
  var locale = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : currentLocale;

  var testNumber = Number(1000.1);
  var localeStringOfNumber = testNumber.toLocaleString(locale);

  return {
    decimalSeparator: localeStringOfNumber.substring(5, 6),
    thousandsSeparator: localeStringOfNumber.substring(1, 2)
  };
}

// Get the decimal value of a currency string based on the user's locale (default)
// or a custom locale
function convertCurrencyToDecimalValue(currencyString) {
  var locale = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : currentLocale;

  var _getSeparatorsBasedOn = getSeparatorsBasedOnLocale(locale),
      decimalSeparator = _getSeparatorsBasedOn.decimalSeparator,
      thousandsSeparator = _getSeparatorsBasedOn.thousandsSeparator;

  if (!currencyString) {
    return Number(0);
  }

  // Convert NBSP to space
  var formattedThousandsSeparator = thousandsSeparator.replace(String.fromCharCode(160), ' ');

  var thousandsSeparatorRegex = new RegExp('[' + formattedThousandsSeparator + ']', 'g');
  var removedThousandsSeparators = currencyString.replace(thousandsSeparatorRegex, '');

  var decimalRegex = new RegExp('[' + decimalSeparator + ']', 'g');
  var decimalCharacterConvertedToDot = removedThousandsSeparators.replace(decimalRegex, '.');

  return Number(decimalCharacterConvertedToDot);
}

// Format a string that represents a currency amount to its locale-formatted value based on
// the user's locale (default) or a custom locale
function formatCurrencyStringForLocale(currencyString, options) {
  var locale = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : currentLocale;

  var decimalValue = convertCurrencyToDecimalValue(currencyString, locale);

  return decimalValue.toLocaleString(locale, options || { minimumFractionDigits: 2 });
}
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(convertBytesToHumanReadableFormat, 'convertBytesToHumanReadableFormat', 'src/utils/formatter.js');

  __REACT_HOT_LOADER__.register(escapeRegularExpression, 'escapeRegularExpression', 'src/utils/formatter.js');

  __REACT_HOT_LOADER__.register(currentLocale, 'currentLocale', 'src/utils/formatter.js');

  __REACT_HOT_LOADER__.register(getSeparatorsBasedOnLocale, 'getSeparatorsBasedOnLocale', 'src/utils/formatter.js');

  __REACT_HOT_LOADER__.register(convertCurrencyToDecimalValue, 'convertCurrencyToDecimalValue', 'src/utils/formatter.js');

  __REACT_HOT_LOADER__.register(formatCurrencyStringForLocale, 'formatCurrencyStringForLocale', 'src/utils/formatter.js');
}();

;
