// txl/utils/ItemBuffer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _clamp = require('./clamp');

var _clamp2 = _interopRequireDefault(_clamp);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var MIN_BUFFER_SIZE = 3;
var MAX_BUFFER_SIZE = 6;

/**
 * ItemBuffer calculates items to be presented in the DOM.
 *
 * +----------------------------------------------+
 * |   Full List                                  |
 * |  +-----------------------------------------+ |
 * |  |0 |1 |2 |3 |4 |5 |6 |7 |8 |9 |10|11|12|  | |
 * |  +-----------------------------------------+ |
 * |      Buffer Slots                            |
 * |     +--------------------+                   |
 * |  <+ |1 |2 |3 |4 |5 |6 |7 | +-------------->  |
 * |     +--------------------+                   |
 * |   0  1  2  3  4  5  6  0  1  2  3  4  5  6   |
 * |                                              |
 * |           Viewport Slots                     |
 * |           +-----+                            |
 * |           |3 |4 |                            |
 * |           +-----+                            |
 * +----------------------------------------------+
 *
 * +----------------------------------------------+
 * |   Full List                                  |
 * |  +-----------------------------------------+ |
 * |  |0 |1 |2 |3 |4 |5 |6 |7 |8 |9 |10|11|12|  | |
 * |  +-----------------------------------------+ |
 * |            Buffer Slots                      |
 * |           +--------------------+             |
 * |  <------+ |3 |4 |5 |6 |7 |8 |9 | +-------->  |
 * |           +--------------------+             |
 * |   0  1  2  3  4  5  6  0  1  2  3  4  5  6   |
 * |                                              |
 * |                  Viewport Slots              |
 * |                 +-----+                      |
 * |                 |5 |6 |                      |
 * |                 +-----+                      |
 * +----------------------------------------------+
 */

var ItemBuffer = function () {
  function ItemBuffer() {
    var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
    (0, _classCallCheck3.default)(this, ItemBuffer);
    var itemsCount = options.itemsCount,
        defaultItemSize = options.defaultItemSize,
        viewportSize = options.viewportSize,
        _options$minSize = options.minSize,
        minSize = _options$minSize === undefined ? MIN_BUFFER_SIZE : _options$minSize,
        _options$maxSize = options.maxSize,
        maxSize = _options$maxSize === undefined ? MAX_BUFFER_SIZE : _options$maxSize;


    this.getItems = this.getItems.bind(this);
    this.itemsCount = itemsCount;
    this.itemSize = defaultItemSize;
    this.viewportSize = viewportSize;
    this.items = [];
    this.maxVisibleItemsCount = Math.ceil(viewportSize / defaultItemSize) + 1;
    this.bufferSize = (0, _clamp2.default)(Math.floor(this.maxVisibleItemsCount / 2), minSize, maxSize);
  }

  /**
   * normalizePosition based on bufferSize and total number of items size
   *
   * @param  {Number} position    current position to be normalized.
   * @param  {Number} bufferSize  size of buffer
   * @param  {Number} total       Total number of items.
   * @return {Number}             normalized position index.
   */


  (0, _createClass3.default)(ItemBuffer, [{
    key: '_normalizePosition',
    value: function _normalizePosition(position, bufferSize, total) {
      if (position < 0) {
        return position + bufferSize;
      }

      if (position >= total) {
        return (total - position) % bufferSize;
      }

      return position;
    }

    /**
     * getItems returns array of item index to be displayed based on the
     * current buffer state.
     *
     * @param  {Number} firstItemIndex  index of the first item in the viewport.
     * @return {Array<Number>}                 Array of index.
     */

  }, {
    key: 'getItems',
    value: function getItems(firstItemIndex) {
      var size = this.maxVisibleItemsCount + this.bufferSize * 2;
      for (var i = 0; i < size; i++) {
        var itemIndex = i + firstItemIndex;
        var bufferPosition = itemIndex % size;
        bufferPosition = this._normalizePosition(bufferPosition, size, this.itemsCount);
        if (itemIndex < this.itemsCount) {
          this.items[bufferPosition] = itemIndex;
        }
      }
      return this.items;
    }
  }]);
  return ItemBuffer;
}();

var _default = ItemBuffer;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(MIN_BUFFER_SIZE, 'MIN_BUFFER_SIZE', 'src/utils/ItemBuffer.js');

  __REACT_HOT_LOADER__.register(MAX_BUFFER_SIZE, 'MAX_BUFFER_SIZE', 'src/utils/ItemBuffer.js');

  __REACT_HOT_LOADER__.register(ItemBuffer, 'ItemBuffer', 'src/utils/ItemBuffer.js');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/utils/ItemBuffer.js');
}();

;
module.exports = exports['default'];
