// txl/utils/propTypes.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _toConsumableArray2 = require('babel-runtime/helpers/toConsumableArray');

var _toConsumableArray3 = _interopRequireDefault(_toConsumableArray2);

var _typeof2 = require('babel-runtime/helpers/typeof');

var _typeof3 = _interopRequireDefault(_typeof2);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _isUndefined = require('lodash/isUndefined');

var _isUndefined2 = _interopRequireDefault(_isUndefined);

var _isBoolean = require('lodash/isBoolean');

var _isBoolean2 = _interopRequireDefault(_isBoolean);

var _range = require('lodash/range');

var _range2 = _interopRequireDefault(_range);

var _CalendarUtils = require('../calendar-picker/_CalendarUtils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ANONYMOUS = 'ANONYMOUS';

function searchFunctionChecker(props, propName) {
  var componentName = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : ANONYMOUS;

  if (!(0, _isUndefined2.default)(props[propName])) {
    if (typeof props[propName] !== 'function') {
      return new Error(propName + ' was of type `' + (0, _typeof3.default)(props[propName]) + '` in ' + componentName + '. ' + 'Was expecting a Function.');
    }

    var returnValue = props[propName]();
    if (!(0, _isBoolean2.default)(returnValue)) {
      return new Error(propName + ' in ' + componentName + ' was a function, but returned a type ' + ('`' + (typeof returnValue === 'undefined' ? 'undefined' : (0, _typeof3.default)(returnValue)) + '`. Was expecting a boolean.'));
    }
  }
  return null;
}

/**
 * Wrapper function for custom propType validators to add ability to include .isRequired
 *
 * @param {Function} validate
 * @return {Function}
 */
var createChainableTypeChecker = function createChainableTypeChecker(validate) {
  function checkType(isRequired, props, propName) {
    var componentName = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : ANONYMOUS;
    var location = arguments[4];
    var propFullName = arguments[5];

    if ((0, _isUndefined2.default)(props[propName])) {
      if (isRequired) {
        return new Error('Required prop `' + (propFullName || propName) + '` was not specified in `' + componentName + '`.');
      }
      return null;
    }
    return validate(props, propName, componentName, location, propFullName);
  }

  var chainedCheckType = checkType.bind(null, false);
  chainedCheckType.isRequired = checkType.bind(null, true);
  return chainedCheckType;
};

/**
 * Custom PropType validation for validating a date range value, e.g., `[ startDate, endDate ]`
 */
var dateRange = createChainableTypeChecker(function (props, propName, componentName) {
  if (props[propName] !== undefined && Array.isArray(props[propName]) && !props[propName].length > 0 // empty array is default prop val, no need to validate
  && !_CalendarUtils.isValidDateRange.apply(undefined, (0, _toConsumableArray3.default)(props[propName]))) {
    return new Error(componentName + '.props.' + propName + ' is an invalid date range.');
  }
  return null;
});

/**
 * Custom PropType validation for validating shape of date range preset obj, e.g.,
 *
 * {
 *   endDate: TxlPropTypes.date,
 *   key: PropTypes.string,
 *   label: PropTypes.string,
 *   startDate: TxlPropTypes.date,
 * }
 */
var dateRangePreset = createChainableTypeChecker(function (props, propName, componentName) {
  var preset = props[propName];

  // wrap in try/catch so any errors thrown in forEach loops are returned from outer func
  try {
    // validate string types
    ['key', 'label'].forEach(function (stringProp) {
      if (preset[stringProp] !== undefined && typeof preset[stringProp] !== 'string') {
        throw new Error(componentName + '.props.' + propName + '.' + stringProp + ' should be a string.');
      }
    });

    // validate date types
    ['endDate', 'startDate'].forEach(function (dateProp) {
      if (preset[dateProp] !== undefined && !(0, _moment2.default)(preset[dateProp]).isValid()) {
        throw new Error(componentName + '.props.' + propName + '.' + dateProp + ' value (\'' + preset[dateProp] + '\')' + 'could not be coerced to moment.');
      }
    });
  } catch (err) {
    return err;
  }
  return null;
});

/**
 * PropType shape for time obj,
 * e.g., { hour: 22, minute: 9, second: 0 }
 */
var timeObj = createChainableTypeChecker(function (props, propName, componentName) {
  var timeObjVal = props[propName];

  // wrap in try/catch so any errors thrown in forEach loops are returned from outer func
  try {
    ['hour', 'minute'].forEach(function (requiredProp) {
      if (timeObjVal[requiredProp] === undefined) {
        throw new Error(componentName + '.props.' + propName + '.' + requiredProp + ' is a required prop.');
      }
    });

    var propRanges = [
    // note: range doesn't include ending, i.e., valid hour is 0..23
    { unit: 'hour', validRange: (0, _range2.default)(0, 24) }, { unit: 'minute', validRange: (0, _range2.default)(0, 60) }, { unit: 'second', validRange: (0, _range2.default)(0, 60) }];

    propRanges.forEach(function (_ref) {
      var unit = _ref.unit,
          validRange = _ref.validRange;

      var val = timeObjVal[unit];
      if (val !== undefined && !(val in validRange)) {
        throw new Error(componentName + '.props.' + propName + '.' + unit + ' must be an integer between ' + ('0 and ' + validRange[validRange.length - 1] + '.'));
      }
    });
  } catch (err) {
    return err;
  }
  return null;
});

var filterFunction = createChainableTypeChecker(searchFunctionChecker);

/**
 * Custom PropType validation for validating a moment|Date|string date prop
 */
var date = createChainableTypeChecker(function (props, propName, componentName) {
  if (props[propName] !== undefined && !(0, _moment2.default)(props[propName]).isValid()) {
    return new Error(componentName + '.props.' + propName + ' value (\'' + props[propName] + '\')' + 'could not be coerced to moment.');
  }
  return null;
});

var _default = {
  date: date,
  dateRange: dateRange,
  dateRangePreset: dateRangePreset,
  filterFunction: filterFunction,
  timeObj: timeObj
};
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ANONYMOUS, 'ANONYMOUS', 'src/utils/propTypes.js');

  __REACT_HOT_LOADER__.register(searchFunctionChecker, 'searchFunctionChecker', 'src/utils/propTypes.js');

  __REACT_HOT_LOADER__.register(createChainableTypeChecker, 'createChainableTypeChecker', 'src/utils/propTypes.js');

  __REACT_HOT_LOADER__.register(dateRange, 'dateRange', 'src/utils/propTypes.js');

  __REACT_HOT_LOADER__.register(dateRangePreset, 'dateRangePreset', 'src/utils/propTypes.js');

  __REACT_HOT_LOADER__.register(timeObj, 'timeObj', 'src/utils/propTypes.js');

  __REACT_HOT_LOADER__.register(filterFunction, 'filterFunction', 'src/utils/propTypes.js');

  __REACT_HOT_LOADER__.register(date, 'date', 'src/utils/propTypes.js');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/utils/propTypes.js');
}();

;
module.exports = exports['default'];
