// txl/notifications/_GhostButton.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _Button = require('../buttons/Button');

var _Button2 = _interopRequireDefault(_Button);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var BASE_STYLES = {
  backgroundColor: 'transparent',
  borderStyle: 'solid',
  borderWidth: '1px'
};

var TxlGhostButton = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlGhostButton, _React$Component);

  function TxlGhostButton() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlGhostButton);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlGhostButton.__proto__ || (0, _getPrototypeOf2.default)(TxlGhostButton)).call.apply(_ref, [this].concat(args)));

    _this.state = {
      hovered: false
    };
    return _this;
  }

  (0, _createClass3.default)(TxlGhostButton, [{
    key: '_handleMouseEnter',
    value: function _handleMouseEnter() {
      this.setState({ hovered: true });
    }
  }, {
    key: '_handleMouseLeave',
    value: function _handleMouseLeave() {
      this.setState({ hovered: false });
    }
  }, {
    key: '_handleClick',
    value: function _handleClick() {
      this.props.onClick({});
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          backgroundColor = _props.backgroundColor,
          color = _props.color,
          children = _props.children,
          id = _props.id;


      var restingStyle = {
        borderColor: color,
        color: color
      };

      var hoveredStyle = {
        backgroundColor: color,
        color: backgroundColor
      };

      var stateStyles = this.state.hovered ? hoveredStyle : restingStyle;

      var buttonStyles = (0, _extends3.default)({}, BASE_STYLES, stateStyles);

      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlGhostButton',
          onMouseEnter: this._handleMouseEnter,
          onMouseLeave: this._handleMouseLeave
        },
        _react2.default.createElement(
          _Button2.default,
          {
            id: id,
            onClick: this._handleClick,
            _style: buttonStyles
          },
          children
        )
      );
    }
  }]);
  return TxlGhostButton;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleMouseEnter', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseEnter'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleMouseLeave', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseLeave'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleClick'), _class2.prototype)), _class2)) || _class;

exports.default = TxlGhostButton;


TxlGhostButton.defaultProps = {
  onClick: _Base.NOOP
};

TxlGhostButton.propTypes = {
  backgroundColor: _react.PropTypes.string.isRequired,
  children: _react.PropTypes.node.isRequired,
  color: _react.PropTypes.string.isRequired,
  id: _react.PropTypes.string,
  onClick: _react.PropTypes.func
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(BASE_STYLES, 'BASE_STYLES', 'src/notifications/_GhostButton.jsx');

  __REACT_HOT_LOADER__.register(TxlGhostButton, 'TxlGhostButton', 'src/notifications/_GhostButton.jsx');
}();

;
module.exports = exports['default'];
