// txl/notifications/_HintDot.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HINT_DOT_WRAPPER_STYLES = exports.HINT_DOT_STYLES = undefined;
exports.getHintPulseStyles = getHintPulseStyles;

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _animations = require('../styles/mixins/animations');

var _theme = require('../styles/theme');

var _helpers = require('../styles/helpers');

var _colorHelpers = require('../styles/_helpers/colorHelpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CUBIC_BEZIER_TIMING = 'cubic-bezier(0.66, 0, 0, 1)';
var HINT_PULSE_KEYFRAME = _radium2.default.keyframes({
  to: {
    boxShadow: '0 0 ' + (0, _helpers.gridUnits)(1) + ' ' + (0, _helpers.gridUnits)(1.5) + ' ' + (0, _colorHelpers.colorToRgbaString)(_theme.COLOR_ACCENT['500'], 0),
    height: (0, _helpers.gridUnits)(3),
    marginLeft: '-50%',
    marginTop: '-50%',
    width: (0, _helpers.gridUnits)(3)
  }
});

var HINT_DOT_STYLES = exports.HINT_DOT_STYLES = {
  borderRadius: '50%',
  boxShadow: '0 0 0 0 ' + (0, _colorHelpers.colorToRgbaString)(_theme.COLOR_ACCENT['500'], 0.7),
  display: 'inline-block',
  height: 0,
  left: 7,
  position: 'absolute',
  top: 7,
  width: 0
};

function getHintPulseStyles() {
  var duration = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 1225;

  return {
    animation: _animations.PLACEHOLDER_ANIMATION_NAME,
    animationDirection: 'normal',
    animationDuration: duration + 'ms',
    animationIterationCount: 'infinite',
    animationName: HINT_PULSE_KEYFRAME,
    animationTimingFunction: CUBIC_BEZIER_TIMING
  };
}

var HINT_DOT_WRAPPER_STYLES = exports.HINT_DOT_WRAPPER_STYLES = {
  alignItems: 'center',
  display: 'inline-block',
  height: (0, _helpers.gridUnits)(3),
  position: 'relative',
  width: (0, _helpers.gridUnits)(3)
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(CUBIC_BEZIER_TIMING, 'CUBIC_BEZIER_TIMING', 'src/notifications/_HintDot.style.js');

  __REACT_HOT_LOADER__.register(HINT_PULSE_KEYFRAME, 'HINT_PULSE_KEYFRAME', 'src/notifications/_HintDot.style.js');

  __REACT_HOT_LOADER__.register(HINT_DOT_STYLES, 'HINT_DOT_STYLES', 'src/notifications/_HintDot.style.js');

  __REACT_HOT_LOADER__.register(getHintPulseStyles, 'getHintPulseStyles', 'src/notifications/_HintDot.style.js');

  __REACT_HOT_LOADER__.register(HINT_DOT_WRAPPER_STYLES, 'HINT_DOT_WRAPPER_STYLES', 'src/notifications/_HintDot.style.js');
}();

;
