// txl/notifications/_PageLevelAlert.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _NoticeOutline = require('../icons/NoticeOutline');

var _NoticeOutline2 = _interopRequireDefault(_NoticeOutline);

var _layers = require('../styles/mixins/layers');

var _theme = require('../styles/theme');

var _type = require('../styles/mixins/type');

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CONTENT_COLOR = _theme.COLOR_ERROR['500c'];
var Y_PADDING = (0, _helpers.gridUnits)(2);
var X_PADDING = (0, _helpers.gridUnits)(4);

var MESSAGE_STYLES = {
  marginLeft: (0, _helpers.gridUnits)(5)
};

var CONTAINER_STYLES = (0, _extends3.default)({}, _layers.LAYER_2, _type.DEFAULT_FONT_FAMILY, _type.TYPE_BASE, {
  backgroundColor: _theme.COLOR_ERROR['500'],
  borderRadius: 2,
  color: CONTENT_COLOR,
  display: 'inline-flex',
  lineHeight: (0, _helpers.gridUnits)(4),
  minWidth: (0, _helpers.gridUnits)(60),
  padding: Y_PADDING + ' ' + X_PADDING,
  pointerEvents: 'initial',
  position: 'relative'
});

var ICON_CONTAINER_STYLE = (0, _extends3.default)({}, (0, _helpers.square)((0, _helpers.gridUnits)(4)), {
  alignItems: 'center',
  display: 'flex',
  left: X_PADDING,
  position: 'absolute',
  top: Y_PADDING
});

var TxlPageLevelAlert = function TxlPageLevelAlert(props) {
  var message = props.message,
      style = props.style;


  return _react2.default.createElement(
    'div',
    {
      'data-component': 'TxlPageLevelAlert',
      style: (0, _extends3.default)({}, CONTAINER_STYLES, style)
    },
    _react2.default.createElement(
      'div',
      { style: ICON_CONTAINER_STYLE },
      _react2.default.createElement(_NoticeOutline2.default, { color: CONTENT_COLOR })
    ),
    _react2.default.createElement(
      'div',
      {
        'data-test-alert-content': true,
        style: MESSAGE_STYLES
      },
      message
    )
  );
};

TxlPageLevelAlert.propTypes = {
  message: _react.PropTypes.string.isRequired,
  style: _react.PropTypes.object
};

var _default = (0, _radium2.default)(TxlPageLevelAlert);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(CONTENT_COLOR, 'CONTENT_COLOR', 'src/notifications/_PageLevelAlert.jsx');

  __REACT_HOT_LOADER__.register(Y_PADDING, 'Y_PADDING', 'src/notifications/_PageLevelAlert.jsx');

  __REACT_HOT_LOADER__.register(X_PADDING, 'X_PADDING', 'src/notifications/_PageLevelAlert.jsx');

  __REACT_HOT_LOADER__.register(MESSAGE_STYLES, 'MESSAGE_STYLES', 'src/notifications/_PageLevelAlert.jsx');

  __REACT_HOT_LOADER__.register(CONTAINER_STYLES, 'CONTAINER_STYLES', 'src/notifications/_PageLevelAlert.jsx');

  __REACT_HOT_LOADER__.register(ICON_CONTAINER_STYLE, 'ICON_CONTAINER_STYLE', 'src/notifications/_PageLevelAlert.jsx');

  __REACT_HOT_LOADER__.register(TxlPageLevelAlert, 'TxlPageLevelAlert', 'src/notifications/_PageLevelAlert.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/notifications/_PageLevelAlert.jsx');
}();

;
module.exports = exports['default'];
