// txl/notifications/_ProductWideNotificationQueue.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _invariant = require('invariant');

var _invariant2 = _interopRequireDefault(_invariant);

var _ProductWideNotification = require('./_ProductWideNotification');

var _ProductWideNotification2 = _interopRequireDefault(_ProductWideNotification);

var _Base = require('../base/Base');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlProductWideNotificationQueue = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlProductWideNotificationQueue, _React$Component);

  function TxlProductWideNotificationQueue() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlProductWideNotificationQueue);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlProductWideNotificationQueue.__proto__ || (0, _getPrototypeOf2.default)(TxlProductWideNotificationQueue)).call.apply(_ref, [this].concat(args)));

    _this._notificationQueue = [];
    _this._busy = false;

    _this.state = {
      currentNotification: null
    };
    return _this;
  }

  (0, _createClass3.default)(TxlProductWideNotificationQueue, [{
    key: 'push',
    value: function push(notification) {
      var _this2 = this;

      (0, _invariant2.default)(notification.message && notification.header, 'Product Wide Notification Queue: Notification is missing message or header ' + 'and will be ignored');

      this._notificationQueue.push(_react2.default.createElement(_ProductWideNotification2.default, {
        message: notification.message,
        id: notification.id,
        header: notification.header,
        type: notification.type,
        onDismiss: function onDismiss() {
          return _this2._onDismiss(notification);
        },
        buttons: notification.buttons
      }));

      this._pullFromQueue();
    }
  }, {
    key: 'removeAll',
    value: function removeAll(callback) {
      var _this3 = this;

      this.setState({ currentNotification: null }, function () {
        _this3._notificationQueue = [];
        _this3._busy = false;
        callback();
      });
    }
  }, {
    key: '_pullFromQueue',
    value: function _pullFromQueue() {
      if (!this._notificationQueue.length || this._busy) {
        return;
      }
      this._busy = true;
      this.setState({
        currentNotification: this._notificationQueue.shift()
      });
    }
  }, {
    key: '_onDismiss',
    value: function _onDismiss(notification) {
      var _this4 = this;

      this.setState({ currentNotification: null }, function () {
        _this4._busy = false;
        _this4._pullFromQueue();

        if (notification.onDismiss) {
          notification.onDismiss();
        }
      });
    }
  }, {
    key: 'render',
    value: function render() {
      if (this.props.hide) {
        return null;
      }

      return _react2.default.createElement(
        'div',
        { 'data-component': 'TxlProductWideNotificationQueue' },
        this.state.currentNotification
      );
    }
  }]);
  return TxlProductWideNotificationQueue;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, 'push', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, 'push'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, 'removeAll', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, 'removeAll'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_pullFromQueue', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_pullFromQueue'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_onDismiss', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_onDismiss'), _class2.prototype)), _class2)) || _class;

exports.default = TxlProductWideNotificationQueue;


TxlProductWideNotificationQueue.propTypes = {
  hide: _react.PropTypes.bool
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlProductWideNotificationQueue, 'TxlProductWideNotificationQueue', 'src/notifications/_ProductWideNotificationQueue.jsx');
}();

;
module.exports = exports['default'];
