// txl/notifications/_ProductWideNotification.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MESSAGE_STYLES = exports.ICON_BASE_STYLE = exports.HEADER_STYLES = exports.CONTENT_CONTAINER_STYLES = exports.BUTTON_SPACER_STYLES = exports.BUTTON_SPACER_CONTAINER_STYLES = exports.BUTTON_CONTAINER_STYLES = exports.BASE_CONTAINER_STYLES = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _helpers = require('../styles/helpers');

var _type = require('../styles/mixins/type');

var _layers = require('../styles/mixins/layers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BASE_CONTAINER_STYLES = (0, _extends3.default)({}, _layers.LAYER_1, _type.TYPE_BASE, {
  display: 'flex',
  padding: (0, _helpers.gridUnits)(4)
});

var BUTTON_CONTAINER_STYLES = {
  display: 'inline-flex',
  marginTop: (0, _helpers.gridUnits)(2)
};

var BUTTON_SPACER_CONTAINER_STYLES = {
  display: 'inline-flex'
};

var BUTTON_SPACER_STYLES = {
  marginRight: (0, _helpers.gridUnits)(2)
};

var CONTENT_CONTAINER_STYLES = {
  display: 'flex',
  flex: '100 1 auto',
  flexDirection: 'column',
  margin: '0 ' + (0, _helpers.gridUnits)(2)
};

var HEADER_STYLES = (0, _extends3.default)({}, _type.HEADING_SMALL, {
  lineHeight: (0, _helpers.gridUnits)(5)
});

var ICON_BASE_STYLE = (0, _extends3.default)({}, (0, _helpers.square)((0, _helpers.gridUnits)(5)), {
  alignItems: 'center',
  display: 'flex',
  justifyContent: 'center',
  padding: 0
});

var MESSAGE_STYLES = (0, _extends3.default)({}, _type.TYPE_BASE, {
  lineHeight: (0, _helpers.gridUnits)(6)
});

exports.BASE_CONTAINER_STYLES = BASE_CONTAINER_STYLES;
exports.BUTTON_CONTAINER_STYLES = BUTTON_CONTAINER_STYLES;
exports.BUTTON_SPACER_CONTAINER_STYLES = BUTTON_SPACER_CONTAINER_STYLES;
exports.BUTTON_SPACER_STYLES = BUTTON_SPACER_STYLES;
exports.CONTENT_CONTAINER_STYLES = CONTENT_CONTAINER_STYLES;
exports.HEADER_STYLES = HEADER_STYLES;
exports.ICON_BASE_STYLE = ICON_BASE_STYLE;
exports.MESSAGE_STYLES = MESSAGE_STYLES;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(BASE_CONTAINER_STYLES, 'BASE_CONTAINER_STYLES', 'src/notifications/_ProductWideNotification.style.js');

  __REACT_HOT_LOADER__.register(BUTTON_CONTAINER_STYLES, 'BUTTON_CONTAINER_STYLES', 'src/notifications/_ProductWideNotification.style.js');

  __REACT_HOT_LOADER__.register(BUTTON_SPACER_CONTAINER_STYLES, 'BUTTON_SPACER_CONTAINER_STYLES', 'src/notifications/_ProductWideNotification.style.js');

  __REACT_HOT_LOADER__.register(BUTTON_SPACER_STYLES, 'BUTTON_SPACER_STYLES', 'src/notifications/_ProductWideNotification.style.js');

  __REACT_HOT_LOADER__.register(CONTENT_CONTAINER_STYLES, 'CONTENT_CONTAINER_STYLES', 'src/notifications/_ProductWideNotification.style.js');

  __REACT_HOT_LOADER__.register(HEADER_STYLES, 'HEADER_STYLES', 'src/notifications/_ProductWideNotification.style.js');

  __REACT_HOT_LOADER__.register(ICON_BASE_STYLE, 'ICON_BASE_STYLE', 'src/notifications/_ProductWideNotification.style.js');

  __REACT_HOT_LOADER__.register(MESSAGE_STYLES, 'MESSAGE_STYLES', 'src/notifications/_ProductWideNotification.style.js');
}();

;
