// txl/notifications/_ProductWideNotification.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _Close = require('../icons/Close');

var _Close2 = _interopRequireDefault(_Close);

var _FlagSolid = require('../icons/FlagSolid');

var _FlagSolid2 = _interopRequireDefault(_FlagSolid);

var _GhostButton = require('./_GhostButton');

var _GhostButton2 = _interopRequireDefault(_GhostButton);

var _colorHelpers = require('../styles/_helpers/colorHelpers');

var _helpers = require('../styles/helpers');

var _ProductWideNotification = require('./_ProductWideNotification.style');

var _NotificationColors = require('./NotificationColors.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlProductWideNotification = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlProductWideNotification, _React$Component);

  function TxlProductWideNotification() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlProductWideNotification);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlProductWideNotification.__proto__ || (0, _getPrototypeOf2.default)(TxlProductWideNotification)).call.apply(_ref, [this].concat(args)));

    _this.state = {
      dismissHovered: false,
      dismissed: false
    };
    return _this;
  }

  (0, _createClass3.default)(TxlProductWideNotification, [{
    key: '_onDismissMouseEnter',
    value: function _onDismissMouseEnter() {
      this.setState({ dismissHovered: true });
    }
  }, {
    key: '_onDismissMouseLeave',
    value: function _onDismissMouseLeave() {
      this.setState({ dismissHovered: false });
    }
  }, {
    key: '_handleDismissClick',
    value: function _handleDismissClick() {
      this.setState({ dismissed: true });
      this.props.onDismiss({});
    }
  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      if (this.state.dismissed) {
        return null;
      }

      var _props = this.props,
          buttons = _props.buttons,
          header = _props.header,
          id = _props.id,
          message = _props.message,
          type = _props.type;


      var mainContainerStyle = (0, _helpers.combineStyles)(_ProductWideNotification.BASE_CONTAINER_STYLES, _NotificationColors.COLORS[type]);

      var dismissOpacity = this.state.dismissHovered ? _NotificationColors.GHOST_ICON_HOVERED_OPACITY : _NotificationColors.GHOST_ICON_OPACITY;
      var dismissColor = (0, _colorHelpers.getColorWithOpacity)(_NotificationColors.COLORS[type].color, dismissOpacity);

      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlProductWideNotification',
          style: mainContainerStyle,
          id: id
        },
        _react2.default.createElement(
          'div',
          { style: _ProductWideNotification.ICON_BASE_STYLE },
          _react2.default.createElement(_FlagSolid2.default, { color: _NotificationColors.COLORS[type].color })
        ),
        _react2.default.createElement(
          'div',
          { style: _ProductWideNotification.CONTENT_CONTAINER_STYLES },
          _react2.default.createElement(
            'div',
            { style: _ProductWideNotification.HEADER_STYLES },
            header
          ),
          _react2.default.createElement(
            'div',
            { style: _ProductWideNotification.MESSAGE_STYLES },
            message
          ),
          buttons.length && _react2.default.createElement(
            'div',
            { style: _ProductWideNotification.BUTTON_CONTAINER_STYLES },
            buttons.map(function (button, index) {
              var needsSpacer = index < buttons.length - 1;

              return _react2.default.createElement(
                'div',
                {
                  style: _ProductWideNotification.BUTTON_SPACER_CONTAINER_STYLES,
                  key: index
                },
                _react2.default.createElement(
                  _GhostButton2.default,
                  (0, _extends3.default)({
                    id: button.id,
                    onClick: function onClick() {
                      return button.onClick({ dismiss: _this2._handleDismissClick });
                    }
                  }, _NotificationColors.COLORS[type]),
                  button.content
                ),
                needsSpacer && _react2.default.createElement('div', { style: _ProductWideNotification.BUTTON_SPACER_STYLES })
              );
            })
          )
        ),
        _react2.default.createElement(
          'div',
          {
            style: _ProductWideNotification.ICON_BASE_STYLE,
            onMouseEnter: this._onDismissMouseEnter,
            onMouseLeave: this._onDismissMouseLeave,
            onClick: this._handleDismissClick
          },
          _react2.default.createElement(
            'a',
            null,
            _react2.default.createElement(_Close2.default, { color: dismissColor })
          )
        )
      );
    }
  }]);
  return TxlProductWideNotification;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_onDismissMouseEnter', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_onDismissMouseEnter'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_onDismissMouseLeave', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_onDismissMouseLeave'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDismissClick', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDismissClick'), _class2.prototype)), _class2)) || _class;

exports.default = TxlProductWideNotification;


TxlProductWideNotification.defaultProps = {
  buttons: [],
  onDismiss: _Base.NOOP,
  type: 'info'
};

TxlProductWideNotification.propTypes = {
  buttons: _react.PropTypes.arrayOf(_react.PropTypes.shape({
    content: _react.PropTypes.node.isRequired,
    id: _react.PropTypes.string,
    onClick: _react.PropTypes.func.isRequired
  })),
  header: _react.PropTypes.node.isRequired,
  id: _react.PropTypes.string,
  message: _react.PropTypes.node.isRequired,
  onDismiss: _react.PropTypes.func,
  type: _react.PropTypes.oneOf(['alert', 'warning', 'info'])
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlProductWideNotification, 'TxlProductWideNotification', 'src/notifications/_ProductWideNotification.jsx');
}();

;
module.exports = exports['default'];
