// txl/notifications/_SystemNotification.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _desc, _value, _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Base = require('../base/Base');

var _SystemNotificationControlled = require('./_SystemNotificationControlled');

var _SystemNotificationControlled2 = _interopRequireDefault(_SystemNotificationControlled);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlSystemNotification = (_class = function (_React$Component) {
  (0, _inherits3.default)(TxlSystemNotification, _React$Component);

  function TxlSystemNotification() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlSystemNotification);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlSystemNotification.__proto__ || (0, _getPrototypeOf2.default)(TxlSystemNotification)).call.apply(_ref, [this].concat(args)));

    _this.state = { dismissed: false };
    return _this;
  }

  (0, _createClass3.default)(TxlSystemNotification, [{
    key: '_onDismiss',
    value: function _onDismiss() {
      this.setState({ dismissed: true });
      this.props.onDismiss({});
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          button = _props.button,
          message = _props.message,
          variant = _props.variant;
      var dismissed = this.state.dismissed;


      return _react2.default.createElement(_SystemNotificationControlled2.default, {
        button: button,
        dismissed: dismissed,
        message: message,
        onDismiss: this._onDismiss,
        variant: variant
      });
    }
  }]);
  return TxlSystemNotification;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class.prototype, '_onDismiss', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_onDismiss'), _class.prototype)), _class);
exports.default = TxlSystemNotification;


TxlSystemNotification.displayName = 'TxlSystemNotification';

TxlSystemNotification.propTypes = {
  button: _react.PropTypes.shape({
    content: _react.PropTypes.node.isRequired,
    id: _react.PropTypes.string,
    onClick: _react.PropTypes.func.isRequired
  }),
  message: _react.PropTypes.string,
  onDismiss: _react.PropTypes.func,
  variant: _react.PropTypes.oneOf(['danger', 'warning', 'info'])
};

TxlSystemNotification.defaultProps = {
  onDismiss: _Base.NOOP,
  variant: 'warning'
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlSystemNotification, 'TxlSystemNotification', 'src/notifications/_SystemNotification.jsx');
}();

;
module.exports = exports['default'];
