// txl/notifications/_SystemNotificationControlled.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _desc, _value, _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Base = require('../base/Base');

var _Close = require('../icons/Close');

var _Close2 = _interopRequireDefault(_Close);

var _HighlightNote = require('./HighlightNote');

var _HighlightNote2 = _interopRequireDefault(_HighlightNote);

var _colorHelpers = require('../styles/_helpers/colorHelpers');

var _GhostButton = require('./_GhostButton');

var _GhostButton2 = _interopRequireDefault(_GhostButton);

var _Underline = require('../layout/_Underline');

var _Underline2 = _interopRequireDefault(_Underline);

var _NotificationColors = require('./NotificationColors.style');

var _SystemNotification = require('./_SystemNotification.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlSystemNotificationControlled = (_class = function (_React$Component) {
  (0, _inherits3.default)(TxlSystemNotificationControlled, _React$Component);

  function TxlSystemNotificationControlled() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlSystemNotificationControlled);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlSystemNotificationControlled.__proto__ || (0, _getPrototypeOf2.default)(TxlSystemNotificationControlled)).call.apply(_ref, [this].concat(args)));

    _this.state = { dismissHovered: false };
    return _this;
  }

  (0, _createClass3.default)(TxlSystemNotificationControlled, [{
    key: '_onDismissMouseEnter',
    value: function _onDismissMouseEnter() {
      this.setState({ dismissHovered: true });
    }
  }, {
    key: '_onDismissMouseLeave',
    value: function _onDismissMouseLeave() {
      this.setState({ dismissHovered: false });
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          button = _props.button,
          dismissed = _props.dismissed,
          onDismiss = _props.onDismiss,
          message = _props.message,
          variant = _props.variant;


      if (dismissed) {
        return null;
      }

      var dismissOpacity = this.state.dismissHovered ? _NotificationColors.GHOST_ICON_HOVERED_OPACITY : _NotificationColors.GHOST_ICON_OPACITY;
      var dismissColor = (0, _colorHelpers.getColorWithOpacity)(_NotificationColors.COLORS[variant].color, dismissOpacity);

      return _react2.default.createElement(
        'div',
        { 'data-component': 'TxlSystemNotificationControlled' },
        _react2.default.createElement(
          _HighlightNote2.default,
          { variant: variant },
          _react2.default.createElement(
            'div',
            { style: _SystemNotification.CONTENT_CONTAINER_STYLE },
            _react2.default.createElement(
              'div',
              { style: _SystemNotification.MESSAGE_CONTAINER_STYLE },
              message
            ),
            button && _react2.default.createElement(
              _GhostButton2.default,
              (0, _extends3.default)({
                id: button.id,
                onClick: function onClick() {
                  return button.onClick({ dismiss: onDismiss });
                }
              }, _NotificationColors.COLORS[variant]),
              button.content
            )
          ),
          _react2.default.createElement(
            'div',
            {
              style: _SystemNotification.DISMISS_ICON_BASE_STYLE,
              onMouseEnter: this._onDismissMouseEnter,
              onMouseLeave: this._onDismissMouseLeave,
              onClick: onDismiss
            },
            _react2.default.createElement(
              'a',
              null,
              _react2.default.createElement(_Close2.default, { color: dismissColor })
            )
          )
        ),
        _react2.default.createElement(_Underline2.default, { color: 'white' })
      );
    }
  }]);
  return TxlSystemNotificationControlled;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class.prototype, '_onDismissMouseEnter', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_onDismissMouseEnter'), _class.prototype), _applyDecoratedDescriptor(_class.prototype, '_onDismissMouseLeave', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class.prototype, '_onDismissMouseLeave'), _class.prototype)), _class);
exports.default = TxlSystemNotificationControlled;


TxlSystemNotificationControlled.displayName = 'TxlSystemNotificationControlled';

TxlSystemNotificationControlled.propTypes = {
  button: _react.PropTypes.shape({
    content: _react.PropTypes.node.isRequired,
    id: _react.PropTypes.string,
    onClick: _react.PropTypes.func.isRequired
  }),
  dismissed: _react.PropTypes.bool,
  message: _react.PropTypes.string,
  onDismiss: _react.PropTypes.func,
  variant: _react.PropTypes.oneOf(['danger', 'warning', 'info'])
};

TxlSystemNotificationControlled.defaultProps = {
  dismissed: false,
  onDismiss: _Base.NOOP,
  variant: 'warning'
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlSystemNotificationControlled, 'TxlSystemNotificationControlled', 'src/notifications/_SystemNotificationControlled.jsx');
}();

;
module.exports = exports['default'];
