// txl/notifications/_SystemNotification.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DISMISS_ICON_BASE_STYLE = exports.MESSAGE_CONTAINER_STYLE = exports.CONTENT_CONTAINER_STYLE = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CONTENT_CONTAINER_STYLE = exports.CONTENT_CONTAINER_STYLE = {
  alignItems: 'center',
  display: 'flex',
  flex: '100 1 auto',
  height: '100%',
  marginLeft: (0, _helpers.gridUnits)(2),
  marginRight: (0, _helpers.gridUnits)(4)
};

var MESSAGE_CONTAINER_STYLE = exports.MESSAGE_CONTAINER_STYLE = {
  paddingRight: (0, _helpers.gridUnits)(2)
};

var DISMISS_ICON_BASE_STYLE = exports.DISMISS_ICON_BASE_STYLE = (0, _extends3.default)({}, (0, _helpers.square)((0, _helpers.gridUnits)(5)), {
  alignItems: 'center',
  display: 'flex',
  justifyContent: 'center'
});
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(CONTENT_CONTAINER_STYLE, 'CONTENT_CONTAINER_STYLE', 'src/notifications/_SystemNotification.style.js');

  __REACT_HOT_LOADER__.register(MESSAGE_CONTAINER_STYLE, 'MESSAGE_CONTAINER_STYLE', 'src/notifications/_SystemNotification.style.js');

  __REACT_HOT_LOADER__.register(DISMISS_ICON_BASE_STYLE, 'DISMISS_ICON_BASE_STYLE', 'src/notifications/_SystemNotification.style.js');
}();

;
