// txl/notifications/NotificationColors.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.COLORS = exports.GHOST_ICON_OPACITY = exports.GHOST_ICON_HOVERED_OPACITY = undefined;

var _theme = require('../styles/theme');

var GHOST_ICON_HOVERED_OPACITY = exports.GHOST_ICON_HOVERED_OPACITY = 1;
var GHOST_ICON_OPACITY = exports.GHOST_ICON_OPACITY = 0.7;

var COLORS = exports.COLORS = {
  alert: {
    backgroundColor: _theme.COLOR_ERROR['500'],
    color: _theme.COLOR_ERROR['500c']
  },
  danger: {
    backgroundColor: _theme.COLOR_ERROR['500'],
    color: _theme.COLOR_ERROR['500c']
  },
  info: {
    backgroundColor: _theme.COLOR_ACCENT['500'],
    color: _theme.COLOR_ACCENT['500c']
  },
  warning: {
    backgroundColor: _theme.COLOR_WARNING['500'],
    color: _theme.COLOR_WARNING['500c']
  }
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(GHOST_ICON_HOVERED_OPACITY, 'GHOST_ICON_HOVERED_OPACITY', 'src/notifications/NotificationColors.style.js');

  __REACT_HOT_LOADER__.register(GHOST_ICON_OPACITY, 'GHOST_ICON_OPACITY', 'src/notifications/NotificationColors.style.js');

  __REACT_HOT_LOADER__.register(COLORS, 'COLORS', 'src/notifications/NotificationColors.style.js');
}();

;
