// txl/notifications/PageLevelAlertsContainer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _PageLevelAlert = require('./_PageLevelAlert');

var _PageLevelAlert2 = _interopRequireDefault(_PageLevelAlert);

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var ALERT_WRAPPER_STYLE = {
  marginBottom: (0, _helpers.gridUnits)(2),
  pointerEvents: 'none'
};

var ALERTS_CONTAINER_STYLES = {
  display: 'inline'
};

var TxlPageLevelAlertsContainer = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlPageLevelAlertsContainer, _React$Component);

  function TxlPageLevelAlertsContainer() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlPageLevelAlertsContainer);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlPageLevelAlertsContainer.__proto__ || (0, _getPrototypeOf2.default)(TxlPageLevelAlertsContainer)).call.apply(_ref, [this].concat(args)));

    _this._alerts = [];
    _this._counter = 0;
    return _this;
  }

  (0, _createClass3.default)(TxlPageLevelAlertsContainer, [{
    key: 'push',
    value: function push(message) {
      if (!(typeof message === 'string' && message)) {
        // eslint-disable-next-line no-console
        // console.warn('TxlPageLevelAlertsContainer: Alert is missing message and will be ignored');
        // is this an error? does it need to verified? use invariant
        return undefined;
      }

      var id = this._counter++;
      this._alerts.push({
        id: id,
        message: message
      });
      this.forceUpdate();

      // returning id of the pushed element.
      return id;
    }
  }, {
    key: 'clear',
    value: function clear(id) {
      this._alerts = this._alerts.filter(function (alert) {
        return alert.id !== id;
      });
      this.forceUpdate();
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlPageLevelAlertContainer',
          style: ALERTS_CONTAINER_STYLES
        },
        this._alerts.map(function (alert) {
          return _react2.default.createElement(
            'div',
            {
              key: alert.id,
              style: ALERT_WRAPPER_STYLE
            },
            _react2.default.createElement(_PageLevelAlert2.default, { message: alert.message })
          );
        })
      );
    }
  }]);
  return TxlPageLevelAlertsContainer;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, 'push', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, 'push'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, 'clear', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, 'clear'), _class2.prototype)), _class2)) || _class;

exports.default = TxlPageLevelAlertsContainer;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ALERT_WRAPPER_STYLE, 'ALERT_WRAPPER_STYLE', 'src/notifications/PageLevelAlertsContainer.jsx');

  __REACT_HOT_LOADER__.register(ALERTS_CONTAINER_STYLES, 'ALERTS_CONTAINER_STYLES', 'src/notifications/PageLevelAlertsContainer.jsx');

  __REACT_HOT_LOADER__.register(TxlPageLevelAlertsContainer, 'TxlPageLevelAlertsContainer', 'src/notifications/PageLevelAlertsContainer.jsx');
}();

;
module.exports = exports['default'];
