// txl/notifications/Toaster.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _Toast = require('./Toast');

var _Toast2 = _interopRequireDefault(_Toast);

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TOASTER_STYLE = {
  display: 'inline-flex',
  height: '0',
  // @TODO: Merge from prop (offset?) into calc when component is updated
  left: '50%',
  overflow: 'visible',
  // @TODO: When this component is updated to not req positioned container, change to position fixed
  position: 'absolute',
  width: '100%'
};

// @NOTE: Safari will do bad things if the order of the translate x/y is changed
var POPUP_KEYFRAME = _radium2.default.keyframes({
  '0%': {
    // 18gu is 2x the height of a toast
    transform: 'translateX(-50%) translateY(' + (0, _helpers.gridUnits)(18) + ')'
  },
  '100%': {
    transform: 'translateX(-50%)'
  }
});

var getPopupStyles = function getPopupStyles() {
  return {
    animation: POPUP_KEYFRAME,
    animationDirection: 'normal',
    animationDuration: '100ms',
    animationFillMode: 'forwards',
    animationIterationCount: '1',
    animationTimingFunction: 'ease-out'
  };
};

var TxlToaster = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlToaster, _React$Component);

  function TxlToaster() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlToaster);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlToaster.__proto__ || (0, _getPrototypeOf2.default)(TxlToaster)).call.apply(_ref, [this].concat(args)));

    _this._notificationQueue = [];
    _this._busy = false;

    _this.state = {
      currentNotification: null
    };
    return _this;
  }

  (0, _createClass3.default)(TxlToaster, [{
    key: 'push',
    value: function push(toast) {
      if (!toast.message || !toast.lifetime) {
        // eslint-disable-next-line no-console
        // console.warn('TxlToaster: Toast is missing message or lifetime data and will be ignored');
        // is this warning important? if so use invariant
        return;
      }

      var toastPositionStyles = (0, _extends3.default)({}, getPopupStyles(), {
        bottom: (0, _helpers.gridUnits)(2),
        position: 'absolute',
        transform: 'translateX(-50%)'
      });

      this._notificationQueue.push(_react2.default.createElement(
        _Toast2.default,
        {
          _style: toastPositionStyles,
          lifetime: toast.lifetime,
          onUndo: toast.onUndo,
          onExpire: this._onToastExpire
        },
        toast.message
      ));
      this._pullFromQueue();
    }
  }, {
    key: '_pullFromQueue',
    value: function _pullFromQueue() {
      if (!this._notificationQueue.length || this._busy) {
        return;
      }
      this._busy = true;
      this.setState({ currentNotification: this._notificationQueue.shift() });
    }
  }, {
    key: '_onToastExpire',
    value: function _onToastExpire() {
      var _this2 = this;

      this.setState({ currentNotification: null }, function () {
        _this2._busy = false;
        _this2._pullFromQueue();
      });
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlToaster',
          style: TOASTER_STYLE
        },
        this.state.currentNotification
      );
    }
  }]);
  return TxlToaster;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, 'push', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, 'push'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_pullFromQueue', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_pullFromQueue'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_onToastExpire', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_onToastExpire'), _class2.prototype)), _class2)) || _class;

exports.default = TxlToaster;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TOASTER_STYLE, 'TOASTER_STYLE', 'src/notifications/Toaster.jsx');

  __REACT_HOT_LOADER__.register(POPUP_KEYFRAME, 'POPUP_KEYFRAME', 'src/notifications/Toaster.jsx');

  __REACT_HOT_LOADER__.register(getPopupStyles, 'getPopupStyles', 'src/notifications/Toaster.jsx');

  __REACT_HOT_LOADER__.register(TxlToaster, 'TxlToaster', 'src/notifications/Toaster.jsx');
}();

;
module.exports = exports['default'];
