// txl/notifications/Toast.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UNDO_CONTAINER_STYLES = exports.UNDO_BUTTON_HOVERED = exports.UNDO_BUTTON_BASE = exports.TOAST_STYLES = exports.TOAST_MESSAGE_STYLES = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

var _layers = require('../styles/mixins/layers');

var _type = require('../styles/mixins/type');

var _ButtonStyleHelpers = require('../buttons/_ButtonStyleHelpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TOAST_MESSAGE_STYLES = (0, _extends3.default)({}, _helpers.TRUNCATE, {
  padding: '0 ' + (0, _helpers.gridUnits)(4)
});

var TOAST_STYLES = (0, _extends3.default)({}, _type.DEFAULT_FONT_FAMILY, _layers.LAYER_2, _type.TYPE_BASE, {
  alignItems: 'center',
  backgroundColor: _theme.COLOR_NEUTRAL['500'],
  borderRadius: 2,
  color: _theme.COLOR_NEUTRAL['500c'],
  display: 'inline-flex',
  height: (0, _helpers.gridUnits)(9),
  justifyContent: 'space-between',
  maxWidth: (0, _helpers.gridUnits)(120),
  minWidth: (0, _helpers.gridUnits)(60)
});

var UNDO_BUTTON_BASE = (0, _extends3.default)({}, _ButtonStyleHelpers.BUTTON_BASE_STYLE, {
  backgroundColor: 'transparent',
  color: _theme.COLOR_WARNING['300'],
  height: (0, _helpers.gridUnits)(5),
  lineHeight: (0, _helpers.gridUnits)(5),
  textDecoration: 'none'
});

var UNDO_BUTTON_HOVERED = {
  color: _theme.COLOR_WARNING['400']
};

var UNDO_CONTAINER_STYLES = {
  margin: '0 ' + (0, _helpers.gridUnits)(4)
};

exports.TOAST_MESSAGE_STYLES = TOAST_MESSAGE_STYLES;
exports.TOAST_STYLES = TOAST_STYLES;
exports.UNDO_BUTTON_BASE = UNDO_BUTTON_BASE;
exports.UNDO_BUTTON_HOVERED = UNDO_BUTTON_HOVERED;
exports.UNDO_CONTAINER_STYLES = UNDO_CONTAINER_STYLES;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TOAST_MESSAGE_STYLES, 'TOAST_MESSAGE_STYLES', 'src/notifications/Toast.style.js');

  __REACT_HOT_LOADER__.register(TOAST_STYLES, 'TOAST_STYLES', 'src/notifications/Toast.style.js');

  __REACT_HOT_LOADER__.register(UNDO_BUTTON_BASE, 'UNDO_BUTTON_BASE', 'src/notifications/Toast.style.js');

  __REACT_HOT_LOADER__.register(UNDO_BUTTON_HOVERED, 'UNDO_BUTTON_HOVERED', 'src/notifications/Toast.style.js');

  __REACT_HOT_LOADER__.register(UNDO_CONTAINER_STYLES, 'UNDO_CONTAINER_STYLES', 'src/notifications/Toast.style.js');
}();

;
