// txl/notifications/Toast.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _once = require('lodash/once');

var _once2 = _interopRequireDefault(_once);

var _Base = require('../base/Base');

var _LoadingIndicator = require('../progress-indicators/LoadingIndicator');

var _LoadingIndicator2 = _interopRequireDefault(_LoadingIndicator);

var _helpers = require('../styles/helpers');

var _theme = require('../styles/theme');

var _Toast = require('./Toast.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlToast = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlToast, _React$Component);

  function TxlToast() {
    var _ref;

    (0, _classCallCheck3.default)(this, TxlToast);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = TxlToast.__proto__ || (0, _getPrototypeOf2.default)(TxlToast)).call.apply(_ref, [this].concat(args)));

    _this.state = {
      expired: false,
      hovered: false,
      loading: false,
      undoing: false
    };

    _this._timeoutId = null;
    _this._remainingLife = 0;
    _this._startTime = null;
    _this._expireFromUndoCallback = (0, _once2.default)(function () {
      _this.setState({
        undoing: false
      }, _this._expire);
    });
    return _this;
  }

  (0, _createClass3.default)(TxlToast, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      this._startTimer();
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      this._endTimer();
      this._remainingLife = 0;
    }
  }, {
    key: '_handleUndo',
    value: function _handleUndo() {
      // Once the undo button is pressed, the wrapping component is required
      // to handle dismissing the notification, since we don't want it to automatically
      // expire the notification before the undo action is complete.

      this.setState({
        loading: true,
        undoing: true
      });

      this.props.onUndo({
        done: this._expireFromUndoCallback,
        error: this._handleUndoError
      });
    }
  }, {
    key: '_handleUndoError',
    value: function _handleUndoError(message) {
      this.setState({
        errorMessage: message,
        loading: false
      });
    }
  }, {
    key: '_handleMouseEnter',
    value: function _handleMouseEnter() {
      this.setState({ hovered: true });
    }
  }, {
    key: '_handleMouseLeave',
    value: function _handleMouseLeave() {
      this.setState({ hovered: false });
    }
  }, {
    key: '_pauseTimer',
    value: function _pauseTimer() {
      this._endTimer();
      var elapsed = Date.now() - this._startTime;
      this._remainingLife = (this._remainingLife || this.props.lifetime) - elapsed;
    }
  }, {
    key: '_unPauseTimer',
    value: function _unPauseTimer() {
      if (this._remainingLife <= 0) {
        return;
      }
      this._startTimer(this._remainingLife);
    }
  }, {
    key: '_startTimer',
    value: function _startTimer(remainingLife) {
      this._startTime = Date.now();

      if (this.props.lifetime) {
        this._timeoutId = setTimeout(this._expire, remainingLife || this.props.lifetime);
      }
    }
  }, {
    key: '_endTimer',
    value: function _endTimer() {
      clearTimeout(this._timeoutId);
    }
  }, {
    key: '_expire',
    value: function _expire() {
      if (this.state.undoing) {
        return;
      }
      this.setState({ expired: true });
      this.props.onExpire({});
    }
  }, {
    key: 'render',
    value: function render() {
      if (this.state.expired) {
        return null;
      }

      var undoContainer = void 0;
      var mouseEvents = void 0;

      if (this.props.lifetime) {
        mouseEvents = {
          onMouseEnter: this._pauseTimer,
          onMouseLeave: this._unPauseTimer
        };
      }

      if (this.props.onUndo) {
        var undoContent = void 0;

        if (this.state.loading) {
          undoContent = _react2.default.createElement(_LoadingIndicator2.default, { size: 'small', color: _theme.COLOR_NEUTRAL['500c'] });
        } else {
          // @TODO: Abstract these into a pattern that will allow us to use the
          // button component. https://jira.corp.tune.com/browse/TXLENG-4
          var undoButtonColor = this.state.hovered ? _Toast.UNDO_BUTTON_HOVERED : undefined;
          var undoButtonStyle = (0, _helpers.combineStyles)(_Toast.UNDO_BUTTON_BASE, undoButtonColor);

          undoContent = _react2.default.createElement(
            'a',
            {
              'data-test-toast-undo-button': true,
              onClick: this._handleUndo,
              onMouseEnter: this._handleMouseEnter,
              onMouseLeave: this._handleMouseLeave,
              style: undoButtonStyle
            },
            'UNDO'
          );
        }

        undoContainer = _react2.default.createElement(
          'div',
          { style: _Toast.UNDO_CONTAINER_STYLES },
          undoContent
        );
      }

      var toastContent = this.state.errorMessage || this.props.children;

      var toastStyles = (0, _helpers.combineStyles)(this.props._style, _Toast.TOAST_STYLES);

      return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, mouseEvents, {
          'data-component': 'TxlToast',
          style: toastStyles
        }),
        _react2.default.createElement(
          'div',
          {
            style: _Toast.TOAST_MESSAGE_STYLES,
            'data-test-toast-content': true
          },
          toastContent
        ),
        undoContainer
      );
    }
  }]);
  return TxlToast;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleUndo', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleUndo'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleUndoError', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleUndoError'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleMouseEnter', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseEnter'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleMouseLeave', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseLeave'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_pauseTimer', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_pauseTimer'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_unPauseTimer', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_unPauseTimer'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_startTimer', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_startTimer'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_endTimer', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_endTimer'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_expire', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_expire'), _class2.prototype)), _class2)) || _class;

exports.default = TxlToast;


TxlToast.propTypes = {
  _style: _react.PropTypes.object,
  children: _react.PropTypes.node.isRequired,
  lifetime: _react.PropTypes.number,
  onExpire: _react.PropTypes.func,
  onUndo: _react.PropTypes.func
};

TxlToast.defaultProps = {
  onExpire: _Base.NOOP
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlToast, 'TxlToast', 'src/notifications/Toast.jsx');
}();

;
module.exports = exports['default'];
