// txl/file-preview/FileDetailsModal.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MODAL_CONTAINER_STYLE = exports.IMAGE_PREVIEW_STYLE = exports.IMAGE_PREVIEW_CONTAINER_STYLE = exports.FILE_PREVIEW_CONTAINER_STYLE = exports.FILE_INFO_STYLE = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _theme = require('../styles/theme');

var _type = require('../styles/mixins/type');

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BASE_PREVIEW_CONTAINER_STYLE = {
  alignItems: 'center',
  backgroundColor: _theme.COLOR_NEUTRAL['50'],
  display: 'flex',
  flex: '1',
  justifyContent: 'center',
  marginRight: (0, _helpers.gridUnits)(2)
};

var IMAGE_PREVIEW_CONTAINER_STYLE = (0, _extends3.default)({}, BASE_PREVIEW_CONTAINER_STYLE);

var FILE_PREVIEW_CONTAINER_STYLE = (0, _extends3.default)({}, BASE_PREVIEW_CONTAINER_STYLE, {
  marginBottom: (0, _helpers.gridUnits)(-4),
  marginLeft: (0, _helpers.gridUnits)(-4),
  marginTop: (0, _helpers.gridUnits)(-4)
});

var FILE_INFO_STYLE = (0, _extends3.default)({}, _type.TYPE_BASE, {
  flex: '1',
  marginLeft: (0, _helpers.gridUnits)(2),
  overflow: 'hidden',
  wordWrap: 'break-word'
});

var MODAL_CONTAINER_STYLE = {
  display: 'flex'
};

var IMAGE_PREVIEW_STYLE = {
  maxHeight: '100%',
  maxWidth: '100%'
};

exports.FILE_INFO_STYLE = FILE_INFO_STYLE;
exports.FILE_PREVIEW_CONTAINER_STYLE = FILE_PREVIEW_CONTAINER_STYLE;
exports.IMAGE_PREVIEW_CONTAINER_STYLE = IMAGE_PREVIEW_CONTAINER_STYLE;
exports.IMAGE_PREVIEW_STYLE = IMAGE_PREVIEW_STYLE;
exports.MODAL_CONTAINER_STYLE = MODAL_CONTAINER_STYLE;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(BASE_PREVIEW_CONTAINER_STYLE, 'BASE_PREVIEW_CONTAINER_STYLE', 'src/file-preview/FileDetailsModal.style.js');

  __REACT_HOT_LOADER__.register(IMAGE_PREVIEW_CONTAINER_STYLE, 'IMAGE_PREVIEW_CONTAINER_STYLE', 'src/file-preview/FileDetailsModal.style.js');

  __REACT_HOT_LOADER__.register(FILE_PREVIEW_CONTAINER_STYLE, 'FILE_PREVIEW_CONTAINER_STYLE', 'src/file-preview/FileDetailsModal.style.js');

  __REACT_HOT_LOADER__.register(FILE_INFO_STYLE, 'FILE_INFO_STYLE', 'src/file-preview/FileDetailsModal.style.js');

  __REACT_HOT_LOADER__.register(MODAL_CONTAINER_STYLE, 'MODAL_CONTAINER_STYLE', 'src/file-preview/FileDetailsModal.style.js');

  __REACT_HOT_LOADER__.register(IMAGE_PREVIEW_STYLE, 'IMAGE_PREVIEW_STYLE', 'src/file-preview/FileDetailsModal.style.js');
}();

;
