// txl/file-preview/FileDetailsModal.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _slicedToArray2 = require('babel-runtime/helpers/slicedToArray');

var _slicedToArray3 = _interopRequireDefault(_slicedToArray2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _Base = require('../base/Base');

var _formatter = require('../utils/formatter');

var _Modal = require('../modals/Modal');

var _Modal2 = _interopRequireDefault(_Modal);

var _Button = require('../buttons/Button');

var _Button2 = _interopRequireDefault(_Button);

var _FilePlaceholder = require('../icons/FilePlaceholder');

var _FilePlaceholder2 = _interopRequireDefault(_FilePlaceholder);

var _theme = require('../styles/theme');

var _FileDetailsModal = require('./FileDetailsModal.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

// @TODO-TRANSLATE
var CLOSE_BUTTON_TEXT = 'Close';
var IMAGE_HEADER_TITLE = 'Image Details';
var FILE_HEADER_TITLE = 'File Details';
var FILE_NAME_LABEL_TEXT = 'File name: ';
var FILE_TYPE_LABEL_TEXT = 'File type: ';
var UPLOADED_ON_LABEL_TEXT = 'Uploaded on: ';
var FILE_SIZE_LABEL_TEXT = 'File size: ';
var DIMENSIONS_LABEL_TEXT = 'Dimensions: ';

var TxlFileDetailsModal = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlFileDetailsModal, _React$Component);

  function TxlFileDetailsModal() {
    (0, _classCallCheck3.default)(this, TxlFileDetailsModal);
    return (0, _possibleConstructorReturn3.default)(this, (TxlFileDetailsModal.__proto__ || (0, _getPrototypeOf2.default)(TxlFileDetailsModal)).apply(this, arguments));
  }

  (0, _createClass3.default)(TxlFileDetailsModal, [{
    key: '_handleFileNameLabelRef',
    value: function _handleFileNameLabelRef(ref) {
      this._fileNameLabel = ref;
    }
  }, {
    key: '_handleFileTypeLabelRef',
    value: function _handleFileTypeLabelRef(ref) {
      this._fileTypeLabel = ref;
    }
  }, {
    key: '_handleFileUploadDateLabelRef',
    value: function _handleFileUploadDateLabelRef(ref) {
      this._fileUploadDateLabel = ref;
    }
  }, {
    key: '_handleFileSizeLabelRef',
    value: function _handleFileSizeLabelRef(ref) {
      this._fileSizeLabel = ref;
    }
  }, {
    key: '_handleDimensionsLabelRef',
    value: function _handleDimensionsLabelRef(ref) {
      this._dimensionsLabel = ref;
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          dateUploaded = _props.dateUploaded,
          file = _props.file,
          fileURL = _props.fileURL,
          isImage = _props.isImage,
          imageHeight = _props.imageHeight,
          imageWidth = _props.imageWidth,
          onDismiss = _props.onDismiss;


      var footer = _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(
          _Button2.default,
          {
            size: 'large',
            variant: 'accent',
            onClick: onDismiss
          },
          CLOSE_BUTTON_TEXT
        )
      );

      var previewStyle = isImage ? _FileDetailsModal.IMAGE_PREVIEW_CONTAINER_STYLE : _FileDetailsModal.FILE_PREVIEW_CONTAINER_STYLE;

      var header = isImage ? IMAGE_HEADER_TITLE : FILE_HEADER_TITLE;
      var matches = file.name.match(/(^[\D\d]+)\.([\w]+)$/i);

      var _ref = matches || [file.name, file.name, ''],
          _ref2 = (0, _slicedToArray3.default)(_ref, 3),
          fileName = _ref2[1],
          fileType = _ref2[2]; // handle matches being null


      var fileSize = (0, _formatter.convertBytesToHumanReadableFormat)(file.size);

      var uploadedOn = (0, _moment2.default)(dateUploaded).format('MMMM DD, YYYY');
      var dimensions = imageWidth + ' x ' + imageHeight;

      return _react2.default.createElement(
        _Modal2.default,
        (0, _extends3.default)({}, this.props, {
          size: 'small',
          header: header,
          footer: footer
        }),
        _react2.default.createElement(
          'div',
          { style: _FileDetailsModal.MODAL_CONTAINER_STYLE },
          _react2.default.createElement(
            'div',
            { style: previewStyle },
            isImage && _react2.default.createElement('img', {
              style: _FileDetailsModal.IMAGE_PREVIEW_STYLE,
              src: fileURL
            }),
            !isImage && _react2.default.createElement(_FilePlaceholder2.default, {
              scale: 5,
              color: _theme.COLOR_NEUTRAL['200']
            })
          ),
          _react2.default.createElement(
            'div',
            { style: _FileDetailsModal.FILE_INFO_STYLE },
            _react2.default.createElement(
              'div',
              null,
              FILE_NAME_LABEL_TEXT,
              _react2.default.createElement(
                'span',
                { ref: this._handleFileNameLabelRef },
                fileName
              )
            ),
            _react2.default.createElement(
              'div',
              null,
              FILE_TYPE_LABEL_TEXT,
              _react2.default.createElement(
                'span',
                { ref: this._handleFileTypeLabelRef },
                fileType
              )
            ),
            _react2.default.createElement(
              'div',
              null,
              UPLOADED_ON_LABEL_TEXT,
              _react2.default.createElement(
                'span',
                { ref: this._handleFileUploadDateLabelRef },
                uploadedOn
              )
            ),
            _react2.default.createElement(
              'div',
              null,
              FILE_SIZE_LABEL_TEXT,
              _react2.default.createElement(
                'span',
                { ref: this._handleFileSizeLabelRef },
                fileSize
              )
            ),
            imageWidth && imageHeight && _react2.default.createElement(
              'div',
              null,
              DIMENSIONS_LABEL_TEXT,
              _react2.default.createElement(
                'span',
                { ref: this._handleDimensionsLabelRef },
                dimensions
              )
            )
          )
        )
      );
    }
  }]);
  return TxlFileDetailsModal;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleFileNameLabelRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleFileNameLabelRef'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleFileTypeLabelRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleFileTypeLabelRef'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleFileUploadDateLabelRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleFileUploadDateLabelRef'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleFileSizeLabelRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleFileSizeLabelRef'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDimensionsLabelRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDimensionsLabelRef'), _class2.prototype)), _class2)) || _class;

exports.default = TxlFileDetailsModal;


TxlFileDetailsModal.displayName = 'TxlFileDetailsModal';

TxlFileDetailsModal.propTypes = {
  dateUploaded: _react.PropTypes.instanceOf(Date),
  file: _react.PropTypes.instanceOf(File),
  fileURL: _react.PropTypes.string,
  imageHeight: _react.PropTypes.number,
  imageWidth: _react.PropTypes.number,
  isImage: _react.PropTypes.bool,
  onDismiss: _react.PropTypes.func
};

TxlFileDetailsModal.defaultProps = {
  dateUploaded: new Date(),
  isImage: false,
  onDismiss: _Base.NOOP
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(CLOSE_BUTTON_TEXT, 'CLOSE_BUTTON_TEXT', 'src/file-preview/FileDetailsModal.jsx');

  __REACT_HOT_LOADER__.register(IMAGE_HEADER_TITLE, 'IMAGE_HEADER_TITLE', 'src/file-preview/FileDetailsModal.jsx');

  __REACT_HOT_LOADER__.register(FILE_HEADER_TITLE, 'FILE_HEADER_TITLE', 'src/file-preview/FileDetailsModal.jsx');

  __REACT_HOT_LOADER__.register(FILE_NAME_LABEL_TEXT, 'FILE_NAME_LABEL_TEXT', 'src/file-preview/FileDetailsModal.jsx');

  __REACT_HOT_LOADER__.register(FILE_TYPE_LABEL_TEXT, 'FILE_TYPE_LABEL_TEXT', 'src/file-preview/FileDetailsModal.jsx');

  __REACT_HOT_LOADER__.register(UPLOADED_ON_LABEL_TEXT, 'UPLOADED_ON_LABEL_TEXT', 'src/file-preview/FileDetailsModal.jsx');

  __REACT_HOT_LOADER__.register(FILE_SIZE_LABEL_TEXT, 'FILE_SIZE_LABEL_TEXT', 'src/file-preview/FileDetailsModal.jsx');

  __REACT_HOT_LOADER__.register(DIMENSIONS_LABEL_TEXT, 'DIMENSIONS_LABEL_TEXT', 'src/file-preview/FileDetailsModal.jsx');

  __REACT_HOT_LOADER__.register(TxlFileDetailsModal, 'TxlFileDetailsModal', 'src/file-preview/FileDetailsModal.jsx');
}();

;
module.exports = exports['default'];
