// txl/file-preview/FilePreview.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UPLOAD_OVERLAY_STYLE = exports.OVERLAY_CONTAINER_STYLE = exports.IMAGE_PREVIEW_STYLE = exports.IMAGE_PREVIEW_CONTAINER_STYLE = exports.FILE_PREVIEW_CONTAINER_STYLE = exports.FILE_PLACEHOLDER_TEXT_STYLE = exports.FILE_PLACEHOLDER_PREVIEW_STYLE = exports.FILE_PLACEHOLDER_PREVIEW_ICON_COLOR = exports.FILE_INPUT_STYLE = exports.DRAG_OVERLAY_STYLE = exports.CONTROLS_OVERLAY_STYLE = exports.CONTROLS_ITEM_STYLE = exports.CANCEL_UPLOAD_ICON_STYLE = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _theme = require('../styles/theme');

var _type = require('../styles/mixins/type');

var _colorHelpers = require('../styles/_helpers/colorHelpers');

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FILE_PREVIEW_CONTAINER_STYLE = {
  height: (0, _helpers.gridUnits)(26),
  position: 'relative',
  width: (0, _helpers.gridUnits)(26)
};

var OVERLAY_CONTAINER_STYLE = {
  backgroundColor: (0, _colorHelpers.getColorWithOpacity)(_theme.COLOR_NEUTRAL['900'], 0.8),
  height: '100%',
  left: '0',
  position: 'absolute',
  top: '0',
  width: '100%'
};

var DRAG_OVERLAY_STYLE = (0, _extends3.default)({}, OVERLAY_CONTAINER_STYLE, {
  backgroundColor: (0, _colorHelpers.getColorWithOpacity)(_theme.COLOR_ACCENT['300'], 0.5),
  pointerEvents: 'none'
});

var CONTROLS_OVERLAY_STYLE = {
  alignItems: 'center',
  boxSizing: 'border-box',
  display: 'flex',
  height: '100%',
  justifyContent: 'center',
  width: '100%'
};

var UPLOAD_OVERLAY_STYLE = {
  bottom: '0',
  boxSizing: 'border-box',
  height: '50%',
  paddingLeft: (0, _helpers.gridUnits)(2),
  paddingRight: (0, _helpers.gridUnits)(2),
  position: 'absolute',
  textAlign: 'center',
  width: '100%'
};

var CANCEL_UPLOAD_ICON_STYLE = {
  marginTop: (0, _helpers.gridUnits)(4)
};

var FILE_INPUT_STYLE = {
  display: 'none'
};

var CONTROLS_ITEM_STYLE = {
  padding: '0px 7.5px'
};

var IMAGE_PREVIEW_STYLE = {
  maxHeight: '100%',
  maxWidth: '100%'
};

var IMAGE_PREVIEW_CONTAINER_STYLE = {
  alignItems: 'center',
  display: 'flex',
  height: '100%',
  justifyContent: 'center',
  width: '100%'
};

var FILE_PLACEHOLDER_PREVIEW_STYLE = {
  boxSizing: 'border-box',
  height: '100%',
  overflowWrap: 'break-word',
  paddingLeft: (0, _helpers.gridUnits)(2),
  paddingRight: (0, _helpers.gridUnits)(2),
  paddingTop: (0, _helpers.gridUnits)(3),
  textAlign: 'center',
  width: '100%'
};

var FILE_PLACEHOLDER_TEXT_STYLE = (0, _extends3.default)({}, _type.TYPE_SMALL, {
  color: _theme.COLOR_NEUTRAL['400'],
  marginTop: (0, _helpers.gridUnits)(2)
});

var FILE_PLACEHOLDER_PREVIEW_ICON_COLOR = _theme.COLOR_NEUTRAL['200'];

exports.CANCEL_UPLOAD_ICON_STYLE = CANCEL_UPLOAD_ICON_STYLE;
exports.CONTROLS_ITEM_STYLE = CONTROLS_ITEM_STYLE;
exports.CONTROLS_OVERLAY_STYLE = CONTROLS_OVERLAY_STYLE;
exports.DRAG_OVERLAY_STYLE = DRAG_OVERLAY_STYLE;
exports.FILE_INPUT_STYLE = FILE_INPUT_STYLE;
exports.FILE_PLACEHOLDER_PREVIEW_ICON_COLOR = FILE_PLACEHOLDER_PREVIEW_ICON_COLOR;
exports.FILE_PLACEHOLDER_PREVIEW_STYLE = FILE_PLACEHOLDER_PREVIEW_STYLE;
exports.FILE_PLACEHOLDER_TEXT_STYLE = FILE_PLACEHOLDER_TEXT_STYLE;
exports.FILE_PREVIEW_CONTAINER_STYLE = FILE_PREVIEW_CONTAINER_STYLE;
exports.IMAGE_PREVIEW_CONTAINER_STYLE = IMAGE_PREVIEW_CONTAINER_STYLE;
exports.IMAGE_PREVIEW_STYLE = IMAGE_PREVIEW_STYLE;
exports.OVERLAY_CONTAINER_STYLE = OVERLAY_CONTAINER_STYLE;
exports.UPLOAD_OVERLAY_STYLE = UPLOAD_OVERLAY_STYLE;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(FILE_PREVIEW_CONTAINER_STYLE, 'FILE_PREVIEW_CONTAINER_STYLE', 'src/file-preview/FilePreview.style.js');

  __REACT_HOT_LOADER__.register(OVERLAY_CONTAINER_STYLE, 'OVERLAY_CONTAINER_STYLE', 'src/file-preview/FilePreview.style.js');

  __REACT_HOT_LOADER__.register(DRAG_OVERLAY_STYLE, 'DRAG_OVERLAY_STYLE', 'src/file-preview/FilePreview.style.js');

  __REACT_HOT_LOADER__.register(CONTROLS_OVERLAY_STYLE, 'CONTROLS_OVERLAY_STYLE', 'src/file-preview/FilePreview.style.js');

  __REACT_HOT_LOADER__.register(UPLOAD_OVERLAY_STYLE, 'UPLOAD_OVERLAY_STYLE', 'src/file-preview/FilePreview.style.js');

  __REACT_HOT_LOADER__.register(CANCEL_UPLOAD_ICON_STYLE, 'CANCEL_UPLOAD_ICON_STYLE', 'src/file-preview/FilePreview.style.js');

  __REACT_HOT_LOADER__.register(FILE_INPUT_STYLE, 'FILE_INPUT_STYLE', 'src/file-preview/FilePreview.style.js');

  __REACT_HOT_LOADER__.register(CONTROLS_ITEM_STYLE, 'CONTROLS_ITEM_STYLE', 'src/file-preview/FilePreview.style.js');

  __REACT_HOT_LOADER__.register(IMAGE_PREVIEW_STYLE, 'IMAGE_PREVIEW_STYLE', 'src/file-preview/FilePreview.style.js');

  __REACT_HOT_LOADER__.register(IMAGE_PREVIEW_CONTAINER_STYLE, 'IMAGE_PREVIEW_CONTAINER_STYLE', 'src/file-preview/FilePreview.style.js');

  __REACT_HOT_LOADER__.register(FILE_PLACEHOLDER_PREVIEW_STYLE, 'FILE_PLACEHOLDER_PREVIEW_STYLE', 'src/file-preview/FilePreview.style.js');

  __REACT_HOT_LOADER__.register(FILE_PLACEHOLDER_TEXT_STYLE, 'FILE_PLACEHOLDER_TEXT_STYLE', 'src/file-preview/FilePreview.style.js');

  __REACT_HOT_LOADER__.register(FILE_PLACEHOLDER_PREVIEW_ICON_COLOR, 'FILE_PLACEHOLDER_PREVIEW_ICON_COLOR', 'src/file-preview/FilePreview.style.js');
}();

;
