// txl/file-preview/FilePreviewControlled.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getOwnPropertyDescriptor = require('babel-runtime/core-js/object/get-own-property-descriptor');

var _getOwnPropertyDescriptor2 = _interopRequireDefault(_getOwnPropertyDescriptor);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _desc, _value, _class2;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Base = require('../base/Base');

var _theme = require('../styles/theme');

var _Close = require('../icons/Close');

var _Close2 = _interopRequireDefault(_Close);

var _EditOutline = require('../icons/EditOutline');

var _EditOutline2 = _interopRequireDefault(_EditOutline);

var _TrashOutline = require('../icons/TrashOutline');

var _TrashOutline2 = _interopRequireDefault(_TrashOutline);

var _PhotoOutline = require('../icons/PhotoOutline');

var _PhotoOutline2 = _interopRequireDefault(_PhotoOutline);

var _ProgressBar = require('../progress-indicators/ProgressBar');

var _ProgressBar2 = _interopRequireDefault(_ProgressBar);

var _FilePreview = require('./FilePreview.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
  var desc = {};
  Object['ke' + 'ys'](descriptor).forEach(function (key) {
    desc[key] = descriptor[key];
  });
  desc.enumerable = !!desc.enumerable;
  desc.configurable = !!desc.configurable;

  if ('value' in desc || desc.initializer) {
    desc.writable = true;
  }

  desc = decorators.slice().reverse().reduce(function (desc, decorator) {
    return decorator(target, property, desc) || desc;
  }, desc);

  if (context && desc.initializer !== void 0) {
    desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
    desc.initializer = undefined;
  }

  if (desc.initializer === void 0) {
    Object['define' + 'Property'](target, property, desc);
    desc = null;
  }

  return desc;
}

var TxlFilePreviewControlled = (0, _radium2.default)(_class = (_class2 = function (_React$Component) {
  (0, _inherits3.default)(TxlFilePreviewControlled, _React$Component);

  function TxlFilePreviewControlled() {
    (0, _classCallCheck3.default)(this, TxlFilePreviewControlled);
    return (0, _possibleConstructorReturn3.default)(this, (TxlFilePreviewControlled.__proto__ || (0, _getPrototypeOf2.default)(TxlFilePreviewControlled)).apply(this, arguments));
  }

  (0, _createClass3.default)(TxlFilePreviewControlled, [{
    key: '_handleMouseEnter',
    value: function _handleMouseEnter(event) {
      this.props.onMouseEnter({ event: event });
    }
  }, {
    key: '_handleMouseLeave',
    value: function _handleMouseLeave(event) {
      this.props.onMouseLeave({ event: event });
    }
  }, {
    key: '_handleDragOver',
    value: function _handleDragOver(event) {
      this.props.onDragOver({ event: event });
    }
  }, {
    key: '_handleDragLeave',
    value: function _handleDragLeave(event) {
      this.props.onDragLeave({ event: event });
    }
  }, {
    key: '_handleDrop',
    value: function _handleDrop(event) {
      this.props.onDrop({ event: event });
    }
  }, {
    key: '_handleDetails',
    value: function _handleDetails() {
      var _props = this.props,
          fileInfo = _props.fileInfo,
          onDetails = _props.onDetails;


      onDetails({ fileInfo: fileInfo });
    }
  }, {
    key: '_handleEdit',
    value: function _handleEdit() {
      var _props2 = this.props,
          fileInfo = _props2.fileInfo,
          onEdit = _props2.onEdit;


      onEdit({ fileInfo: fileInfo });
    }
  }, {
    key: '_handleDelete',
    value: function _handleDelete() {
      var _props3 = this.props,
          fileInfo = _props3.fileInfo,
          onDelete = _props3.onDelete;


      onDelete({ fileInfo: fileInfo });
    }
  }, {
    key: '_handleCancel',
    value: function _handleCancel() {
      var _props4 = this.props,
          fileInfo = _props4.fileInfo,
          onCancel = _props4.onCancel;


      onCancel({ fileInfo: fileInfo });
    }
  }, {
    key: '_handleFilePreviewWrapperRef',
    value: function _handleFilePreviewWrapperRef(ref) {
      this._filePreviewWrapper = ref;
    }
  }, {
    key: '_handleUploadingOverlayRef',
    value: function _handleUploadingOverlayRef(ref) {
      this._uploadingOverlay = ref;
    }
  }, {
    key: '_handleProgressBarRef',
    value: function _handleProgressBarRef(ref) {
      this._progressBar = ref;
    }
  }, {
    key: '_handleCancelWrapperRef',
    value: function _handleCancelWrapperRef(ref) {
      this._cancelWrapper = ref;
    }
  }, {
    key: '_handleCancelButtonRef',
    value: function _handleCancelButtonRef(ref) {
      this._cancelButton = ref;
    }
  }, {
    key: '_handleEditOverlayRef',
    value: function _handleEditOverlayRef(ref) {
      this._editOverlay = ref;
    }
  }, {
    key: '_handleDetailsWrapperRef',
    value: function _handleDetailsWrapperRef(ref) {
      this._detailsWrapper = ref;
    }
  }, {
    key: '_handleEditWrapperRef',
    value: function _handleEditWrapperRef(ref) {
      this._editWrapper = ref;
    }
  }, {
    key: '_handleDeleteWrapperRef',
    value: function _handleDeleteWrapperRef(ref) {
      this._deleteWrapper = ref;
    }
  }, {
    key: '_handleDragOverlayRef',
    value: function _handleDragOverlayRef(ref) {
      this._dragOverlay = ref;
    }
  }, {
    key: 'render',
    value: function render() {
      var _props5 = this.props,
          dragging = _props5.dragging,
          fileInfo = _props5.fileInfo,
          hovered = _props5.hovered,
          preview = _props5.preview;


      var isUploading = fileInfo ? fileInfo.uploading : false;

      var percentUploaded = fileInfo ? fileInfo.percentUploaded : 0;
      var showUploadOverlay = isUploading;
      var showControlsOverlay = hovered && this.props.fileInfo !== undefined && !isUploading;
      var showDragOverlay = dragging && this.props.fileInfo !== undefined && !isUploading;

      var COLOR_APP_THEME = (0, _theme.determineColorAppTheme)(this.context.txlAppTheme);
      var ICON_COLOR = COLOR_APP_THEME['500c'];

      return _react2.default.createElement(
        'div',
        {
          'data-component': 'TxlFilePreviewControlled',
          ref: this._handleFilePreviewWrapperRef,
          style: _FilePreview.FILE_PREVIEW_CONTAINER_STYLE,
          onDragLeave: this._handleDragLeave,
          onDragOver: this._handleDragOver,
          onDrop: this._handleDrop,
          onMouseEnter: this._handleMouseEnter,
          onMouseLeave: this._handleMouseLeave
        },
        preview,
        showUploadOverlay && _react2.default.createElement(
          'div',
          {
            ref: this._handleUploadingOverlayRef,
            style: _FilePreview.OVERLAY_CONTAINER_STYLE
          },
          _react2.default.createElement(
            'div',
            { style: _FilePreview.UPLOAD_OVERLAY_STYLE },
            _react2.default.createElement(_ProgressBar2.default, {
              ref: this._handleProgressBarRef,
              percentage: percentUploaded
            }),
            _react2.default.createElement(
              'div',
              {
                onClick: this._handleCancel,
                ref: this._handleCancelWrapperRef,
                style: _FilePreview.CANCEL_UPLOAD_ICON_STYLE
              },
              _react2.default.createElement(_Close2.default, {
                color: ICON_COLOR,
                ref: this._handleCancelButtonRef
              })
            )
          )
        ),
        showControlsOverlay && _react2.default.createElement(
          'div',
          {
            ref: this._handleEditOverlayRef,
            style: _FilePreview.OVERLAY_CONTAINER_STYLE
          },
          _react2.default.createElement(
            'div',
            { style: _FilePreview.CONTROLS_OVERLAY_STYLE },
            _react2.default.createElement(
              'div',
              {
                ref: this._handleDetailsWrapperRef,
                style: _FilePreview.CONTROLS_ITEM_STYLE,
                onClick: this._handleDetails
              },
              _react2.default.createElement(_PhotoOutline2.default, { color: ICON_COLOR })
            ),
            _react2.default.createElement(
              'div',
              {
                ref: this._handleEditWrapperRef,
                style: _FilePreview.CONTROLS_ITEM_STYLE, onClick: this._handleEdit
              },
              _react2.default.createElement(_EditOutline2.default, { color: ICON_COLOR })
            ),
            _react2.default.createElement(
              'div',
              {
                ref: this._handleDeleteWrapperRef,
                style: _FilePreview.CONTROLS_ITEM_STYLE, onClick: this._handleDelete
              },
              _react2.default.createElement(_TrashOutline2.default, { color: ICON_COLOR })
            )
          )
        ),
        showDragOverlay && _react2.default.createElement('div', {
          ref: this._handleDragOverlayRef,
          style: _FilePreview.DRAG_OVERLAY_STYLE
        })
      );
    }
  }]);
  return TxlFilePreviewControlled;
}(_react2.default.Component), (_applyDecoratedDescriptor(_class2.prototype, '_handleMouseEnter', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseEnter'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleMouseLeave', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleMouseLeave'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDragOver', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDragOver'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDragLeave', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDragLeave'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDrop', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDrop'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDetails', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDetails'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleEdit', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleEdit'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDelete', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDelete'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleCancel', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleCancel'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleFilePreviewWrapperRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleFilePreviewWrapperRef'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleUploadingOverlayRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleUploadingOverlayRef'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleProgressBarRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleProgressBarRef'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleCancelWrapperRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleCancelWrapperRef'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleCancelButtonRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleCancelButtonRef'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleEditOverlayRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleEditOverlayRef'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDetailsWrapperRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDetailsWrapperRef'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleEditWrapperRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleEditWrapperRef'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDeleteWrapperRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDeleteWrapperRef'), _class2.prototype), _applyDecoratedDescriptor(_class2.prototype, '_handleDragOverlayRef', [_Base.autobind], (0, _getOwnPropertyDescriptor2.default)(_class2.prototype, '_handleDragOverlayRef'), _class2.prototype)), _class2)) || _class;

exports.default = TxlFilePreviewControlled;


TxlFilePreviewControlled.displayName = 'TxlFilePreviewControlled';

TxlFilePreviewControlled.propTypes = {
  dragging: _react.PropTypes.bool,
  fileInfo: _react.PropTypes.shape({
    file: _react.PropTypes.instanceOf(File),
    percentUploaded: _react.PropTypes.number,
    uploading: _react.PropTypes.bool
  }),
  hovered: _react.PropTypes.bool,
  onCancel: _react.PropTypes.func,
  onDelete: _react.PropTypes.func,
  onDetails: _react.PropTypes.func,
  onDragLeave: _react.PropTypes.func,
  onDragOver: _react.PropTypes.func,
  onDrop: _react.PropTypes.func,
  onEdit: _react.PropTypes.func,
  onMouseEnter: _react.PropTypes.func,
  onMouseLeave: _react.PropTypes.func,
  preview: _react.PropTypes.object
};

TxlFilePreviewControlled.defaultProps = {
  onCancel: _Base.NOOP,
  onDelete: _Base.NOOP,
  onDetails: _Base.NOOP,
  onDragLeave: _Base.NOOP,
  onDragOver: _Base.NOOP,
  onDrop: _Base.NOOP,
  onEdit: _Base.NOOP,
  onMouseEnter: _Base.NOOP,
  onMouseLeave: _Base.NOOP
};

TxlFilePreviewControlled.contextTypes = {
  txlAppTheme: _theme.contextThemePropType
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(TxlFilePreviewControlled, 'TxlFilePreviewControlled', 'src/file-preview/FilePreviewControlled.jsx');
}();

;
module.exports = exports['default'];
