// txl/navigation/records/Group.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _immutable = require('immutable');

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _DefaultTemplate = require('../DefaultTemplate');

var _DefaultTemplate2 = _interopRequireDefault(_DefaultTemplate);

var _Item = require('./Item');

var _Item2 = _interopRequireDefault(_Item);

var _nextId = require('./nextId');

var _nextId2 = _interopRequireDefault(_nextId);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Group = function (_Record) {
  (0, _inherits3.default)(Group, _Record);

  function Group(args) {
    (0, _classCallCheck3.default)(this, Group);

    var id = args.id || (0, _nextId2.default)();
    return (0, _possibleConstructorReturn3.default)(this, (Group.__proto__ || (0, _getPrototypeOf2.default)(Group)).call(this, (0, _extends3.default)({ id: id }, args)));
  }

  (0, _createClass3.default)(Group, [{
    key: 'update',
    value: function update(predicate, _update) {
      var items = this.items.map(function (item) {
        if (predicate != null && !predicate(item)) {
          return null;
        }
        return _update(item);
      });

      if (items.filter(function (i) {
        return i;
      }).size) {
        var result = this.items.mergeWith(function (oldVal, newVal) {
          return newVal == null ? oldVal : newVal;
        }, items);
        return this.set('items', result);
      }

      return this;
    }
  }]);
  return Group;
}((0, _immutable.Record)({
  active: false,
  display: null,
  expanded: false,
  hovered: false,
  icon: function icon() {
    return null;
  },
  id: 0,
  items: (0, _immutable.List)(),
  meta: (0, _immutable.Map)(),
  name: '',
  template: _DefaultTemplate2.default,
  templateProps: {}
}));

var _default = Group;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(Group, 'Group', 'src/navigation/records/Group.js');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/navigation/records/Group.js');
}();

;
module.exports = exports['default'];
