// txl/progress-indicators/DotsLoadingIndicator.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _theme = require('../styles/theme');

var _animations = require('../styles/mixins/animations');

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DOT_COUNT = 5;

var STYLES = {
  container: {
    alignItems: 'center',
    display: 'flex',
    height: (0, _helpers.gridUnits)(2),
    justifyContent: 'space-between',
    width: (0, _helpers.gridUnits)(16)
  },
  dot: (0, _extends3.default)({}, (0, _helpers.square)('8px'), (0, _animations.getPulseStyles)(), {
    borderRadius: '50%',
    opacity: 0
  })
};

var shadeMap = {
  base: _theme.COLOR_ACCENT['500'],
  dark: _theme.COLOR_NEUTRAL['700'],
  light: _theme.COLOR_NEUTRAL['50']
};

function getDotStyles(index, shade) {
  var delayValue = _animations.PULSE_DURATION / DOT_COUNT * (index + 1);
  return (0, _extends3.default)({}, STYLES.dot, {
    animationDelay: delayValue + 'ms',
    backgroundColor: shadeMap[shade]
  });
}

var TxlDotsLoadingIndicator = function TxlDotsLoadingIndicator(_ref) {
  var shade = _ref.shade;

  var dots = [];

  for (var i = 0; i < DOT_COUNT; i++) {
    dots.push(_react2.default.createElement('div', {
      key: i,
      style: getDotStyles(i, shade),
      'data-test-dot-element': true
    }));
  }

  return _react2.default.createElement(
    'div',
    {
      'data-component': 'TxlDotsLoadingIndicator',
      style: STYLES.container
    },
    dots
  );
};

TxlDotsLoadingIndicator.propTypes = {
  shade: _react.PropTypes.oneOf(['light', 'dark', 'base'])
};

TxlDotsLoadingIndicator.defaultProps = {
  shade: 'dark'
};

var _default = (0, _radium2.default)(TxlDotsLoadingIndicator);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(DOT_COUNT, 'DOT_COUNT', 'src/progress-indicators/DotsLoadingIndicator.jsx');

  __REACT_HOT_LOADER__.register(STYLES, 'STYLES', 'src/progress-indicators/DotsLoadingIndicator.jsx');

  __REACT_HOT_LOADER__.register(shadeMap, 'shadeMap', 'src/progress-indicators/DotsLoadingIndicator.jsx');

  __REACT_HOT_LOADER__.register(getDotStyles, 'getDotStyles', 'src/progress-indicators/DotsLoadingIndicator.jsx');

  __REACT_HOT_LOADER__.register(TxlDotsLoadingIndicator, 'TxlDotsLoadingIndicator', 'src/progress-indicators/DotsLoadingIndicator.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/progress-indicators/DotsLoadingIndicator.jsx');
}();

;
module.exports = exports['default'];
