// txl/progress-indicators/LoadingIndicator.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _helpers = require('../styles/helpers');

var _animations = require('../styles/mixins/animations');

var _theme = require('../styles/theme');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function getBorderColors(color) {
  return {
    borderBottomColor: color,
    borderLeftColor: color,
    borderTopColor: color
  };
}

var STYLES = {
  base: (0, _extends3.default)({}, (0, _animations.getSpinStyles)(), {
    borderRadius: '50%',
    borderRightColor: 'transparent',
    borderStyle: 'solid',
    borderWidth: 2
  }),
  shade: {
    base: getBorderColors(_theme.COLOR_ACCENT['500']),
    dark: getBorderColors(_theme.COLOR_NEUTRAL['700']),
    light: getBorderColors(_theme.COLOR_NEUTRAL['50'])
  },
  size: {
    large: (0, _helpers.square)('63px'),
    medium: (0, _helpers.square)('33px'),
    small: (0, _helpers.square)('13px')
  }
};

var TxlLoadingIndicator = function TxlLoadingIndicator(props) {
  var color = props.color,
      shade = props.shade,
      size = props.size;


  var borderColor = color ? getBorderColors(color) : STYLES.shade[shade];
  var styles = (0, _helpers.combineStyles)(STYLES.base, STYLES.size[size], borderColor);

  return _react2.default.createElement('div', {
    'data-component': 'TxlLoadingIndicator',
    style: styles
  });
};

TxlLoadingIndicator.propTypes = {
  color: _react.PropTypes.string,
  shade: _react.PropTypes.oneOf(['light', 'dark', 'base']),
  size: _react.PropTypes.oneOf(['small', 'medium', 'large'])
};

TxlLoadingIndicator.defaultProps = {
  shade: 'dark',
  size: 'large'
};

var _default = (0, _radium2.default)(TxlLoadingIndicator);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(getBorderColors, 'getBorderColors', 'src/progress-indicators/LoadingIndicator.jsx');

  __REACT_HOT_LOADER__.register(STYLES, 'STYLES', 'src/progress-indicators/LoadingIndicator.jsx');

  __REACT_HOT_LOADER__.register(TxlLoadingIndicator, 'TxlLoadingIndicator', 'src/progress-indicators/LoadingIndicator.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/progress-indicators/LoadingIndicator.jsx');
}();

;
module.exports = exports['default'];
