// txl/progress-indicators/ProgressBar.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _theme = require('../styles/theme');

var _helpers = require('../styles/helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BORDER_RADIUS = '10px';
var BAR_STYLE = {
  borderRadius: BORDER_RADIUS,
  height: '3px',
  minWidth: (0, _helpers.gridUnits)(16)
};

var PROGRESS_FILL_STYLE = {
  borderBottomLeftRadius: BORDER_RADIUS,
  borderTopLeftRadius: BORDER_RADIUS,
  display: 'block',
  height: '100%',
  overflow: 'hidden'
};

var shadeMap = {
  background: {
    base: _theme.COLOR_ACCENT['300'],
    dark: _theme.COLOR_NEUTRAL['200'],
    light: _theme.COLOR_NEUTRAL['200']
  },
  foreground: {
    base: _theme.COLOR_ACCENT['500'],
    dark: _theme.COLOR_NEUTRAL['500'],
    light: _theme.COLOR_NEUTRAL['50']
  }
};

var TxlProgressBar = function TxlProgressBar(props) {
  var percentage = props.percentage,
      shade = props.shade;


  var CURRENT_PROGRESS_FILL_STYLE = (0, _extends3.default)({}, PROGRESS_FILL_STYLE, {
    backgroundColor: shadeMap.foreground[shade],
    width: Math.min(percentage, 100) + '%'
  });

  var BACKGROUND_BAR_STYLE = (0, _extends3.default)({}, BAR_STYLE, {
    backgroundColor: shadeMap.background[shade]
  });

  return _react2.default.createElement(
    'div',
    {
      'data-component': 'TxlProgressBar',
      style: BACKGROUND_BAR_STYLE
    },
    _react2.default.createElement('span', { style: CURRENT_PROGRESS_FILL_STYLE })
  );
};

TxlProgressBar.propTypes = {
  percentage: _react.PropTypes.number,
  shade: _react.PropTypes.oneOf(['light', 'dark', 'base'])
};

TxlProgressBar.defaultProps = {
  percentage: 0,
  shade: 'dark'
};

var _default = (0, _radium2.default)(TxlProgressBar);

exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(BORDER_RADIUS, 'BORDER_RADIUS', 'src/progress-indicators/ProgressBar.jsx');

  __REACT_HOT_LOADER__.register(BAR_STYLE, 'BAR_STYLE', 'src/progress-indicators/ProgressBar.jsx');

  __REACT_HOT_LOADER__.register(PROGRESS_FILL_STYLE, 'PROGRESS_FILL_STYLE', 'src/progress-indicators/ProgressBar.jsx');

  __REACT_HOT_LOADER__.register(shadeMap, 'shadeMap', 'src/progress-indicators/ProgressBar.jsx');

  __REACT_HOT_LOADER__.register(TxlProgressBar, 'TxlProgressBar', 'src/progress-indicators/ProgressBar.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/progress-indicators/ProgressBar.jsx');
}();

;
module.exports = exports['default'];
