// txl/action-bar/BulkActionBar.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _FlexContainer = require('../layout/FlexContainer');

var _FlexContainer2 = _interopRequireDefault(_FlexContainer);

var _FlexBox = require('../layout/FlexBox');

var _FlexBox2 = _interopRequireDefault(_FlexBox);

var _FlexItem = require('../layout/FlexItem');

var _FlexItem2 = _interopRequireDefault(_FlexItem);

var _BulkActionBar = require('./BulkActionBar.style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// @TODO-TRANSLATE
var COUNT_STRING = '{0} selected';

var TxlBulkActionBar = function TxlBulkActionBar(_ref) {
  var children = _ref.children,
      _ref$context = _ref.context,
      context = _ref$context === undefined ? 'table' : _ref$context,
      _ref$items = _ref.items,
      items = _ref$items === undefined ? [] : _ref$items;

  var hidden = context === 'table' && !items.length;
  var showCount = context === 'table' && items.length;
  var containerStyles = hidden ? _BulkActionBar.BAR_HIDDEN : _BulkActionBar.BAR;

  return _react2.default.createElement(
    _FlexContainer2.default,
    { style: containerStyles },
    _react2.default.createElement(
      _FlexBox2.default,
      {
        'data-txl-action-bar-count': true,
        flexBasis: _BulkActionBar.INTERCOM_WIDTH,
        style: _BulkActionBar.COUNT
      },
      showCount && COUNT_STRING.replace('{0}', items.length)
    ),
    _react2.default.createElement(
      _FlexBox2.default,
      {
        flexGrow: '1',
        justifyContent: 'center'
      },
      children
    ),
    _react2.default.createElement(_FlexItem2.default, { flexBasis: _BulkActionBar.INTERCOM_WIDTH })
  );
};

TxlBulkActionBar.propTypes = {
  context: require('react').PropTypes.oneOf(['table', 'page']),
  items: require('react').PropTypes.arrayOf(require('react').PropTypes.any),
  children: require('react').PropTypes.any.isRequired
};
var _default = TxlBulkActionBar;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(COUNT_STRING, 'COUNT_STRING', 'src/action-bar/BulkActionBar.jsx');

  __REACT_HOT_LOADER__.register(TxlBulkActionBar, 'TxlBulkActionBar', 'src/action-bar/BulkActionBar.jsx');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/action-bar/BulkActionBar.jsx');
}();

;
module.exports = exports['default'];
