// txl/action-bar/BulkActionBar.style.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.COUNT = exports.BAR_HIDDEN = exports.BAR = exports.INTERCOM_WIDTH = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _helpers = require('../styles/helpers');

var _layers = require('../styles/mixins/layers');

var _constants = require('../styles/constants');

var _type = require('../styles/mixins/type');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// known Intercom icon width + its margins. Hardcoded as px since Intercom knows nothing of grid units
var INTERCOM_WIDTH = exports.INTERCOM_WIDTH = '100px'; // 60px icon width + 2 * 20px margin.

var BAR = exports.BAR = (0, _extends3.default)({}, _type.DEFAULT_FONT_FAMILY, _type.TYPE_BASE, _layers.LAYER_1, {
  alignItems: 'center',
  backgroundColor: 'white',
  bottom: '0',
  height: (0, _helpers.gridUnits)(9),
  left: '0',
  position: 'fixed',
  right: '0',
  transition: 'transform 0.3s ease-in-out',
  willChange: true,
  zIndex: _constants.Z_INDEXES.actionBar
});

var BAR_HIDDEN = exports.BAR_HIDDEN = (0, _extends3.default)({}, BAR, {
  transform: 'translateY(' + (0, _helpers.gridUnits)(9) + ')'
});

var COUNT = exports.COUNT = {
  alignItems: 'center',
  height: (0, _helpers.gridUnits)(5),
  justifyContent: 'center',
  minWidth: (0, _helpers.gridUnits)(10),
  padding: (0, _helpers.gridUnits)(2)
};
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(INTERCOM_WIDTH, 'INTERCOM_WIDTH', 'src/action-bar/BulkActionBar.style.js');

  __REACT_HOT_LOADER__.register(BAR, 'BAR', 'src/action-bar/BulkActionBar.style.js');

  __REACT_HOT_LOADER__.register(BAR_HIDDEN, 'BAR_HIDDEN', 'src/action-bar/BulkActionBar.style.js');

  __REACT_HOT_LOADER__.register(COUNT, 'COUNT', 'src/action-bar/BulkActionBar.style.js');
}();

;
