// txl/constants/KeyCodes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
/**
 * Frequently used keyCodes will be exported as readable constants
 *
 * @module KeyCodes
 *
 * To add more keyCode constants
 * @see https://css-tricks.com/snippets/javascript/javascript-keycodes/
 */
var ARROW_DOWN = exports.ARROW_DOWN = 40;

var ARROW_LEFT = exports.ARROW_LEFT = 37;

var ARROW_RIGHT = exports.ARROW_RIGHT = 39;

var ARROW_UP = exports.ARROW_UP = 38;

var BACKSPACE = exports.BACKSPACE = 8;

var COMMA = exports.COMMA = 188;

var END = exports.END = 35;

var ENTER = exports.ENTER = 13;

var ESCAPE = exports.ESCAPE = 27;

var HOME = exports.HOME = 36;

var PAGE_DOWN = exports.PAGE_DOWN = 34;

var PAGE_UP = exports.PAGE_UP = 33;

var TAB = exports.TAB = 9;

var SPACEBAR = exports.SPACEBAR = 32;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(ARROW_DOWN, "ARROW_DOWN", "src/constants/KeyCodes.js");

  __REACT_HOT_LOADER__.register(ARROW_LEFT, "ARROW_LEFT", "src/constants/KeyCodes.js");

  __REACT_HOT_LOADER__.register(ARROW_RIGHT, "ARROW_RIGHT", "src/constants/KeyCodes.js");

  __REACT_HOT_LOADER__.register(ARROW_UP, "ARROW_UP", "src/constants/KeyCodes.js");

  __REACT_HOT_LOADER__.register(BACKSPACE, "BACKSPACE", "src/constants/KeyCodes.js");

  __REACT_HOT_LOADER__.register(COMMA, "COMMA", "src/constants/KeyCodes.js");

  __REACT_HOT_LOADER__.register(END, "END", "src/constants/KeyCodes.js");

  __REACT_HOT_LOADER__.register(ENTER, "ENTER", "src/constants/KeyCodes.js");

  __REACT_HOT_LOADER__.register(ESCAPE, "ESCAPE", "src/constants/KeyCodes.js");

  __REACT_HOT_LOADER__.register(HOME, "HOME", "src/constants/KeyCodes.js");

  __REACT_HOT_LOADER__.register(PAGE_DOWN, "PAGE_DOWN", "src/constants/KeyCodes.js");

  __REACT_HOT_LOADER__.register(PAGE_UP, "PAGE_UP", "src/constants/KeyCodes.js");

  __REACT_HOT_LOADER__.register(TAB, "TAB", "src/constants/KeyCodes.js");

  __REACT_HOT_LOADER__.register(SPACEBAR, "SPACEBAR", "src/constants/KeyCodes.js");
}();

;
